// @xmtp/proto/ts/dist/cjs/mls/api/v1/mls.pb.js
"use strict";
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v1.181.2
//   protoc               v3.21.12
// source: mls/api/v1/mls.proto
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MlsApiClientImpl = exports.MlsApiServiceName = exports.GetNewestGroupMessageResponse_Response = exports.GetNewestGroupMessageResponse = exports.GetNewestGroupMessageRequest = exports.BatchQueryCommitLogResponse = exports.BatchQueryCommitLogRequest = exports.QueryCommitLogResponse = exports.QueryCommitLogRequest = exports.PublishCommitLogRequest = exports.BatchPublishCommitLogRequest = exports.SubscribeWelcomeMessagesRequest_Filter = exports.SubscribeWelcomeMessagesRequest = exports.SubscribeGroupMessagesRequest_Filter = exports.SubscribeGroupMessagesRequest = exports.QueryWelcomeMessagesResponse = exports.QueryWelcomeMessagesRequest = exports.QueryGroupMessagesResponse = exports.QueryGroupMessagesRequest = exports.PagingInfo = exports.GetIdentityUpdatesResponse_WalletUpdates = exports.GetIdentityUpdatesResponse_Update = exports.GetIdentityUpdatesResponse_RevokedInstallationUpdate = exports.GetIdentityUpdatesResponse_NewInstallationUpdate = exports.GetIdentityUpdatesResponse = exports.GetIdentityUpdatesRequest = exports.RevokeInstallationRequest = exports.FetchKeyPackagesResponse_KeyPackage = exports.FetchKeyPackagesResponse = exports.FetchKeyPackagesRequest = exports.UploadKeyPackageRequest = exports.RegisterInstallationResponse = exports.RegisterInstallationRequest = exports.KeyPackageUpload = exports.SendWelcomeMessagesRequest = exports.SendGroupMessagesRequest = exports.GroupMessageInput_V1 = exports.GroupMessageInput = exports.GroupMessage_V1 = exports.GroupMessage = exports.WelcomeMetadata = exports.WelcomeMessageInput_WelcomePointer = exports.WelcomeMessageInput_V1 = exports.WelcomeMessageInput = exports.WelcomeMessage_WelcomePointer = exports.WelcomeMessage_V1 = exports.WelcomeMessage = exports.SortDirection = exports.protobufPackage = void 0;
exports.sortDirectionFromJSON = sortDirectionFromJSON;
exports.sortDirectionToJSON = sortDirectionToJSON;
/* eslint-disable */
const long_1 = __importDefault(require("long"));
const minimal_1 = __importDefault(require("protobufjs/minimal"));
const operators_1 = require("rxjs/operators");
const empty_pb_1 = require("../../../google/protobuf/empty.pb");
const signature_pb_1 = require("../../../identity/associations/signature.pb");
const signature_pb_2 = require("../../../message_contents/signature.pb");
const commit_log_pb_1 = require("../../message_contents/commit_log.pb");
const welcome_pointer_pb_1 = require("../../message_contents/welcome_pointer.pb");
const wrapper_encryption_pb_1 = require("../../message_contents/wrapper_encryption.pb");
exports.protobufPackage = "xmtp.mls.api.v1";
/** Message API */
/** Sort direction for queries */
var SortDirection;
(function (SortDirection) {
    SortDirection[SortDirection["SORT_DIRECTION_UNSPECIFIED"] = 0] = "SORT_DIRECTION_UNSPECIFIED";
    SortDirection[SortDirection["SORT_DIRECTION_ASCENDING"] = 1] = "SORT_DIRECTION_ASCENDING";
    SortDirection[SortDirection["SORT_DIRECTION_DESCENDING"] = 2] = "SORT_DIRECTION_DESCENDING";
    SortDirection[SortDirection["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(SortDirection || (exports.SortDirection = SortDirection = {}));
function sortDirectionFromJSON(object) {
    switch (object) {
        case 0:
        case "SORT_DIRECTION_UNSPECIFIED":
            return SortDirection.SORT_DIRECTION_UNSPECIFIED;
        case 1:
        case "SORT_DIRECTION_ASCENDING":
            return SortDirection.SORT_DIRECTION_ASCENDING;
        case 2:
        case "SORT_DIRECTION_DESCENDING":
            return SortDirection.SORT_DIRECTION_DESCENDING;
        case -1:
        case "UNRECOGNIZED":
        default:
            return SortDirection.UNRECOGNIZED;
    }
}
function sortDirectionToJSON(object) {
    switch (object) {
        case SortDirection.SORT_DIRECTION_UNSPECIFIED:
            return "SORT_DIRECTION_UNSPECIFIED";
        case SortDirection.SORT_DIRECTION_ASCENDING:
            return "SORT_DIRECTION_ASCENDING";
        case SortDirection.SORT_DIRECTION_DESCENDING:
            return "SORT_DIRECTION_DESCENDING";
        case SortDirection.UNRECOGNIZED:
        default:
            return "UNRECOGNIZED";
    }
}
function createBaseWelcomeMessage() {
    return { v1: undefined, welcomePointer: undefined };
}
exports.WelcomeMessage = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.v1 !== undefined) {
            exports.WelcomeMessage_V1.encode(message.v1, writer.uint32(10).fork()).ldelim();
        }
        if (message.welcomePointer !== undefined) {
            exports.WelcomeMessage_WelcomePointer.encode(message.welcomePointer, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseWelcomeMessage();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.v1 = exports.WelcomeMessage_V1.decode(reader, reader.uint32());
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.welcomePointer = exports.WelcomeMessage_WelcomePointer.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            v1: isSet(object.v1) ? exports.WelcomeMessage_V1.fromJSON(object.v1) : undefined,
            welcomePointer: isSet(object.welcomePointer)
                ? exports.WelcomeMessage_WelcomePointer.fromJSON(object.welcomePointer)
                : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.v1 !== undefined) {
            obj.v1 = exports.WelcomeMessage_V1.toJSON(message.v1);
        }
        if (message.welcomePointer !== undefined) {
            obj.welcomePointer = exports.WelcomeMessage_WelcomePointer.toJSON(message.welcomePointer);
        }
        return obj;
    },
    create(base) {
        return exports.WelcomeMessage.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseWelcomeMessage();
        message.v1 = (object.v1 !== undefined && object.v1 !== null) ? exports.WelcomeMessage_V1.fromPartial(object.v1) : undefined;
        message.welcomePointer = (object.welcomePointer !== undefined && object.welcomePointer !== null)
            ? exports.WelcomeMessage_WelcomePointer.fromPartial(object.welcomePointer)
            : undefined;
        return message;
    },
};
function createBaseWelcomeMessage_V1() {
    return {
        id: long_1.default.UZERO,
        createdNs: long_1.default.UZERO,
        installationKey: new Uint8Array(0),
        data: new Uint8Array(0),
        hpkePublicKey: new Uint8Array(0),
        wrapperAlgorithm: 0,
        welcomeMetadata: new Uint8Array(0),
    };
}
exports.WelcomeMessage_V1 = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (!message.id.equals(long_1.default.UZERO)) {
            writer.uint32(8).uint64(message.id);
        }
        if (!message.createdNs.equals(long_1.default.UZERO)) {
            writer.uint32(16).uint64(message.createdNs);
        }
        if (message.installationKey.length !== 0) {
            writer.uint32(26).bytes(message.installationKey);
        }
        if (message.data.length !== 0) {
            writer.uint32(34).bytes(message.data);
        }
        if (message.hpkePublicKey.length !== 0) {
            writer.uint32(42).bytes(message.hpkePublicKey);
        }
        if (message.wrapperAlgorithm !== 0) {
            writer.uint32(48).int32(message.wrapperAlgorithm);
        }
        if (message.welcomeMetadata.length !== 0) {
            writer.uint32(58).bytes(message.welcomeMetadata);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseWelcomeMessage_V1();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 8) {
                        break;
                    }
                    message.id = reader.uint64();
                    continue;
                case 2:
                    if (tag !== 16) {
                        break;
                    }
                    message.createdNs = reader.uint64();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.installationKey = reader.bytes();
                    continue;
                case 4:
                    if (tag !== 34) {
                        break;
                    }
                    message.data = reader.bytes();
                    continue;
                case 5:
                    if (tag !== 42) {
                        break;
                    }
                    message.hpkePublicKey = reader.bytes();
                    continue;
                case 6:
                    if (tag !== 48) {
                        break;
                    }
                    message.wrapperAlgorithm = reader.int32();
                    continue;
                case 7:
                    if (tag !== 58) {
                        break;
                    }
                    message.welcomeMetadata = reader.bytes();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            id: isSet(object.id) ? long_1.default.fromValue(object.id) : long_1.default.UZERO,
            createdNs: isSet(object.createdNs) ? long_1.default.fromValue(object.createdNs) : long_1.default.UZERO,
            installationKey: isSet(object.installationKey) ? bytesFromBase64(object.installationKey) : new Uint8Array(0),
            data: isSet(object.data) ? bytesFromBase64(object.data) : new Uint8Array(0),
            hpkePublicKey: isSet(object.hpkePublicKey) ? bytesFromBase64(object.hpkePublicKey) : new Uint8Array(0),
            wrapperAlgorithm: isSet(object.wrapperAlgorithm) ? (0, wrapper_encryption_pb_1.welcomeWrapperAlgorithmFromJSON)(object.wrapperAlgorithm) : 0,
            welcomeMetadata: isSet(object.welcomeMetadata) ? bytesFromBase64(object.welcomeMetadata) : new Uint8Array(0),
        };
    },
    toJSON(message) {
        const obj = {};
        if (!message.id.equals(long_1.default.UZERO)) {
            obj.id = (message.id || long_1.default.UZERO).toString();
        }
        if (!message.createdNs.equals(long_1.default.UZERO)) {
            obj.createdNs = (message.createdNs || long_1.default.UZERO).toString();
        }
        if (message.installationKey.length !== 0) {
            obj.installationKey = base64FromBytes(message.installationKey);
        }
        if (message.data.length !== 0) {
            obj.data = base64FromBytes(message.data);
        }
        if (message.hpkePublicKey.length !== 0) {
            obj.hpkePublicKey = base64FromBytes(message.hpkePublicKey);
        }
        if (message.wrapperAlgorithm !== 0) {
            obj.wrapperAlgorithm = (0, wrapper_encryption_pb_1.welcomeWrapperAlgorithmToJSON)(message.wrapperAlgorithm);
        }
        if (message.welcomeMetadata.length !== 0) {
            obj.welcomeMetadata = base64FromBytes(message.welcomeMetadata);
        }
        return obj;
    },
    create(base) {
        return exports.WelcomeMessage_V1.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseWelcomeMessage_V1();
        message.id = (object.id !== undefined && object.id !== null) ? long_1.default.fromValue(object.id) : long_1.default.UZERO;
        message.createdNs = (object.createdNs !== undefined && object.createdNs !== null)
            ? long_1.default.fromValue(object.createdNs)
            : long_1.default.UZERO;
        message.installationKey = object.installationKey ?? new Uint8Array(0);
        message.data = object.data ?? new Uint8Array(0);
        message.hpkePublicKey = object.hpkePublicKey ?? new Uint8Array(0);
        message.wrapperAlgorithm = object.wrapperAlgorithm ?? 0;
        message.welcomeMetadata = object.welcomeMetadata ?? new Uint8Array(0);
        return message;
    },
};
function createBaseWelcomeMessage_WelcomePointer() {
    return {
        id: long_1.default.UZERO,
        createdNs: long_1.default.UZERO,
        installationKey: new Uint8Array(0),
        welcomePointer: new Uint8Array(0),
        hpkePublicKey: new Uint8Array(0),
        wrapperAlgorithm: 0,
    };
}
exports.WelcomeMessage_WelcomePointer = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (!message.id.equals(long_1.default.UZERO)) {
            writer.uint32(8).uint64(message.id);
        }
        if (!message.createdNs.equals(long_1.default.UZERO)) {
            writer.uint32(16).uint64(message.createdNs);
        }
        if (message.installationKey.length !== 0) {
            writer.uint32(26).bytes(message.installationKey);
        }
        if (message.welcomePointer.length !== 0) {
            writer.uint32(34).bytes(message.welcomePointer);
        }
        if (message.hpkePublicKey.length !== 0) {
            writer.uint32(42).bytes(message.hpkePublicKey);
        }
        if (message.wrapperAlgorithm !== 0) {
            writer.uint32(48).int32(message.wrapperAlgorithm);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseWelcomeMessage_WelcomePointer();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 8) {
                        break;
                    }
                    message.id = reader.uint64();
                    continue;
                case 2:
                    if (tag !== 16) {
                        break;
                    }
                    message.createdNs = reader.uint64();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.installationKey = reader.bytes();
                    continue;
                case 4:
                    if (tag !== 34) {
                        break;
                    }
                    message.welcomePointer = reader.bytes();
                    continue;
                case 5:
                    if (tag !== 42) {
                        break;
                    }
                    message.hpkePublicKey = reader.bytes();
                    continue;
                case 6:
                    if (tag !== 48) {
                        break;
                    }
                    message.wrapperAlgorithm = reader.int32();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            id: isSet(object.id) ? long_1.default.fromValue(object.id) : long_1.default.UZERO,
            createdNs: isSet(object.createdNs) ? long_1.default.fromValue(object.createdNs) : long_1.default.UZERO,
            installationKey: isSet(object.installationKey) ? bytesFromBase64(object.installationKey) : new Uint8Array(0),
            welcomePointer: isSet(object.welcomePointer) ? bytesFromBase64(object.welcomePointer) : new Uint8Array(0),
            hpkePublicKey: isSet(object.hpkePublicKey) ? bytesFromBase64(object.hpkePublicKey) : new Uint8Array(0),
            wrapperAlgorithm: isSet(object.wrapperAlgorithm)
                ? (0, welcome_pointer_pb_1.welcomePointerWrapperAlgorithmFromJSON)(object.wrapperAlgorithm)
                : 0,
        };
    },
    toJSON(message) {
        const obj = {};
        if (!message.id.equals(long_1.default.UZERO)) {
            obj.id = (message.id || long_1.default.UZERO).toString();
        }
        if (!message.createdNs.equals(long_1.default.UZERO)) {
            obj.createdNs = (message.createdNs || long_1.default.UZERO).toString();
        }
        if (message.installationKey.length !== 0) {
            obj.installationKey = base64FromBytes(message.installationKey);
        }
        if (message.welcomePointer.length !== 0) {
            obj.welcomePointer = base64FromBytes(message.welcomePointer);
        }
        if (message.hpkePublicKey.length !== 0) {
            obj.hpkePublicKey = base64FromBytes(message.hpkePublicKey);
        }
        if (message.wrapperAlgorithm !== 0) {
            obj.wrapperAlgorithm = (0, welcome_pointer_pb_1.welcomePointerWrapperAlgorithmToJSON)(message.wrapperAlgorithm);
        }
        return obj;
    },
    create(base) {
        return exports.WelcomeMessage_WelcomePointer.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseWelcomeMessage_WelcomePointer();
        message.id = (object.id !== undefined && object.id !== null) ? long_1.default.fromValue(object.id) : long_1.default.UZERO;
        message.createdNs = (object.createdNs !== undefined && object.createdNs !== null)
            ? long_1.default.fromValue(object.createdNs)
            : long_1.default.UZERO;
        message.installationKey = object.installationKey ?? new Uint8Array(0);
        message.welcomePointer = object.welcomePointer ?? new Uint8Array(0);
        message.hpkePublicKey = object.hpkePublicKey ?? new Uint8Array(0);
        message.wrapperAlgorithm = object.wrapperAlgorithm ?? 0;
        return message;
    },
};
function createBaseWelcomeMessageInput() {
    return { v1: undefined, welcomePointer: undefined };
}
exports.WelcomeMessageInput = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.v1 !== undefined) {
            exports.WelcomeMessageInput_V1.encode(message.v1, writer.uint32(10).fork()).ldelim();
        }
        if (message.welcomePointer !== undefined) {
            exports.WelcomeMessageInput_WelcomePointer.encode(message.welcomePointer, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseWelcomeMessageInput();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.v1 = exports.WelcomeMessageInput_V1.decode(reader, reader.uint32());
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.welcomePointer = exports.WelcomeMessageInput_WelcomePointer.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            v1: isSet(object.v1) ? exports.WelcomeMessageInput_V1.fromJSON(object.v1) : undefined,
            welcomePointer: isSet(object.welcomePointer)
                ? exports.WelcomeMessageInput_WelcomePointer.fromJSON(object.welcomePointer)
                : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.v1 !== undefined) {
            obj.v1 = exports.WelcomeMessageInput_V1.toJSON(message.v1);
        }
        if (message.welcomePointer !== undefined) {
            obj.welcomePointer = exports.WelcomeMessageInput_WelcomePointer.toJSON(message.welcomePointer);
        }
        return obj;
    },
    create(base) {
        return exports.WelcomeMessageInput.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseWelcomeMessageInput();
        message.v1 = (object.v1 !== undefined && object.v1 !== null)
            ? exports.WelcomeMessageInput_V1.fromPartial(object.v1)
            : undefined;
        message.welcomePointer = (object.welcomePointer !== undefined && object.welcomePointer !== null)
            ? exports.WelcomeMessageInput_WelcomePointer.fromPartial(object.welcomePointer)
            : undefined;
        return message;
    },
};
function createBaseWelcomeMessageInput_V1() {
    return {
        installationKey: new Uint8Array(0),
        data: new Uint8Array(0),
        hpkePublicKey: new Uint8Array(0),
        wrapperAlgorithm: 0,
        welcomeMetadata: new Uint8Array(0),
    };
}
exports.WelcomeMessageInput_V1 = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.installationKey.length !== 0) {
            writer.uint32(10).bytes(message.installationKey);
        }
        if (message.data.length !== 0) {
            writer.uint32(18).bytes(message.data);
        }
        if (message.hpkePublicKey.length !== 0) {
            writer.uint32(26).bytes(message.hpkePublicKey);
        }
        if (message.wrapperAlgorithm !== 0) {
            writer.uint32(32).int32(message.wrapperAlgorithm);
        }
        if (message.welcomeMetadata.length !== 0) {
            writer.uint32(58).bytes(message.welcomeMetadata);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseWelcomeMessageInput_V1();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.installationKey = reader.bytes();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.data = reader.bytes();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.hpkePublicKey = reader.bytes();
                    continue;
                case 4:
                    if (tag !== 32) {
                        break;
                    }
                    message.wrapperAlgorithm = reader.int32();
                    continue;
                case 7:
                    if (tag !== 58) {
                        break;
                    }
                    message.welcomeMetadata = reader.bytes();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            installationKey: isSet(object.installationKey) ? bytesFromBase64(object.installationKey) : new Uint8Array(0),
            data: isSet(object.data) ? bytesFromBase64(object.data) : new Uint8Array(0),
            hpkePublicKey: isSet(object.hpkePublicKey) ? bytesFromBase64(object.hpkePublicKey) : new Uint8Array(0),
            wrapperAlgorithm: isSet(object.wrapperAlgorithm) ? (0, wrapper_encryption_pb_1.welcomeWrapperAlgorithmFromJSON)(object.wrapperAlgorithm) : 0,
            welcomeMetadata: isSet(object.welcomeMetadata) ? bytesFromBase64(object.welcomeMetadata) : new Uint8Array(0),
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.installationKey.length !== 0) {
            obj.installationKey = base64FromBytes(message.installationKey);
        }
        if (message.data.length !== 0) {
            obj.data = base64FromBytes(message.data);
        }
        if (message.hpkePublicKey.length !== 0) {
            obj.hpkePublicKey = base64FromBytes(message.hpkePublicKey);
        }
        if (message.wrapperAlgorithm !== 0) {
            obj.wrapperAlgorithm = (0, wrapper_encryption_pb_1.welcomeWrapperAlgorithmToJSON)(message.wrapperAlgorithm);
        }
        if (message.welcomeMetadata.length !== 0) {
            obj.welcomeMetadata = base64FromBytes(message.welcomeMetadata);
        }
        return obj;
    },
    create(base) {
        return exports.WelcomeMessageInput_V1.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseWelcomeMessageInput_V1();
        message.installationKey = object.installationKey ?? new Uint8Array(0);
        message.data = object.data ?? new Uint8Array(0);
        message.hpkePublicKey = object.hpkePublicKey ?? new Uint8Array(0);
        message.wrapperAlgorithm = object.wrapperAlgorithm ?? 0;
        message.welcomeMetadata = object.welcomeMetadata ?? new Uint8Array(0);
        return message;
    },
};
function createBaseWelcomeMessageInput_WelcomePointer() {
    return {
        installationKey: new Uint8Array(0),
        welcomePointer: new Uint8Array(0),
        hpkePublicKey: new Uint8Array(0),
        wrapperAlgorithm: 0,
    };
}
exports.WelcomeMessageInput_WelcomePointer = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.installationKey.length !== 0) {
            writer.uint32(10).bytes(message.installationKey);
        }
        if (message.welcomePointer.length !== 0) {
            writer.uint32(18).bytes(message.welcomePointer);
        }
        if (message.hpkePublicKey.length !== 0) {
            writer.uint32(26).bytes(message.hpkePublicKey);
        }
        if (message.wrapperAlgorithm !== 0) {
            writer.uint32(32).int32(message.wrapperAlgorithm);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseWelcomeMessageInput_WelcomePointer();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.installationKey = reader.bytes();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.welcomePointer = reader.bytes();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.hpkePublicKey = reader.bytes();
                    continue;
                case 4:
                    if (tag !== 32) {
                        break;
                    }
                    message.wrapperAlgorithm = reader.int32();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            installationKey: isSet(object.installationKey) ? bytesFromBase64(object.installationKey) : new Uint8Array(0),
            welcomePointer: isSet(object.welcomePointer) ? bytesFromBase64(object.welcomePointer) : new Uint8Array(0),
            hpkePublicKey: isSet(object.hpkePublicKey) ? bytesFromBase64(object.hpkePublicKey) : new Uint8Array(0),
            wrapperAlgorithm: isSet(object.wrapperAlgorithm)
                ? (0, welcome_pointer_pb_1.welcomePointerWrapperAlgorithmFromJSON)(object.wrapperAlgorithm)
                : 0,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.installationKey.length !== 0) {
            obj.installationKey = base64FromBytes(message.installationKey);
        }
        if (message.welcomePointer.length !== 0) {
            obj.welcomePointer = base64FromBytes(message.welcomePointer);
        }
        if (message.hpkePublicKey.length !== 0) {
            obj.hpkePublicKey = base64FromBytes(message.hpkePublicKey);
        }
        if (message.wrapperAlgorithm !== 0) {
            obj.wrapperAlgorithm = (0, welcome_pointer_pb_1.welcomePointerWrapperAlgorithmToJSON)(message.wrapperAlgorithm);
        }
        return obj;
    },
    create(base) {
        return exports.WelcomeMessageInput_WelcomePointer.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseWelcomeMessageInput_WelcomePointer();
        message.installationKey = object.installationKey ?? new Uint8Array(0);
        message.welcomePointer = object.welcomePointer ?? new Uint8Array(0);
        message.hpkePublicKey = object.hpkePublicKey ?? new Uint8Array(0);
        message.wrapperAlgorithm = object.wrapperAlgorithm ?? 0;
        return message;
    },
};
function createBaseWelcomeMetadata() {
    return { messageCursor: long_1.default.UZERO };
}
exports.WelcomeMetadata = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (!message.messageCursor.equals(long_1.default.UZERO)) {
            writer.uint32(8).uint64(message.messageCursor);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseWelcomeMetadata();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 8) {
                        break;
                    }
                    message.messageCursor = reader.uint64();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { messageCursor: isSet(object.messageCursor) ? long_1.default.fromValue(object.messageCursor) : long_1.default.UZERO };
    },
    toJSON(message) {
        const obj = {};
        if (!message.messageCursor.equals(long_1.default.UZERO)) {
            obj.messageCursor = (message.messageCursor || long_1.default.UZERO).toString();
        }
        return obj;
    },
    create(base) {
        return exports.WelcomeMetadata.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseWelcomeMetadata();
        message.messageCursor = (object.messageCursor !== undefined && object.messageCursor !== null)
            ? long_1.default.fromValue(object.messageCursor)
            : long_1.default.UZERO;
        return message;
    },
};
function createBaseGroupMessage() {
    return { v1: undefined };
}
exports.GroupMessage = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.v1 !== undefined) {
            exports.GroupMessage_V1.encode(message.v1, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGroupMessage();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.v1 = exports.GroupMessage_V1.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { v1: isSet(object.v1) ? exports.GroupMessage_V1.fromJSON(object.v1) : undefined };
    },
    toJSON(message) {
        const obj = {};
        if (message.v1 !== undefined) {
            obj.v1 = exports.GroupMessage_V1.toJSON(message.v1);
        }
        return obj;
    },
    create(base) {
        return exports.GroupMessage.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGroupMessage();
        message.v1 = (object.v1 !== undefined && object.v1 !== null) ? exports.GroupMessage_V1.fromPartial(object.v1) : undefined;
        return message;
    },
};
function createBaseGroupMessage_V1() {
    return {
        id: long_1.default.UZERO,
        createdNs: long_1.default.UZERO,
        groupId: new Uint8Array(0),
        data: new Uint8Array(0),
        senderHmac: new Uint8Array(0),
        shouldPush: false,
        isCommit: false,
    };
}
exports.GroupMessage_V1 = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (!message.id.equals(long_1.default.UZERO)) {
            writer.uint32(8).uint64(message.id);
        }
        if (!message.createdNs.equals(long_1.default.UZERO)) {
            writer.uint32(16).uint64(message.createdNs);
        }
        if (message.groupId.length !== 0) {
            writer.uint32(26).bytes(message.groupId);
        }
        if (message.data.length !== 0) {
            writer.uint32(34).bytes(message.data);
        }
        if (message.senderHmac.length !== 0) {
            writer.uint32(42).bytes(message.senderHmac);
        }
        if (message.shouldPush !== false) {
            writer.uint32(48).bool(message.shouldPush);
        }
        if (message.isCommit !== false) {
            writer.uint32(56).bool(message.isCommit);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGroupMessage_V1();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 8) {
                        break;
                    }
                    message.id = reader.uint64();
                    continue;
                case 2:
                    if (tag !== 16) {
                        break;
                    }
                    message.createdNs = reader.uint64();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.groupId = reader.bytes();
                    continue;
                case 4:
                    if (tag !== 34) {
                        break;
                    }
                    message.data = reader.bytes();
                    continue;
                case 5:
                    if (tag !== 42) {
                        break;
                    }
                    message.senderHmac = reader.bytes();
                    continue;
                case 6:
                    if (tag !== 48) {
                        break;
                    }
                    message.shouldPush = reader.bool();
                    continue;
                case 7:
                    if (tag !== 56) {
                        break;
                    }
                    message.isCommit = reader.bool();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            id: isSet(object.id) ? long_1.default.fromValue(object.id) : long_1.default.UZERO,
            createdNs: isSet(object.createdNs) ? long_1.default.fromValue(object.createdNs) : long_1.default.UZERO,
            groupId: isSet(object.groupId) ? bytesFromBase64(object.groupId) : new Uint8Array(0),
            data: isSet(object.data) ? bytesFromBase64(object.data) : new Uint8Array(0),
            senderHmac: isSet(object.senderHmac) ? bytesFromBase64(object.senderHmac) : new Uint8Array(0),
            shouldPush: isSet(object.shouldPush) ? globalThis.Boolean(object.shouldPush) : false,
            isCommit: isSet(object.isCommit) ? globalThis.Boolean(object.isCommit) : false,
        };
    },
    toJSON(message) {
        const obj = {};
        if (!message.id.equals(long_1.default.UZERO)) {
            obj.id = (message.id || long_1.default.UZERO).toString();
        }
        if (!message.createdNs.equals(long_1.default.UZERO)) {
            obj.createdNs = (message.createdNs || long_1.default.UZERO).toString();
        }
        if (message.groupId.length !== 0) {
            obj.groupId = base64FromBytes(message.groupId);
        }
        if (message.data.length !== 0) {
            obj.data = base64FromBytes(message.data);
        }
        if (message.senderHmac.length !== 0) {
            obj.senderHmac = base64FromBytes(message.senderHmac);
        }
        if (message.shouldPush !== false) {
            obj.shouldPush = message.shouldPush;
        }
        if (message.isCommit !== false) {
            obj.isCommit = message.isCommit;
        }
        return obj;
    },
    create(base) {
        return exports.GroupMessage_V1.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGroupMessage_V1();
        message.id = (object.id !== undefined && object.id !== null) ? long_1.default.fromValue(object.id) : long_1.default.UZERO;
        message.createdNs = (object.createdNs !== undefined && object.createdNs !== null)
            ? long_1.default.fromValue(object.createdNs)
            : long_1.default.UZERO;
        message.groupId = object.groupId ?? new Uint8Array(0);
        message.data = object.data ?? new Uint8Array(0);
        message.senderHmac = object.senderHmac ?? new Uint8Array(0);
        message.shouldPush = object.shouldPush ?? false;
        message.isCommit = object.isCommit ?? false;
        return message;
    },
};
function createBaseGroupMessageInput() {
    return { v1: undefined };
}
exports.GroupMessageInput = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.v1 !== undefined) {
            exports.GroupMessageInput_V1.encode(message.v1, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGroupMessageInput();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.v1 = exports.GroupMessageInput_V1.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { v1: isSet(object.v1) ? exports.GroupMessageInput_V1.fromJSON(object.v1) : undefined };
    },
    toJSON(message) {
        const obj = {};
        if (message.v1 !== undefined) {
            obj.v1 = exports.GroupMessageInput_V1.toJSON(message.v1);
        }
        return obj;
    },
    create(base) {
        return exports.GroupMessageInput.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGroupMessageInput();
        message.v1 = (object.v1 !== undefined && object.v1 !== null)
            ? exports.GroupMessageInput_V1.fromPartial(object.v1)
            : undefined;
        return message;
    },
};
function createBaseGroupMessageInput_V1() {
    return { data: new Uint8Array(0), senderHmac: new Uint8Array(0), shouldPush: false };
}
exports.GroupMessageInput_V1 = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.data.length !== 0) {
            writer.uint32(10).bytes(message.data);
        }
        if (message.senderHmac.length !== 0) {
            writer.uint32(18).bytes(message.senderHmac);
        }
        if (message.shouldPush !== false) {
            writer.uint32(24).bool(message.shouldPush);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGroupMessageInput_V1();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.data = reader.bytes();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.senderHmac = reader.bytes();
                    continue;
                case 3:
                    if (tag !== 24) {
                        break;
                    }
                    message.shouldPush = reader.bool();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            data: isSet(object.data) ? bytesFromBase64(object.data) : new Uint8Array(0),
            senderHmac: isSet(object.senderHmac) ? bytesFromBase64(object.senderHmac) : new Uint8Array(0),
            shouldPush: isSet(object.shouldPush) ? globalThis.Boolean(object.shouldPush) : false,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.data.length !== 0) {
            obj.data = base64FromBytes(message.data);
        }
        if (message.senderHmac.length !== 0) {
            obj.senderHmac = base64FromBytes(message.senderHmac);
        }
        if (message.shouldPush !== false) {
            obj.shouldPush = message.shouldPush;
        }
        return obj;
    },
    create(base) {
        return exports.GroupMessageInput_V1.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGroupMessageInput_V1();
        message.data = object.data ?? new Uint8Array(0);
        message.senderHmac = object.senderHmac ?? new Uint8Array(0);
        message.shouldPush = object.shouldPush ?? false;
        return message;
    },
};
function createBaseSendGroupMessagesRequest() {
    return { messages: [] };
}
exports.SendGroupMessagesRequest = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        for (const v of message.messages) {
            exports.GroupMessageInput.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSendGroupMessagesRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.messages.push(exports.GroupMessageInput.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            messages: globalThis.Array.isArray(object?.messages)
                ? object.messages.map((e) => exports.GroupMessageInput.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.messages?.length) {
            obj.messages = message.messages.map((e) => exports.GroupMessageInput.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return exports.SendGroupMessagesRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseSendGroupMessagesRequest();
        message.messages = object.messages?.map((e) => exports.GroupMessageInput.fromPartial(e)) || [];
        return message;
    },
};
function createBaseSendWelcomeMessagesRequest() {
    return { messages: [] };
}
exports.SendWelcomeMessagesRequest = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        for (const v of message.messages) {
            exports.WelcomeMessageInput.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSendWelcomeMessagesRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.messages.push(exports.WelcomeMessageInput.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            messages: globalThis.Array.isArray(object?.messages)
                ? object.messages.map((e) => exports.WelcomeMessageInput.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.messages?.length) {
            obj.messages = message.messages.map((e) => exports.WelcomeMessageInput.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return exports.SendWelcomeMessagesRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseSendWelcomeMessagesRequest();
        message.messages = object.messages?.map((e) => exports.WelcomeMessageInput.fromPartial(e)) || [];
        return message;
    },
};
function createBaseKeyPackageUpload() {
    return { keyPackageTlsSerialized: new Uint8Array(0) };
}
exports.KeyPackageUpload = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.keyPackageTlsSerialized.length !== 0) {
            writer.uint32(10).bytes(message.keyPackageTlsSerialized);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseKeyPackageUpload();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.keyPackageTlsSerialized = reader.bytes();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            keyPackageTlsSerialized: isSet(object.keyPackageTlsSerialized)
                ? bytesFromBase64(object.keyPackageTlsSerialized)
                : new Uint8Array(0),
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.keyPackageTlsSerialized.length !== 0) {
            obj.keyPackageTlsSerialized = base64FromBytes(message.keyPackageTlsSerialized);
        }
        return obj;
    },
    create(base) {
        return exports.KeyPackageUpload.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseKeyPackageUpload();
        message.keyPackageTlsSerialized = object.keyPackageTlsSerialized ?? new Uint8Array(0);
        return message;
    },
};
function createBaseRegisterInstallationRequest() {
    return { keyPackage: undefined, isInboxIdCredential: false };
}
exports.RegisterInstallationRequest = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.keyPackage !== undefined) {
            exports.KeyPackageUpload.encode(message.keyPackage, writer.uint32(10).fork()).ldelim();
        }
        if (message.isInboxIdCredential !== false) {
            writer.uint32(16).bool(message.isInboxIdCredential);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseRegisterInstallationRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.keyPackage = exports.KeyPackageUpload.decode(reader, reader.uint32());
                    continue;
                case 2:
                    if (tag !== 16) {
                        break;
                    }
                    message.isInboxIdCredential = reader.bool();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            keyPackage: isSet(object.keyPackage) ? exports.KeyPackageUpload.fromJSON(object.keyPackage) : undefined,
            isInboxIdCredential: isSet(object.isInboxIdCredential) ? globalThis.Boolean(object.isInboxIdCredential) : false,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.keyPackage !== undefined) {
            obj.keyPackage = exports.KeyPackageUpload.toJSON(message.keyPackage);
        }
        if (message.isInboxIdCredential !== false) {
            obj.isInboxIdCredential = message.isInboxIdCredential;
        }
        return obj;
    },
    create(base) {
        return exports.RegisterInstallationRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseRegisterInstallationRequest();
        message.keyPackage = (object.keyPackage !== undefined && object.keyPackage !== null)
            ? exports.KeyPackageUpload.fromPartial(object.keyPackage)
            : undefined;
        message.isInboxIdCredential = object.isInboxIdCredential ?? false;
        return message;
    },
};
function createBaseRegisterInstallationResponse() {
    return { installationKey: new Uint8Array(0) };
}
exports.RegisterInstallationResponse = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.installationKey.length !== 0) {
            writer.uint32(10).bytes(message.installationKey);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseRegisterInstallationResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.installationKey = reader.bytes();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            installationKey: isSet(object.installationKey) ? bytesFromBase64(object.installationKey) : new Uint8Array(0),
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.installationKey.length !== 0) {
            obj.installationKey = base64FromBytes(message.installationKey);
        }
        return obj;
    },
    create(base) {
        return exports.RegisterInstallationResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseRegisterInstallationResponse();
        message.installationKey = object.installationKey ?? new Uint8Array(0);
        return message;
    },
};
function createBaseUploadKeyPackageRequest() {
    return { keyPackage: undefined, isInboxIdCredential: false };
}
exports.UploadKeyPackageRequest = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.keyPackage !== undefined) {
            exports.KeyPackageUpload.encode(message.keyPackage, writer.uint32(10).fork()).ldelim();
        }
        if (message.isInboxIdCredential !== false) {
            writer.uint32(16).bool(message.isInboxIdCredential);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseUploadKeyPackageRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.keyPackage = exports.KeyPackageUpload.decode(reader, reader.uint32());
                    continue;
                case 2:
                    if (tag !== 16) {
                        break;
                    }
                    message.isInboxIdCredential = reader.bool();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            keyPackage: isSet(object.keyPackage) ? exports.KeyPackageUpload.fromJSON(object.keyPackage) : undefined,
            isInboxIdCredential: isSet(object.isInboxIdCredential) ? globalThis.Boolean(object.isInboxIdCredential) : false,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.keyPackage !== undefined) {
            obj.keyPackage = exports.KeyPackageUpload.toJSON(message.keyPackage);
        }
        if (message.isInboxIdCredential !== false) {
            obj.isInboxIdCredential = message.isInboxIdCredential;
        }
        return obj;
    },
    create(base) {
        return exports.UploadKeyPackageRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseUploadKeyPackageRequest();
        message.keyPackage = (object.keyPackage !== undefined && object.keyPackage !== null)
            ? exports.KeyPackageUpload.fromPartial(object.keyPackage)
            : undefined;
        message.isInboxIdCredential = object.isInboxIdCredential ?? false;
        return message;
    },
};
function createBaseFetchKeyPackagesRequest() {
    return { installationKeys: [] };
}
exports.FetchKeyPackagesRequest = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        for (const v of message.installationKeys) {
            writer.uint32(10).bytes(v);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseFetchKeyPackagesRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.installationKeys.push(reader.bytes());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            installationKeys: globalThis.Array.isArray(object?.installationKeys)
                ? object.installationKeys.map((e) => bytesFromBase64(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.installationKeys?.length) {
            obj.installationKeys = message.installationKeys.map((e) => base64FromBytes(e));
        }
        return obj;
    },
    create(base) {
        return exports.FetchKeyPackagesRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseFetchKeyPackagesRequest();
        message.installationKeys = object.installationKeys?.map((e) => e) || [];
        return message;
    },
};
function createBaseFetchKeyPackagesResponse() {
    return { keyPackages: [] };
}
exports.FetchKeyPackagesResponse = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        for (const v of message.keyPackages) {
            exports.FetchKeyPackagesResponse_KeyPackage.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseFetchKeyPackagesResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.keyPackages.push(exports.FetchKeyPackagesResponse_KeyPackage.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            keyPackages: globalThis.Array.isArray(object?.keyPackages)
                ? object.keyPackages.map((e) => exports.FetchKeyPackagesResponse_KeyPackage.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.keyPackages?.length) {
            obj.keyPackages = message.keyPackages.map((e) => exports.FetchKeyPackagesResponse_KeyPackage.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return exports.FetchKeyPackagesResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseFetchKeyPackagesResponse();
        message.keyPackages = object.keyPackages?.map((e) => exports.FetchKeyPackagesResponse_KeyPackage.fromPartial(e)) || [];
        return message;
    },
};
function createBaseFetchKeyPackagesResponse_KeyPackage() {
    return { keyPackageTlsSerialized: new Uint8Array(0) };
}
exports.FetchKeyPackagesResponse_KeyPackage = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.keyPackageTlsSerialized.length !== 0) {
            writer.uint32(10).bytes(message.keyPackageTlsSerialized);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseFetchKeyPackagesResponse_KeyPackage();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.keyPackageTlsSerialized = reader.bytes();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            keyPackageTlsSerialized: isSet(object.keyPackageTlsSerialized)
                ? bytesFromBase64(object.keyPackageTlsSerialized)
                : new Uint8Array(0),
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.keyPackageTlsSerialized.length !== 0) {
            obj.keyPackageTlsSerialized = base64FromBytes(message.keyPackageTlsSerialized);
        }
        return obj;
    },
    create(base) {
        return exports.FetchKeyPackagesResponse_KeyPackage.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseFetchKeyPackagesResponse_KeyPackage();
        message.keyPackageTlsSerialized = object.keyPackageTlsSerialized ?? new Uint8Array(0);
        return message;
    },
};
function createBaseRevokeInstallationRequest() {
    return { installationKey: new Uint8Array(0), walletSignature: undefined };
}
exports.RevokeInstallationRequest = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.installationKey.length !== 0) {
            writer.uint32(10).bytes(message.installationKey);
        }
        if (message.walletSignature !== undefined) {
            signature_pb_2.Signature.encode(message.walletSignature, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseRevokeInstallationRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.installationKey = reader.bytes();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.walletSignature = signature_pb_2.Signature.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            installationKey: isSet(object.installationKey) ? bytesFromBase64(object.installationKey) : new Uint8Array(0),
            walletSignature: isSet(object.walletSignature) ? signature_pb_2.Signature.fromJSON(object.walletSignature) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.installationKey.length !== 0) {
            obj.installationKey = base64FromBytes(message.installationKey);
        }
        if (message.walletSignature !== undefined) {
            obj.walletSignature = signature_pb_2.Signature.toJSON(message.walletSignature);
        }
        return obj;
    },
    create(base) {
        return exports.RevokeInstallationRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseRevokeInstallationRequest();
        message.installationKey = object.installationKey ?? new Uint8Array(0);
        message.walletSignature = (object.walletSignature !== undefined && object.walletSignature !== null)
            ? signature_pb_2.Signature.fromPartial(object.walletSignature)
            : undefined;
        return message;
    },
};
function createBaseGetIdentityUpdatesRequest() {
    return { accountAddresses: [], startTimeNs: long_1.default.UZERO };
}
exports.GetIdentityUpdatesRequest = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        for (const v of message.accountAddresses) {
            writer.uint32(10).string(v);
        }
        if (!message.startTimeNs.equals(long_1.default.UZERO)) {
            writer.uint32(16).uint64(message.startTimeNs);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetIdentityUpdatesRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.accountAddresses.push(reader.string());
                    continue;
                case 2:
                    if (tag !== 16) {
                        break;
                    }
                    message.startTimeNs = reader.uint64();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            accountAddresses: globalThis.Array.isArray(object?.accountAddresses)
                ? object.accountAddresses.map((e) => globalThis.String(e))
                : [],
            startTimeNs: isSet(object.startTimeNs) ? long_1.default.fromValue(object.startTimeNs) : long_1.default.UZERO,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.accountAddresses?.length) {
            obj.accountAddresses = message.accountAddresses;
        }
        if (!message.startTimeNs.equals(long_1.default.UZERO)) {
            obj.startTimeNs = (message.startTimeNs || long_1.default.UZERO).toString();
        }
        return obj;
    },
    create(base) {
        return exports.GetIdentityUpdatesRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGetIdentityUpdatesRequest();
        message.accountAddresses = object.accountAddresses?.map((e) => e) || [];
        message.startTimeNs = (object.startTimeNs !== undefined && object.startTimeNs !== null)
            ? long_1.default.fromValue(object.startTimeNs)
            : long_1.default.UZERO;
        return message;
    },
};
function createBaseGetIdentityUpdatesResponse() {
    return { updates: [] };
}
exports.GetIdentityUpdatesResponse = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        for (const v of message.updates) {
            exports.GetIdentityUpdatesResponse_WalletUpdates.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetIdentityUpdatesResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.updates.push(exports.GetIdentityUpdatesResponse_WalletUpdates.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            updates: globalThis.Array.isArray(object?.updates)
                ? object.updates.map((e) => exports.GetIdentityUpdatesResponse_WalletUpdates.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.updates?.length) {
            obj.updates = message.updates.map((e) => exports.GetIdentityUpdatesResponse_WalletUpdates.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return exports.GetIdentityUpdatesResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGetIdentityUpdatesResponse();
        message.updates = object.updates?.map((e) => exports.GetIdentityUpdatesResponse_WalletUpdates.fromPartial(e)) || [];
        return message;
    },
};
function createBaseGetIdentityUpdatesResponse_NewInstallationUpdate() {
    return { installationKey: new Uint8Array(0), credentialIdentity: new Uint8Array(0) };
}
exports.GetIdentityUpdatesResponse_NewInstallationUpdate = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.installationKey.length !== 0) {
            writer.uint32(10).bytes(message.installationKey);
        }
        if (message.credentialIdentity.length !== 0) {
            writer.uint32(18).bytes(message.credentialIdentity);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetIdentityUpdatesResponse_NewInstallationUpdate();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.installationKey = reader.bytes();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.credentialIdentity = reader.bytes();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            installationKey: isSet(object.installationKey) ? bytesFromBase64(object.installationKey) : new Uint8Array(0),
            credentialIdentity: isSet(object.credentialIdentity)
                ? bytesFromBase64(object.credentialIdentity)
                : new Uint8Array(0),
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.installationKey.length !== 0) {
            obj.installationKey = base64FromBytes(message.installationKey);
        }
        if (message.credentialIdentity.length !== 0) {
            obj.credentialIdentity = base64FromBytes(message.credentialIdentity);
        }
        return obj;
    },
    create(base) {
        return exports.GetIdentityUpdatesResponse_NewInstallationUpdate.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGetIdentityUpdatesResponse_NewInstallationUpdate();
        message.installationKey = object.installationKey ?? new Uint8Array(0);
        message.credentialIdentity = object.credentialIdentity ?? new Uint8Array(0);
        return message;
    },
};
function createBaseGetIdentityUpdatesResponse_RevokedInstallationUpdate() {
    return { installationKey: new Uint8Array(0) };
}
exports.GetIdentityUpdatesResponse_RevokedInstallationUpdate = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.installationKey.length !== 0) {
            writer.uint32(10).bytes(message.installationKey);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetIdentityUpdatesResponse_RevokedInstallationUpdate();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.installationKey = reader.bytes();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            installationKey: isSet(object.installationKey) ? bytesFromBase64(object.installationKey) : new Uint8Array(0),
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.installationKey.length !== 0) {
            obj.installationKey = base64FromBytes(message.installationKey);
        }
        return obj;
    },
    create(base) {
        return exports.GetIdentityUpdatesResponse_RevokedInstallationUpdate.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGetIdentityUpdatesResponse_RevokedInstallationUpdate();
        message.installationKey = object.installationKey ?? new Uint8Array(0);
        return message;
    },
};
function createBaseGetIdentityUpdatesResponse_Update() {
    return { timestampNs: long_1.default.UZERO, newInstallation: undefined, revokedInstallation: undefined };
}
exports.GetIdentityUpdatesResponse_Update = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (!message.timestampNs.equals(long_1.default.UZERO)) {
            writer.uint32(8).uint64(message.timestampNs);
        }
        if (message.newInstallation !== undefined) {
            exports.GetIdentityUpdatesResponse_NewInstallationUpdate.encode(message.newInstallation, writer.uint32(18).fork())
                .ldelim();
        }
        if (message.revokedInstallation !== undefined) {
            exports.GetIdentityUpdatesResponse_RevokedInstallationUpdate.encode(message.revokedInstallation, writer.uint32(26).fork())
                .ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetIdentityUpdatesResponse_Update();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 8) {
                        break;
                    }
                    message.timestampNs = reader.uint64();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.newInstallation = exports.GetIdentityUpdatesResponse_NewInstallationUpdate.decode(reader, reader.uint32());
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.revokedInstallation = exports.GetIdentityUpdatesResponse_RevokedInstallationUpdate.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            timestampNs: isSet(object.timestampNs) ? long_1.default.fromValue(object.timestampNs) : long_1.default.UZERO,
            newInstallation: isSet(object.newInstallation)
                ? exports.GetIdentityUpdatesResponse_NewInstallationUpdate.fromJSON(object.newInstallation)
                : undefined,
            revokedInstallation: isSet(object.revokedInstallation)
                ? exports.GetIdentityUpdatesResponse_RevokedInstallationUpdate.fromJSON(object.revokedInstallation)
                : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (!message.timestampNs.equals(long_1.default.UZERO)) {
            obj.timestampNs = (message.timestampNs || long_1.default.UZERO).toString();
        }
        if (message.newInstallation !== undefined) {
            obj.newInstallation = exports.GetIdentityUpdatesResponse_NewInstallationUpdate.toJSON(message.newInstallation);
        }
        if (message.revokedInstallation !== undefined) {
            obj.revokedInstallation = exports.GetIdentityUpdatesResponse_RevokedInstallationUpdate.toJSON(message.revokedInstallation);
        }
        return obj;
    },
    create(base) {
        return exports.GetIdentityUpdatesResponse_Update.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGetIdentityUpdatesResponse_Update();
        message.timestampNs = (object.timestampNs !== undefined && object.timestampNs !== null)
            ? long_1.default.fromValue(object.timestampNs)
            : long_1.default.UZERO;
        message.newInstallation = (object.newInstallation !== undefined && object.newInstallation !== null)
            ? exports.GetIdentityUpdatesResponse_NewInstallationUpdate.fromPartial(object.newInstallation)
            : undefined;
        message.revokedInstallation = (object.revokedInstallation !== undefined && object.revokedInstallation !== null)
            ? exports.GetIdentityUpdatesResponse_RevokedInstallationUpdate.fromPartial(object.revokedInstallation)
            : undefined;
        return message;
    },
};
function createBaseGetIdentityUpdatesResponse_WalletUpdates() {
    return { updates: [] };
}
exports.GetIdentityUpdatesResponse_WalletUpdates = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        for (const v of message.updates) {
            exports.GetIdentityUpdatesResponse_Update.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetIdentityUpdatesResponse_WalletUpdates();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.updates.push(exports.GetIdentityUpdatesResponse_Update.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            updates: globalThis.Array.isArray(object?.updates)
                ? object.updates.map((e) => exports.GetIdentityUpdatesResponse_Update.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.updates?.length) {
            obj.updates = message.updates.map((e) => exports.GetIdentityUpdatesResponse_Update.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return exports.GetIdentityUpdatesResponse_WalletUpdates.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGetIdentityUpdatesResponse_WalletUpdates();
        message.updates = object.updates?.map((e) => exports.GetIdentityUpdatesResponse_Update.fromPartial(e)) || [];
        return message;
    },
};
function createBasePagingInfo() {
    return { direction: 0, limit: 0, idCursor: long_1.default.UZERO };
}
exports.PagingInfo = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.direction !== 0) {
            writer.uint32(8).int32(message.direction);
        }
        if (message.limit !== 0) {
            writer.uint32(16).uint32(message.limit);
        }
        if (!message.idCursor.equals(long_1.default.UZERO)) {
            writer.uint32(24).uint64(message.idCursor);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBasePagingInfo();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 8) {
                        break;
                    }
                    message.direction = reader.int32();
                    continue;
                case 2:
                    if (tag !== 16) {
                        break;
                    }
                    message.limit = reader.uint32();
                    continue;
                case 3:
                    if (tag !== 24) {
                        break;
                    }
                    message.idCursor = reader.uint64();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            direction: isSet(object.direction) ? sortDirectionFromJSON(object.direction) : 0,
            limit: isSet(object.limit) ? globalThis.Number(object.limit) : 0,
            idCursor: isSet(object.idCursor) ? long_1.default.fromValue(object.idCursor) : long_1.default.UZERO,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.direction !== 0) {
            obj.direction = sortDirectionToJSON(message.direction);
        }
        if (message.limit !== 0) {
            obj.limit = Math.round(message.limit);
        }
        if (!message.idCursor.equals(long_1.default.UZERO)) {
            obj.idCursor = (message.idCursor || long_1.default.UZERO).toString();
        }
        return obj;
    },
    create(base) {
        return exports.PagingInfo.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBasePagingInfo();
        message.direction = object.direction ?? 0;
        message.limit = object.limit ?? 0;
        message.idCursor = (object.idCursor !== undefined && object.idCursor !== null)
            ? long_1.default.fromValue(object.idCursor)
            : long_1.default.UZERO;
        return message;
    },
};
function createBaseQueryGroupMessagesRequest() {
    return { groupId: new Uint8Array(0), pagingInfo: undefined };
}
exports.QueryGroupMessagesRequest = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.groupId.length !== 0) {
            writer.uint32(10).bytes(message.groupId);
        }
        if (message.pagingInfo !== undefined) {
            exports.PagingInfo.encode(message.pagingInfo, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryGroupMessagesRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.groupId = reader.bytes();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.pagingInfo = exports.PagingInfo.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            groupId: isSet(object.groupId) ? bytesFromBase64(object.groupId) : new Uint8Array(0),
            pagingInfo: isSet(object.pagingInfo) ? exports.PagingInfo.fromJSON(object.pagingInfo) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.groupId.length !== 0) {
            obj.groupId = base64FromBytes(message.groupId);
        }
        if (message.pagingInfo !== undefined) {
            obj.pagingInfo = exports.PagingInfo.toJSON(message.pagingInfo);
        }
        return obj;
    },
    create(base) {
        return exports.QueryGroupMessagesRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseQueryGroupMessagesRequest();
        message.groupId = object.groupId ?? new Uint8Array(0);
        message.pagingInfo = (object.pagingInfo !== undefined && object.pagingInfo !== null)
            ? exports.PagingInfo.fromPartial(object.pagingInfo)
            : undefined;
        return message;
    },
};
function createBaseQueryGroupMessagesResponse() {
    return { messages: [], pagingInfo: undefined };
}
exports.QueryGroupMessagesResponse = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        for (const v of message.messages) {
            exports.GroupMessage.encode(v, writer.uint32(10).fork()).ldelim();
        }
        if (message.pagingInfo !== undefined) {
            exports.PagingInfo.encode(message.pagingInfo, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryGroupMessagesResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.messages.push(exports.GroupMessage.decode(reader, reader.uint32()));
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.pagingInfo = exports.PagingInfo.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            messages: globalThis.Array.isArray(object?.messages)
                ? object.messages.map((e) => exports.GroupMessage.fromJSON(e))
                : [],
            pagingInfo: isSet(object.pagingInfo) ? exports.PagingInfo.fromJSON(object.pagingInfo) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.messages?.length) {
            obj.messages = message.messages.map((e) => exports.GroupMessage.toJSON(e));
        }
        if (message.pagingInfo !== undefined) {
            obj.pagingInfo = exports.PagingInfo.toJSON(message.pagingInfo);
        }
        return obj;
    },
    create(base) {
        return exports.QueryGroupMessagesResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseQueryGroupMessagesResponse();
        message.messages = object.messages?.map((e) => exports.GroupMessage.fromPartial(e)) || [];
        message.pagingInfo = (object.pagingInfo !== undefined && object.pagingInfo !== null)
            ? exports.PagingInfo.fromPartial(object.pagingInfo)
            : undefined;
        return message;
    },
};
function createBaseQueryWelcomeMessagesRequest() {
    return { installationKey: new Uint8Array(0), pagingInfo: undefined };
}
exports.QueryWelcomeMessagesRequest = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.installationKey.length !== 0) {
            writer.uint32(10).bytes(message.installationKey);
        }
        if (message.pagingInfo !== undefined) {
            exports.PagingInfo.encode(message.pagingInfo, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryWelcomeMessagesRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.installationKey = reader.bytes();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.pagingInfo = exports.PagingInfo.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            installationKey: isSet(object.installationKey) ? bytesFromBase64(object.installationKey) : new Uint8Array(0),
            pagingInfo: isSet(object.pagingInfo) ? exports.PagingInfo.fromJSON(object.pagingInfo) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.installationKey.length !== 0) {
            obj.installationKey = base64FromBytes(message.installationKey);
        }
        if (message.pagingInfo !== undefined) {
            obj.pagingInfo = exports.PagingInfo.toJSON(message.pagingInfo);
        }
        return obj;
    },
    create(base) {
        return exports.QueryWelcomeMessagesRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseQueryWelcomeMessagesRequest();
        message.installationKey = object.installationKey ?? new Uint8Array(0);
        message.pagingInfo = (object.pagingInfo !== undefined && object.pagingInfo !== null)
            ? exports.PagingInfo.fromPartial(object.pagingInfo)
            : undefined;
        return message;
    },
};
function createBaseQueryWelcomeMessagesResponse() {
    return { messages: [], pagingInfo: undefined };
}
exports.QueryWelcomeMessagesResponse = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        for (const v of message.messages) {
            exports.WelcomeMessage.encode(v, writer.uint32(10).fork()).ldelim();
        }
        if (message.pagingInfo !== undefined) {
            exports.PagingInfo.encode(message.pagingInfo, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryWelcomeMessagesResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.messages.push(exports.WelcomeMessage.decode(reader, reader.uint32()));
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.pagingInfo = exports.PagingInfo.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            messages: globalThis.Array.isArray(object?.messages)
                ? object.messages.map((e) => exports.WelcomeMessage.fromJSON(e))
                : [],
            pagingInfo: isSet(object.pagingInfo) ? exports.PagingInfo.fromJSON(object.pagingInfo) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.messages?.length) {
            obj.messages = message.messages.map((e) => exports.WelcomeMessage.toJSON(e));
        }
        if (message.pagingInfo !== undefined) {
            obj.pagingInfo = exports.PagingInfo.toJSON(message.pagingInfo);
        }
        return obj;
    },
    create(base) {
        return exports.QueryWelcomeMessagesResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseQueryWelcomeMessagesResponse();
        message.messages = object.messages?.map((e) => exports.WelcomeMessage.fromPartial(e)) || [];
        message.pagingInfo = (object.pagingInfo !== undefined && object.pagingInfo !== null)
            ? exports.PagingInfo.fromPartial(object.pagingInfo)
            : undefined;
        return message;
    },
};
function createBaseSubscribeGroupMessagesRequest() {
    return { filters: [] };
}
exports.SubscribeGroupMessagesRequest = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        for (const v of message.filters) {
            exports.SubscribeGroupMessagesRequest_Filter.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSubscribeGroupMessagesRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.filters.push(exports.SubscribeGroupMessagesRequest_Filter.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            filters: globalThis.Array.isArray(object?.filters)
                ? object.filters.map((e) => exports.SubscribeGroupMessagesRequest_Filter.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.filters?.length) {
            obj.filters = message.filters.map((e) => exports.SubscribeGroupMessagesRequest_Filter.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return exports.SubscribeGroupMessagesRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseSubscribeGroupMessagesRequest();
        message.filters = object.filters?.map((e) => exports.SubscribeGroupMessagesRequest_Filter.fromPartial(e)) || [];
        return message;
    },
};
function createBaseSubscribeGroupMessagesRequest_Filter() {
    return { groupId: new Uint8Array(0), idCursor: long_1.default.UZERO };
}
exports.SubscribeGroupMessagesRequest_Filter = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.groupId.length !== 0) {
            writer.uint32(10).bytes(message.groupId);
        }
        if (!message.idCursor.equals(long_1.default.UZERO)) {
            writer.uint32(16).uint64(message.idCursor);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSubscribeGroupMessagesRequest_Filter();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.groupId = reader.bytes();
                    continue;
                case 2:
                    if (tag !== 16) {
                        break;
                    }
                    message.idCursor = reader.uint64();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            groupId: isSet(object.groupId) ? bytesFromBase64(object.groupId) : new Uint8Array(0),
            idCursor: isSet(object.idCursor) ? long_1.default.fromValue(object.idCursor) : long_1.default.UZERO,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.groupId.length !== 0) {
            obj.groupId = base64FromBytes(message.groupId);
        }
        if (!message.idCursor.equals(long_1.default.UZERO)) {
            obj.idCursor = (message.idCursor || long_1.default.UZERO).toString();
        }
        return obj;
    },
    create(base) {
        return exports.SubscribeGroupMessagesRequest_Filter.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseSubscribeGroupMessagesRequest_Filter();
        message.groupId = object.groupId ?? new Uint8Array(0);
        message.idCursor = (object.idCursor !== undefined && object.idCursor !== null)
            ? long_1.default.fromValue(object.idCursor)
            : long_1.default.UZERO;
        return message;
    },
};
function createBaseSubscribeWelcomeMessagesRequest() {
    return { filters: [] };
}
exports.SubscribeWelcomeMessagesRequest = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        for (const v of message.filters) {
            exports.SubscribeWelcomeMessagesRequest_Filter.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSubscribeWelcomeMessagesRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.filters.push(exports.SubscribeWelcomeMessagesRequest_Filter.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            filters: globalThis.Array.isArray(object?.filters)
                ? object.filters.map((e) => exports.SubscribeWelcomeMessagesRequest_Filter.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.filters?.length) {
            obj.filters = message.filters.map((e) => exports.SubscribeWelcomeMessagesRequest_Filter.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return exports.SubscribeWelcomeMessagesRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseSubscribeWelcomeMessagesRequest();
        message.filters = object.filters?.map((e) => exports.SubscribeWelcomeMessagesRequest_Filter.fromPartial(e)) || [];
        return message;
    },
};
function createBaseSubscribeWelcomeMessagesRequest_Filter() {
    return { installationKey: new Uint8Array(0), idCursor: long_1.default.UZERO };
}
exports.SubscribeWelcomeMessagesRequest_Filter = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.installationKey.length !== 0) {
            writer.uint32(10).bytes(message.installationKey);
        }
        if (!message.idCursor.equals(long_1.default.UZERO)) {
            writer.uint32(16).uint64(message.idCursor);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSubscribeWelcomeMessagesRequest_Filter();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.installationKey = reader.bytes();
                    continue;
                case 2:
                    if (tag !== 16) {
                        break;
                    }
                    message.idCursor = reader.uint64();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            installationKey: isSet(object.installationKey) ? bytesFromBase64(object.installationKey) : new Uint8Array(0),
            idCursor: isSet(object.idCursor) ? long_1.default.fromValue(object.idCursor) : long_1.default.UZERO,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.installationKey.length !== 0) {
            obj.installationKey = base64FromBytes(message.installationKey);
        }
        if (!message.idCursor.equals(long_1.default.UZERO)) {
            obj.idCursor = (message.idCursor || long_1.default.UZERO).toString();
        }
        return obj;
    },
    create(base) {
        return exports.SubscribeWelcomeMessagesRequest_Filter.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseSubscribeWelcomeMessagesRequest_Filter();
        message.installationKey = object.installationKey ?? new Uint8Array(0);
        message.idCursor = (object.idCursor !== undefined && object.idCursor !== null)
            ? long_1.default.fromValue(object.idCursor)
            : long_1.default.UZERO;
        return message;
    },
};
function createBaseBatchPublishCommitLogRequest() {
    return { requests: [] };
}
exports.BatchPublishCommitLogRequest = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        for (const v of message.requests) {
            exports.PublishCommitLogRequest.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseBatchPublishCommitLogRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.requests.push(exports.PublishCommitLogRequest.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            requests: globalThis.Array.isArray(object?.requests)
                ? object.requests.map((e) => exports.PublishCommitLogRequest.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.requests?.length) {
            obj.requests = message.requests.map((e) => exports.PublishCommitLogRequest.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return exports.BatchPublishCommitLogRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseBatchPublishCommitLogRequest();
        message.requests = object.requests?.map((e) => exports.PublishCommitLogRequest.fromPartial(e)) || [];
        return message;
    },
};
function createBasePublishCommitLogRequest() {
    return { groupId: new Uint8Array(0), serializedCommitLogEntry: new Uint8Array(0), signature: undefined };
}
exports.PublishCommitLogRequest = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.groupId.length !== 0) {
            writer.uint32(10).bytes(message.groupId);
        }
        if (message.serializedCommitLogEntry.length !== 0) {
            writer.uint32(18).bytes(message.serializedCommitLogEntry);
        }
        if (message.signature !== undefined) {
            signature_pb_1.RecoverableEd25519Signature.encode(message.signature, writer.uint32(26).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBasePublishCommitLogRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.groupId = reader.bytes();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.serializedCommitLogEntry = reader.bytes();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.signature = signature_pb_1.RecoverableEd25519Signature.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            groupId: isSet(object.groupId) ? bytesFromBase64(object.groupId) : new Uint8Array(0),
            serializedCommitLogEntry: isSet(object.serializedCommitLogEntry)
                ? bytesFromBase64(object.serializedCommitLogEntry)
                : new Uint8Array(0),
            signature: isSet(object.signature) ? signature_pb_1.RecoverableEd25519Signature.fromJSON(object.signature) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.groupId.length !== 0) {
            obj.groupId = base64FromBytes(message.groupId);
        }
        if (message.serializedCommitLogEntry.length !== 0) {
            obj.serializedCommitLogEntry = base64FromBytes(message.serializedCommitLogEntry);
        }
        if (message.signature !== undefined) {
            obj.signature = signature_pb_1.RecoverableEd25519Signature.toJSON(message.signature);
        }
        return obj;
    },
    create(base) {
        return exports.PublishCommitLogRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBasePublishCommitLogRequest();
        message.groupId = object.groupId ?? new Uint8Array(0);
        message.serializedCommitLogEntry = object.serializedCommitLogEntry ?? new Uint8Array(0);
        message.signature = (object.signature !== undefined && object.signature !== null)
            ? signature_pb_1.RecoverableEd25519Signature.fromPartial(object.signature)
            : undefined;
        return message;
    },
};
function createBaseQueryCommitLogRequest() {
    return { groupId: new Uint8Array(0), pagingInfo: undefined };
}
exports.QueryCommitLogRequest = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.groupId.length !== 0) {
            writer.uint32(10).bytes(message.groupId);
        }
        if (message.pagingInfo !== undefined) {
            exports.PagingInfo.encode(message.pagingInfo, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryCommitLogRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.groupId = reader.bytes();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.pagingInfo = exports.PagingInfo.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            groupId: isSet(object.groupId) ? bytesFromBase64(object.groupId) : new Uint8Array(0),
            pagingInfo: isSet(object.pagingInfo) ? exports.PagingInfo.fromJSON(object.pagingInfo) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.groupId.length !== 0) {
            obj.groupId = base64FromBytes(message.groupId);
        }
        if (message.pagingInfo !== undefined) {
            obj.pagingInfo = exports.PagingInfo.toJSON(message.pagingInfo);
        }
        return obj;
    },
    create(base) {
        return exports.QueryCommitLogRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseQueryCommitLogRequest();
        message.groupId = object.groupId ?? new Uint8Array(0);
        message.pagingInfo = (object.pagingInfo !== undefined && object.pagingInfo !== null)
            ? exports.PagingInfo.fromPartial(object.pagingInfo)
            : undefined;
        return message;
    },
};
function createBaseQueryCommitLogResponse() {
    return { groupId: new Uint8Array(0), commitLogEntries: [], pagingInfo: undefined };
}
exports.QueryCommitLogResponse = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.groupId.length !== 0) {
            writer.uint32(10).bytes(message.groupId);
        }
        for (const v of message.commitLogEntries) {
            commit_log_pb_1.CommitLogEntry.encode(v, writer.uint32(18).fork()).ldelim();
        }
        if (message.pagingInfo !== undefined) {
            exports.PagingInfo.encode(message.pagingInfo, writer.uint32(26).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryCommitLogResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.groupId = reader.bytes();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.commitLogEntries.push(commit_log_pb_1.CommitLogEntry.decode(reader, reader.uint32()));
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.pagingInfo = exports.PagingInfo.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            groupId: isSet(object.groupId) ? bytesFromBase64(object.groupId) : new Uint8Array(0),
            commitLogEntries: globalThis.Array.isArray(object?.commitLogEntries)
                ? object.commitLogEntries.map((e) => commit_log_pb_1.CommitLogEntry.fromJSON(e))
                : [],
            pagingInfo: isSet(object.pagingInfo) ? exports.PagingInfo.fromJSON(object.pagingInfo) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.groupId.length !== 0) {
            obj.groupId = base64FromBytes(message.groupId);
        }
        if (message.commitLogEntries?.length) {
            obj.commitLogEntries = message.commitLogEntries.map((e) => commit_log_pb_1.CommitLogEntry.toJSON(e));
        }
        if (message.pagingInfo !== undefined) {
            obj.pagingInfo = exports.PagingInfo.toJSON(message.pagingInfo);
        }
        return obj;
    },
    create(base) {
        return exports.QueryCommitLogResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseQueryCommitLogResponse();
        message.groupId = object.groupId ?? new Uint8Array(0);
        message.commitLogEntries = object.commitLogEntries?.map((e) => commit_log_pb_1.CommitLogEntry.fromPartial(e)) || [];
        message.pagingInfo = (object.pagingInfo !== undefined && object.pagingInfo !== null)
            ? exports.PagingInfo.fromPartial(object.pagingInfo)
            : undefined;
        return message;
    },
};
function createBaseBatchQueryCommitLogRequest() {
    return { requests: [] };
}
exports.BatchQueryCommitLogRequest = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        for (const v of message.requests) {
            exports.QueryCommitLogRequest.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseBatchQueryCommitLogRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.requests.push(exports.QueryCommitLogRequest.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            requests: globalThis.Array.isArray(object?.requests)
                ? object.requests.map((e) => exports.QueryCommitLogRequest.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.requests?.length) {
            obj.requests = message.requests.map((e) => exports.QueryCommitLogRequest.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return exports.BatchQueryCommitLogRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseBatchQueryCommitLogRequest();
        message.requests = object.requests?.map((e) => exports.QueryCommitLogRequest.fromPartial(e)) || [];
        return message;
    },
};
function createBaseBatchQueryCommitLogResponse() {
    return { responses: [] };
}
exports.BatchQueryCommitLogResponse = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        for (const v of message.responses) {
            exports.QueryCommitLogResponse.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseBatchQueryCommitLogResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.responses.push(exports.QueryCommitLogResponse.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            responses: globalThis.Array.isArray(object?.responses)
                ? object.responses.map((e) => exports.QueryCommitLogResponse.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.responses?.length) {
            obj.responses = message.responses.map((e) => exports.QueryCommitLogResponse.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return exports.BatchQueryCommitLogResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseBatchQueryCommitLogResponse();
        message.responses = object.responses?.map((e) => exports.QueryCommitLogResponse.fromPartial(e)) || [];
        return message;
    },
};
function createBaseGetNewestGroupMessageRequest() {
    return { groupIds: [], includeContent: false };
}
exports.GetNewestGroupMessageRequest = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        for (const v of message.groupIds) {
            writer.uint32(10).bytes(v);
        }
        if (message.includeContent !== false) {
            writer.uint32(16).bool(message.includeContent);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetNewestGroupMessageRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.groupIds.push(reader.bytes());
                    continue;
                case 2:
                    if (tag !== 16) {
                        break;
                    }
                    message.includeContent = reader.bool();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            groupIds: globalThis.Array.isArray(object?.groupIds) ? object.groupIds.map((e) => bytesFromBase64(e)) : [],
            includeContent: isSet(object.includeContent) ? globalThis.Boolean(object.includeContent) : false,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.groupIds?.length) {
            obj.groupIds = message.groupIds.map((e) => base64FromBytes(e));
        }
        if (message.includeContent !== false) {
            obj.includeContent = message.includeContent;
        }
        return obj;
    },
    create(base) {
        return exports.GetNewestGroupMessageRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGetNewestGroupMessageRequest();
        message.groupIds = object.groupIds?.map((e) => e) || [];
        message.includeContent = object.includeContent ?? false;
        return message;
    },
};
function createBaseGetNewestGroupMessageResponse() {
    return { responses: [] };
}
exports.GetNewestGroupMessageResponse = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        for (const v of message.responses) {
            exports.GetNewestGroupMessageResponse_Response.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetNewestGroupMessageResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.responses.push(exports.GetNewestGroupMessageResponse_Response.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            responses: globalThis.Array.isArray(object?.responses)
                ? object.responses.map((e) => exports.GetNewestGroupMessageResponse_Response.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.responses?.length) {
            obj.responses = message.responses.map((e) => exports.GetNewestGroupMessageResponse_Response.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return exports.GetNewestGroupMessageResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGetNewestGroupMessageResponse();
        message.responses = object.responses?.map((e) => exports.GetNewestGroupMessageResponse_Response.fromPartial(e)) || [];
        return message;
    },
};
function createBaseGetNewestGroupMessageResponse_Response() {
    return { groupMessage: undefined };
}
exports.GetNewestGroupMessageResponse_Response = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.groupMessage !== undefined) {
            exports.GroupMessage.encode(message.groupMessage, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetNewestGroupMessageResponse_Response();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.groupMessage = exports.GroupMessage.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { groupMessage: isSet(object.groupMessage) ? exports.GroupMessage.fromJSON(object.groupMessage) : undefined };
    },
    toJSON(message) {
        const obj = {};
        if (message.groupMessage !== undefined) {
            obj.groupMessage = exports.GroupMessage.toJSON(message.groupMessage);
        }
        return obj;
    },
    create(base) {
        return exports.GetNewestGroupMessageResponse_Response.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGetNewestGroupMessageResponse_Response();
        message.groupMessage = (object.groupMessage !== undefined && object.groupMessage !== null)
            ? exports.GroupMessage.fromPartial(object.groupMessage)
            : undefined;
        return message;
    },
};
exports.MlsApiServiceName = "xmtp.mls.api.v1.MlsApi";
class MlsApiClientImpl {
    rpc;
    service;
    constructor(rpc, opts) {
        this.service = opts?.service || exports.MlsApiServiceName;
        this.rpc = rpc;
        this.SendGroupMessages = this.SendGroupMessages.bind(this);
        this.SendWelcomeMessages = this.SendWelcomeMessages.bind(this);
        this.RegisterInstallation = this.RegisterInstallation.bind(this);
        this.UploadKeyPackage = this.UploadKeyPackage.bind(this);
        this.FetchKeyPackages = this.FetchKeyPackages.bind(this);
        this.RevokeInstallation = this.RevokeInstallation.bind(this);
        this.GetIdentityUpdates = this.GetIdentityUpdates.bind(this);
        this.QueryGroupMessages = this.QueryGroupMessages.bind(this);
        this.QueryWelcomeMessages = this.QueryWelcomeMessages.bind(this);
        this.SubscribeGroupMessages = this.SubscribeGroupMessages.bind(this);
        this.SubscribeWelcomeMessages = this.SubscribeWelcomeMessages.bind(this);
        this.BatchPublishCommitLog = this.BatchPublishCommitLog.bind(this);
        this.BatchQueryCommitLog = this.BatchQueryCommitLog.bind(this);
        this.GetNewestGroupMessage = this.GetNewestGroupMessage.bind(this);
    }
    SendGroupMessages(request) {
        const data = exports.SendGroupMessagesRequest.encode(request).finish();
        const promise = this.rpc.request(this.service, "SendGroupMessages", data);
        return promise.then((data) => empty_pb_1.Empty.decode(minimal_1.default.Reader.create(data)));
    }
    SendWelcomeMessages(request) {
        const data = exports.SendWelcomeMessagesRequest.encode(request).finish();
        const promise = this.rpc.request(this.service, "SendWelcomeMessages", data);
        return promise.then((data) => empty_pb_1.Empty.decode(minimal_1.default.Reader.create(data)));
    }
    RegisterInstallation(request) {
        const data = exports.RegisterInstallationRequest.encode(request).finish();
        const promise = this.rpc.request(this.service, "RegisterInstallation", data);
        return promise.then((data) => exports.RegisterInstallationResponse.decode(minimal_1.default.Reader.create(data)));
    }
    UploadKeyPackage(request) {
        const data = exports.UploadKeyPackageRequest.encode(request).finish();
        const promise = this.rpc.request(this.service, "UploadKeyPackage", data);
        return promise.then((data) => empty_pb_1.Empty.decode(minimal_1.default.Reader.create(data)));
    }
    FetchKeyPackages(request) {
        const data = exports.FetchKeyPackagesRequest.encode(request).finish();
        const promise = this.rpc.request(this.service, "FetchKeyPackages", data);
        return promise.then((data) => exports.FetchKeyPackagesResponse.decode(minimal_1.default.Reader.create(data)));
    }
    RevokeInstallation(request) {
        const data = exports.RevokeInstallationRequest.encode(request).finish();
        const promise = this.rpc.request(this.service, "RevokeInstallation", data);
        return promise.then((data) => empty_pb_1.Empty.decode(minimal_1.default.Reader.create(data)));
    }
    GetIdentityUpdates(request) {
        const data = exports.GetIdentityUpdatesRequest.encode(request).finish();
        const promise = this.rpc.request(this.service, "GetIdentityUpdates", data);
        return promise.then((data) => exports.GetIdentityUpdatesResponse.decode(minimal_1.default.Reader.create(data)));
    }
    QueryGroupMessages(request) {
        const data = exports.QueryGroupMessagesRequest.encode(request).finish();
        const promise = this.rpc.request(this.service, "QueryGroupMessages", data);
        return promise.then((data) => exports.QueryGroupMessagesResponse.decode(minimal_1.default.Reader.create(data)));
    }
    QueryWelcomeMessages(request) {
        const data = exports.QueryWelcomeMessagesRequest.encode(request).finish();
        const promise = this.rpc.request(this.service, "QueryWelcomeMessages", data);
        return promise.then((data) => exports.QueryWelcomeMessagesResponse.decode(minimal_1.default.Reader.create(data)));
    }
    SubscribeGroupMessages(request) {
        const data = exports.SubscribeGroupMessagesRequest.encode(request).finish();
        const result = this.rpc.serverStreamingRequest(this.service, "SubscribeGroupMessages", data);
        return result.pipe((0, operators_1.map)((data) => exports.GroupMessage.decode(minimal_1.default.Reader.create(data))));
    }
    SubscribeWelcomeMessages(request) {
        const data = exports.SubscribeWelcomeMessagesRequest.encode(request).finish();
        const result = this.rpc.serverStreamingRequest(this.service, "SubscribeWelcomeMessages", data);
        return result.pipe((0, operators_1.map)((data) => exports.WelcomeMessage.decode(minimal_1.default.Reader.create(data))));
    }
    BatchPublishCommitLog(request) {
        const data = exports.BatchPublishCommitLogRequest.encode(request).finish();
        const promise = this.rpc.request(this.service, "BatchPublishCommitLog", data);
        return promise.then((data) => empty_pb_1.Empty.decode(minimal_1.default.Reader.create(data)));
    }
    BatchQueryCommitLog(request) {
        const data = exports.BatchQueryCommitLogRequest.encode(request).finish();
        const promise = this.rpc.request(this.service, "BatchQueryCommitLog", data);
        return promise.then((data) => exports.BatchQueryCommitLogResponse.decode(minimal_1.default.Reader.create(data)));
    }
    GetNewestGroupMessage(request) {
        const data = exports.GetNewestGroupMessageRequest.encode(request).finish();
        const promise = this.rpc.request(this.service, "GetNewestGroupMessage", data);
        return promise.then((data) => exports.GetNewestGroupMessageResponse.decode(minimal_1.default.Reader.create(data)));
    }
}
exports.MlsApiClientImpl = MlsApiClientImpl;
function bytesFromBase64(b64) {
    if (globalThis.Buffer) {
        return Uint8Array.from(globalThis.Buffer.from(b64, "base64"));
    }
    else {
        const bin = globalThis.atob(b64);
        const arr = new Uint8Array(bin.length);
        for (let i = 0; i < bin.length; ++i) {
            arr[i] = bin.charCodeAt(i);
        }
        return arr;
    }
}
function base64FromBytes(arr) {
    if (globalThis.Buffer) {
        return globalThis.Buffer.from(arr).toString("base64");
    }
    else {
        const bin = [];
        arr.forEach((byte) => {
            bin.push(globalThis.String.fromCharCode(byte));
        });
        return globalThis.btoa(bin.join(""));
    }
}
if (minimal_1.default.util.Long !== long_1.default) {
    minimal_1.default.util.Long = long_1.default;
    minimal_1.default.configure();
}
function isSet(value) {
    return value !== null && value !== undefined;
}
