// @xmtp/proto/ts/dist/cjs/mls/database/intents.pb.js
"use strict";
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v1.181.2
//   protoc               v3.21.12
// source: mls/database/intents.proto
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CommitPendingProposalsData_V1 = exports.CommitPendingProposalsData = exports.ProposeGroupContextExtensionData_V1 = exports.ProposeGroupContextExtensionData = exports.ProposeMemberUpdateData_V1 = exports.ProposeMemberUpdateData = exports.PostCommitAction_SendWelcomes = exports.PostCommitAction_Installation = exports.PostCommitAction = exports.UpdatePermissionData_V1 = exports.UpdatePermissionData = exports.UpdateAdminListsData_V1 = exports.UpdateAdminListsData = exports.UpdateMetadataData_V1 = exports.UpdateMetadataData = exports.ReaddInstallationsData_V1 = exports.ReaddInstallationsData = exports.UpdateGroupMembershipData_V1_MembershipUpdatesEntry = exports.UpdateGroupMembershipData_V1 = exports.UpdateGroupMembershipData = exports.RemoveMembersData_V1 = exports.RemoveMembersData = exports.AddMembersData_V1 = exports.AddMembersData = exports.AddressesOrInstallationIds = exports.InstallationIds = exports.AccountAddresses = exports.SendMessageData_V1 = exports.SendMessageData = exports.PermissionPolicyOption = exports.PermissionUpdateType = exports.AdminListUpdateType = exports.protobufPackage = void 0;
exports.adminListUpdateTypeFromJSON = adminListUpdateTypeFromJSON;
exports.adminListUpdateTypeToJSON = adminListUpdateTypeToJSON;
exports.permissionUpdateTypeFromJSON = permissionUpdateTypeFromJSON;
exports.permissionUpdateTypeToJSON = permissionUpdateTypeToJSON;
exports.permissionPolicyOptionFromJSON = permissionPolicyOptionFromJSON;
exports.permissionPolicyOptionToJSON = permissionPolicyOptionToJSON;
/* eslint-disable */
const long_1 = __importDefault(require("long"));
const minimal_1 = __importDefault(require("protobufjs/minimal"));
const welcome_pointer_pb_1 = require("../message_contents/welcome_pointer.pb");
const wrapper_encryption_pb_1 = require("../message_contents/wrapper_encryption.pb");
exports.protobufPackage = "xmtp.mls.database";
/** Type of update to admin lists */
var AdminListUpdateType;
(function (AdminListUpdateType) {
    AdminListUpdateType[AdminListUpdateType["ADMIN_LIST_UPDATE_TYPE_UNSPECIFIED"] = 0] = "ADMIN_LIST_UPDATE_TYPE_UNSPECIFIED";
    AdminListUpdateType[AdminListUpdateType["ADMIN_LIST_UPDATE_TYPE_ADD_ADMIN"] = 1] = "ADMIN_LIST_UPDATE_TYPE_ADD_ADMIN";
    AdminListUpdateType[AdminListUpdateType["ADMIN_LIST_UPDATE_TYPE_REMOVE_ADMIN"] = 2] = "ADMIN_LIST_UPDATE_TYPE_REMOVE_ADMIN";
    AdminListUpdateType[AdminListUpdateType["ADMIN_LIST_UPDATE_TYPE_ADD_SUPER_ADMIN"] = 3] = "ADMIN_LIST_UPDATE_TYPE_ADD_SUPER_ADMIN";
    AdminListUpdateType[AdminListUpdateType["ADMIN_LIST_UPDATE_TYPE_REMOVE_SUPER_ADMIN"] = 4] = "ADMIN_LIST_UPDATE_TYPE_REMOVE_SUPER_ADMIN";
    AdminListUpdateType[AdminListUpdateType["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(AdminListUpdateType || (exports.AdminListUpdateType = AdminListUpdateType = {}));
function adminListUpdateTypeFromJSON(object) {
    switch (object) {
        case 0:
        case "ADMIN_LIST_UPDATE_TYPE_UNSPECIFIED":
            return AdminListUpdateType.ADMIN_LIST_UPDATE_TYPE_UNSPECIFIED;
        case 1:
        case "ADMIN_LIST_UPDATE_TYPE_ADD_ADMIN":
            return AdminListUpdateType.ADMIN_LIST_UPDATE_TYPE_ADD_ADMIN;
        case 2:
        case "ADMIN_LIST_UPDATE_TYPE_REMOVE_ADMIN":
            return AdminListUpdateType.ADMIN_LIST_UPDATE_TYPE_REMOVE_ADMIN;
        case 3:
        case "ADMIN_LIST_UPDATE_TYPE_ADD_SUPER_ADMIN":
            return AdminListUpdateType.ADMIN_LIST_UPDATE_TYPE_ADD_SUPER_ADMIN;
        case 4:
        case "ADMIN_LIST_UPDATE_TYPE_REMOVE_SUPER_ADMIN":
            return AdminListUpdateType.ADMIN_LIST_UPDATE_TYPE_REMOVE_SUPER_ADMIN;
        case -1:
        case "UNRECOGNIZED":
        default:
            return AdminListUpdateType.UNRECOGNIZED;
    }
}
function adminListUpdateTypeToJSON(object) {
    switch (object) {
        case AdminListUpdateType.ADMIN_LIST_UPDATE_TYPE_UNSPECIFIED:
            return "ADMIN_LIST_UPDATE_TYPE_UNSPECIFIED";
        case AdminListUpdateType.ADMIN_LIST_UPDATE_TYPE_ADD_ADMIN:
            return "ADMIN_LIST_UPDATE_TYPE_ADD_ADMIN";
        case AdminListUpdateType.ADMIN_LIST_UPDATE_TYPE_REMOVE_ADMIN:
            return "ADMIN_LIST_UPDATE_TYPE_REMOVE_ADMIN";
        case AdminListUpdateType.ADMIN_LIST_UPDATE_TYPE_ADD_SUPER_ADMIN:
            return "ADMIN_LIST_UPDATE_TYPE_ADD_SUPER_ADMIN";
        case AdminListUpdateType.ADMIN_LIST_UPDATE_TYPE_REMOVE_SUPER_ADMIN:
            return "ADMIN_LIST_UPDATE_TYPE_REMOVE_SUPER_ADMIN";
        case AdminListUpdateType.UNRECOGNIZED:
        default:
            return "UNRECOGNIZED";
    }
}
/** Type of Permission to Update */
var PermissionUpdateType;
(function (PermissionUpdateType) {
    PermissionUpdateType[PermissionUpdateType["PERMISSION_UPDATE_TYPE_UNSPECIFIED"] = 0] = "PERMISSION_UPDATE_TYPE_UNSPECIFIED";
    PermissionUpdateType[PermissionUpdateType["PERMISSION_UPDATE_TYPE_ADD_MEMBER"] = 1] = "PERMISSION_UPDATE_TYPE_ADD_MEMBER";
    PermissionUpdateType[PermissionUpdateType["PERMISSION_UPDATE_TYPE_REMOVE_MEMBER"] = 2] = "PERMISSION_UPDATE_TYPE_REMOVE_MEMBER";
    PermissionUpdateType[PermissionUpdateType["PERMISSION_UPDATE_TYPE_ADD_ADMIN"] = 3] = "PERMISSION_UPDATE_TYPE_ADD_ADMIN";
    PermissionUpdateType[PermissionUpdateType["PERMISSION_UPDATE_TYPE_REMOVE_ADMIN"] = 4] = "PERMISSION_UPDATE_TYPE_REMOVE_ADMIN";
    PermissionUpdateType[PermissionUpdateType["PERMISSION_UPDATE_TYPE_UPDATE_METADATA"] = 5] = "PERMISSION_UPDATE_TYPE_UPDATE_METADATA";
    PermissionUpdateType[PermissionUpdateType["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(PermissionUpdateType || (exports.PermissionUpdateType = PermissionUpdateType = {}));
function permissionUpdateTypeFromJSON(object) {
    switch (object) {
        case 0:
        case "PERMISSION_UPDATE_TYPE_UNSPECIFIED":
            return PermissionUpdateType.PERMISSION_UPDATE_TYPE_UNSPECIFIED;
        case 1:
        case "PERMISSION_UPDATE_TYPE_ADD_MEMBER":
            return PermissionUpdateType.PERMISSION_UPDATE_TYPE_ADD_MEMBER;
        case 2:
        case "PERMISSION_UPDATE_TYPE_REMOVE_MEMBER":
            return PermissionUpdateType.PERMISSION_UPDATE_TYPE_REMOVE_MEMBER;
        case 3:
        case "PERMISSION_UPDATE_TYPE_ADD_ADMIN":
            return PermissionUpdateType.PERMISSION_UPDATE_TYPE_ADD_ADMIN;
        case 4:
        case "PERMISSION_UPDATE_TYPE_REMOVE_ADMIN":
            return PermissionUpdateType.PERMISSION_UPDATE_TYPE_REMOVE_ADMIN;
        case 5:
        case "PERMISSION_UPDATE_TYPE_UPDATE_METADATA":
            return PermissionUpdateType.PERMISSION_UPDATE_TYPE_UPDATE_METADATA;
        case -1:
        case "UNRECOGNIZED":
        default:
            return PermissionUpdateType.UNRECOGNIZED;
    }
}
function permissionUpdateTypeToJSON(object) {
    switch (object) {
        case PermissionUpdateType.PERMISSION_UPDATE_TYPE_UNSPECIFIED:
            return "PERMISSION_UPDATE_TYPE_UNSPECIFIED";
        case PermissionUpdateType.PERMISSION_UPDATE_TYPE_ADD_MEMBER:
            return "PERMISSION_UPDATE_TYPE_ADD_MEMBER";
        case PermissionUpdateType.PERMISSION_UPDATE_TYPE_REMOVE_MEMBER:
            return "PERMISSION_UPDATE_TYPE_REMOVE_MEMBER";
        case PermissionUpdateType.PERMISSION_UPDATE_TYPE_ADD_ADMIN:
            return "PERMISSION_UPDATE_TYPE_ADD_ADMIN";
        case PermissionUpdateType.PERMISSION_UPDATE_TYPE_REMOVE_ADMIN:
            return "PERMISSION_UPDATE_TYPE_REMOVE_ADMIN";
        case PermissionUpdateType.PERMISSION_UPDATE_TYPE_UPDATE_METADATA:
            return "PERMISSION_UPDATE_TYPE_UPDATE_METADATA";
        case PermissionUpdateType.UNRECOGNIZED:
        default:
            return "UNRECOGNIZED";
    }
}
/** Permission Policy */
var PermissionPolicyOption;
(function (PermissionPolicyOption) {
    PermissionPolicyOption[PermissionPolicyOption["PERMISSION_POLICY_OPTION_UNSPECIFIED"] = 0] = "PERMISSION_POLICY_OPTION_UNSPECIFIED";
    PermissionPolicyOption[PermissionPolicyOption["PERMISSION_POLICY_OPTION_ALLOW"] = 1] = "PERMISSION_POLICY_OPTION_ALLOW";
    PermissionPolicyOption[PermissionPolicyOption["PERMISSION_POLICY_OPTION_DENY"] = 2] = "PERMISSION_POLICY_OPTION_DENY";
    PermissionPolicyOption[PermissionPolicyOption["PERMISSION_POLICY_OPTION_ADMIN_ONLY"] = 3] = "PERMISSION_POLICY_OPTION_ADMIN_ONLY";
    PermissionPolicyOption[PermissionPolicyOption["PERMISSION_POLICY_OPTION_SUPER_ADMIN_ONLY"] = 4] = "PERMISSION_POLICY_OPTION_SUPER_ADMIN_ONLY";
    PermissionPolicyOption[PermissionPolicyOption["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(PermissionPolicyOption || (exports.PermissionPolicyOption = PermissionPolicyOption = {}));
function permissionPolicyOptionFromJSON(object) {
    switch (object) {
        case 0:
        case "PERMISSION_POLICY_OPTION_UNSPECIFIED":
            return PermissionPolicyOption.PERMISSION_POLICY_OPTION_UNSPECIFIED;
        case 1:
        case "PERMISSION_POLICY_OPTION_ALLOW":
            return PermissionPolicyOption.PERMISSION_POLICY_OPTION_ALLOW;
        case 2:
        case "PERMISSION_POLICY_OPTION_DENY":
            return PermissionPolicyOption.PERMISSION_POLICY_OPTION_DENY;
        case 3:
        case "PERMISSION_POLICY_OPTION_ADMIN_ONLY":
            return PermissionPolicyOption.PERMISSION_POLICY_OPTION_ADMIN_ONLY;
        case 4:
        case "PERMISSION_POLICY_OPTION_SUPER_ADMIN_ONLY":
            return PermissionPolicyOption.PERMISSION_POLICY_OPTION_SUPER_ADMIN_ONLY;
        case -1:
        case "UNRECOGNIZED":
        default:
            return PermissionPolicyOption.UNRECOGNIZED;
    }
}
function permissionPolicyOptionToJSON(object) {
    switch (object) {
        case PermissionPolicyOption.PERMISSION_POLICY_OPTION_UNSPECIFIED:
            return "PERMISSION_POLICY_OPTION_UNSPECIFIED";
        case PermissionPolicyOption.PERMISSION_POLICY_OPTION_ALLOW:
            return "PERMISSION_POLICY_OPTION_ALLOW";
        case PermissionPolicyOption.PERMISSION_POLICY_OPTION_DENY:
            return "PERMISSION_POLICY_OPTION_DENY";
        case PermissionPolicyOption.PERMISSION_POLICY_OPTION_ADMIN_ONLY:
            return "PERMISSION_POLICY_OPTION_ADMIN_ONLY";
        case PermissionPolicyOption.PERMISSION_POLICY_OPTION_SUPER_ADMIN_ONLY:
            return "PERMISSION_POLICY_OPTION_SUPER_ADMIN_ONLY";
        case PermissionPolicyOption.UNRECOGNIZED:
        default:
            return "UNRECOGNIZED";
    }
}
function createBaseSendMessageData() {
    return { v1: undefined };
}
exports.SendMessageData = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.v1 !== undefined) {
            exports.SendMessageData_V1.encode(message.v1, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSendMessageData();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.v1 = exports.SendMessageData_V1.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { v1: isSet(object.v1) ? exports.SendMessageData_V1.fromJSON(object.v1) : undefined };
    },
    toJSON(message) {
        const obj = {};
        if (message.v1 !== undefined) {
            obj.v1 = exports.SendMessageData_V1.toJSON(message.v1);
        }
        return obj;
    },
    create(base) {
        return exports.SendMessageData.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseSendMessageData();
        message.v1 = (object.v1 !== undefined && object.v1 !== null)
            ? exports.SendMessageData_V1.fromPartial(object.v1)
            : undefined;
        return message;
    },
};
function createBaseSendMessageData_V1() {
    return { payloadBytes: new Uint8Array(0) };
}
exports.SendMessageData_V1 = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.payloadBytes.length !== 0) {
            writer.uint32(10).bytes(message.payloadBytes);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSendMessageData_V1();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.payloadBytes = reader.bytes();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { payloadBytes: isSet(object.payloadBytes) ? bytesFromBase64(object.payloadBytes) : new Uint8Array(0) };
    },
    toJSON(message) {
        const obj = {};
        if (message.payloadBytes.length !== 0) {
            obj.payloadBytes = base64FromBytes(message.payloadBytes);
        }
        return obj;
    },
    create(base) {
        return exports.SendMessageData_V1.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseSendMessageData_V1();
        message.payloadBytes = object.payloadBytes ?? new Uint8Array(0);
        return message;
    },
};
function createBaseAccountAddresses() {
    return { accountAddresses: [] };
}
exports.AccountAddresses = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        for (const v of message.accountAddresses) {
            writer.uint32(10).string(v);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseAccountAddresses();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.accountAddresses.push(reader.string());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            accountAddresses: globalThis.Array.isArray(object?.accountAddresses)
                ? object.accountAddresses.map((e) => globalThis.String(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.accountAddresses?.length) {
            obj.accountAddresses = message.accountAddresses;
        }
        return obj;
    },
    create(base) {
        return exports.AccountAddresses.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseAccountAddresses();
        message.accountAddresses = object.accountAddresses?.map((e) => e) || [];
        return message;
    },
};
function createBaseInstallationIds() {
    return { installationIds: [] };
}
exports.InstallationIds = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        for (const v of message.installationIds) {
            writer.uint32(10).bytes(v);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseInstallationIds();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.installationIds.push(reader.bytes());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            installationIds: globalThis.Array.isArray(object?.installationIds)
                ? object.installationIds.map((e) => bytesFromBase64(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.installationIds?.length) {
            obj.installationIds = message.installationIds.map((e) => base64FromBytes(e));
        }
        return obj;
    },
    create(base) {
        return exports.InstallationIds.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseInstallationIds();
        message.installationIds = object.installationIds?.map((e) => e) || [];
        return message;
    },
};
function createBaseAddressesOrInstallationIds() {
    return { accountAddresses: undefined, installationIds: undefined };
}
exports.AddressesOrInstallationIds = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.accountAddresses !== undefined) {
            exports.AccountAddresses.encode(message.accountAddresses, writer.uint32(10).fork()).ldelim();
        }
        if (message.installationIds !== undefined) {
            exports.InstallationIds.encode(message.installationIds, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseAddressesOrInstallationIds();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.accountAddresses = exports.AccountAddresses.decode(reader, reader.uint32());
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.installationIds = exports.InstallationIds.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            accountAddresses: isSet(object.accountAddresses) ? exports.AccountAddresses.fromJSON(object.accountAddresses) : undefined,
            installationIds: isSet(object.installationIds) ? exports.InstallationIds.fromJSON(object.installationIds) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.accountAddresses !== undefined) {
            obj.accountAddresses = exports.AccountAddresses.toJSON(message.accountAddresses);
        }
        if (message.installationIds !== undefined) {
            obj.installationIds = exports.InstallationIds.toJSON(message.installationIds);
        }
        return obj;
    },
    create(base) {
        return exports.AddressesOrInstallationIds.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseAddressesOrInstallationIds();
        message.accountAddresses = (object.accountAddresses !== undefined && object.accountAddresses !== null)
            ? exports.AccountAddresses.fromPartial(object.accountAddresses)
            : undefined;
        message.installationIds = (object.installationIds !== undefined && object.installationIds !== null)
            ? exports.InstallationIds.fromPartial(object.installationIds)
            : undefined;
        return message;
    },
};
function createBaseAddMembersData() {
    return { v1: undefined };
}
exports.AddMembersData = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.v1 !== undefined) {
            exports.AddMembersData_V1.encode(message.v1, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseAddMembersData();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.v1 = exports.AddMembersData_V1.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { v1: isSet(object.v1) ? exports.AddMembersData_V1.fromJSON(object.v1) : undefined };
    },
    toJSON(message) {
        const obj = {};
        if (message.v1 !== undefined) {
            obj.v1 = exports.AddMembersData_V1.toJSON(message.v1);
        }
        return obj;
    },
    create(base) {
        return exports.AddMembersData.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseAddMembersData();
        message.v1 = (object.v1 !== undefined && object.v1 !== null) ? exports.AddMembersData_V1.fromPartial(object.v1) : undefined;
        return message;
    },
};
function createBaseAddMembersData_V1() {
    return { addressesOrInstallationIds: undefined };
}
exports.AddMembersData_V1 = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.addressesOrInstallationIds !== undefined) {
            exports.AddressesOrInstallationIds.encode(message.addressesOrInstallationIds, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseAddMembersData_V1();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.addressesOrInstallationIds = exports.AddressesOrInstallationIds.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            addressesOrInstallationIds: isSet(object.addressesOrInstallationIds)
                ? exports.AddressesOrInstallationIds.fromJSON(object.addressesOrInstallationIds)
                : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.addressesOrInstallationIds !== undefined) {
            obj.addressesOrInstallationIds = exports.AddressesOrInstallationIds.toJSON(message.addressesOrInstallationIds);
        }
        return obj;
    },
    create(base) {
        return exports.AddMembersData_V1.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseAddMembersData_V1();
        message.addressesOrInstallationIds =
            (object.addressesOrInstallationIds !== undefined && object.addressesOrInstallationIds !== null)
                ? exports.AddressesOrInstallationIds.fromPartial(object.addressesOrInstallationIds)
                : undefined;
        return message;
    },
};
function createBaseRemoveMembersData() {
    return { v1: undefined };
}
exports.RemoveMembersData = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.v1 !== undefined) {
            exports.RemoveMembersData_V1.encode(message.v1, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseRemoveMembersData();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.v1 = exports.RemoveMembersData_V1.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { v1: isSet(object.v1) ? exports.RemoveMembersData_V1.fromJSON(object.v1) : undefined };
    },
    toJSON(message) {
        const obj = {};
        if (message.v1 !== undefined) {
            obj.v1 = exports.RemoveMembersData_V1.toJSON(message.v1);
        }
        return obj;
    },
    create(base) {
        return exports.RemoveMembersData.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseRemoveMembersData();
        message.v1 = (object.v1 !== undefined && object.v1 !== null)
            ? exports.RemoveMembersData_V1.fromPartial(object.v1)
            : undefined;
        return message;
    },
};
function createBaseRemoveMembersData_V1() {
    return { addressesOrInstallationIds: undefined };
}
exports.RemoveMembersData_V1 = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.addressesOrInstallationIds !== undefined) {
            exports.AddressesOrInstallationIds.encode(message.addressesOrInstallationIds, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseRemoveMembersData_V1();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.addressesOrInstallationIds = exports.AddressesOrInstallationIds.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            addressesOrInstallationIds: isSet(object.addressesOrInstallationIds)
                ? exports.AddressesOrInstallationIds.fromJSON(object.addressesOrInstallationIds)
                : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.addressesOrInstallationIds !== undefined) {
            obj.addressesOrInstallationIds = exports.AddressesOrInstallationIds.toJSON(message.addressesOrInstallationIds);
        }
        return obj;
    },
    create(base) {
        return exports.RemoveMembersData_V1.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseRemoveMembersData_V1();
        message.addressesOrInstallationIds =
            (object.addressesOrInstallationIds !== undefined && object.addressesOrInstallationIds !== null)
                ? exports.AddressesOrInstallationIds.fromPartial(object.addressesOrInstallationIds)
                : undefined;
        return message;
    },
};
function createBaseUpdateGroupMembershipData() {
    return { v1: undefined };
}
exports.UpdateGroupMembershipData = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.v1 !== undefined) {
            exports.UpdateGroupMembershipData_V1.encode(message.v1, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseUpdateGroupMembershipData();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.v1 = exports.UpdateGroupMembershipData_V1.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { v1: isSet(object.v1) ? exports.UpdateGroupMembershipData_V1.fromJSON(object.v1) : undefined };
    },
    toJSON(message) {
        const obj = {};
        if (message.v1 !== undefined) {
            obj.v1 = exports.UpdateGroupMembershipData_V1.toJSON(message.v1);
        }
        return obj;
    },
    create(base) {
        return exports.UpdateGroupMembershipData.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseUpdateGroupMembershipData();
        message.v1 = (object.v1 !== undefined && object.v1 !== null)
            ? exports.UpdateGroupMembershipData_V1.fromPartial(object.v1)
            : undefined;
        return message;
    },
};
function createBaseUpdateGroupMembershipData_V1() {
    return { membershipUpdates: {}, removedMembers: [], failedInstallations: [] };
}
exports.UpdateGroupMembershipData_V1 = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        Object.entries(message.membershipUpdates).forEach(([key, value]) => {
            exports.UpdateGroupMembershipData_V1_MembershipUpdatesEntry.encode({ key: key, value }, writer.uint32(10).fork())
                .ldelim();
        });
        for (const v of message.removedMembers) {
            writer.uint32(18).string(v);
        }
        for (const v of message.failedInstallations) {
            writer.uint32(26).bytes(v);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseUpdateGroupMembershipData_V1();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    const entry1 = exports.UpdateGroupMembershipData_V1_MembershipUpdatesEntry.decode(reader, reader.uint32());
                    if (entry1.value !== undefined) {
                        message.membershipUpdates[entry1.key] = entry1.value;
                    }
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.removedMembers.push(reader.string());
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.failedInstallations.push(reader.bytes());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            membershipUpdates: isObject(object.membershipUpdates)
                ? Object.entries(object.membershipUpdates).reduce((acc, [key, value]) => {
                    acc[key] = long_1.default.fromValue(value);
                    return acc;
                }, {})
                : {},
            removedMembers: globalThis.Array.isArray(object?.removedMembers)
                ? object.removedMembers.map((e) => globalThis.String(e))
                : [],
            failedInstallations: globalThis.Array.isArray(object?.failedInstallations)
                ? object.failedInstallations.map((e) => bytesFromBase64(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.membershipUpdates) {
            const entries = Object.entries(message.membershipUpdates);
            if (entries.length > 0) {
                obj.membershipUpdates = {};
                entries.forEach(([k, v]) => {
                    obj.membershipUpdates[k] = v.toString();
                });
            }
        }
        if (message.removedMembers?.length) {
            obj.removedMembers = message.removedMembers;
        }
        if (message.failedInstallations?.length) {
            obj.failedInstallations = message.failedInstallations.map((e) => base64FromBytes(e));
        }
        return obj;
    },
    create(base) {
        return exports.UpdateGroupMembershipData_V1.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseUpdateGroupMembershipData_V1();
        message.membershipUpdates = Object.entries(object.membershipUpdates ?? {}).reduce((acc, [key, value]) => {
            if (value !== undefined) {
                acc[key] = long_1.default.fromValue(value);
            }
            return acc;
        }, {});
        message.removedMembers = object.removedMembers?.map((e) => e) || [];
        message.failedInstallations = object.failedInstallations?.map((e) => e) || [];
        return message;
    },
};
function createBaseUpdateGroupMembershipData_V1_MembershipUpdatesEntry() {
    return { key: "", value: long_1.default.UZERO };
}
exports.UpdateGroupMembershipData_V1_MembershipUpdatesEntry = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.key !== "") {
            writer.uint32(10).string(message.key);
        }
        if (!message.value.equals(long_1.default.UZERO)) {
            writer.uint32(16).uint64(message.value);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseUpdateGroupMembershipData_V1_MembershipUpdatesEntry();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.key = reader.string();
                    continue;
                case 2:
                    if (tag !== 16) {
                        break;
                    }
                    message.value = reader.uint64();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            key: isSet(object.key) ? globalThis.String(object.key) : "",
            value: isSet(object.value) ? long_1.default.fromValue(object.value) : long_1.default.UZERO,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.key !== "") {
            obj.key = message.key;
        }
        if (!message.value.equals(long_1.default.UZERO)) {
            obj.value = (message.value || long_1.default.UZERO).toString();
        }
        return obj;
    },
    create(base) {
        return exports.UpdateGroupMembershipData_V1_MembershipUpdatesEntry.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseUpdateGroupMembershipData_V1_MembershipUpdatesEntry();
        message.key = object.key ?? "";
        message.value = (object.value !== undefined && object.value !== null) ? long_1.default.fromValue(object.value) : long_1.default.UZERO;
        return message;
    },
};
function createBaseReaddInstallationsData() {
    return { v1: undefined };
}
exports.ReaddInstallationsData = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.v1 !== undefined) {
            exports.ReaddInstallationsData_V1.encode(message.v1, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseReaddInstallationsData();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.v1 = exports.ReaddInstallationsData_V1.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { v1: isSet(object.v1) ? exports.ReaddInstallationsData_V1.fromJSON(object.v1) : undefined };
    },
    toJSON(message) {
        const obj = {};
        if (message.v1 !== undefined) {
            obj.v1 = exports.ReaddInstallationsData_V1.toJSON(message.v1);
        }
        return obj;
    },
    create(base) {
        return exports.ReaddInstallationsData.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseReaddInstallationsData();
        message.v1 = (object.v1 !== undefined && object.v1 !== null)
            ? exports.ReaddInstallationsData_V1.fromPartial(object.v1)
            : undefined;
        return message;
    },
};
function createBaseReaddInstallationsData_V1() {
    return { readdedInstallations: [] };
}
exports.ReaddInstallationsData_V1 = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        for (const v of message.readdedInstallations) {
            writer.uint32(10).bytes(v);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseReaddInstallationsData_V1();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.readdedInstallations.push(reader.bytes());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            readdedInstallations: globalThis.Array.isArray(object?.readdedInstallations)
                ? object.readdedInstallations.map((e) => bytesFromBase64(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.readdedInstallations?.length) {
            obj.readdedInstallations = message.readdedInstallations.map((e) => base64FromBytes(e));
        }
        return obj;
    },
    create(base) {
        return exports.ReaddInstallationsData_V1.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseReaddInstallationsData_V1();
        message.readdedInstallations = object.readdedInstallations?.map((e) => e) || [];
        return message;
    },
};
function createBaseUpdateMetadataData() {
    return { v1: undefined };
}
exports.UpdateMetadataData = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.v1 !== undefined) {
            exports.UpdateMetadataData_V1.encode(message.v1, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseUpdateMetadataData();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.v1 = exports.UpdateMetadataData_V1.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { v1: isSet(object.v1) ? exports.UpdateMetadataData_V1.fromJSON(object.v1) : undefined };
    },
    toJSON(message) {
        const obj = {};
        if (message.v1 !== undefined) {
            obj.v1 = exports.UpdateMetadataData_V1.toJSON(message.v1);
        }
        return obj;
    },
    create(base) {
        return exports.UpdateMetadataData.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseUpdateMetadataData();
        message.v1 = (object.v1 !== undefined && object.v1 !== null)
            ? exports.UpdateMetadataData_V1.fromPartial(object.v1)
            : undefined;
        return message;
    },
};
function createBaseUpdateMetadataData_V1() {
    return { fieldName: "", fieldValue: "" };
}
exports.UpdateMetadataData_V1 = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.fieldName !== "") {
            writer.uint32(10).string(message.fieldName);
        }
        if (message.fieldValue !== "") {
            writer.uint32(18).string(message.fieldValue);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseUpdateMetadataData_V1();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.fieldName = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.fieldValue = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            fieldName: isSet(object.fieldName) ? globalThis.String(object.fieldName) : "",
            fieldValue: isSet(object.fieldValue) ? globalThis.String(object.fieldValue) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.fieldName !== "") {
            obj.fieldName = message.fieldName;
        }
        if (message.fieldValue !== "") {
            obj.fieldValue = message.fieldValue;
        }
        return obj;
    },
    create(base) {
        return exports.UpdateMetadataData_V1.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseUpdateMetadataData_V1();
        message.fieldName = object.fieldName ?? "";
        message.fieldValue = object.fieldValue ?? "";
        return message;
    },
};
function createBaseUpdateAdminListsData() {
    return { v1: undefined };
}
exports.UpdateAdminListsData = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.v1 !== undefined) {
            exports.UpdateAdminListsData_V1.encode(message.v1, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseUpdateAdminListsData();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.v1 = exports.UpdateAdminListsData_V1.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { v1: isSet(object.v1) ? exports.UpdateAdminListsData_V1.fromJSON(object.v1) : undefined };
    },
    toJSON(message) {
        const obj = {};
        if (message.v1 !== undefined) {
            obj.v1 = exports.UpdateAdminListsData_V1.toJSON(message.v1);
        }
        return obj;
    },
    create(base) {
        return exports.UpdateAdminListsData.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseUpdateAdminListsData();
        message.v1 = (object.v1 !== undefined && object.v1 !== null)
            ? exports.UpdateAdminListsData_V1.fromPartial(object.v1)
            : undefined;
        return message;
    },
};
function createBaseUpdateAdminListsData_V1() {
    return { adminListUpdateType: 0, inboxId: "" };
}
exports.UpdateAdminListsData_V1 = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.adminListUpdateType !== 0) {
            writer.uint32(8).int32(message.adminListUpdateType);
        }
        if (message.inboxId !== "") {
            writer.uint32(18).string(message.inboxId);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseUpdateAdminListsData_V1();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 8) {
                        break;
                    }
                    message.adminListUpdateType = reader.int32();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.inboxId = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            adminListUpdateType: isSet(object.adminListUpdateType)
                ? adminListUpdateTypeFromJSON(object.adminListUpdateType)
                : 0,
            inboxId: isSet(object.inboxId) ? globalThis.String(object.inboxId) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.adminListUpdateType !== 0) {
            obj.adminListUpdateType = adminListUpdateTypeToJSON(message.adminListUpdateType);
        }
        if (message.inboxId !== "") {
            obj.inboxId = message.inboxId;
        }
        return obj;
    },
    create(base) {
        return exports.UpdateAdminListsData_V1.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseUpdateAdminListsData_V1();
        message.adminListUpdateType = object.adminListUpdateType ?? 0;
        message.inboxId = object.inboxId ?? "";
        return message;
    },
};
function createBaseUpdatePermissionData() {
    return { v1: undefined };
}
exports.UpdatePermissionData = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.v1 !== undefined) {
            exports.UpdatePermissionData_V1.encode(message.v1, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseUpdatePermissionData();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.v1 = exports.UpdatePermissionData_V1.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { v1: isSet(object.v1) ? exports.UpdatePermissionData_V1.fromJSON(object.v1) : undefined };
    },
    toJSON(message) {
        const obj = {};
        if (message.v1 !== undefined) {
            obj.v1 = exports.UpdatePermissionData_V1.toJSON(message.v1);
        }
        return obj;
    },
    create(base) {
        return exports.UpdatePermissionData.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseUpdatePermissionData();
        message.v1 = (object.v1 !== undefined && object.v1 !== null)
            ? exports.UpdatePermissionData_V1.fromPartial(object.v1)
            : undefined;
        return message;
    },
};
function createBaseUpdatePermissionData_V1() {
    return { permissionUpdateType: 0, permissionPolicyOption: 0, metadataFieldName: undefined };
}
exports.UpdatePermissionData_V1 = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.permissionUpdateType !== 0) {
            writer.uint32(8).int32(message.permissionUpdateType);
        }
        if (message.permissionPolicyOption !== 0) {
            writer.uint32(16).int32(message.permissionPolicyOption);
        }
        if (message.metadataFieldName !== undefined) {
            writer.uint32(26).string(message.metadataFieldName);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseUpdatePermissionData_V1();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 8) {
                        break;
                    }
                    message.permissionUpdateType = reader.int32();
                    continue;
                case 2:
                    if (tag !== 16) {
                        break;
                    }
                    message.permissionPolicyOption = reader.int32();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.metadataFieldName = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            permissionUpdateType: isSet(object.permissionUpdateType)
                ? permissionUpdateTypeFromJSON(object.permissionUpdateType)
                : 0,
            permissionPolicyOption: isSet(object.permissionPolicyOption)
                ? permissionPolicyOptionFromJSON(object.permissionPolicyOption)
                : 0,
            metadataFieldName: isSet(object.metadataFieldName) ? globalThis.String(object.metadataFieldName) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.permissionUpdateType !== 0) {
            obj.permissionUpdateType = permissionUpdateTypeToJSON(message.permissionUpdateType);
        }
        if (message.permissionPolicyOption !== 0) {
            obj.permissionPolicyOption = permissionPolicyOptionToJSON(message.permissionPolicyOption);
        }
        if (message.metadataFieldName !== undefined) {
            obj.metadataFieldName = message.metadataFieldName;
        }
        return obj;
    },
    create(base) {
        return exports.UpdatePermissionData_V1.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseUpdatePermissionData_V1();
        message.permissionUpdateType = object.permissionUpdateType ?? 0;
        message.permissionPolicyOption = object.permissionPolicyOption ?? 0;
        message.metadataFieldName = object.metadataFieldName ?? undefined;
        return message;
    },
};
function createBasePostCommitAction() {
    return { sendWelcomes: undefined };
}
exports.PostCommitAction = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.sendWelcomes !== undefined) {
            exports.PostCommitAction_SendWelcomes.encode(message.sendWelcomes, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBasePostCommitAction();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.sendWelcomes = exports.PostCommitAction_SendWelcomes.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            sendWelcomes: isSet(object.sendWelcomes)
                ? exports.PostCommitAction_SendWelcomes.fromJSON(object.sendWelcomes)
                : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.sendWelcomes !== undefined) {
            obj.sendWelcomes = exports.PostCommitAction_SendWelcomes.toJSON(message.sendWelcomes);
        }
        return obj;
    },
    create(base) {
        return exports.PostCommitAction.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBasePostCommitAction();
        message.sendWelcomes = (object.sendWelcomes !== undefined && object.sendWelcomes !== null)
            ? exports.PostCommitAction_SendWelcomes.fromPartial(object.sendWelcomes)
            : undefined;
        return message;
    },
};
function createBasePostCommitAction_Installation() {
    return {
        installationKey: new Uint8Array(0),
        hpkePublicKey: new Uint8Array(0),
        welcomeWrapperAlgorithm: 0,
        welcomePointeeEncryptionAeadTypes: undefined,
    };
}
exports.PostCommitAction_Installation = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.installationKey.length !== 0) {
            writer.uint32(10).bytes(message.installationKey);
        }
        if (message.hpkePublicKey.length !== 0) {
            writer.uint32(18).bytes(message.hpkePublicKey);
        }
        if (message.welcomeWrapperAlgorithm !== 0) {
            writer.uint32(24).int32(message.welcomeWrapperAlgorithm);
        }
        if (message.welcomePointeeEncryptionAeadTypes !== undefined) {
            welcome_pointer_pb_1.WelcomePointeeEncryptionAeadTypesExtension.encode(message.welcomePointeeEncryptionAeadTypes, writer.uint32(34).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBasePostCommitAction_Installation();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.installationKey = reader.bytes();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.hpkePublicKey = reader.bytes();
                    continue;
                case 3:
                    if (tag !== 24) {
                        break;
                    }
                    message.welcomeWrapperAlgorithm = reader.int32();
                    continue;
                case 4:
                    if (tag !== 34) {
                        break;
                    }
                    message.welcomePointeeEncryptionAeadTypes = welcome_pointer_pb_1.WelcomePointeeEncryptionAeadTypesExtension.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            installationKey: isSet(object.installationKey) ? bytesFromBase64(object.installationKey) : new Uint8Array(0),
            hpkePublicKey: isSet(object.hpkePublicKey) ? bytesFromBase64(object.hpkePublicKey) : new Uint8Array(0),
            welcomeWrapperAlgorithm: isSet(object.welcomeWrapperAlgorithm)
                ? (0, wrapper_encryption_pb_1.welcomeWrapperAlgorithmFromJSON)(object.welcomeWrapperAlgorithm)
                : 0,
            welcomePointeeEncryptionAeadTypes: isSet(object.welcomePointeeEncryptionAeadTypes)
                ? welcome_pointer_pb_1.WelcomePointeeEncryptionAeadTypesExtension.fromJSON(object.welcomePointeeEncryptionAeadTypes)
                : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.installationKey.length !== 0) {
            obj.installationKey = base64FromBytes(message.installationKey);
        }
        if (message.hpkePublicKey.length !== 0) {
            obj.hpkePublicKey = base64FromBytes(message.hpkePublicKey);
        }
        if (message.welcomeWrapperAlgorithm !== 0) {
            obj.welcomeWrapperAlgorithm = (0, wrapper_encryption_pb_1.welcomeWrapperAlgorithmToJSON)(message.welcomeWrapperAlgorithm);
        }
        if (message.welcomePointeeEncryptionAeadTypes !== undefined) {
            obj.welcomePointeeEncryptionAeadTypes = welcome_pointer_pb_1.WelcomePointeeEncryptionAeadTypesExtension.toJSON(message.welcomePointeeEncryptionAeadTypes);
        }
        return obj;
    },
    create(base) {
        return exports.PostCommitAction_Installation.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBasePostCommitAction_Installation();
        message.installationKey = object.installationKey ?? new Uint8Array(0);
        message.hpkePublicKey = object.hpkePublicKey ?? new Uint8Array(0);
        message.welcomeWrapperAlgorithm = object.welcomeWrapperAlgorithm ?? 0;
        message.welcomePointeeEncryptionAeadTypes =
            (object.welcomePointeeEncryptionAeadTypes !== undefined && object.welcomePointeeEncryptionAeadTypes !== null)
                ? welcome_pointer_pb_1.WelcomePointeeEncryptionAeadTypesExtension.fromPartial(object.welcomePointeeEncryptionAeadTypes)
                : undefined;
        return message;
    },
};
function createBasePostCommitAction_SendWelcomes() {
    return { installations: [], welcomeMessage: new Uint8Array(0) };
}
exports.PostCommitAction_SendWelcomes = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        for (const v of message.installations) {
            exports.PostCommitAction_Installation.encode(v, writer.uint32(10).fork()).ldelim();
        }
        if (message.welcomeMessage.length !== 0) {
            writer.uint32(18).bytes(message.welcomeMessage);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBasePostCommitAction_SendWelcomes();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.installations.push(exports.PostCommitAction_Installation.decode(reader, reader.uint32()));
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.welcomeMessage = reader.bytes();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            installations: globalThis.Array.isArray(object?.installations)
                ? object.installations.map((e) => exports.PostCommitAction_Installation.fromJSON(e))
                : [],
            welcomeMessage: isSet(object.welcomeMessage) ? bytesFromBase64(object.welcomeMessage) : new Uint8Array(0),
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.installations?.length) {
            obj.installations = message.installations.map((e) => exports.PostCommitAction_Installation.toJSON(e));
        }
        if (message.welcomeMessage.length !== 0) {
            obj.welcomeMessage = base64FromBytes(message.welcomeMessage);
        }
        return obj;
    },
    create(base) {
        return exports.PostCommitAction_SendWelcomes.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBasePostCommitAction_SendWelcomes();
        message.installations = object.installations?.map((e) => exports.PostCommitAction_Installation.fromPartial(e)) || [];
        message.welcomeMessage = object.welcomeMessage ?? new Uint8Array(0);
        return message;
    },
};
function createBaseProposeMemberUpdateData() {
    return { v1: undefined };
}
exports.ProposeMemberUpdateData = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.v1 !== undefined) {
            exports.ProposeMemberUpdateData_V1.encode(message.v1, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseProposeMemberUpdateData();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.v1 = exports.ProposeMemberUpdateData_V1.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { v1: isSet(object.v1) ? exports.ProposeMemberUpdateData_V1.fromJSON(object.v1) : undefined };
    },
    toJSON(message) {
        const obj = {};
        if (message.v1 !== undefined) {
            obj.v1 = exports.ProposeMemberUpdateData_V1.toJSON(message.v1);
        }
        return obj;
    },
    create(base) {
        return exports.ProposeMemberUpdateData.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseProposeMemberUpdateData();
        message.v1 = (object.v1 !== undefined && object.v1 !== null)
            ? exports.ProposeMemberUpdateData_V1.fromPartial(object.v1)
            : undefined;
        return message;
    },
};
function createBaseProposeMemberUpdateData_V1() {
    return { addInboxIds: [], removeInboxIds: [] };
}
exports.ProposeMemberUpdateData_V1 = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        for (const v of message.addInboxIds) {
            writer.uint32(10).bytes(v);
        }
        for (const v of message.removeInboxIds) {
            writer.uint32(18).bytes(v);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseProposeMemberUpdateData_V1();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.addInboxIds.push(reader.bytes());
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.removeInboxIds.push(reader.bytes());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            addInboxIds: globalThis.Array.isArray(object?.addInboxIds)
                ? object.addInboxIds.map((e) => bytesFromBase64(e))
                : [],
            removeInboxIds: globalThis.Array.isArray(object?.removeInboxIds)
                ? object.removeInboxIds.map((e) => bytesFromBase64(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.addInboxIds?.length) {
            obj.addInboxIds = message.addInboxIds.map((e) => base64FromBytes(e));
        }
        if (message.removeInboxIds?.length) {
            obj.removeInboxIds = message.removeInboxIds.map((e) => base64FromBytes(e));
        }
        return obj;
    },
    create(base) {
        return exports.ProposeMemberUpdateData_V1.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseProposeMemberUpdateData_V1();
        message.addInboxIds = object.addInboxIds?.map((e) => e) || [];
        message.removeInboxIds = object.removeInboxIds?.map((e) => e) || [];
        return message;
    },
};
function createBaseProposeGroupContextExtensionData() {
    return { v1: undefined };
}
exports.ProposeGroupContextExtensionData = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.v1 !== undefined) {
            exports.ProposeGroupContextExtensionData_V1.encode(message.v1, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseProposeGroupContextExtensionData();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.v1 = exports.ProposeGroupContextExtensionData_V1.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { v1: isSet(object.v1) ? exports.ProposeGroupContextExtensionData_V1.fromJSON(object.v1) : undefined };
    },
    toJSON(message) {
        const obj = {};
        if (message.v1 !== undefined) {
            obj.v1 = exports.ProposeGroupContextExtensionData_V1.toJSON(message.v1);
        }
        return obj;
    },
    create(base) {
        return exports.ProposeGroupContextExtensionData.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseProposeGroupContextExtensionData();
        message.v1 = (object.v1 !== undefined && object.v1 !== null)
            ? exports.ProposeGroupContextExtensionData_V1.fromPartial(object.v1)
            : undefined;
        return message;
    },
};
function createBaseProposeGroupContextExtensionData_V1() {
    return { groupContextExtension: new Uint8Array(0) };
}
exports.ProposeGroupContextExtensionData_V1 = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.groupContextExtension.length !== 0) {
            writer.uint32(10).bytes(message.groupContextExtension);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseProposeGroupContextExtensionData_V1();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.groupContextExtension = reader.bytes();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            groupContextExtension: isSet(object.groupContextExtension)
                ? bytesFromBase64(object.groupContextExtension)
                : new Uint8Array(0),
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.groupContextExtension.length !== 0) {
            obj.groupContextExtension = base64FromBytes(message.groupContextExtension);
        }
        return obj;
    },
    create(base) {
        return exports.ProposeGroupContextExtensionData_V1.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseProposeGroupContextExtensionData_V1();
        message.groupContextExtension = object.groupContextExtension ?? new Uint8Array(0);
        return message;
    },
};
function createBaseCommitPendingProposalsData() {
    return { v1: undefined };
}
exports.CommitPendingProposalsData = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.v1 !== undefined) {
            exports.CommitPendingProposalsData_V1.encode(message.v1, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseCommitPendingProposalsData();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.v1 = exports.CommitPendingProposalsData_V1.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { v1: isSet(object.v1) ? exports.CommitPendingProposalsData_V1.fromJSON(object.v1) : undefined };
    },
    toJSON(message) {
        const obj = {};
        if (message.v1 !== undefined) {
            obj.v1 = exports.CommitPendingProposalsData_V1.toJSON(message.v1);
        }
        return obj;
    },
    create(base) {
        return exports.CommitPendingProposalsData.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseCommitPendingProposalsData();
        message.v1 = (object.v1 !== undefined && object.v1 !== null)
            ? exports.CommitPendingProposalsData_V1.fromPartial(object.v1)
            : undefined;
        return message;
    },
};
function createBaseCommitPendingProposalsData_V1() {
    return { proposalHashes: [] };
}
exports.CommitPendingProposalsData_V1 = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        for (const v of message.proposalHashes) {
            writer.uint32(10).bytes(v);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseCommitPendingProposalsData_V1();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.proposalHashes.push(reader.bytes());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            proposalHashes: globalThis.Array.isArray(object?.proposalHashes)
                ? object.proposalHashes.map((e) => bytesFromBase64(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.proposalHashes?.length) {
            obj.proposalHashes = message.proposalHashes.map((e) => base64FromBytes(e));
        }
        return obj;
    },
    create(base) {
        return exports.CommitPendingProposalsData_V1.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseCommitPendingProposalsData_V1();
        message.proposalHashes = object.proposalHashes?.map((e) => e) || [];
        return message;
    },
};
function bytesFromBase64(b64) {
    if (globalThis.Buffer) {
        return Uint8Array.from(globalThis.Buffer.from(b64, "base64"));
    }
    else {
        const bin = globalThis.atob(b64);
        const arr = new Uint8Array(bin.length);
        for (let i = 0; i < bin.length; ++i) {
            arr[i] = bin.charCodeAt(i);
        }
        return arr;
    }
}
function base64FromBytes(arr) {
    if (globalThis.Buffer) {
        return globalThis.Buffer.from(arr).toString("base64");
    }
    else {
        const bin = [];
        arr.forEach((byte) => {
            bin.push(globalThis.String.fromCharCode(byte));
        });
        return globalThis.btoa(bin.join(""));
    }
}
if (minimal_1.default.util.Long !== long_1.default) {
    minimal_1.default.util.Long = long_1.default;
    minimal_1.default.configure();
}
function isObject(value) {
    return typeof value === "object" && value !== null;
}
function isSet(value) {
    return value !== null && value !== undefined;
}
