// @xmtp/proto/ts/dist/cjs/mls_validation/v1/service.pb.js
"use strict";
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v1.181.2
//   protoc               v3.21.12
// source: mls_validation/v1/service.proto
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ValidationApiClientImpl = exports.ValidationApiServiceName = exports.GetAssociationStateResponse = exports.GetAssociationStateRequest = exports.ValidateGroupMessagesResponse_ValidationResponse = exports.ValidateGroupMessagesResponse = exports.ValidateGroupMessagesRequest_GroupMessage = exports.ValidateGroupMessagesRequest = exports.ValidateKeyPackagesResponse_ValidationResponse = exports.ValidateKeyPackagesResponse = exports.ValidateKeyPackagesRequest_KeyPackage = exports.ValidateKeyPackagesRequest = exports.ValidateInboxIdKeyPackagesResponse_Response = exports.ValidateInboxIdKeyPackagesResponse = exports.ValidateInboxIdKeyPackagesRequest_KeyPackage = exports.ValidateInboxIdKeyPackagesRequest = exports.protobufPackage = void 0;
/* eslint-disable */
const long_1 = __importDefault(require("long"));
const minimal_1 = __importDefault(require("protobufjs/minimal"));
const identity_pb_1 = require("../../identity/api/v1/identity.pb");
const association_pb_1 = require("../../identity/associations/association.pb");
const credential_pb_1 = require("../../identity/credential.pb");
exports.protobufPackage = "xmtp.mls_validation.v1";
function createBaseValidateInboxIdKeyPackagesRequest() {
    return { keyPackages: [] };
}
exports.ValidateInboxIdKeyPackagesRequest = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        for (const v of message.keyPackages) {
            exports.ValidateInboxIdKeyPackagesRequest_KeyPackage.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseValidateInboxIdKeyPackagesRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.keyPackages.push(exports.ValidateInboxIdKeyPackagesRequest_KeyPackage.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            keyPackages: globalThis.Array.isArray(object?.keyPackages)
                ? object.keyPackages.map((e) => exports.ValidateInboxIdKeyPackagesRequest_KeyPackage.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.keyPackages?.length) {
            obj.keyPackages = message.keyPackages.map((e) => exports.ValidateInboxIdKeyPackagesRequest_KeyPackage.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return exports.ValidateInboxIdKeyPackagesRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseValidateInboxIdKeyPackagesRequest();
        message.keyPackages = object.keyPackages?.map((e) => exports.ValidateInboxIdKeyPackagesRequest_KeyPackage.fromPartial(e)) ||
            [];
        return message;
    },
};
function createBaseValidateInboxIdKeyPackagesRequest_KeyPackage() {
    return { keyPackageBytesTlsSerialized: new Uint8Array(0), isInboxIdCredential: false };
}
exports.ValidateInboxIdKeyPackagesRequest_KeyPackage = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.keyPackageBytesTlsSerialized.length !== 0) {
            writer.uint32(10).bytes(message.keyPackageBytesTlsSerialized);
        }
        if (message.isInboxIdCredential !== false) {
            writer.uint32(16).bool(message.isInboxIdCredential);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseValidateInboxIdKeyPackagesRequest_KeyPackage();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.keyPackageBytesTlsSerialized = reader.bytes();
                    continue;
                case 2:
                    if (tag !== 16) {
                        break;
                    }
                    message.isInboxIdCredential = reader.bool();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            keyPackageBytesTlsSerialized: isSet(object.keyPackageBytesTlsSerialized)
                ? bytesFromBase64(object.keyPackageBytesTlsSerialized)
                : new Uint8Array(0),
            isInboxIdCredential: isSet(object.isInboxIdCredential) ? globalThis.Boolean(object.isInboxIdCredential) : false,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.keyPackageBytesTlsSerialized.length !== 0) {
            obj.keyPackageBytesTlsSerialized = base64FromBytes(message.keyPackageBytesTlsSerialized);
        }
        if (message.isInboxIdCredential !== false) {
            obj.isInboxIdCredential = message.isInboxIdCredential;
        }
        return obj;
    },
    create(base) {
        return exports.ValidateInboxIdKeyPackagesRequest_KeyPackage.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseValidateInboxIdKeyPackagesRequest_KeyPackage();
        message.keyPackageBytesTlsSerialized = object.keyPackageBytesTlsSerialized ?? new Uint8Array(0);
        message.isInboxIdCredential = object.isInboxIdCredential ?? false;
        return message;
    },
};
function createBaseValidateInboxIdKeyPackagesResponse() {
    return { responses: [] };
}
exports.ValidateInboxIdKeyPackagesResponse = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        for (const v of message.responses) {
            exports.ValidateInboxIdKeyPackagesResponse_Response.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseValidateInboxIdKeyPackagesResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.responses.push(exports.ValidateInboxIdKeyPackagesResponse_Response.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            responses: globalThis.Array.isArray(object?.responses)
                ? object.responses.map((e) => exports.ValidateInboxIdKeyPackagesResponse_Response.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.responses?.length) {
            obj.responses = message.responses.map((e) => exports.ValidateInboxIdKeyPackagesResponse_Response.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return exports.ValidateInboxIdKeyPackagesResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseValidateInboxIdKeyPackagesResponse();
        message.responses = object.responses?.map((e) => exports.ValidateInboxIdKeyPackagesResponse_Response.fromPartial(e)) || [];
        return message;
    },
};
function createBaseValidateInboxIdKeyPackagesResponse_Response() {
    return {
        isOk: false,
        errorMessage: "",
        credential: undefined,
        installationPublicKey: new Uint8Array(0),
        expiration: long_1.default.UZERO,
    };
}
exports.ValidateInboxIdKeyPackagesResponse_Response = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.isOk !== false) {
            writer.uint32(8).bool(message.isOk);
        }
        if (message.errorMessage !== "") {
            writer.uint32(18).string(message.errorMessage);
        }
        if (message.credential !== undefined) {
            credential_pb_1.MlsCredential.encode(message.credential, writer.uint32(26).fork()).ldelim();
        }
        if (message.installationPublicKey.length !== 0) {
            writer.uint32(34).bytes(message.installationPublicKey);
        }
        if (!message.expiration.equals(long_1.default.UZERO)) {
            writer.uint32(40).uint64(message.expiration);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseValidateInboxIdKeyPackagesResponse_Response();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 8) {
                        break;
                    }
                    message.isOk = reader.bool();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.errorMessage = reader.string();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.credential = credential_pb_1.MlsCredential.decode(reader, reader.uint32());
                    continue;
                case 4:
                    if (tag !== 34) {
                        break;
                    }
                    message.installationPublicKey = reader.bytes();
                    continue;
                case 5:
                    if (tag !== 40) {
                        break;
                    }
                    message.expiration = reader.uint64();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            isOk: isSet(object.isOk) ? globalThis.Boolean(object.isOk) : false,
            errorMessage: isSet(object.errorMessage) ? globalThis.String(object.errorMessage) : "",
            credential: isSet(object.credential) ? credential_pb_1.MlsCredential.fromJSON(object.credential) : undefined,
            installationPublicKey: isSet(object.installationPublicKey)
                ? bytesFromBase64(object.installationPublicKey)
                : new Uint8Array(0),
            expiration: isSet(object.expiration) ? long_1.default.fromValue(object.expiration) : long_1.default.UZERO,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.isOk !== false) {
            obj.isOk = message.isOk;
        }
        if (message.errorMessage !== "") {
            obj.errorMessage = message.errorMessage;
        }
        if (message.credential !== undefined) {
            obj.credential = credential_pb_1.MlsCredential.toJSON(message.credential);
        }
        if (message.installationPublicKey.length !== 0) {
            obj.installationPublicKey = base64FromBytes(message.installationPublicKey);
        }
        if (!message.expiration.equals(long_1.default.UZERO)) {
            obj.expiration = (message.expiration || long_1.default.UZERO).toString();
        }
        return obj;
    },
    create(base) {
        return exports.ValidateInboxIdKeyPackagesResponse_Response.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseValidateInboxIdKeyPackagesResponse_Response();
        message.isOk = object.isOk ?? false;
        message.errorMessage = object.errorMessage ?? "";
        message.credential = (object.credential !== undefined && object.credential !== null)
            ? credential_pb_1.MlsCredential.fromPartial(object.credential)
            : undefined;
        message.installationPublicKey = object.installationPublicKey ?? new Uint8Array(0);
        message.expiration = (object.expiration !== undefined && object.expiration !== null)
            ? long_1.default.fromValue(object.expiration)
            : long_1.default.UZERO;
        return message;
    },
};
function createBaseValidateKeyPackagesRequest() {
    return { keyPackages: [] };
}
exports.ValidateKeyPackagesRequest = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        for (const v of message.keyPackages) {
            exports.ValidateKeyPackagesRequest_KeyPackage.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseValidateKeyPackagesRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.keyPackages.push(exports.ValidateKeyPackagesRequest_KeyPackage.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            keyPackages: globalThis.Array.isArray(object?.keyPackages)
                ? object.keyPackages.map((e) => exports.ValidateKeyPackagesRequest_KeyPackage.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.keyPackages?.length) {
            obj.keyPackages = message.keyPackages.map((e) => exports.ValidateKeyPackagesRequest_KeyPackage.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return exports.ValidateKeyPackagesRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseValidateKeyPackagesRequest();
        message.keyPackages = object.keyPackages?.map((e) => exports.ValidateKeyPackagesRequest_KeyPackage.fromPartial(e)) || [];
        return message;
    },
};
function createBaseValidateKeyPackagesRequest_KeyPackage() {
    return { keyPackageBytesTlsSerialized: new Uint8Array(0), isInboxIdCredential: false };
}
exports.ValidateKeyPackagesRequest_KeyPackage = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.keyPackageBytesTlsSerialized.length !== 0) {
            writer.uint32(10).bytes(message.keyPackageBytesTlsSerialized);
        }
        if (message.isInboxIdCredential !== false) {
            writer.uint32(16).bool(message.isInboxIdCredential);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseValidateKeyPackagesRequest_KeyPackage();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.keyPackageBytesTlsSerialized = reader.bytes();
                    continue;
                case 2:
                    if (tag !== 16) {
                        break;
                    }
                    message.isInboxIdCredential = reader.bool();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            keyPackageBytesTlsSerialized: isSet(object.keyPackageBytesTlsSerialized)
                ? bytesFromBase64(object.keyPackageBytesTlsSerialized)
                : new Uint8Array(0),
            isInboxIdCredential: isSet(object.isInboxIdCredential) ? globalThis.Boolean(object.isInboxIdCredential) : false,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.keyPackageBytesTlsSerialized.length !== 0) {
            obj.keyPackageBytesTlsSerialized = base64FromBytes(message.keyPackageBytesTlsSerialized);
        }
        if (message.isInboxIdCredential !== false) {
            obj.isInboxIdCredential = message.isInboxIdCredential;
        }
        return obj;
    },
    create(base) {
        return exports.ValidateKeyPackagesRequest_KeyPackage.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseValidateKeyPackagesRequest_KeyPackage();
        message.keyPackageBytesTlsSerialized = object.keyPackageBytesTlsSerialized ?? new Uint8Array(0);
        message.isInboxIdCredential = object.isInboxIdCredential ?? false;
        return message;
    },
};
function createBaseValidateKeyPackagesResponse() {
    return { responses: [] };
}
exports.ValidateKeyPackagesResponse = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        for (const v of message.responses) {
            exports.ValidateKeyPackagesResponse_ValidationResponse.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseValidateKeyPackagesResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.responses.push(exports.ValidateKeyPackagesResponse_ValidationResponse.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            responses: globalThis.Array.isArray(object?.responses)
                ? object.responses.map((e) => exports.ValidateKeyPackagesResponse_ValidationResponse.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.responses?.length) {
            obj.responses = message.responses.map((e) => exports.ValidateKeyPackagesResponse_ValidationResponse.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return exports.ValidateKeyPackagesResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseValidateKeyPackagesResponse();
        message.responses = object.responses?.map((e) => exports.ValidateKeyPackagesResponse_ValidationResponse.fromPartial(e)) ||
            [];
        return message;
    },
};
function createBaseValidateKeyPackagesResponse_ValidationResponse() {
    return {
        isOk: false,
        errorMessage: "",
        installationId: new Uint8Array(0),
        accountAddress: "",
        credentialIdentityBytes: new Uint8Array(0),
        expiration: long_1.default.UZERO,
    };
}
exports.ValidateKeyPackagesResponse_ValidationResponse = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.isOk !== false) {
            writer.uint32(8).bool(message.isOk);
        }
        if (message.errorMessage !== "") {
            writer.uint32(18).string(message.errorMessage);
        }
        if (message.installationId.length !== 0) {
            writer.uint32(26).bytes(message.installationId);
        }
        if (message.accountAddress !== "") {
            writer.uint32(34).string(message.accountAddress);
        }
        if (message.credentialIdentityBytes.length !== 0) {
            writer.uint32(42).bytes(message.credentialIdentityBytes);
        }
        if (!message.expiration.equals(long_1.default.UZERO)) {
            writer.uint32(48).uint64(message.expiration);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseValidateKeyPackagesResponse_ValidationResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 8) {
                        break;
                    }
                    message.isOk = reader.bool();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.errorMessage = reader.string();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.installationId = reader.bytes();
                    continue;
                case 4:
                    if (tag !== 34) {
                        break;
                    }
                    message.accountAddress = reader.string();
                    continue;
                case 5:
                    if (tag !== 42) {
                        break;
                    }
                    message.credentialIdentityBytes = reader.bytes();
                    continue;
                case 6:
                    if (tag !== 48) {
                        break;
                    }
                    message.expiration = reader.uint64();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            isOk: isSet(object.isOk) ? globalThis.Boolean(object.isOk) : false,
            errorMessage: isSet(object.errorMessage) ? globalThis.String(object.errorMessage) : "",
            installationId: isSet(object.installationId) ? bytesFromBase64(object.installationId) : new Uint8Array(0),
            accountAddress: isSet(object.accountAddress) ? globalThis.String(object.accountAddress) : "",
            credentialIdentityBytes: isSet(object.credentialIdentityBytes)
                ? bytesFromBase64(object.credentialIdentityBytes)
                : new Uint8Array(0),
            expiration: isSet(object.expiration) ? long_1.default.fromValue(object.expiration) : long_1.default.UZERO,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.isOk !== false) {
            obj.isOk = message.isOk;
        }
        if (message.errorMessage !== "") {
            obj.errorMessage = message.errorMessage;
        }
        if (message.installationId.length !== 0) {
            obj.installationId = base64FromBytes(message.installationId);
        }
        if (message.accountAddress !== "") {
            obj.accountAddress = message.accountAddress;
        }
        if (message.credentialIdentityBytes.length !== 0) {
            obj.credentialIdentityBytes = base64FromBytes(message.credentialIdentityBytes);
        }
        if (!message.expiration.equals(long_1.default.UZERO)) {
            obj.expiration = (message.expiration || long_1.default.UZERO).toString();
        }
        return obj;
    },
    create(base) {
        return exports.ValidateKeyPackagesResponse_ValidationResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseValidateKeyPackagesResponse_ValidationResponse();
        message.isOk = object.isOk ?? false;
        message.errorMessage = object.errorMessage ?? "";
        message.installationId = object.installationId ?? new Uint8Array(0);
        message.accountAddress = object.accountAddress ?? "";
        message.credentialIdentityBytes = object.credentialIdentityBytes ?? new Uint8Array(0);
        message.expiration = (object.expiration !== undefined && object.expiration !== null)
            ? long_1.default.fromValue(object.expiration)
            : long_1.default.UZERO;
        return message;
    },
};
function createBaseValidateGroupMessagesRequest() {
    return { groupMessages: [] };
}
exports.ValidateGroupMessagesRequest = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        for (const v of message.groupMessages) {
            exports.ValidateGroupMessagesRequest_GroupMessage.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseValidateGroupMessagesRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.groupMessages.push(exports.ValidateGroupMessagesRequest_GroupMessage.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            groupMessages: globalThis.Array.isArray(object?.groupMessages)
                ? object.groupMessages.map((e) => exports.ValidateGroupMessagesRequest_GroupMessage.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.groupMessages?.length) {
            obj.groupMessages = message.groupMessages.map((e) => exports.ValidateGroupMessagesRequest_GroupMessage.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return exports.ValidateGroupMessagesRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseValidateGroupMessagesRequest();
        message.groupMessages =
            object.groupMessages?.map((e) => exports.ValidateGroupMessagesRequest_GroupMessage.fromPartial(e)) || [];
        return message;
    },
};
function createBaseValidateGroupMessagesRequest_GroupMessage() {
    return { groupMessageBytesTlsSerialized: new Uint8Array(0) };
}
exports.ValidateGroupMessagesRequest_GroupMessage = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.groupMessageBytesTlsSerialized.length !== 0) {
            writer.uint32(10).bytes(message.groupMessageBytesTlsSerialized);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseValidateGroupMessagesRequest_GroupMessage();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.groupMessageBytesTlsSerialized = reader.bytes();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            groupMessageBytesTlsSerialized: isSet(object.groupMessageBytesTlsSerialized)
                ? bytesFromBase64(object.groupMessageBytesTlsSerialized)
                : new Uint8Array(0),
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.groupMessageBytesTlsSerialized.length !== 0) {
            obj.groupMessageBytesTlsSerialized = base64FromBytes(message.groupMessageBytesTlsSerialized);
        }
        return obj;
    },
    create(base) {
        return exports.ValidateGroupMessagesRequest_GroupMessage.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseValidateGroupMessagesRequest_GroupMessage();
        message.groupMessageBytesTlsSerialized = object.groupMessageBytesTlsSerialized ?? new Uint8Array(0);
        return message;
    },
};
function createBaseValidateGroupMessagesResponse() {
    return { responses: [] };
}
exports.ValidateGroupMessagesResponse = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        for (const v of message.responses) {
            exports.ValidateGroupMessagesResponse_ValidationResponse.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseValidateGroupMessagesResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.responses.push(exports.ValidateGroupMessagesResponse_ValidationResponse.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            responses: globalThis.Array.isArray(object?.responses)
                ? object.responses.map((e) => exports.ValidateGroupMessagesResponse_ValidationResponse.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.responses?.length) {
            obj.responses = message.responses.map((e) => exports.ValidateGroupMessagesResponse_ValidationResponse.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return exports.ValidateGroupMessagesResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseValidateGroupMessagesResponse();
        message.responses = object.responses?.map((e) => exports.ValidateGroupMessagesResponse_ValidationResponse.fromPartial(e)) ||
            [];
        return message;
    },
};
function createBaseValidateGroupMessagesResponse_ValidationResponse() {
    return { isOk: false, errorMessage: "", groupId: "", isCommit: false };
}
exports.ValidateGroupMessagesResponse_ValidationResponse = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.isOk !== false) {
            writer.uint32(8).bool(message.isOk);
        }
        if (message.errorMessage !== "") {
            writer.uint32(18).string(message.errorMessage);
        }
        if (message.groupId !== "") {
            writer.uint32(26).string(message.groupId);
        }
        if (message.isCommit !== false) {
            writer.uint32(32).bool(message.isCommit);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseValidateGroupMessagesResponse_ValidationResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 8) {
                        break;
                    }
                    message.isOk = reader.bool();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.errorMessage = reader.string();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.groupId = reader.string();
                    continue;
                case 4:
                    if (tag !== 32) {
                        break;
                    }
                    message.isCommit = reader.bool();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            isOk: isSet(object.isOk) ? globalThis.Boolean(object.isOk) : false,
            errorMessage: isSet(object.errorMessage) ? globalThis.String(object.errorMessage) : "",
            groupId: isSet(object.groupId) ? globalThis.String(object.groupId) : "",
            isCommit: isSet(object.isCommit) ? globalThis.Boolean(object.isCommit) : false,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.isOk !== false) {
            obj.isOk = message.isOk;
        }
        if (message.errorMessage !== "") {
            obj.errorMessage = message.errorMessage;
        }
        if (message.groupId !== "") {
            obj.groupId = message.groupId;
        }
        if (message.isCommit !== false) {
            obj.isCommit = message.isCommit;
        }
        return obj;
    },
    create(base) {
        return exports.ValidateGroupMessagesResponse_ValidationResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseValidateGroupMessagesResponse_ValidationResponse();
        message.isOk = object.isOk ?? false;
        message.errorMessage = object.errorMessage ?? "";
        message.groupId = object.groupId ?? "";
        message.isCommit = object.isCommit ?? false;
        return message;
    },
};
function createBaseGetAssociationStateRequest() {
    return { oldUpdates: [], newUpdates: [] };
}
exports.GetAssociationStateRequest = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        for (const v of message.oldUpdates) {
            association_pb_1.IdentityUpdate.encode(v, writer.uint32(10).fork()).ldelim();
        }
        for (const v of message.newUpdates) {
            association_pb_1.IdentityUpdate.encode(v, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetAssociationStateRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.oldUpdates.push(association_pb_1.IdentityUpdate.decode(reader, reader.uint32()));
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.newUpdates.push(association_pb_1.IdentityUpdate.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            oldUpdates: globalThis.Array.isArray(object?.oldUpdates)
                ? object.oldUpdates.map((e) => association_pb_1.IdentityUpdate.fromJSON(e))
                : [],
            newUpdates: globalThis.Array.isArray(object?.newUpdates)
                ? object.newUpdates.map((e) => association_pb_1.IdentityUpdate.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.oldUpdates?.length) {
            obj.oldUpdates = message.oldUpdates.map((e) => association_pb_1.IdentityUpdate.toJSON(e));
        }
        if (message.newUpdates?.length) {
            obj.newUpdates = message.newUpdates.map((e) => association_pb_1.IdentityUpdate.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return exports.GetAssociationStateRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGetAssociationStateRequest();
        message.oldUpdates = object.oldUpdates?.map((e) => association_pb_1.IdentityUpdate.fromPartial(e)) || [];
        message.newUpdates = object.newUpdates?.map((e) => association_pb_1.IdentityUpdate.fromPartial(e)) || [];
        return message;
    },
};
function createBaseGetAssociationStateResponse() {
    return { associationState: undefined, stateDiff: undefined };
}
exports.GetAssociationStateResponse = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.associationState !== undefined) {
            association_pb_1.AssociationState.encode(message.associationState, writer.uint32(10).fork()).ldelim();
        }
        if (message.stateDiff !== undefined) {
            association_pb_1.AssociationStateDiff.encode(message.stateDiff, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetAssociationStateResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.associationState = association_pb_1.AssociationState.decode(reader, reader.uint32());
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.stateDiff = association_pb_1.AssociationStateDiff.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            associationState: isSet(object.associationState) ? association_pb_1.AssociationState.fromJSON(object.associationState) : undefined,
            stateDiff: isSet(object.stateDiff) ? association_pb_1.AssociationStateDiff.fromJSON(object.stateDiff) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.associationState !== undefined) {
            obj.associationState = association_pb_1.AssociationState.toJSON(message.associationState);
        }
        if (message.stateDiff !== undefined) {
            obj.stateDiff = association_pb_1.AssociationStateDiff.toJSON(message.stateDiff);
        }
        return obj;
    },
    create(base) {
        return exports.GetAssociationStateResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGetAssociationStateResponse();
        message.associationState = (object.associationState !== undefined && object.associationState !== null)
            ? association_pb_1.AssociationState.fromPartial(object.associationState)
            : undefined;
        message.stateDiff = (object.stateDiff !== undefined && object.stateDiff !== null)
            ? association_pb_1.AssociationStateDiff.fromPartial(object.stateDiff)
            : undefined;
        return message;
    },
};
exports.ValidationApiServiceName = "xmtp.mls_validation.v1.ValidationApi";
class ValidationApiClientImpl {
    rpc;
    service;
    constructor(rpc, opts) {
        this.service = opts?.service || exports.ValidationApiServiceName;
        this.rpc = rpc;
        this.ValidateGroupMessages = this.ValidateGroupMessages.bind(this);
        this.GetAssociationState = this.GetAssociationState.bind(this);
        this.ValidateInboxIdKeyPackages = this.ValidateInboxIdKeyPackages.bind(this);
        this.VerifySmartContractWalletSignatures = this.VerifySmartContractWalletSignatures.bind(this);
    }
    ValidateGroupMessages(request) {
        const data = exports.ValidateGroupMessagesRequest.encode(request).finish();
        const promise = this.rpc.request(this.service, "ValidateGroupMessages", data);
        return promise.then((data) => exports.ValidateGroupMessagesResponse.decode(minimal_1.default.Reader.create(data)));
    }
    GetAssociationState(request) {
        const data = exports.GetAssociationStateRequest.encode(request).finish();
        const promise = this.rpc.request(this.service, "GetAssociationState", data);
        return promise.then((data) => exports.GetAssociationStateResponse.decode(minimal_1.default.Reader.create(data)));
    }
    ValidateInboxIdKeyPackages(request) {
        const data = exports.ValidateKeyPackagesRequest.encode(request).finish();
        const promise = this.rpc.request(this.service, "ValidateInboxIdKeyPackages", data);
        return promise.then((data) => exports.ValidateInboxIdKeyPackagesResponse.decode(minimal_1.default.Reader.create(data)));
    }
    VerifySmartContractWalletSignatures(request) {
        const data = identity_pb_1.VerifySmartContractWalletSignaturesRequest.encode(request).finish();
        const promise = this.rpc.request(this.service, "VerifySmartContractWalletSignatures", data);
        return promise.then((data) => identity_pb_1.VerifySmartContractWalletSignaturesResponse.decode(minimal_1.default.Reader.create(data)));
    }
}
exports.ValidationApiClientImpl = ValidationApiClientImpl;
function bytesFromBase64(b64) {
    if (globalThis.Buffer) {
        return Uint8Array.from(globalThis.Buffer.from(b64, "base64"));
    }
    else {
        const bin = globalThis.atob(b64);
        const arr = new Uint8Array(bin.length);
        for (let i = 0; i < bin.length; ++i) {
            arr[i] = bin.charCodeAt(i);
        }
        return arr;
    }
}
function base64FromBytes(arr) {
    if (globalThis.Buffer) {
        return globalThis.Buffer.from(arr).toString("base64");
    }
    else {
        const bin = [];
        arr.forEach((byte) => {
            bin.push(globalThis.String.fromCharCode(byte));
        });
        return globalThis.btoa(bin.join(""));
    }
}
if (minimal_1.default.util.Long !== long_1.default) {
    minimal_1.default.util.Long = long_1.default;
    minimal_1.default.configure();
}
function isSet(value) {
    return value !== null && value !== undefined;
}
