// @xmtp/proto/ts/dist/esm/keystore_api/v1/keystore.pb.js
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v1.181.2
//   protoc               v3.21.12
// source: keystore_api/v1/keystore.proto
/* eslint-disable */
import Long from "long";
import _m0 from "protobufjs/minimal";
import { Ciphertext } from "../../message_contents/ciphertext.pb";
import { ConversationReference } from "../../message_contents/conversation_reference.pb";
import { ConsentProofPayload, InvitationV1, InvitationV1_Context } from "../../message_contents/invitation.pb";
import { PrivateKeyBundleV1 } from "../../message_contents/private_key.pb";
import { PrivatePreferencesAction } from "../../message_contents/private_preferences.pb";
import { PublicKeyBundle, SignedPublicKeyBundle } from "../../message_contents/public_key.pb";
export const protobufPackage = "xmtp.keystore_api.v1";
/** Message content encoding structures */
/** Application-specific error codes for the Keystore API. */
export var ErrorCode;
(function (ErrorCode) {
    ErrorCode[ErrorCode["ERROR_CODE_UNSPECIFIED"] = 0] = "ERROR_CODE_UNSPECIFIED";
    ErrorCode[ErrorCode["ERROR_CODE_INVALID_INPUT"] = 1] = "ERROR_CODE_INVALID_INPUT";
    ErrorCode[ErrorCode["ERROR_CODE_NO_MATCHING_PREKEY"] = 2] = "ERROR_CODE_NO_MATCHING_PREKEY";
    ErrorCode[ErrorCode["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(ErrorCode || (ErrorCode = {}));
export function errorCodeFromJSON(object) {
    switch (object) {
        case 0:
        case "ERROR_CODE_UNSPECIFIED":
            return ErrorCode.ERROR_CODE_UNSPECIFIED;
        case 1:
        case "ERROR_CODE_INVALID_INPUT":
            return ErrorCode.ERROR_CODE_INVALID_INPUT;
        case 2:
        case "ERROR_CODE_NO_MATCHING_PREKEY":
            return ErrorCode.ERROR_CODE_NO_MATCHING_PREKEY;
        case -1:
        case "UNRECOGNIZED":
        default:
            return ErrorCode.UNRECOGNIZED;
    }
}
export function errorCodeToJSON(object) {
    switch (object) {
        case ErrorCode.ERROR_CODE_UNSPECIFIED:
            return "ERROR_CODE_UNSPECIFIED";
        case ErrorCode.ERROR_CODE_INVALID_INPUT:
            return "ERROR_CODE_INVALID_INPUT";
        case ErrorCode.ERROR_CODE_NO_MATCHING_PREKEY:
            return "ERROR_CODE_NO_MATCHING_PREKEY";
        case ErrorCode.UNRECOGNIZED:
        default:
            return "UNRECOGNIZED";
    }
}
/** JobType is used to specify the type of job the caller would like info on */
export var JobType;
(function (JobType) {
    JobType[JobType["JOB_TYPE_UNSPECIFIED"] = 0] = "JOB_TYPE_UNSPECIFIED";
    JobType[JobType["JOB_TYPE_REFRESH_V1"] = 1] = "JOB_TYPE_REFRESH_V1";
    JobType[JobType["JOB_TYPE_REFRESH_V2"] = 2] = "JOB_TYPE_REFRESH_V2";
    JobType[JobType["JOB_TYPE_REFRESH_PPPP"] = 3] = "JOB_TYPE_REFRESH_PPPP";
    JobType[JobType["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(JobType || (JobType = {}));
export function jobTypeFromJSON(object) {
    switch (object) {
        case 0:
        case "JOB_TYPE_UNSPECIFIED":
            return JobType.JOB_TYPE_UNSPECIFIED;
        case 1:
        case "JOB_TYPE_REFRESH_V1":
            return JobType.JOB_TYPE_REFRESH_V1;
        case 2:
        case "JOB_TYPE_REFRESH_V2":
            return JobType.JOB_TYPE_REFRESH_V2;
        case 3:
        case "JOB_TYPE_REFRESH_PPPP":
            return JobType.JOB_TYPE_REFRESH_PPPP;
        case -1:
        case "UNRECOGNIZED":
        default:
            return JobType.UNRECOGNIZED;
    }
}
export function jobTypeToJSON(object) {
    switch (object) {
        case JobType.JOB_TYPE_UNSPECIFIED:
            return "JOB_TYPE_UNSPECIFIED";
        case JobType.JOB_TYPE_REFRESH_V1:
            return "JOB_TYPE_REFRESH_V1";
        case JobType.JOB_TYPE_REFRESH_V2:
            return "JOB_TYPE_REFRESH_V2";
        case JobType.JOB_TYPE_REFRESH_PPPP:
            return "JOB_TYPE_REFRESH_PPPP";
        case JobType.UNRECOGNIZED:
        default:
            return "UNRECOGNIZED";
    }
}
/** Status of the Keystore for the specified wallet address */
export var GetKeystoreStatusResponse_KeystoreStatus;
(function (GetKeystoreStatusResponse_KeystoreStatus) {
    GetKeystoreStatusResponse_KeystoreStatus[GetKeystoreStatusResponse_KeystoreStatus["KEYSTORE_STATUS_UNSPECIFIED"] = 0] = "KEYSTORE_STATUS_UNSPECIFIED";
    GetKeystoreStatusResponse_KeystoreStatus[GetKeystoreStatusResponse_KeystoreStatus["KEYSTORE_STATUS_UNINITIALIZED"] = 1] = "KEYSTORE_STATUS_UNINITIALIZED";
    GetKeystoreStatusResponse_KeystoreStatus[GetKeystoreStatusResponse_KeystoreStatus["KEYSTORE_STATUS_INITIALIZED"] = 2] = "KEYSTORE_STATUS_INITIALIZED";
    GetKeystoreStatusResponse_KeystoreStatus[GetKeystoreStatusResponse_KeystoreStatus["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(GetKeystoreStatusResponse_KeystoreStatus || (GetKeystoreStatusResponse_KeystoreStatus = {}));
export function getKeystoreStatusResponse_KeystoreStatusFromJSON(object) {
    switch (object) {
        case 0:
        case "KEYSTORE_STATUS_UNSPECIFIED":
            return GetKeystoreStatusResponse_KeystoreStatus.KEYSTORE_STATUS_UNSPECIFIED;
        case 1:
        case "KEYSTORE_STATUS_UNINITIALIZED":
            return GetKeystoreStatusResponse_KeystoreStatus.KEYSTORE_STATUS_UNINITIALIZED;
        case 2:
        case "KEYSTORE_STATUS_INITIALIZED":
            return GetKeystoreStatusResponse_KeystoreStatus.KEYSTORE_STATUS_INITIALIZED;
        case -1:
        case "UNRECOGNIZED":
        default:
            return GetKeystoreStatusResponse_KeystoreStatus.UNRECOGNIZED;
    }
}
export function getKeystoreStatusResponse_KeystoreStatusToJSON(object) {
    switch (object) {
        case GetKeystoreStatusResponse_KeystoreStatus.KEYSTORE_STATUS_UNSPECIFIED:
            return "KEYSTORE_STATUS_UNSPECIFIED";
        case GetKeystoreStatusResponse_KeystoreStatus.KEYSTORE_STATUS_UNINITIALIZED:
            return "KEYSTORE_STATUS_UNINITIALIZED";
        case GetKeystoreStatusResponse_KeystoreStatus.KEYSTORE_STATUS_INITIALIZED:
            return "KEYSTORE_STATUS_INITIALIZED";
        case GetKeystoreStatusResponse_KeystoreStatus.UNRECOGNIZED:
        default:
            return "UNRECOGNIZED";
    }
}
function createBaseKeystoreError() {
    return { message: "", code: 0 };
}
export const KeystoreError = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.message !== "") {
            writer.uint32(10).string(message.message);
        }
        if (message.code !== 0) {
            writer.uint32(16).int32(message.code);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseKeystoreError();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.message = reader.string();
                    continue;
                case 2:
                    if (tag !== 16) {
                        break;
                    }
                    message.code = reader.int32();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            message: isSet(object.message) ? globalThis.String(object.message) : "",
            code: isSet(object.code) ? errorCodeFromJSON(object.code) : 0,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.message !== "") {
            obj.message = message.message;
        }
        if (message.code !== 0) {
            obj.code = errorCodeToJSON(message.code);
        }
        return obj;
    },
    create(base) {
        return KeystoreError.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseKeystoreError();
        message.message = object.message ?? "";
        message.code = object.code ?? 0;
        return message;
    },
};
function createBaseDecryptV1Request() {
    return { requests: [] };
}
export const DecryptV1Request = {
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.requests) {
            DecryptV1Request_Request.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseDecryptV1Request();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.requests.push(DecryptV1Request_Request.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            requests: globalThis.Array.isArray(object?.requests)
                ? object.requests.map((e) => DecryptV1Request_Request.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.requests?.length) {
            obj.requests = message.requests.map((e) => DecryptV1Request_Request.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return DecryptV1Request.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseDecryptV1Request();
        message.requests = object.requests?.map((e) => DecryptV1Request_Request.fromPartial(e)) || [];
        return message;
    },
};
function createBaseDecryptV1Request_Request() {
    return { payload: undefined, peerKeys: undefined, headerBytes: new Uint8Array(0), isSender: false };
}
export const DecryptV1Request_Request = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.payload !== undefined) {
            Ciphertext.encode(message.payload, writer.uint32(10).fork()).ldelim();
        }
        if (message.peerKeys !== undefined) {
            PublicKeyBundle.encode(message.peerKeys, writer.uint32(18).fork()).ldelim();
        }
        if (message.headerBytes.length !== 0) {
            writer.uint32(26).bytes(message.headerBytes);
        }
        if (message.isSender !== false) {
            writer.uint32(32).bool(message.isSender);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseDecryptV1Request_Request();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.payload = Ciphertext.decode(reader, reader.uint32());
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.peerKeys = PublicKeyBundle.decode(reader, reader.uint32());
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.headerBytes = reader.bytes();
                    continue;
                case 4:
                    if (tag !== 32) {
                        break;
                    }
                    message.isSender = reader.bool();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            payload: isSet(object.payload) ? Ciphertext.fromJSON(object.payload) : undefined,
            peerKeys: isSet(object.peerKeys) ? PublicKeyBundle.fromJSON(object.peerKeys) : undefined,
            headerBytes: isSet(object.headerBytes) ? bytesFromBase64(object.headerBytes) : new Uint8Array(0),
            isSender: isSet(object.isSender) ? globalThis.Boolean(object.isSender) : false,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.payload !== undefined) {
            obj.payload = Ciphertext.toJSON(message.payload);
        }
        if (message.peerKeys !== undefined) {
            obj.peerKeys = PublicKeyBundle.toJSON(message.peerKeys);
        }
        if (message.headerBytes.length !== 0) {
            obj.headerBytes = base64FromBytes(message.headerBytes);
        }
        if (message.isSender !== false) {
            obj.isSender = message.isSender;
        }
        return obj;
    },
    create(base) {
        return DecryptV1Request_Request.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseDecryptV1Request_Request();
        message.payload = (object.payload !== undefined && object.payload !== null)
            ? Ciphertext.fromPartial(object.payload)
            : undefined;
        message.peerKeys = (object.peerKeys !== undefined && object.peerKeys !== null)
            ? PublicKeyBundle.fromPartial(object.peerKeys)
            : undefined;
        message.headerBytes = object.headerBytes ?? new Uint8Array(0);
        message.isSender = object.isSender ?? false;
        return message;
    },
};
function createBaseDecryptResponse() {
    return { responses: [] };
}
export const DecryptResponse = {
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.responses) {
            DecryptResponse_Response.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseDecryptResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.responses.push(DecryptResponse_Response.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            responses: globalThis.Array.isArray(object?.responses)
                ? object.responses.map((e) => DecryptResponse_Response.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.responses?.length) {
            obj.responses = message.responses.map((e) => DecryptResponse_Response.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return DecryptResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseDecryptResponse();
        message.responses = object.responses?.map((e) => DecryptResponse_Response.fromPartial(e)) || [];
        return message;
    },
};
function createBaseDecryptResponse_Response() {
    return { result: undefined, error: undefined };
}
export const DecryptResponse_Response = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.result !== undefined) {
            DecryptResponse_Response_Success.encode(message.result, writer.uint32(10).fork()).ldelim();
        }
        if (message.error !== undefined) {
            KeystoreError.encode(message.error, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseDecryptResponse_Response();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.result = DecryptResponse_Response_Success.decode(reader, reader.uint32());
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.error = KeystoreError.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            result: isSet(object.result) ? DecryptResponse_Response_Success.fromJSON(object.result) : undefined,
            error: isSet(object.error) ? KeystoreError.fromJSON(object.error) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.result !== undefined) {
            obj.result = DecryptResponse_Response_Success.toJSON(message.result);
        }
        if (message.error !== undefined) {
            obj.error = KeystoreError.toJSON(message.error);
        }
        return obj;
    },
    create(base) {
        return DecryptResponse_Response.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseDecryptResponse_Response();
        message.result = (object.result !== undefined && object.result !== null)
            ? DecryptResponse_Response_Success.fromPartial(object.result)
            : undefined;
        message.error = (object.error !== undefined && object.error !== null)
            ? KeystoreError.fromPartial(object.error)
            : undefined;
        return message;
    },
};
function createBaseDecryptResponse_Response_Success() {
    return { decrypted: new Uint8Array(0) };
}
export const DecryptResponse_Response_Success = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.decrypted.length !== 0) {
            writer.uint32(10).bytes(message.decrypted);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseDecryptResponse_Response_Success();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.decrypted = reader.bytes();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { decrypted: isSet(object.decrypted) ? bytesFromBase64(object.decrypted) : new Uint8Array(0) };
    },
    toJSON(message) {
        const obj = {};
        if (message.decrypted.length !== 0) {
            obj.decrypted = base64FromBytes(message.decrypted);
        }
        return obj;
    },
    create(base) {
        return DecryptResponse_Response_Success.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseDecryptResponse_Response_Success();
        message.decrypted = object.decrypted ?? new Uint8Array(0);
        return message;
    },
};
function createBaseDecryptV2Request() {
    return { requests: [] };
}
export const DecryptV2Request = {
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.requests) {
            DecryptV2Request_Request.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseDecryptV2Request();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.requests.push(DecryptV2Request_Request.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            requests: globalThis.Array.isArray(object?.requests)
                ? object.requests.map((e) => DecryptV2Request_Request.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.requests?.length) {
            obj.requests = message.requests.map((e) => DecryptV2Request_Request.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return DecryptV2Request.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseDecryptV2Request();
        message.requests = object.requests?.map((e) => DecryptV2Request_Request.fromPartial(e)) || [];
        return message;
    },
};
function createBaseDecryptV2Request_Request() {
    return { payload: undefined, headerBytes: new Uint8Array(0), contentTopic: "" };
}
export const DecryptV2Request_Request = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.payload !== undefined) {
            Ciphertext.encode(message.payload, writer.uint32(10).fork()).ldelim();
        }
        if (message.headerBytes.length !== 0) {
            writer.uint32(18).bytes(message.headerBytes);
        }
        if (message.contentTopic !== "") {
            writer.uint32(26).string(message.contentTopic);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseDecryptV2Request_Request();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.payload = Ciphertext.decode(reader, reader.uint32());
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.headerBytes = reader.bytes();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.contentTopic = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            payload: isSet(object.payload) ? Ciphertext.fromJSON(object.payload) : undefined,
            headerBytes: isSet(object.headerBytes) ? bytesFromBase64(object.headerBytes) : new Uint8Array(0),
            contentTopic: isSet(object.contentTopic) ? globalThis.String(object.contentTopic) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.payload !== undefined) {
            obj.payload = Ciphertext.toJSON(message.payload);
        }
        if (message.headerBytes.length !== 0) {
            obj.headerBytes = base64FromBytes(message.headerBytes);
        }
        if (message.contentTopic !== "") {
            obj.contentTopic = message.contentTopic;
        }
        return obj;
    },
    create(base) {
        return DecryptV2Request_Request.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseDecryptV2Request_Request();
        message.payload = (object.payload !== undefined && object.payload !== null)
            ? Ciphertext.fromPartial(object.payload)
            : undefined;
        message.headerBytes = object.headerBytes ?? new Uint8Array(0);
        message.contentTopic = object.contentTopic ?? "";
        return message;
    },
};
function createBaseEncryptV1Request() {
    return { requests: [] };
}
export const EncryptV1Request = {
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.requests) {
            EncryptV1Request_Request.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseEncryptV1Request();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.requests.push(EncryptV1Request_Request.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            requests: globalThis.Array.isArray(object?.requests)
                ? object.requests.map((e) => EncryptV1Request_Request.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.requests?.length) {
            obj.requests = message.requests.map((e) => EncryptV1Request_Request.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return EncryptV1Request.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseEncryptV1Request();
        message.requests = object.requests?.map((e) => EncryptV1Request_Request.fromPartial(e)) || [];
        return message;
    },
};
function createBaseEncryptV1Request_Request() {
    return { recipient: undefined, payload: new Uint8Array(0), headerBytes: new Uint8Array(0) };
}
export const EncryptV1Request_Request = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.recipient !== undefined) {
            PublicKeyBundle.encode(message.recipient, writer.uint32(10).fork()).ldelim();
        }
        if (message.payload.length !== 0) {
            writer.uint32(18).bytes(message.payload);
        }
        if (message.headerBytes.length !== 0) {
            writer.uint32(26).bytes(message.headerBytes);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseEncryptV1Request_Request();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.recipient = PublicKeyBundle.decode(reader, reader.uint32());
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.payload = reader.bytes();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.headerBytes = reader.bytes();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            recipient: isSet(object.recipient) ? PublicKeyBundle.fromJSON(object.recipient) : undefined,
            payload: isSet(object.payload) ? bytesFromBase64(object.payload) : new Uint8Array(0),
            headerBytes: isSet(object.headerBytes) ? bytesFromBase64(object.headerBytes) : new Uint8Array(0),
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.recipient !== undefined) {
            obj.recipient = PublicKeyBundle.toJSON(message.recipient);
        }
        if (message.payload.length !== 0) {
            obj.payload = base64FromBytes(message.payload);
        }
        if (message.headerBytes.length !== 0) {
            obj.headerBytes = base64FromBytes(message.headerBytes);
        }
        return obj;
    },
    create(base) {
        return EncryptV1Request_Request.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseEncryptV1Request_Request();
        message.recipient = (object.recipient !== undefined && object.recipient !== null)
            ? PublicKeyBundle.fromPartial(object.recipient)
            : undefined;
        message.payload = object.payload ?? new Uint8Array(0);
        message.headerBytes = object.headerBytes ?? new Uint8Array(0);
        return message;
    },
};
function createBaseEncryptResponse() {
    return { responses: [] };
}
export const EncryptResponse = {
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.responses) {
            EncryptResponse_Response.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseEncryptResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.responses.push(EncryptResponse_Response.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            responses: globalThis.Array.isArray(object?.responses)
                ? object.responses.map((e) => EncryptResponse_Response.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.responses?.length) {
            obj.responses = message.responses.map((e) => EncryptResponse_Response.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return EncryptResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseEncryptResponse();
        message.responses = object.responses?.map((e) => EncryptResponse_Response.fromPartial(e)) || [];
        return message;
    },
};
function createBaseEncryptResponse_Response() {
    return { result: undefined, error: undefined };
}
export const EncryptResponse_Response = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.result !== undefined) {
            EncryptResponse_Response_Success.encode(message.result, writer.uint32(10).fork()).ldelim();
        }
        if (message.error !== undefined) {
            KeystoreError.encode(message.error, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseEncryptResponse_Response();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.result = EncryptResponse_Response_Success.decode(reader, reader.uint32());
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.error = KeystoreError.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            result: isSet(object.result) ? EncryptResponse_Response_Success.fromJSON(object.result) : undefined,
            error: isSet(object.error) ? KeystoreError.fromJSON(object.error) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.result !== undefined) {
            obj.result = EncryptResponse_Response_Success.toJSON(message.result);
        }
        if (message.error !== undefined) {
            obj.error = KeystoreError.toJSON(message.error);
        }
        return obj;
    },
    create(base) {
        return EncryptResponse_Response.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseEncryptResponse_Response();
        message.result = (object.result !== undefined && object.result !== null)
            ? EncryptResponse_Response_Success.fromPartial(object.result)
            : undefined;
        message.error = (object.error !== undefined && object.error !== null)
            ? KeystoreError.fromPartial(object.error)
            : undefined;
        return message;
    },
};
function createBaseEncryptResponse_Response_Success() {
    return { encrypted: undefined, senderHmac: new Uint8Array(0) };
}
export const EncryptResponse_Response_Success = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.encrypted !== undefined) {
            Ciphertext.encode(message.encrypted, writer.uint32(10).fork()).ldelim();
        }
        if (message.senderHmac.length !== 0) {
            writer.uint32(18).bytes(message.senderHmac);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseEncryptResponse_Response_Success();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.encrypted = Ciphertext.decode(reader, reader.uint32());
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.senderHmac = reader.bytes();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            encrypted: isSet(object.encrypted) ? Ciphertext.fromJSON(object.encrypted) : undefined,
            senderHmac: isSet(object.senderHmac) ? bytesFromBase64(object.senderHmac) : new Uint8Array(0),
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.encrypted !== undefined) {
            obj.encrypted = Ciphertext.toJSON(message.encrypted);
        }
        if (message.senderHmac.length !== 0) {
            obj.senderHmac = base64FromBytes(message.senderHmac);
        }
        return obj;
    },
    create(base) {
        return EncryptResponse_Response_Success.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseEncryptResponse_Response_Success();
        message.encrypted = (object.encrypted !== undefined && object.encrypted !== null)
            ? Ciphertext.fromPartial(object.encrypted)
            : undefined;
        message.senderHmac = object.senderHmac ?? new Uint8Array(0);
        return message;
    },
};
function createBaseEncryptV2Request() {
    return { requests: [] };
}
export const EncryptV2Request = {
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.requests) {
            EncryptV2Request_Request.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseEncryptV2Request();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.requests.push(EncryptV2Request_Request.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            requests: globalThis.Array.isArray(object?.requests)
                ? object.requests.map((e) => EncryptV2Request_Request.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.requests?.length) {
            obj.requests = message.requests.map((e) => EncryptV2Request_Request.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return EncryptV2Request.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseEncryptV2Request();
        message.requests = object.requests?.map((e) => EncryptV2Request_Request.fromPartial(e)) || [];
        return message;
    },
};
function createBaseEncryptV2Request_Request() {
    return { payload: new Uint8Array(0), headerBytes: new Uint8Array(0), contentTopic: "" };
}
export const EncryptV2Request_Request = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.payload.length !== 0) {
            writer.uint32(10).bytes(message.payload);
        }
        if (message.headerBytes.length !== 0) {
            writer.uint32(18).bytes(message.headerBytes);
        }
        if (message.contentTopic !== "") {
            writer.uint32(26).string(message.contentTopic);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseEncryptV2Request_Request();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.payload = reader.bytes();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.headerBytes = reader.bytes();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.contentTopic = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            payload: isSet(object.payload) ? bytesFromBase64(object.payload) : new Uint8Array(0),
            headerBytes: isSet(object.headerBytes) ? bytesFromBase64(object.headerBytes) : new Uint8Array(0),
            contentTopic: isSet(object.contentTopic) ? globalThis.String(object.contentTopic) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.payload.length !== 0) {
            obj.payload = base64FromBytes(message.payload);
        }
        if (message.headerBytes.length !== 0) {
            obj.headerBytes = base64FromBytes(message.headerBytes);
        }
        if (message.contentTopic !== "") {
            obj.contentTopic = message.contentTopic;
        }
        return obj;
    },
    create(base) {
        return EncryptV2Request_Request.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseEncryptV2Request_Request();
        message.payload = object.payload ?? new Uint8Array(0);
        message.headerBytes = object.headerBytes ?? new Uint8Array(0);
        message.contentTopic = object.contentTopic ?? "";
        return message;
    },
};
function createBaseSelfEncryptRequest() {
    return { requests: [] };
}
export const SelfEncryptRequest = {
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.requests) {
            SelfEncryptRequest_Request.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSelfEncryptRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.requests.push(SelfEncryptRequest_Request.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            requests: globalThis.Array.isArray(object?.requests)
                ? object.requests.map((e) => SelfEncryptRequest_Request.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.requests?.length) {
            obj.requests = message.requests.map((e) => SelfEncryptRequest_Request.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return SelfEncryptRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseSelfEncryptRequest();
        message.requests = object.requests?.map((e) => SelfEncryptRequest_Request.fromPartial(e)) || [];
        return message;
    },
};
function createBaseSelfEncryptRequest_Request() {
    return { payload: new Uint8Array(0) };
}
export const SelfEncryptRequest_Request = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.payload.length !== 0) {
            writer.uint32(10).bytes(message.payload);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSelfEncryptRequest_Request();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.payload = reader.bytes();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { payload: isSet(object.payload) ? bytesFromBase64(object.payload) : new Uint8Array(0) };
    },
    toJSON(message) {
        const obj = {};
        if (message.payload.length !== 0) {
            obj.payload = base64FromBytes(message.payload);
        }
        return obj;
    },
    create(base) {
        return SelfEncryptRequest_Request.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseSelfEncryptRequest_Request();
        message.payload = object.payload ?? new Uint8Array(0);
        return message;
    },
};
function createBaseSelfEncryptResponse() {
    return { responses: [] };
}
export const SelfEncryptResponse = {
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.responses) {
            SelfEncryptResponse_Response.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSelfEncryptResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.responses.push(SelfEncryptResponse_Response.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            responses: globalThis.Array.isArray(object?.responses)
                ? object.responses.map((e) => SelfEncryptResponse_Response.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.responses?.length) {
            obj.responses = message.responses.map((e) => SelfEncryptResponse_Response.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return SelfEncryptResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseSelfEncryptResponse();
        message.responses = object.responses?.map((e) => SelfEncryptResponse_Response.fromPartial(e)) || [];
        return message;
    },
};
function createBaseSelfEncryptResponse_Response() {
    return { result: undefined, error: undefined };
}
export const SelfEncryptResponse_Response = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.result !== undefined) {
            SelfEncryptResponse_Response_Success.encode(message.result, writer.uint32(10).fork()).ldelim();
        }
        if (message.error !== undefined) {
            KeystoreError.encode(message.error, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSelfEncryptResponse_Response();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.result = SelfEncryptResponse_Response_Success.decode(reader, reader.uint32());
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.error = KeystoreError.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            result: isSet(object.result) ? SelfEncryptResponse_Response_Success.fromJSON(object.result) : undefined,
            error: isSet(object.error) ? KeystoreError.fromJSON(object.error) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.result !== undefined) {
            obj.result = SelfEncryptResponse_Response_Success.toJSON(message.result);
        }
        if (message.error !== undefined) {
            obj.error = KeystoreError.toJSON(message.error);
        }
        return obj;
    },
    create(base) {
        return SelfEncryptResponse_Response.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseSelfEncryptResponse_Response();
        message.result = (object.result !== undefined && object.result !== null)
            ? SelfEncryptResponse_Response_Success.fromPartial(object.result)
            : undefined;
        message.error = (object.error !== undefined && object.error !== null)
            ? KeystoreError.fromPartial(object.error)
            : undefined;
        return message;
    },
};
function createBaseSelfEncryptResponse_Response_Success() {
    return { encrypted: new Uint8Array(0) };
}
export const SelfEncryptResponse_Response_Success = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.encrypted.length !== 0) {
            writer.uint32(10).bytes(message.encrypted);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSelfEncryptResponse_Response_Success();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.encrypted = reader.bytes();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { encrypted: isSet(object.encrypted) ? bytesFromBase64(object.encrypted) : new Uint8Array(0) };
    },
    toJSON(message) {
        const obj = {};
        if (message.encrypted.length !== 0) {
            obj.encrypted = base64FromBytes(message.encrypted);
        }
        return obj;
    },
    create(base) {
        return SelfEncryptResponse_Response_Success.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseSelfEncryptResponse_Response_Success();
        message.encrypted = object.encrypted ?? new Uint8Array(0);
        return message;
    },
};
function createBaseSelfDecryptRequest() {
    return { requests: [] };
}
export const SelfDecryptRequest = {
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.requests) {
            SelfDecryptRequest_Request.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSelfDecryptRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.requests.push(SelfDecryptRequest_Request.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            requests: globalThis.Array.isArray(object?.requests)
                ? object.requests.map((e) => SelfDecryptRequest_Request.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.requests?.length) {
            obj.requests = message.requests.map((e) => SelfDecryptRequest_Request.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return SelfDecryptRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseSelfDecryptRequest();
        message.requests = object.requests?.map((e) => SelfDecryptRequest_Request.fromPartial(e)) || [];
        return message;
    },
};
function createBaseSelfDecryptRequest_Request() {
    return { payload: new Uint8Array(0) };
}
export const SelfDecryptRequest_Request = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.payload.length !== 0) {
            writer.uint32(10).bytes(message.payload);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSelfDecryptRequest_Request();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.payload = reader.bytes();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { payload: isSet(object.payload) ? bytesFromBase64(object.payload) : new Uint8Array(0) };
    },
    toJSON(message) {
        const obj = {};
        if (message.payload.length !== 0) {
            obj.payload = base64FromBytes(message.payload);
        }
        return obj;
    },
    create(base) {
        return SelfDecryptRequest_Request.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseSelfDecryptRequest_Request();
        message.payload = object.payload ?? new Uint8Array(0);
        return message;
    },
};
function createBaseGetPrivatePreferencesTopicIdentifierResponse() {
    return { identifier: "" };
}
export const GetPrivatePreferencesTopicIdentifierResponse = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.identifier !== "") {
            writer.uint32(10).string(message.identifier);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetPrivatePreferencesTopicIdentifierResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.identifier = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { identifier: isSet(object.identifier) ? globalThis.String(object.identifier) : "" };
    },
    toJSON(message) {
        const obj = {};
        if (message.identifier !== "") {
            obj.identifier = message.identifier;
        }
        return obj;
    },
    create(base) {
        return GetPrivatePreferencesTopicIdentifierResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGetPrivatePreferencesTopicIdentifierResponse();
        message.identifier = object.identifier ?? "";
        return message;
    },
};
function createBaseCreateInviteRequest() {
    return { context: undefined, recipient: undefined, createdNs: Long.UZERO, consentProof: undefined };
}
export const CreateInviteRequest = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.context !== undefined) {
            InvitationV1_Context.encode(message.context, writer.uint32(10).fork()).ldelim();
        }
        if (message.recipient !== undefined) {
            SignedPublicKeyBundle.encode(message.recipient, writer.uint32(18).fork()).ldelim();
        }
        if (!message.createdNs.equals(Long.UZERO)) {
            writer.uint32(24).uint64(message.createdNs);
        }
        if (message.consentProof !== undefined) {
            ConsentProofPayload.encode(message.consentProof, writer.uint32(34).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseCreateInviteRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.context = InvitationV1_Context.decode(reader, reader.uint32());
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.recipient = SignedPublicKeyBundle.decode(reader, reader.uint32());
                    continue;
                case 3:
                    if (tag !== 24) {
                        break;
                    }
                    message.createdNs = reader.uint64();
                    continue;
                case 4:
                    if (tag !== 34) {
                        break;
                    }
                    message.consentProof = ConsentProofPayload.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            context: isSet(object.context) ? InvitationV1_Context.fromJSON(object.context) : undefined,
            recipient: isSet(object.recipient) ? SignedPublicKeyBundle.fromJSON(object.recipient) : undefined,
            createdNs: isSet(object.createdNs) ? Long.fromValue(object.createdNs) : Long.UZERO,
            consentProof: isSet(object.consentProof) ? ConsentProofPayload.fromJSON(object.consentProof) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.context !== undefined) {
            obj.context = InvitationV1_Context.toJSON(message.context);
        }
        if (message.recipient !== undefined) {
            obj.recipient = SignedPublicKeyBundle.toJSON(message.recipient);
        }
        if (!message.createdNs.equals(Long.UZERO)) {
            obj.createdNs = (message.createdNs || Long.UZERO).toString();
        }
        if (message.consentProof !== undefined) {
            obj.consentProof = ConsentProofPayload.toJSON(message.consentProof);
        }
        return obj;
    },
    create(base) {
        return CreateInviteRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseCreateInviteRequest();
        message.context = (object.context !== undefined && object.context !== null)
            ? InvitationV1_Context.fromPartial(object.context)
            : undefined;
        message.recipient = (object.recipient !== undefined && object.recipient !== null)
            ? SignedPublicKeyBundle.fromPartial(object.recipient)
            : undefined;
        message.createdNs = (object.createdNs !== undefined && object.createdNs !== null)
            ? Long.fromValue(object.createdNs)
            : Long.UZERO;
        message.consentProof = (object.consentProof !== undefined && object.consentProof !== null)
            ? ConsentProofPayload.fromPartial(object.consentProof)
            : undefined;
        return message;
    },
};
function createBaseCreateInviteResponse() {
    return { conversation: undefined, payload: new Uint8Array(0) };
}
export const CreateInviteResponse = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.conversation !== undefined) {
            ConversationReference.encode(message.conversation, writer.uint32(10).fork()).ldelim();
        }
        if (message.payload.length !== 0) {
            writer.uint32(18).bytes(message.payload);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseCreateInviteResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.conversation = ConversationReference.decode(reader, reader.uint32());
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.payload = reader.bytes();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            conversation: isSet(object.conversation) ? ConversationReference.fromJSON(object.conversation) : undefined,
            payload: isSet(object.payload) ? bytesFromBase64(object.payload) : new Uint8Array(0),
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.conversation !== undefined) {
            obj.conversation = ConversationReference.toJSON(message.conversation);
        }
        if (message.payload.length !== 0) {
            obj.payload = base64FromBytes(message.payload);
        }
        return obj;
    },
    create(base) {
        return CreateInviteResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseCreateInviteResponse();
        message.conversation = (object.conversation !== undefined && object.conversation !== null)
            ? ConversationReference.fromPartial(object.conversation)
            : undefined;
        message.payload = object.payload ?? new Uint8Array(0);
        return message;
    },
};
function createBaseSaveInvitesRequest() {
    return { requests: [] };
}
export const SaveInvitesRequest = {
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.requests) {
            SaveInvitesRequest_Request.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSaveInvitesRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.requests.push(SaveInvitesRequest_Request.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            requests: globalThis.Array.isArray(object?.requests)
                ? object.requests.map((e) => SaveInvitesRequest_Request.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.requests?.length) {
            obj.requests = message.requests.map((e) => SaveInvitesRequest_Request.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return SaveInvitesRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseSaveInvitesRequest();
        message.requests = object.requests?.map((e) => SaveInvitesRequest_Request.fromPartial(e)) || [];
        return message;
    },
};
function createBaseSaveInvitesRequest_Request() {
    return { contentTopic: "", timestampNs: Long.UZERO, payload: new Uint8Array(0) };
}
export const SaveInvitesRequest_Request = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.contentTopic !== "") {
            writer.uint32(10).string(message.contentTopic);
        }
        if (!message.timestampNs.equals(Long.UZERO)) {
            writer.uint32(16).uint64(message.timestampNs);
        }
        if (message.payload.length !== 0) {
            writer.uint32(26).bytes(message.payload);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSaveInvitesRequest_Request();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.contentTopic = reader.string();
                    continue;
                case 2:
                    if (tag !== 16) {
                        break;
                    }
                    message.timestampNs = reader.uint64();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.payload = reader.bytes();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            contentTopic: isSet(object.contentTopic) ? globalThis.String(object.contentTopic) : "",
            timestampNs: isSet(object.timestampNs) ? Long.fromValue(object.timestampNs) : Long.UZERO,
            payload: isSet(object.payload) ? bytesFromBase64(object.payload) : new Uint8Array(0),
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.contentTopic !== "") {
            obj.contentTopic = message.contentTopic;
        }
        if (!message.timestampNs.equals(Long.UZERO)) {
            obj.timestampNs = (message.timestampNs || Long.UZERO).toString();
        }
        if (message.payload.length !== 0) {
            obj.payload = base64FromBytes(message.payload);
        }
        return obj;
    },
    create(base) {
        return SaveInvitesRequest_Request.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseSaveInvitesRequest_Request();
        message.contentTopic = object.contentTopic ?? "";
        message.timestampNs = (object.timestampNs !== undefined && object.timestampNs !== null)
            ? Long.fromValue(object.timestampNs)
            : Long.UZERO;
        message.payload = object.payload ?? new Uint8Array(0);
        return message;
    },
};
function createBaseSaveInvitesResponse() {
    return { responses: [] };
}
export const SaveInvitesResponse = {
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.responses) {
            SaveInvitesResponse_Response.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSaveInvitesResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.responses.push(SaveInvitesResponse_Response.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            responses: globalThis.Array.isArray(object?.responses)
                ? object.responses.map((e) => SaveInvitesResponse_Response.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.responses?.length) {
            obj.responses = message.responses.map((e) => SaveInvitesResponse_Response.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return SaveInvitesResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseSaveInvitesResponse();
        message.responses = object.responses?.map((e) => SaveInvitesResponse_Response.fromPartial(e)) || [];
        return message;
    },
};
function createBaseSaveInvitesResponse_Response() {
    return { result: undefined, error: undefined };
}
export const SaveInvitesResponse_Response = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.result !== undefined) {
            SaveInvitesResponse_Response_Success.encode(message.result, writer.uint32(10).fork()).ldelim();
        }
        if (message.error !== undefined) {
            KeystoreError.encode(message.error, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSaveInvitesResponse_Response();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.result = SaveInvitesResponse_Response_Success.decode(reader, reader.uint32());
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.error = KeystoreError.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            result: isSet(object.result) ? SaveInvitesResponse_Response_Success.fromJSON(object.result) : undefined,
            error: isSet(object.error) ? KeystoreError.fromJSON(object.error) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.result !== undefined) {
            obj.result = SaveInvitesResponse_Response_Success.toJSON(message.result);
        }
        if (message.error !== undefined) {
            obj.error = KeystoreError.toJSON(message.error);
        }
        return obj;
    },
    create(base) {
        return SaveInvitesResponse_Response.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseSaveInvitesResponse_Response();
        message.result = (object.result !== undefined && object.result !== null)
            ? SaveInvitesResponse_Response_Success.fromPartial(object.result)
            : undefined;
        message.error = (object.error !== undefined && object.error !== null)
            ? KeystoreError.fromPartial(object.error)
            : undefined;
        return message;
    },
};
function createBaseSaveInvitesResponse_Response_Success() {
    return { conversation: undefined };
}
export const SaveInvitesResponse_Response_Success = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.conversation !== undefined) {
            ConversationReference.encode(message.conversation, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSaveInvitesResponse_Response_Success();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.conversation = ConversationReference.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            conversation: isSet(object.conversation) ? ConversationReference.fromJSON(object.conversation) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.conversation !== undefined) {
            obj.conversation = ConversationReference.toJSON(message.conversation);
        }
        return obj;
    },
    create(base) {
        return SaveInvitesResponse_Response_Success.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseSaveInvitesResponse_Response_Success();
        message.conversation = (object.conversation !== undefined && object.conversation !== null)
            ? ConversationReference.fromPartial(object.conversation)
            : undefined;
        return message;
    },
};
function createBaseCreateAuthTokenRequest() {
    return { timestampNs: undefined };
}
export const CreateAuthTokenRequest = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.timestampNs !== undefined) {
            writer.uint32(8).uint64(message.timestampNs);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseCreateAuthTokenRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 8) {
                        break;
                    }
                    message.timestampNs = reader.uint64();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { timestampNs: isSet(object.timestampNs) ? Long.fromValue(object.timestampNs) : undefined };
    },
    toJSON(message) {
        const obj = {};
        if (message.timestampNs !== undefined) {
            obj.timestampNs = (message.timestampNs || Long.UZERO).toString();
        }
        return obj;
    },
    create(base) {
        return CreateAuthTokenRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseCreateAuthTokenRequest();
        message.timestampNs = (object.timestampNs !== undefined && object.timestampNs !== null)
            ? Long.fromValue(object.timestampNs)
            : undefined;
        return message;
    },
};
function createBaseSaveV1ConversationsRequest() {
    return { conversations: [] };
}
export const SaveV1ConversationsRequest = {
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.conversations) {
            ConversationReference.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSaveV1ConversationsRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.conversations.push(ConversationReference.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            conversations: globalThis.Array.isArray(object?.conversations)
                ? object.conversations.map((e) => ConversationReference.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.conversations?.length) {
            obj.conversations = message.conversations.map((e) => ConversationReference.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return SaveV1ConversationsRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseSaveV1ConversationsRequest();
        message.conversations = object.conversations?.map((e) => ConversationReference.fromPartial(e)) || [];
        return message;
    },
};
function createBaseSaveV1ConversationsResponse() {
    return {};
}
export const SaveV1ConversationsResponse = {
    encode(_, writer = _m0.Writer.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSaveV1ConversationsResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return SaveV1ConversationsResponse.fromPartial(base ?? {});
    },
    fromPartial(_) {
        const message = createBaseSaveV1ConversationsResponse();
        return message;
    },
};
function createBaseGetConversationsResponse() {
    return { conversations: [] };
}
export const GetConversationsResponse = {
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.conversations) {
            ConversationReference.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetConversationsResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.conversations.push(ConversationReference.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            conversations: globalThis.Array.isArray(object?.conversations)
                ? object.conversations.map((e) => ConversationReference.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.conversations?.length) {
            obj.conversations = message.conversations.map((e) => ConversationReference.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return GetConversationsResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGetConversationsResponse();
        message.conversations = object.conversations?.map((e) => ConversationReference.fromPartial(e)) || [];
        return message;
    },
};
function createBaseGetKeystoreStatusRequest() {
    return { walletAddress: "" };
}
export const GetKeystoreStatusRequest = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.walletAddress !== "") {
            writer.uint32(10).string(message.walletAddress);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetKeystoreStatusRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.walletAddress = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { walletAddress: isSet(object.walletAddress) ? globalThis.String(object.walletAddress) : "" };
    },
    toJSON(message) {
        const obj = {};
        if (message.walletAddress !== "") {
            obj.walletAddress = message.walletAddress;
        }
        return obj;
    },
    create(base) {
        return GetKeystoreStatusRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGetKeystoreStatusRequest();
        message.walletAddress = object.walletAddress ?? "";
        return message;
    },
};
function createBaseGetKeystoreStatusResponse() {
    return { status: 0 };
}
export const GetKeystoreStatusResponse = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.status !== 0) {
            writer.uint32(8).int32(message.status);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetKeystoreStatusResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 8) {
                        break;
                    }
                    message.status = reader.int32();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { status: isSet(object.status) ? getKeystoreStatusResponse_KeystoreStatusFromJSON(object.status) : 0 };
    },
    toJSON(message) {
        const obj = {};
        if (message.status !== 0) {
            obj.status = getKeystoreStatusResponse_KeystoreStatusToJSON(message.status);
        }
        return obj;
    },
    create(base) {
        return GetKeystoreStatusResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGetKeystoreStatusResponse();
        message.status = object.status ?? 0;
        return message;
    },
};
function createBaseInitKeystoreRequest() {
    return { v1: undefined };
}
export const InitKeystoreRequest = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.v1 !== undefined) {
            PrivateKeyBundleV1.encode(message.v1, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseInitKeystoreRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.v1 = PrivateKeyBundleV1.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { v1: isSet(object.v1) ? PrivateKeyBundleV1.fromJSON(object.v1) : undefined };
    },
    toJSON(message) {
        const obj = {};
        if (message.v1 !== undefined) {
            obj.v1 = PrivateKeyBundleV1.toJSON(message.v1);
        }
        return obj;
    },
    create(base) {
        return InitKeystoreRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseInitKeystoreRequest();
        message.v1 = (object.v1 !== undefined && object.v1 !== null)
            ? PrivateKeyBundleV1.fromPartial(object.v1)
            : undefined;
        return message;
    },
};
function createBaseInitKeystoreResponse() {
    return { error: undefined };
}
export const InitKeystoreResponse = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.error !== undefined) {
            KeystoreError.encode(message.error, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseInitKeystoreResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.error = KeystoreError.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { error: isSet(object.error) ? KeystoreError.fromJSON(object.error) : undefined };
    },
    toJSON(message) {
        const obj = {};
        if (message.error !== undefined) {
            obj.error = KeystoreError.toJSON(message.error);
        }
        return obj;
    },
    create(base) {
        return InitKeystoreResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseInitKeystoreResponse();
        message.error = (object.error !== undefined && object.error !== null)
            ? KeystoreError.fromPartial(object.error)
            : undefined;
        return message;
    },
};
function createBaseSignDigestRequest() {
    return { digest: new Uint8Array(0), identityKey: undefined, prekeyIndex: undefined };
}
export const SignDigestRequest = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.digest.length !== 0) {
            writer.uint32(10).bytes(message.digest);
        }
        if (message.identityKey !== undefined) {
            writer.uint32(16).bool(message.identityKey);
        }
        if (message.prekeyIndex !== undefined) {
            writer.uint32(24).uint32(message.prekeyIndex);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSignDigestRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.digest = reader.bytes();
                    continue;
                case 2:
                    if (tag !== 16) {
                        break;
                    }
                    message.identityKey = reader.bool();
                    continue;
                case 3:
                    if (tag !== 24) {
                        break;
                    }
                    message.prekeyIndex = reader.uint32();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            digest: isSet(object.digest) ? bytesFromBase64(object.digest) : new Uint8Array(0),
            identityKey: isSet(object.identityKey) ? globalThis.Boolean(object.identityKey) : undefined,
            prekeyIndex: isSet(object.prekeyIndex) ? globalThis.Number(object.prekeyIndex) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.digest.length !== 0) {
            obj.digest = base64FromBytes(message.digest);
        }
        if (message.identityKey !== undefined) {
            obj.identityKey = message.identityKey;
        }
        if (message.prekeyIndex !== undefined) {
            obj.prekeyIndex = Math.round(message.prekeyIndex);
        }
        return obj;
    },
    create(base) {
        return SignDigestRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseSignDigestRequest();
        message.digest = object.digest ?? new Uint8Array(0);
        message.identityKey = object.identityKey ?? undefined;
        message.prekeyIndex = object.prekeyIndex ?? undefined;
        return message;
    },
};
function createBaseGetRefreshJobRequest() {
    return { jobType: 0 };
}
export const GetRefreshJobRequest = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.jobType !== 0) {
            writer.uint32(8).int32(message.jobType);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetRefreshJobRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 8) {
                        break;
                    }
                    message.jobType = reader.int32();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { jobType: isSet(object.jobType) ? jobTypeFromJSON(object.jobType) : 0 };
    },
    toJSON(message) {
        const obj = {};
        if (message.jobType !== 0) {
            obj.jobType = jobTypeToJSON(message.jobType);
        }
        return obj;
    },
    create(base) {
        return GetRefreshJobRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGetRefreshJobRequest();
        message.jobType = object.jobType ?? 0;
        return message;
    },
};
function createBaseGetRefreshJobResponse() {
    return { lastRunNs: Long.ZERO };
}
export const GetRefreshJobResponse = {
    encode(message, writer = _m0.Writer.create()) {
        if (!message.lastRunNs.equals(Long.ZERO)) {
            writer.uint32(8).int64(message.lastRunNs);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetRefreshJobResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 8) {
                        break;
                    }
                    message.lastRunNs = reader.int64();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { lastRunNs: isSet(object.lastRunNs) ? Long.fromValue(object.lastRunNs) : Long.ZERO };
    },
    toJSON(message) {
        const obj = {};
        if (!message.lastRunNs.equals(Long.ZERO)) {
            obj.lastRunNs = (message.lastRunNs || Long.ZERO).toString();
        }
        return obj;
    },
    create(base) {
        return GetRefreshJobResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGetRefreshJobResponse();
        message.lastRunNs = (object.lastRunNs !== undefined && object.lastRunNs !== null)
            ? Long.fromValue(object.lastRunNs)
            : Long.ZERO;
        return message;
    },
};
function createBaseSetRefeshJobRequest() {
    return { jobType: 0, lastRunNs: Long.ZERO };
}
export const SetRefeshJobRequest = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.jobType !== 0) {
            writer.uint32(8).int32(message.jobType);
        }
        if (!message.lastRunNs.equals(Long.ZERO)) {
            writer.uint32(16).int64(message.lastRunNs);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSetRefeshJobRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 8) {
                        break;
                    }
                    message.jobType = reader.int32();
                    continue;
                case 2:
                    if (tag !== 16) {
                        break;
                    }
                    message.lastRunNs = reader.int64();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            jobType: isSet(object.jobType) ? jobTypeFromJSON(object.jobType) : 0,
            lastRunNs: isSet(object.lastRunNs) ? Long.fromValue(object.lastRunNs) : Long.ZERO,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.jobType !== 0) {
            obj.jobType = jobTypeToJSON(message.jobType);
        }
        if (!message.lastRunNs.equals(Long.ZERO)) {
            obj.lastRunNs = (message.lastRunNs || Long.ZERO).toString();
        }
        return obj;
    },
    create(base) {
        return SetRefeshJobRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseSetRefeshJobRequest();
        message.jobType = object.jobType ?? 0;
        message.lastRunNs = (object.lastRunNs !== undefined && object.lastRunNs !== null)
            ? Long.fromValue(object.lastRunNs)
            : Long.ZERO;
        return message;
    },
};
function createBaseSetRefreshJobResponse() {
    return {};
}
export const SetRefreshJobResponse = {
    encode(_, writer = _m0.Writer.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSetRefreshJobResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return SetRefreshJobResponse.fromPartial(base ?? {});
    },
    fromPartial(_) {
        const message = createBaseSetRefreshJobResponse();
        return message;
    },
};
function createBaseTopicMap() {
    return { topics: {} };
}
export const TopicMap = {
    encode(message, writer = _m0.Writer.create()) {
        Object.entries(message.topics).forEach(([key, value]) => {
            TopicMap_TopicsEntry.encode({ key: key, value }, writer.uint32(10).fork()).ldelim();
        });
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseTopicMap();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    const entry1 = TopicMap_TopicsEntry.decode(reader, reader.uint32());
                    if (entry1.value !== undefined) {
                        message.topics[entry1.key] = entry1.value;
                    }
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            topics: isObject(object.topics)
                ? Object.entries(object.topics).reduce((acc, [key, value]) => {
                    acc[key] = TopicMap_TopicData.fromJSON(value);
                    return acc;
                }, {})
                : {},
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.topics) {
            const entries = Object.entries(message.topics);
            if (entries.length > 0) {
                obj.topics = {};
                entries.forEach(([k, v]) => {
                    obj.topics[k] = TopicMap_TopicData.toJSON(v);
                });
            }
        }
        return obj;
    },
    create(base) {
        return TopicMap.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseTopicMap();
        message.topics = Object.entries(object.topics ?? {}).reduce((acc, [key, value]) => {
            if (value !== undefined) {
                acc[key] = TopicMap_TopicData.fromPartial(value);
            }
            return acc;
        }, {});
        return message;
    },
};
function createBaseTopicMap_TopicData() {
    return { createdNs: Long.UZERO, peerAddress: "", invitation: undefined };
}
export const TopicMap_TopicData = {
    encode(message, writer = _m0.Writer.create()) {
        if (!message.createdNs.equals(Long.UZERO)) {
            writer.uint32(8).uint64(message.createdNs);
        }
        if (message.peerAddress !== "") {
            writer.uint32(18).string(message.peerAddress);
        }
        if (message.invitation !== undefined) {
            InvitationV1.encode(message.invitation, writer.uint32(26).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseTopicMap_TopicData();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 8) {
                        break;
                    }
                    message.createdNs = reader.uint64();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.peerAddress = reader.string();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.invitation = InvitationV1.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            createdNs: isSet(object.createdNs) ? Long.fromValue(object.createdNs) : Long.UZERO,
            peerAddress: isSet(object.peerAddress) ? globalThis.String(object.peerAddress) : "",
            invitation: isSet(object.invitation) ? InvitationV1.fromJSON(object.invitation) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (!message.createdNs.equals(Long.UZERO)) {
            obj.createdNs = (message.createdNs || Long.UZERO).toString();
        }
        if (message.peerAddress !== "") {
            obj.peerAddress = message.peerAddress;
        }
        if (message.invitation !== undefined) {
            obj.invitation = InvitationV1.toJSON(message.invitation);
        }
        return obj;
    },
    create(base) {
        return TopicMap_TopicData.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseTopicMap_TopicData();
        message.createdNs = (object.createdNs !== undefined && object.createdNs !== null)
            ? Long.fromValue(object.createdNs)
            : Long.UZERO;
        message.peerAddress = object.peerAddress ?? "";
        message.invitation = (object.invitation !== undefined && object.invitation !== null)
            ? InvitationV1.fromPartial(object.invitation)
            : undefined;
        return message;
    },
};
function createBaseTopicMap_TopicsEntry() {
    return { key: "", value: undefined };
}
export const TopicMap_TopicsEntry = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.key !== "") {
            writer.uint32(10).string(message.key);
        }
        if (message.value !== undefined) {
            TopicMap_TopicData.encode(message.value, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseTopicMap_TopicsEntry();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.key = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.value = TopicMap_TopicData.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            key: isSet(object.key) ? globalThis.String(object.key) : "",
            value: isSet(object.value) ? TopicMap_TopicData.fromJSON(object.value) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.key !== "") {
            obj.key = message.key;
        }
        if (message.value !== undefined) {
            obj.value = TopicMap_TopicData.toJSON(message.value);
        }
        return obj;
    },
    create(base) {
        return TopicMap_TopicsEntry.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseTopicMap_TopicsEntry();
        message.key = object.key ?? "";
        message.value = (object.value !== undefined && object.value !== null)
            ? TopicMap_TopicData.fromPartial(object.value)
            : undefined;
        return message;
    },
};
function createBaseGetConversationHmacKeysRequest() {
    return { topics: [] };
}
export const GetConversationHmacKeysRequest = {
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.topics) {
            writer.uint32(10).string(v);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetConversationHmacKeysRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.topics.push(reader.string());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            topics: globalThis.Array.isArray(object?.topics) ? object.topics.map((e) => globalThis.String(e)) : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.topics?.length) {
            obj.topics = message.topics;
        }
        return obj;
    },
    create(base) {
        return GetConversationHmacKeysRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGetConversationHmacKeysRequest();
        message.topics = object.topics?.map((e) => e) || [];
        return message;
    },
};
function createBaseGetConversationHmacKeysResponse() {
    return { hmacKeys: {} };
}
export const GetConversationHmacKeysResponse = {
    encode(message, writer = _m0.Writer.create()) {
        Object.entries(message.hmacKeys).forEach(([key, value]) => {
            GetConversationHmacKeysResponse_HmacKeysEntry.encode({ key: key, value }, writer.uint32(10).fork())
                .ldelim();
        });
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetConversationHmacKeysResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    const entry1 = GetConversationHmacKeysResponse_HmacKeysEntry.decode(reader, reader.uint32());
                    if (entry1.value !== undefined) {
                        message.hmacKeys[entry1.key] = entry1.value;
                    }
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            hmacKeys: isObject(object.hmacKeys)
                ? Object.entries(object.hmacKeys).reduce((acc, [key, value]) => {
                    acc[key] = GetConversationHmacKeysResponse_HmacKeys.fromJSON(value);
                    return acc;
                }, {})
                : {},
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.hmacKeys) {
            const entries = Object.entries(message.hmacKeys);
            if (entries.length > 0) {
                obj.hmacKeys = {};
                entries.forEach(([k, v]) => {
                    obj.hmacKeys[k] = GetConversationHmacKeysResponse_HmacKeys.toJSON(v);
                });
            }
        }
        return obj;
    },
    create(base) {
        return GetConversationHmacKeysResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGetConversationHmacKeysResponse();
        message.hmacKeys = Object.entries(object.hmacKeys ?? {}).reduce((acc, [key, value]) => {
            if (value !== undefined) {
                acc[key] = GetConversationHmacKeysResponse_HmacKeys.fromPartial(value);
            }
            return acc;
        }, {});
        return message;
    },
};
function createBaseGetConversationHmacKeysResponse_HmacKeyData() {
    return { thirtyDayPeriodsSinceEpoch: 0, hmacKey: new Uint8Array(0) };
}
export const GetConversationHmacKeysResponse_HmacKeyData = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.thirtyDayPeriodsSinceEpoch !== 0) {
            writer.uint32(8).int32(message.thirtyDayPeriodsSinceEpoch);
        }
        if (message.hmacKey.length !== 0) {
            writer.uint32(18).bytes(message.hmacKey);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetConversationHmacKeysResponse_HmacKeyData();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 8) {
                        break;
                    }
                    message.thirtyDayPeriodsSinceEpoch = reader.int32();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.hmacKey = reader.bytes();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            thirtyDayPeriodsSinceEpoch: isSet(object.thirtyDayPeriodsSinceEpoch)
                ? globalThis.Number(object.thirtyDayPeriodsSinceEpoch)
                : 0,
            hmacKey: isSet(object.hmacKey) ? bytesFromBase64(object.hmacKey) : new Uint8Array(0),
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.thirtyDayPeriodsSinceEpoch !== 0) {
            obj.thirtyDayPeriodsSinceEpoch = Math.round(message.thirtyDayPeriodsSinceEpoch);
        }
        if (message.hmacKey.length !== 0) {
            obj.hmacKey = base64FromBytes(message.hmacKey);
        }
        return obj;
    },
    create(base) {
        return GetConversationHmacKeysResponse_HmacKeyData.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGetConversationHmacKeysResponse_HmacKeyData();
        message.thirtyDayPeriodsSinceEpoch = object.thirtyDayPeriodsSinceEpoch ?? 0;
        message.hmacKey = object.hmacKey ?? new Uint8Array(0);
        return message;
    },
};
function createBaseGetConversationHmacKeysResponse_HmacKeys() {
    return { values: [] };
}
export const GetConversationHmacKeysResponse_HmacKeys = {
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.values) {
            GetConversationHmacKeysResponse_HmacKeyData.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetConversationHmacKeysResponse_HmacKeys();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.values.push(GetConversationHmacKeysResponse_HmacKeyData.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            values: globalThis.Array.isArray(object?.values)
                ? object.values.map((e) => GetConversationHmacKeysResponse_HmacKeyData.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.values?.length) {
            obj.values = message.values.map((e) => GetConversationHmacKeysResponse_HmacKeyData.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return GetConversationHmacKeysResponse_HmacKeys.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGetConversationHmacKeysResponse_HmacKeys();
        message.values = object.values?.map((e) => GetConversationHmacKeysResponse_HmacKeyData.fromPartial(e)) || [];
        return message;
    },
};
function createBaseGetConversationHmacKeysResponse_HmacKeysEntry() {
    return { key: "", value: undefined };
}
export const GetConversationHmacKeysResponse_HmacKeysEntry = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.key !== "") {
            writer.uint32(10).string(message.key);
        }
        if (message.value !== undefined) {
            GetConversationHmacKeysResponse_HmacKeys.encode(message.value, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetConversationHmacKeysResponse_HmacKeysEntry();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.key = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.value = GetConversationHmacKeysResponse_HmacKeys.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            key: isSet(object.key) ? globalThis.String(object.key) : "",
            value: isSet(object.value) ? GetConversationHmacKeysResponse_HmacKeys.fromJSON(object.value) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.key !== "") {
            obj.key = message.key;
        }
        if (message.value !== undefined) {
            obj.value = GetConversationHmacKeysResponse_HmacKeys.toJSON(message.value);
        }
        return obj;
    },
    create(base) {
        return GetConversationHmacKeysResponse_HmacKeysEntry.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGetConversationHmacKeysResponse_HmacKeysEntry();
        message.key = object.key ?? "";
        message.value = (object.value !== undefined && object.value !== null)
            ? GetConversationHmacKeysResponse_HmacKeys.fromPartial(object.value)
            : undefined;
        return message;
    },
};
function createBasePrivatePreferencesActionMap() {
    return { actions: {} };
}
export const PrivatePreferencesActionMap = {
    encode(message, writer = _m0.Writer.create()) {
        Object.entries(message.actions).forEach(([key, value]) => {
            PrivatePreferencesActionMap_ActionsEntry.encode({ key: key, value }, writer.uint32(10).fork()).ldelim();
        });
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBasePrivatePreferencesActionMap();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    const entry1 = PrivatePreferencesActionMap_ActionsEntry.decode(reader, reader.uint32());
                    if (entry1.value !== undefined) {
                        message.actions[entry1.key] = entry1.value;
                    }
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            actions: isObject(object.actions)
                ? Object.entries(object.actions).reduce((acc, [key, value]) => {
                    acc[key] = PrivatePreferencesAction.fromJSON(value);
                    return acc;
                }, {})
                : {},
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.actions) {
            const entries = Object.entries(message.actions);
            if (entries.length > 0) {
                obj.actions = {};
                entries.forEach(([k, v]) => {
                    obj.actions[k] = PrivatePreferencesAction.toJSON(v);
                });
            }
        }
        return obj;
    },
    create(base) {
        return PrivatePreferencesActionMap.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBasePrivatePreferencesActionMap();
        message.actions = Object.entries(object.actions ?? {}).reduce((acc, [key, value]) => {
            if (value !== undefined) {
                acc[key] = PrivatePreferencesAction.fromPartial(value);
            }
            return acc;
        }, {});
        return message;
    },
};
function createBasePrivatePreferencesActionMap_ActionsEntry() {
    return { key: "", value: undefined };
}
export const PrivatePreferencesActionMap_ActionsEntry = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.key !== "") {
            writer.uint32(10).string(message.key);
        }
        if (message.value !== undefined) {
            PrivatePreferencesAction.encode(message.value, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBasePrivatePreferencesActionMap_ActionsEntry();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.key = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.value = PrivatePreferencesAction.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            key: isSet(object.key) ? globalThis.String(object.key) : "",
            value: isSet(object.value) ? PrivatePreferencesAction.fromJSON(object.value) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.key !== "") {
            obj.key = message.key;
        }
        if (message.value !== undefined) {
            obj.value = PrivatePreferencesAction.toJSON(message.value);
        }
        return obj;
    },
    create(base) {
        return PrivatePreferencesActionMap_ActionsEntry.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBasePrivatePreferencesActionMap_ActionsEntry();
        message.key = object.key ?? "";
        message.value = (object.value !== undefined && object.value !== null)
            ? PrivatePreferencesAction.fromPartial(object.value)
            : undefined;
        return message;
    },
};
function bytesFromBase64(b64) {
    if (globalThis.Buffer) {
        return Uint8Array.from(globalThis.Buffer.from(b64, "base64"));
    }
    else {
        const bin = globalThis.atob(b64);
        const arr = new Uint8Array(bin.length);
        for (let i = 0; i < bin.length; ++i) {
            arr[i] = bin.charCodeAt(i);
        }
        return arr;
    }
}
function base64FromBytes(arr) {
    if (globalThis.Buffer) {
        return globalThis.Buffer.from(arr).toString("base64");
    }
    else {
        const bin = [];
        arr.forEach((byte) => {
            bin.push(globalThis.String.fromCharCode(byte));
        });
        return globalThis.btoa(bin.join(""));
    }
}
if (_m0.util.Long !== Long) {
    _m0.util.Long = Long;
    _m0.configure();
}
function isObject(value) {
    return typeof value === "object" && value !== null;
}
function isSet(value) {
    return value !== null && value !== undefined;
}
