// @xmtp/proto/ts/dist/esm/identity/api/v1/identity.pb.js
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v1.181.2
//   protoc               v3.21.12
// source: identity/api/v1/identity.proto
/* eslint-disable */
import Long from "long";
import _m0 from "protobufjs/minimal";
import { identifierKindFromJSON, identifierKindToJSON, IdentityUpdate, } from "../../associations/association.pb";
export const protobufPackage = "xmtp.identity.api.v1";
function createBaseVerifySmartContractWalletSignaturesRequest() {
    return { signatures: [] };
}
export const VerifySmartContractWalletSignaturesRequest = {
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.signatures) {
            VerifySmartContractWalletSignatureRequestSignature.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseVerifySmartContractWalletSignaturesRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.signatures.push(VerifySmartContractWalletSignatureRequestSignature.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            signatures: globalThis.Array.isArray(object?.signatures)
                ? object.signatures.map((e) => VerifySmartContractWalletSignatureRequestSignature.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.signatures?.length) {
            obj.signatures = message.signatures.map((e) => VerifySmartContractWalletSignatureRequestSignature.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return VerifySmartContractWalletSignaturesRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseVerifySmartContractWalletSignaturesRequest();
        message.signatures =
            object.signatures?.map((e) => VerifySmartContractWalletSignatureRequestSignature.fromPartial(e)) || [];
        return message;
    },
};
function createBaseVerifySmartContractWalletSignatureRequestSignature() {
    return { accountId: "", blockNumber: undefined, signature: new Uint8Array(0), hash: new Uint8Array(0) };
}
export const VerifySmartContractWalletSignatureRequestSignature = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.accountId !== "") {
            writer.uint32(10).string(message.accountId);
        }
        if (message.blockNumber !== undefined) {
            writer.uint32(16).uint64(message.blockNumber);
        }
        if (message.signature.length !== 0) {
            writer.uint32(26).bytes(message.signature);
        }
        if (message.hash.length !== 0) {
            writer.uint32(34).bytes(message.hash);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseVerifySmartContractWalletSignatureRequestSignature();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.accountId = reader.string();
                    continue;
                case 2:
                    if (tag !== 16) {
                        break;
                    }
                    message.blockNumber = reader.uint64();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.signature = reader.bytes();
                    continue;
                case 4:
                    if (tag !== 34) {
                        break;
                    }
                    message.hash = reader.bytes();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            accountId: isSet(object.accountId) ? globalThis.String(object.accountId) : "",
            blockNumber: isSet(object.blockNumber) ? Long.fromValue(object.blockNumber) : undefined,
            signature: isSet(object.signature) ? bytesFromBase64(object.signature) : new Uint8Array(0),
            hash: isSet(object.hash) ? bytesFromBase64(object.hash) : new Uint8Array(0),
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.accountId !== "") {
            obj.accountId = message.accountId;
        }
        if (message.blockNumber !== undefined) {
            obj.blockNumber = (message.blockNumber || Long.UZERO).toString();
        }
        if (message.signature.length !== 0) {
            obj.signature = base64FromBytes(message.signature);
        }
        if (message.hash.length !== 0) {
            obj.hash = base64FromBytes(message.hash);
        }
        return obj;
    },
    create(base) {
        return VerifySmartContractWalletSignatureRequestSignature.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseVerifySmartContractWalletSignatureRequestSignature();
        message.accountId = object.accountId ?? "";
        message.blockNumber = (object.blockNumber !== undefined && object.blockNumber !== null)
            ? Long.fromValue(object.blockNumber)
            : undefined;
        message.signature = object.signature ?? new Uint8Array(0);
        message.hash = object.hash ?? new Uint8Array(0);
        return message;
    },
};
function createBaseVerifySmartContractWalletSignaturesResponse() {
    return { responses: [] };
}
export const VerifySmartContractWalletSignaturesResponse = {
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.responses) {
            VerifySmartContractWalletSignaturesResponse_ValidationResponse.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseVerifySmartContractWalletSignaturesResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.responses.push(VerifySmartContractWalletSignaturesResponse_ValidationResponse.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            responses: globalThis.Array.isArray(object?.responses)
                ? object.responses.map((e) => VerifySmartContractWalletSignaturesResponse_ValidationResponse.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.responses?.length) {
            obj.responses = message.responses.map((e) => VerifySmartContractWalletSignaturesResponse_ValidationResponse.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return VerifySmartContractWalletSignaturesResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseVerifySmartContractWalletSignaturesResponse();
        message.responses =
            object.responses?.map((e) => VerifySmartContractWalletSignaturesResponse_ValidationResponse.fromPartial(e)) || [];
        return message;
    },
};
function createBaseVerifySmartContractWalletSignaturesResponse_ValidationResponse() {
    return { isValid: false, blockNumber: undefined, error: undefined };
}
export const VerifySmartContractWalletSignaturesResponse_ValidationResponse = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.isValid !== false) {
            writer.uint32(8).bool(message.isValid);
        }
        if (message.blockNumber !== undefined) {
            writer.uint32(16).uint64(message.blockNumber);
        }
        if (message.error !== undefined) {
            writer.uint32(26).string(message.error);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseVerifySmartContractWalletSignaturesResponse_ValidationResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 8) {
                        break;
                    }
                    message.isValid = reader.bool();
                    continue;
                case 2:
                    if (tag !== 16) {
                        break;
                    }
                    message.blockNumber = reader.uint64();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.error = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            isValid: isSet(object.isValid) ? globalThis.Boolean(object.isValid) : false,
            blockNumber: isSet(object.blockNumber) ? Long.fromValue(object.blockNumber) : undefined,
            error: isSet(object.error) ? globalThis.String(object.error) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.isValid !== false) {
            obj.isValid = message.isValid;
        }
        if (message.blockNumber !== undefined) {
            obj.blockNumber = (message.blockNumber || Long.UZERO).toString();
        }
        if (message.error !== undefined) {
            obj.error = message.error;
        }
        return obj;
    },
    create(base) {
        return VerifySmartContractWalletSignaturesResponse_ValidationResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseVerifySmartContractWalletSignaturesResponse_ValidationResponse();
        message.isValid = object.isValid ?? false;
        message.blockNumber = (object.blockNumber !== undefined && object.blockNumber !== null)
            ? Long.fromValue(object.blockNumber)
            : undefined;
        message.error = object.error ?? undefined;
        return message;
    },
};
function createBasePublishIdentityUpdateRequest() {
    return { identityUpdate: undefined };
}
export const PublishIdentityUpdateRequest = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.identityUpdate !== undefined) {
            IdentityUpdate.encode(message.identityUpdate, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBasePublishIdentityUpdateRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.identityUpdate = IdentityUpdate.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            identityUpdate: isSet(object.identityUpdate) ? IdentityUpdate.fromJSON(object.identityUpdate) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.identityUpdate !== undefined) {
            obj.identityUpdate = IdentityUpdate.toJSON(message.identityUpdate);
        }
        return obj;
    },
    create(base) {
        return PublishIdentityUpdateRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBasePublishIdentityUpdateRequest();
        message.identityUpdate = (object.identityUpdate !== undefined && object.identityUpdate !== null)
            ? IdentityUpdate.fromPartial(object.identityUpdate)
            : undefined;
        return message;
    },
};
function createBasePublishIdentityUpdateResponse() {
    return {};
}
export const PublishIdentityUpdateResponse = {
    encode(_, writer = _m0.Writer.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBasePublishIdentityUpdateResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return PublishIdentityUpdateResponse.fromPartial(base ?? {});
    },
    fromPartial(_) {
        const message = createBasePublishIdentityUpdateResponse();
        return message;
    },
};
function createBaseGetIdentityUpdatesRequest() {
    return { requests: [] };
}
export const GetIdentityUpdatesRequest = {
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.requests) {
            GetIdentityUpdatesRequest_Request.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetIdentityUpdatesRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.requests.push(GetIdentityUpdatesRequest_Request.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            requests: globalThis.Array.isArray(object?.requests)
                ? object.requests.map((e) => GetIdentityUpdatesRequest_Request.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.requests?.length) {
            obj.requests = message.requests.map((e) => GetIdentityUpdatesRequest_Request.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return GetIdentityUpdatesRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGetIdentityUpdatesRequest();
        message.requests = object.requests?.map((e) => GetIdentityUpdatesRequest_Request.fromPartial(e)) || [];
        return message;
    },
};
function createBaseGetIdentityUpdatesRequest_Request() {
    return { inboxId: "", sequenceId: Long.UZERO };
}
export const GetIdentityUpdatesRequest_Request = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.inboxId !== "") {
            writer.uint32(10).string(message.inboxId);
        }
        if (!message.sequenceId.equals(Long.UZERO)) {
            writer.uint32(16).uint64(message.sequenceId);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetIdentityUpdatesRequest_Request();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.inboxId = reader.string();
                    continue;
                case 2:
                    if (tag !== 16) {
                        break;
                    }
                    message.sequenceId = reader.uint64();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            inboxId: isSet(object.inboxId) ? globalThis.String(object.inboxId) : "",
            sequenceId: isSet(object.sequenceId) ? Long.fromValue(object.sequenceId) : Long.UZERO,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.inboxId !== "") {
            obj.inboxId = message.inboxId;
        }
        if (!message.sequenceId.equals(Long.UZERO)) {
            obj.sequenceId = (message.sequenceId || Long.UZERO).toString();
        }
        return obj;
    },
    create(base) {
        return GetIdentityUpdatesRequest_Request.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGetIdentityUpdatesRequest_Request();
        message.inboxId = object.inboxId ?? "";
        message.sequenceId = (object.sequenceId !== undefined && object.sequenceId !== null)
            ? Long.fromValue(object.sequenceId)
            : Long.UZERO;
        return message;
    },
};
function createBaseGetIdentityUpdatesResponse() {
    return { responses: [] };
}
export const GetIdentityUpdatesResponse = {
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.responses) {
            GetIdentityUpdatesResponse_Response.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetIdentityUpdatesResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.responses.push(GetIdentityUpdatesResponse_Response.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            responses: globalThis.Array.isArray(object?.responses)
                ? object.responses.map((e) => GetIdentityUpdatesResponse_Response.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.responses?.length) {
            obj.responses = message.responses.map((e) => GetIdentityUpdatesResponse_Response.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return GetIdentityUpdatesResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGetIdentityUpdatesResponse();
        message.responses = object.responses?.map((e) => GetIdentityUpdatesResponse_Response.fromPartial(e)) || [];
        return message;
    },
};
function createBaseGetIdentityUpdatesResponse_IdentityUpdateLog() {
    return { sequenceId: Long.UZERO, serverTimestampNs: Long.UZERO, update: undefined };
}
export const GetIdentityUpdatesResponse_IdentityUpdateLog = {
    encode(message, writer = _m0.Writer.create()) {
        if (!message.sequenceId.equals(Long.UZERO)) {
            writer.uint32(8).uint64(message.sequenceId);
        }
        if (!message.serverTimestampNs.equals(Long.UZERO)) {
            writer.uint32(16).uint64(message.serverTimestampNs);
        }
        if (message.update !== undefined) {
            IdentityUpdate.encode(message.update, writer.uint32(26).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetIdentityUpdatesResponse_IdentityUpdateLog();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 8) {
                        break;
                    }
                    message.sequenceId = reader.uint64();
                    continue;
                case 2:
                    if (tag !== 16) {
                        break;
                    }
                    message.serverTimestampNs = reader.uint64();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.update = IdentityUpdate.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            sequenceId: isSet(object.sequenceId) ? Long.fromValue(object.sequenceId) : Long.UZERO,
            serverTimestampNs: isSet(object.serverTimestampNs) ? Long.fromValue(object.serverTimestampNs) : Long.UZERO,
            update: isSet(object.update) ? IdentityUpdate.fromJSON(object.update) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (!message.sequenceId.equals(Long.UZERO)) {
            obj.sequenceId = (message.sequenceId || Long.UZERO).toString();
        }
        if (!message.serverTimestampNs.equals(Long.UZERO)) {
            obj.serverTimestampNs = (message.serverTimestampNs || Long.UZERO).toString();
        }
        if (message.update !== undefined) {
            obj.update = IdentityUpdate.toJSON(message.update);
        }
        return obj;
    },
    create(base) {
        return GetIdentityUpdatesResponse_IdentityUpdateLog.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGetIdentityUpdatesResponse_IdentityUpdateLog();
        message.sequenceId = (object.sequenceId !== undefined && object.sequenceId !== null)
            ? Long.fromValue(object.sequenceId)
            : Long.UZERO;
        message.serverTimestampNs = (object.serverTimestampNs !== undefined && object.serverTimestampNs !== null)
            ? Long.fromValue(object.serverTimestampNs)
            : Long.UZERO;
        message.update = (object.update !== undefined && object.update !== null)
            ? IdentityUpdate.fromPartial(object.update)
            : undefined;
        return message;
    },
};
function createBaseGetIdentityUpdatesResponse_Response() {
    return { inboxId: "", updates: [] };
}
export const GetIdentityUpdatesResponse_Response = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.inboxId !== "") {
            writer.uint32(10).string(message.inboxId);
        }
        for (const v of message.updates) {
            GetIdentityUpdatesResponse_IdentityUpdateLog.encode(v, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetIdentityUpdatesResponse_Response();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.inboxId = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.updates.push(GetIdentityUpdatesResponse_IdentityUpdateLog.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            inboxId: isSet(object.inboxId) ? globalThis.String(object.inboxId) : "",
            updates: globalThis.Array.isArray(object?.updates)
                ? object.updates.map((e) => GetIdentityUpdatesResponse_IdentityUpdateLog.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.inboxId !== "") {
            obj.inboxId = message.inboxId;
        }
        if (message.updates?.length) {
            obj.updates = message.updates.map((e) => GetIdentityUpdatesResponse_IdentityUpdateLog.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return GetIdentityUpdatesResponse_Response.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGetIdentityUpdatesResponse_Response();
        message.inboxId = object.inboxId ?? "";
        message.updates = object.updates?.map((e) => GetIdentityUpdatesResponse_IdentityUpdateLog.fromPartial(e)) || [];
        return message;
    },
};
function createBaseGetInboxIdsRequest() {
    return { requests: [] };
}
export const GetInboxIdsRequest = {
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.requests) {
            GetInboxIdsRequest_Request.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetInboxIdsRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.requests.push(GetInboxIdsRequest_Request.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            requests: globalThis.Array.isArray(object?.requests)
                ? object.requests.map((e) => GetInboxIdsRequest_Request.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.requests?.length) {
            obj.requests = message.requests.map((e) => GetInboxIdsRequest_Request.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return GetInboxIdsRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGetInboxIdsRequest();
        message.requests = object.requests?.map((e) => GetInboxIdsRequest_Request.fromPartial(e)) || [];
        return message;
    },
};
function createBaseGetInboxIdsRequest_Request() {
    return { identifier: "", identifierKind: 0 };
}
export const GetInboxIdsRequest_Request = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.identifier !== "") {
            writer.uint32(10).string(message.identifier);
        }
        if (message.identifierKind !== 0) {
            writer.uint32(16).int32(message.identifierKind);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetInboxIdsRequest_Request();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.identifier = reader.string();
                    continue;
                case 2:
                    if (tag !== 16) {
                        break;
                    }
                    message.identifierKind = reader.int32();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            identifier: isSet(object.identifier) ? globalThis.String(object.identifier) : "",
            identifierKind: isSet(object.identifierKind) ? identifierKindFromJSON(object.identifierKind) : 0,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.identifier !== "") {
            obj.identifier = message.identifier;
        }
        if (message.identifierKind !== 0) {
            obj.identifierKind = identifierKindToJSON(message.identifierKind);
        }
        return obj;
    },
    create(base) {
        return GetInboxIdsRequest_Request.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGetInboxIdsRequest_Request();
        message.identifier = object.identifier ?? "";
        message.identifierKind = object.identifierKind ?? 0;
        return message;
    },
};
function createBaseGetInboxIdsResponse() {
    return { responses: [] };
}
export const GetInboxIdsResponse = {
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.responses) {
            GetInboxIdsResponse_Response.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetInboxIdsResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.responses.push(GetInboxIdsResponse_Response.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            responses: globalThis.Array.isArray(object?.responses)
                ? object.responses.map((e) => GetInboxIdsResponse_Response.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.responses?.length) {
            obj.responses = message.responses.map((e) => GetInboxIdsResponse_Response.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return GetInboxIdsResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGetInboxIdsResponse();
        message.responses = object.responses?.map((e) => GetInboxIdsResponse_Response.fromPartial(e)) || [];
        return message;
    },
};
function createBaseGetInboxIdsResponse_Response() {
    return { identifier: "", inboxId: undefined, identifierKind: 0 };
}
export const GetInboxIdsResponse_Response = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.identifier !== "") {
            writer.uint32(10).string(message.identifier);
        }
        if (message.inboxId !== undefined) {
            writer.uint32(18).string(message.inboxId);
        }
        if (message.identifierKind !== 0) {
            writer.uint32(24).int32(message.identifierKind);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetInboxIdsResponse_Response();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.identifier = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.inboxId = reader.string();
                    continue;
                case 3:
                    if (tag !== 24) {
                        break;
                    }
                    message.identifierKind = reader.int32();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            identifier: isSet(object.identifier) ? globalThis.String(object.identifier) : "",
            inboxId: isSet(object.inboxId) ? globalThis.String(object.inboxId) : undefined,
            identifierKind: isSet(object.identifierKind) ? identifierKindFromJSON(object.identifierKind) : 0,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.identifier !== "") {
            obj.identifier = message.identifier;
        }
        if (message.inboxId !== undefined) {
            obj.inboxId = message.inboxId;
        }
        if (message.identifierKind !== 0) {
            obj.identifierKind = identifierKindToJSON(message.identifierKind);
        }
        return obj;
    },
    create(base) {
        return GetInboxIdsResponse_Response.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGetInboxIdsResponse_Response();
        message.identifier = object.identifier ?? "";
        message.inboxId = object.inboxId ?? undefined;
        message.identifierKind = object.identifierKind ?? 0;
        return message;
    },
};
export const IdentityApiServiceName = "xmtp.identity.api.v1.IdentityApi";
export class IdentityApiClientImpl {
    rpc;
    service;
    constructor(rpc, opts) {
        this.service = opts?.service || IdentityApiServiceName;
        this.rpc = rpc;
        this.PublishIdentityUpdate = this.PublishIdentityUpdate.bind(this);
        this.GetIdentityUpdates = this.GetIdentityUpdates.bind(this);
        this.GetInboxIds = this.GetInboxIds.bind(this);
        this.VerifySmartContractWalletSignatures = this.VerifySmartContractWalletSignatures.bind(this);
    }
    PublishIdentityUpdate(request) {
        const data = PublishIdentityUpdateRequest.encode(request).finish();
        const promise = this.rpc.request(this.service, "PublishIdentityUpdate", data);
        return promise.then((data) => PublishIdentityUpdateResponse.decode(_m0.Reader.create(data)));
    }
    GetIdentityUpdates(request) {
        const data = GetIdentityUpdatesRequest.encode(request).finish();
        const promise = this.rpc.request(this.service, "GetIdentityUpdates", data);
        return promise.then((data) => GetIdentityUpdatesResponse.decode(_m0.Reader.create(data)));
    }
    GetInboxIds(request) {
        const data = GetInboxIdsRequest.encode(request).finish();
        const promise = this.rpc.request(this.service, "GetInboxIds", data);
        return promise.then((data) => GetInboxIdsResponse.decode(_m0.Reader.create(data)));
    }
    VerifySmartContractWalletSignatures(request) {
        const data = VerifySmartContractWalletSignaturesRequest.encode(request).finish();
        const promise = this.rpc.request(this.service, "VerifySmartContractWalletSignatures", data);
        return promise.then((data) => VerifySmartContractWalletSignaturesResponse.decode(_m0.Reader.create(data)));
    }
}
function bytesFromBase64(b64) {
    if (globalThis.Buffer) {
        return Uint8Array.from(globalThis.Buffer.from(b64, "base64"));
    }
    else {
        const bin = globalThis.atob(b64);
        const arr = new Uint8Array(bin.length);
        for (let i = 0; i < bin.length; ++i) {
            arr[i] = bin.charCodeAt(i);
        }
        return arr;
    }
}
function base64FromBytes(arr) {
    if (globalThis.Buffer) {
        return globalThis.Buffer.from(arr).toString("base64");
    }
    else {
        const bin = [];
        arr.forEach((byte) => {
            bin.push(globalThis.String.fromCharCode(byte));
        });
        return globalThis.btoa(bin.join(""));
    }
}
if (_m0.util.Long !== Long) {
    _m0.util.Long = Long;
    _m0.configure();
}
function isSet(value) {
    return value !== null && value !== undefined;
}
