// @xmtp/proto/ts/dist/esm/mls/api/v1/mls.pb.js
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v1.181.2
//   protoc               v3.21.12
// source: mls/api/v1/mls.proto
/* eslint-disable */
import Long from "long";
import _m0 from "protobufjs/minimal";
import { map } from "rxjs/operators";
import { Empty } from "../../../google/protobuf/empty.pb";
import { RecoverableEd25519Signature } from "../../../identity/associations/signature.pb";
import { Signature } from "../../../message_contents/signature.pb";
import { CommitLogEntry } from "../../message_contents/commit_log.pb";
import { welcomePointerWrapperAlgorithmFromJSON, welcomePointerWrapperAlgorithmToJSON, } from "../../message_contents/welcome_pointer.pb";
import { welcomeWrapperAlgorithmFromJSON, welcomeWrapperAlgorithmToJSON, } from "../../message_contents/wrapper_encryption.pb";
export const protobufPackage = "xmtp.mls.api.v1";
/** Message API */
/** Sort direction for queries */
export var SortDirection;
(function (SortDirection) {
    SortDirection[SortDirection["SORT_DIRECTION_UNSPECIFIED"] = 0] = "SORT_DIRECTION_UNSPECIFIED";
    SortDirection[SortDirection["SORT_DIRECTION_ASCENDING"] = 1] = "SORT_DIRECTION_ASCENDING";
    SortDirection[SortDirection["SORT_DIRECTION_DESCENDING"] = 2] = "SORT_DIRECTION_DESCENDING";
    SortDirection[SortDirection["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(SortDirection || (SortDirection = {}));
export function sortDirectionFromJSON(object) {
    switch (object) {
        case 0:
        case "SORT_DIRECTION_UNSPECIFIED":
            return SortDirection.SORT_DIRECTION_UNSPECIFIED;
        case 1:
        case "SORT_DIRECTION_ASCENDING":
            return SortDirection.SORT_DIRECTION_ASCENDING;
        case 2:
        case "SORT_DIRECTION_DESCENDING":
            return SortDirection.SORT_DIRECTION_DESCENDING;
        case -1:
        case "UNRECOGNIZED":
        default:
            return SortDirection.UNRECOGNIZED;
    }
}
export function sortDirectionToJSON(object) {
    switch (object) {
        case SortDirection.SORT_DIRECTION_UNSPECIFIED:
            return "SORT_DIRECTION_UNSPECIFIED";
        case SortDirection.SORT_DIRECTION_ASCENDING:
            return "SORT_DIRECTION_ASCENDING";
        case SortDirection.SORT_DIRECTION_DESCENDING:
            return "SORT_DIRECTION_DESCENDING";
        case SortDirection.UNRECOGNIZED:
        default:
            return "UNRECOGNIZED";
    }
}
function createBaseWelcomeMessage() {
    return { v1: undefined, welcomePointer: undefined };
}
export const WelcomeMessage = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.v1 !== undefined) {
            WelcomeMessage_V1.encode(message.v1, writer.uint32(10).fork()).ldelim();
        }
        if (message.welcomePointer !== undefined) {
            WelcomeMessage_WelcomePointer.encode(message.welcomePointer, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseWelcomeMessage();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.v1 = WelcomeMessage_V1.decode(reader, reader.uint32());
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.welcomePointer = WelcomeMessage_WelcomePointer.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            v1: isSet(object.v1) ? WelcomeMessage_V1.fromJSON(object.v1) : undefined,
            welcomePointer: isSet(object.welcomePointer)
                ? WelcomeMessage_WelcomePointer.fromJSON(object.welcomePointer)
                : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.v1 !== undefined) {
            obj.v1 = WelcomeMessage_V1.toJSON(message.v1);
        }
        if (message.welcomePointer !== undefined) {
            obj.welcomePointer = WelcomeMessage_WelcomePointer.toJSON(message.welcomePointer);
        }
        return obj;
    },
    create(base) {
        return WelcomeMessage.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseWelcomeMessage();
        message.v1 = (object.v1 !== undefined && object.v1 !== null) ? WelcomeMessage_V1.fromPartial(object.v1) : undefined;
        message.welcomePointer = (object.welcomePointer !== undefined && object.welcomePointer !== null)
            ? WelcomeMessage_WelcomePointer.fromPartial(object.welcomePointer)
            : undefined;
        return message;
    },
};
function createBaseWelcomeMessage_V1() {
    return {
        id: Long.UZERO,
        createdNs: Long.UZERO,
        installationKey: new Uint8Array(0),
        data: new Uint8Array(0),
        hpkePublicKey: new Uint8Array(0),
        wrapperAlgorithm: 0,
        welcomeMetadata: new Uint8Array(0),
    };
}
export const WelcomeMessage_V1 = {
    encode(message, writer = _m0.Writer.create()) {
        if (!message.id.equals(Long.UZERO)) {
            writer.uint32(8).uint64(message.id);
        }
        if (!message.createdNs.equals(Long.UZERO)) {
            writer.uint32(16).uint64(message.createdNs);
        }
        if (message.installationKey.length !== 0) {
            writer.uint32(26).bytes(message.installationKey);
        }
        if (message.data.length !== 0) {
            writer.uint32(34).bytes(message.data);
        }
        if (message.hpkePublicKey.length !== 0) {
            writer.uint32(42).bytes(message.hpkePublicKey);
        }
        if (message.wrapperAlgorithm !== 0) {
            writer.uint32(48).int32(message.wrapperAlgorithm);
        }
        if (message.welcomeMetadata.length !== 0) {
            writer.uint32(58).bytes(message.welcomeMetadata);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseWelcomeMessage_V1();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 8) {
                        break;
                    }
                    message.id = reader.uint64();
                    continue;
                case 2:
                    if (tag !== 16) {
                        break;
                    }
                    message.createdNs = reader.uint64();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.installationKey = reader.bytes();
                    continue;
                case 4:
                    if (tag !== 34) {
                        break;
                    }
                    message.data = reader.bytes();
                    continue;
                case 5:
                    if (tag !== 42) {
                        break;
                    }
                    message.hpkePublicKey = reader.bytes();
                    continue;
                case 6:
                    if (tag !== 48) {
                        break;
                    }
                    message.wrapperAlgorithm = reader.int32();
                    continue;
                case 7:
                    if (tag !== 58) {
                        break;
                    }
                    message.welcomeMetadata = reader.bytes();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            id: isSet(object.id) ? Long.fromValue(object.id) : Long.UZERO,
            createdNs: isSet(object.createdNs) ? Long.fromValue(object.createdNs) : Long.UZERO,
            installationKey: isSet(object.installationKey) ? bytesFromBase64(object.installationKey) : new Uint8Array(0),
            data: isSet(object.data) ? bytesFromBase64(object.data) : new Uint8Array(0),
            hpkePublicKey: isSet(object.hpkePublicKey) ? bytesFromBase64(object.hpkePublicKey) : new Uint8Array(0),
            wrapperAlgorithm: isSet(object.wrapperAlgorithm) ? welcomeWrapperAlgorithmFromJSON(object.wrapperAlgorithm) : 0,
            welcomeMetadata: isSet(object.welcomeMetadata) ? bytesFromBase64(object.welcomeMetadata) : new Uint8Array(0),
        };
    },
    toJSON(message) {
        const obj = {};
        if (!message.id.equals(Long.UZERO)) {
            obj.id = (message.id || Long.UZERO).toString();
        }
        if (!message.createdNs.equals(Long.UZERO)) {
            obj.createdNs = (message.createdNs || Long.UZERO).toString();
        }
        if (message.installationKey.length !== 0) {
            obj.installationKey = base64FromBytes(message.installationKey);
        }
        if (message.data.length !== 0) {
            obj.data = base64FromBytes(message.data);
        }
        if (message.hpkePublicKey.length !== 0) {
            obj.hpkePublicKey = base64FromBytes(message.hpkePublicKey);
        }
        if (message.wrapperAlgorithm !== 0) {
            obj.wrapperAlgorithm = welcomeWrapperAlgorithmToJSON(message.wrapperAlgorithm);
        }
        if (message.welcomeMetadata.length !== 0) {
            obj.welcomeMetadata = base64FromBytes(message.welcomeMetadata);
        }
        return obj;
    },
    create(base) {
        return WelcomeMessage_V1.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseWelcomeMessage_V1();
        message.id = (object.id !== undefined && object.id !== null) ? Long.fromValue(object.id) : Long.UZERO;
        message.createdNs = (object.createdNs !== undefined && object.createdNs !== null)
            ? Long.fromValue(object.createdNs)
            : Long.UZERO;
        message.installationKey = object.installationKey ?? new Uint8Array(0);
        message.data = object.data ?? new Uint8Array(0);
        message.hpkePublicKey = object.hpkePublicKey ?? new Uint8Array(0);
        message.wrapperAlgorithm = object.wrapperAlgorithm ?? 0;
        message.welcomeMetadata = object.welcomeMetadata ?? new Uint8Array(0);
        return message;
    },
};
function createBaseWelcomeMessage_WelcomePointer() {
    return {
        id: Long.UZERO,
        createdNs: Long.UZERO,
        installationKey: new Uint8Array(0),
        welcomePointer: new Uint8Array(0),
        hpkePublicKey: new Uint8Array(0),
        wrapperAlgorithm: 0,
    };
}
export const WelcomeMessage_WelcomePointer = {
    encode(message, writer = _m0.Writer.create()) {
        if (!message.id.equals(Long.UZERO)) {
            writer.uint32(8).uint64(message.id);
        }
        if (!message.createdNs.equals(Long.UZERO)) {
            writer.uint32(16).uint64(message.createdNs);
        }
        if (message.installationKey.length !== 0) {
            writer.uint32(26).bytes(message.installationKey);
        }
        if (message.welcomePointer.length !== 0) {
            writer.uint32(34).bytes(message.welcomePointer);
        }
        if (message.hpkePublicKey.length !== 0) {
            writer.uint32(42).bytes(message.hpkePublicKey);
        }
        if (message.wrapperAlgorithm !== 0) {
            writer.uint32(48).int32(message.wrapperAlgorithm);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseWelcomeMessage_WelcomePointer();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 8) {
                        break;
                    }
                    message.id = reader.uint64();
                    continue;
                case 2:
                    if (tag !== 16) {
                        break;
                    }
                    message.createdNs = reader.uint64();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.installationKey = reader.bytes();
                    continue;
                case 4:
                    if (tag !== 34) {
                        break;
                    }
                    message.welcomePointer = reader.bytes();
                    continue;
                case 5:
                    if (tag !== 42) {
                        break;
                    }
                    message.hpkePublicKey = reader.bytes();
                    continue;
                case 6:
                    if (tag !== 48) {
                        break;
                    }
                    message.wrapperAlgorithm = reader.int32();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            id: isSet(object.id) ? Long.fromValue(object.id) : Long.UZERO,
            createdNs: isSet(object.createdNs) ? Long.fromValue(object.createdNs) : Long.UZERO,
            installationKey: isSet(object.installationKey) ? bytesFromBase64(object.installationKey) : new Uint8Array(0),
            welcomePointer: isSet(object.welcomePointer) ? bytesFromBase64(object.welcomePointer) : new Uint8Array(0),
            hpkePublicKey: isSet(object.hpkePublicKey) ? bytesFromBase64(object.hpkePublicKey) : new Uint8Array(0),
            wrapperAlgorithm: isSet(object.wrapperAlgorithm)
                ? welcomePointerWrapperAlgorithmFromJSON(object.wrapperAlgorithm)
                : 0,
        };
    },
    toJSON(message) {
        const obj = {};
        if (!message.id.equals(Long.UZERO)) {
            obj.id = (message.id || Long.UZERO).toString();
        }
        if (!message.createdNs.equals(Long.UZERO)) {
            obj.createdNs = (message.createdNs || Long.UZERO).toString();
        }
        if (message.installationKey.length !== 0) {
            obj.installationKey = base64FromBytes(message.installationKey);
        }
        if (message.welcomePointer.length !== 0) {
            obj.welcomePointer = base64FromBytes(message.welcomePointer);
        }
        if (message.hpkePublicKey.length !== 0) {
            obj.hpkePublicKey = base64FromBytes(message.hpkePublicKey);
        }
        if (message.wrapperAlgorithm !== 0) {
            obj.wrapperAlgorithm = welcomePointerWrapperAlgorithmToJSON(message.wrapperAlgorithm);
        }
        return obj;
    },
    create(base) {
        return WelcomeMessage_WelcomePointer.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseWelcomeMessage_WelcomePointer();
        message.id = (object.id !== undefined && object.id !== null) ? Long.fromValue(object.id) : Long.UZERO;
        message.createdNs = (object.createdNs !== undefined && object.createdNs !== null)
            ? Long.fromValue(object.createdNs)
            : Long.UZERO;
        message.installationKey = object.installationKey ?? new Uint8Array(0);
        message.welcomePointer = object.welcomePointer ?? new Uint8Array(0);
        message.hpkePublicKey = object.hpkePublicKey ?? new Uint8Array(0);
        message.wrapperAlgorithm = object.wrapperAlgorithm ?? 0;
        return message;
    },
};
function createBaseWelcomeMessageInput() {
    return { v1: undefined, welcomePointer: undefined };
}
export const WelcomeMessageInput = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.v1 !== undefined) {
            WelcomeMessageInput_V1.encode(message.v1, writer.uint32(10).fork()).ldelim();
        }
        if (message.welcomePointer !== undefined) {
            WelcomeMessageInput_WelcomePointer.encode(message.welcomePointer, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseWelcomeMessageInput();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.v1 = WelcomeMessageInput_V1.decode(reader, reader.uint32());
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.welcomePointer = WelcomeMessageInput_WelcomePointer.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            v1: isSet(object.v1) ? WelcomeMessageInput_V1.fromJSON(object.v1) : undefined,
            welcomePointer: isSet(object.welcomePointer)
                ? WelcomeMessageInput_WelcomePointer.fromJSON(object.welcomePointer)
                : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.v1 !== undefined) {
            obj.v1 = WelcomeMessageInput_V1.toJSON(message.v1);
        }
        if (message.welcomePointer !== undefined) {
            obj.welcomePointer = WelcomeMessageInput_WelcomePointer.toJSON(message.welcomePointer);
        }
        return obj;
    },
    create(base) {
        return WelcomeMessageInput.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseWelcomeMessageInput();
        message.v1 = (object.v1 !== undefined && object.v1 !== null)
            ? WelcomeMessageInput_V1.fromPartial(object.v1)
            : undefined;
        message.welcomePointer = (object.welcomePointer !== undefined && object.welcomePointer !== null)
            ? WelcomeMessageInput_WelcomePointer.fromPartial(object.welcomePointer)
            : undefined;
        return message;
    },
};
function createBaseWelcomeMessageInput_V1() {
    return {
        installationKey: new Uint8Array(0),
        data: new Uint8Array(0),
        hpkePublicKey: new Uint8Array(0),
        wrapperAlgorithm: 0,
        welcomeMetadata: new Uint8Array(0),
    };
}
export const WelcomeMessageInput_V1 = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.installationKey.length !== 0) {
            writer.uint32(10).bytes(message.installationKey);
        }
        if (message.data.length !== 0) {
            writer.uint32(18).bytes(message.data);
        }
        if (message.hpkePublicKey.length !== 0) {
            writer.uint32(26).bytes(message.hpkePublicKey);
        }
        if (message.wrapperAlgorithm !== 0) {
            writer.uint32(32).int32(message.wrapperAlgorithm);
        }
        if (message.welcomeMetadata.length !== 0) {
            writer.uint32(58).bytes(message.welcomeMetadata);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseWelcomeMessageInput_V1();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.installationKey = reader.bytes();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.data = reader.bytes();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.hpkePublicKey = reader.bytes();
                    continue;
                case 4:
                    if (tag !== 32) {
                        break;
                    }
                    message.wrapperAlgorithm = reader.int32();
                    continue;
                case 7:
                    if (tag !== 58) {
                        break;
                    }
                    message.welcomeMetadata = reader.bytes();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            installationKey: isSet(object.installationKey) ? bytesFromBase64(object.installationKey) : new Uint8Array(0),
            data: isSet(object.data) ? bytesFromBase64(object.data) : new Uint8Array(0),
            hpkePublicKey: isSet(object.hpkePublicKey) ? bytesFromBase64(object.hpkePublicKey) : new Uint8Array(0),
            wrapperAlgorithm: isSet(object.wrapperAlgorithm) ? welcomeWrapperAlgorithmFromJSON(object.wrapperAlgorithm) : 0,
            welcomeMetadata: isSet(object.welcomeMetadata) ? bytesFromBase64(object.welcomeMetadata) : new Uint8Array(0),
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.installationKey.length !== 0) {
            obj.installationKey = base64FromBytes(message.installationKey);
        }
        if (message.data.length !== 0) {
            obj.data = base64FromBytes(message.data);
        }
        if (message.hpkePublicKey.length !== 0) {
            obj.hpkePublicKey = base64FromBytes(message.hpkePublicKey);
        }
        if (message.wrapperAlgorithm !== 0) {
            obj.wrapperAlgorithm = welcomeWrapperAlgorithmToJSON(message.wrapperAlgorithm);
        }
        if (message.welcomeMetadata.length !== 0) {
            obj.welcomeMetadata = base64FromBytes(message.welcomeMetadata);
        }
        return obj;
    },
    create(base) {
        return WelcomeMessageInput_V1.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseWelcomeMessageInput_V1();
        message.installationKey = object.installationKey ?? new Uint8Array(0);
        message.data = object.data ?? new Uint8Array(0);
        message.hpkePublicKey = object.hpkePublicKey ?? new Uint8Array(0);
        message.wrapperAlgorithm = object.wrapperAlgorithm ?? 0;
        message.welcomeMetadata = object.welcomeMetadata ?? new Uint8Array(0);
        return message;
    },
};
function createBaseWelcomeMessageInput_WelcomePointer() {
    return {
        installationKey: new Uint8Array(0),
        welcomePointer: new Uint8Array(0),
        hpkePublicKey: new Uint8Array(0),
        wrapperAlgorithm: 0,
    };
}
export const WelcomeMessageInput_WelcomePointer = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.installationKey.length !== 0) {
            writer.uint32(10).bytes(message.installationKey);
        }
        if (message.welcomePointer.length !== 0) {
            writer.uint32(18).bytes(message.welcomePointer);
        }
        if (message.hpkePublicKey.length !== 0) {
            writer.uint32(26).bytes(message.hpkePublicKey);
        }
        if (message.wrapperAlgorithm !== 0) {
            writer.uint32(32).int32(message.wrapperAlgorithm);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseWelcomeMessageInput_WelcomePointer();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.installationKey = reader.bytes();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.welcomePointer = reader.bytes();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.hpkePublicKey = reader.bytes();
                    continue;
                case 4:
                    if (tag !== 32) {
                        break;
                    }
                    message.wrapperAlgorithm = reader.int32();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            installationKey: isSet(object.installationKey) ? bytesFromBase64(object.installationKey) : new Uint8Array(0),
            welcomePointer: isSet(object.welcomePointer) ? bytesFromBase64(object.welcomePointer) : new Uint8Array(0),
            hpkePublicKey: isSet(object.hpkePublicKey) ? bytesFromBase64(object.hpkePublicKey) : new Uint8Array(0),
            wrapperAlgorithm: isSet(object.wrapperAlgorithm)
                ? welcomePointerWrapperAlgorithmFromJSON(object.wrapperAlgorithm)
                : 0,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.installationKey.length !== 0) {
            obj.installationKey = base64FromBytes(message.installationKey);
        }
        if (message.welcomePointer.length !== 0) {
            obj.welcomePointer = base64FromBytes(message.welcomePointer);
        }
        if (message.hpkePublicKey.length !== 0) {
            obj.hpkePublicKey = base64FromBytes(message.hpkePublicKey);
        }
        if (message.wrapperAlgorithm !== 0) {
            obj.wrapperAlgorithm = welcomePointerWrapperAlgorithmToJSON(message.wrapperAlgorithm);
        }
        return obj;
    },
    create(base) {
        return WelcomeMessageInput_WelcomePointer.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseWelcomeMessageInput_WelcomePointer();
        message.installationKey = object.installationKey ?? new Uint8Array(0);
        message.welcomePointer = object.welcomePointer ?? new Uint8Array(0);
        message.hpkePublicKey = object.hpkePublicKey ?? new Uint8Array(0);
        message.wrapperAlgorithm = object.wrapperAlgorithm ?? 0;
        return message;
    },
};
function createBaseWelcomeMetadata() {
    return { messageCursor: Long.UZERO };
}
export const WelcomeMetadata = {
    encode(message, writer = _m0.Writer.create()) {
        if (!message.messageCursor.equals(Long.UZERO)) {
            writer.uint32(8).uint64(message.messageCursor);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseWelcomeMetadata();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 8) {
                        break;
                    }
                    message.messageCursor = reader.uint64();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { messageCursor: isSet(object.messageCursor) ? Long.fromValue(object.messageCursor) : Long.UZERO };
    },
    toJSON(message) {
        const obj = {};
        if (!message.messageCursor.equals(Long.UZERO)) {
            obj.messageCursor = (message.messageCursor || Long.UZERO).toString();
        }
        return obj;
    },
    create(base) {
        return WelcomeMetadata.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseWelcomeMetadata();
        message.messageCursor = (object.messageCursor !== undefined && object.messageCursor !== null)
            ? Long.fromValue(object.messageCursor)
            : Long.UZERO;
        return message;
    },
};
function createBaseGroupMessage() {
    return { v1: undefined };
}
export const GroupMessage = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.v1 !== undefined) {
            GroupMessage_V1.encode(message.v1, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGroupMessage();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.v1 = GroupMessage_V1.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { v1: isSet(object.v1) ? GroupMessage_V1.fromJSON(object.v1) : undefined };
    },
    toJSON(message) {
        const obj = {};
        if (message.v1 !== undefined) {
            obj.v1 = GroupMessage_V1.toJSON(message.v1);
        }
        return obj;
    },
    create(base) {
        return GroupMessage.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGroupMessage();
        message.v1 = (object.v1 !== undefined && object.v1 !== null) ? GroupMessage_V1.fromPartial(object.v1) : undefined;
        return message;
    },
};
function createBaseGroupMessage_V1() {
    return {
        id: Long.UZERO,
        createdNs: Long.UZERO,
        groupId: new Uint8Array(0),
        data: new Uint8Array(0),
        senderHmac: new Uint8Array(0),
        shouldPush: false,
        isCommit: false,
    };
}
export const GroupMessage_V1 = {
    encode(message, writer = _m0.Writer.create()) {
        if (!message.id.equals(Long.UZERO)) {
            writer.uint32(8).uint64(message.id);
        }
        if (!message.createdNs.equals(Long.UZERO)) {
            writer.uint32(16).uint64(message.createdNs);
        }
        if (message.groupId.length !== 0) {
            writer.uint32(26).bytes(message.groupId);
        }
        if (message.data.length !== 0) {
            writer.uint32(34).bytes(message.data);
        }
        if (message.senderHmac.length !== 0) {
            writer.uint32(42).bytes(message.senderHmac);
        }
        if (message.shouldPush !== false) {
            writer.uint32(48).bool(message.shouldPush);
        }
        if (message.isCommit !== false) {
            writer.uint32(56).bool(message.isCommit);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGroupMessage_V1();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 8) {
                        break;
                    }
                    message.id = reader.uint64();
                    continue;
                case 2:
                    if (tag !== 16) {
                        break;
                    }
                    message.createdNs = reader.uint64();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.groupId = reader.bytes();
                    continue;
                case 4:
                    if (tag !== 34) {
                        break;
                    }
                    message.data = reader.bytes();
                    continue;
                case 5:
                    if (tag !== 42) {
                        break;
                    }
                    message.senderHmac = reader.bytes();
                    continue;
                case 6:
                    if (tag !== 48) {
                        break;
                    }
                    message.shouldPush = reader.bool();
                    continue;
                case 7:
                    if (tag !== 56) {
                        break;
                    }
                    message.isCommit = reader.bool();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            id: isSet(object.id) ? Long.fromValue(object.id) : Long.UZERO,
            createdNs: isSet(object.createdNs) ? Long.fromValue(object.createdNs) : Long.UZERO,
            groupId: isSet(object.groupId) ? bytesFromBase64(object.groupId) : new Uint8Array(0),
            data: isSet(object.data) ? bytesFromBase64(object.data) : new Uint8Array(0),
            senderHmac: isSet(object.senderHmac) ? bytesFromBase64(object.senderHmac) : new Uint8Array(0),
            shouldPush: isSet(object.shouldPush) ? globalThis.Boolean(object.shouldPush) : false,
            isCommit: isSet(object.isCommit) ? globalThis.Boolean(object.isCommit) : false,
        };
    },
    toJSON(message) {
        const obj = {};
        if (!message.id.equals(Long.UZERO)) {
            obj.id = (message.id || Long.UZERO).toString();
        }
        if (!message.createdNs.equals(Long.UZERO)) {
            obj.createdNs = (message.createdNs || Long.UZERO).toString();
        }
        if (message.groupId.length !== 0) {
            obj.groupId = base64FromBytes(message.groupId);
        }
        if (message.data.length !== 0) {
            obj.data = base64FromBytes(message.data);
        }
        if (message.senderHmac.length !== 0) {
            obj.senderHmac = base64FromBytes(message.senderHmac);
        }
        if (message.shouldPush !== false) {
            obj.shouldPush = message.shouldPush;
        }
        if (message.isCommit !== false) {
            obj.isCommit = message.isCommit;
        }
        return obj;
    },
    create(base) {
        return GroupMessage_V1.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGroupMessage_V1();
        message.id = (object.id !== undefined && object.id !== null) ? Long.fromValue(object.id) : Long.UZERO;
        message.createdNs = (object.createdNs !== undefined && object.createdNs !== null)
            ? Long.fromValue(object.createdNs)
            : Long.UZERO;
        message.groupId = object.groupId ?? new Uint8Array(0);
        message.data = object.data ?? new Uint8Array(0);
        message.senderHmac = object.senderHmac ?? new Uint8Array(0);
        message.shouldPush = object.shouldPush ?? false;
        message.isCommit = object.isCommit ?? false;
        return message;
    },
};
function createBaseGroupMessageInput() {
    return { v1: undefined };
}
export const GroupMessageInput = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.v1 !== undefined) {
            GroupMessageInput_V1.encode(message.v1, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGroupMessageInput();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.v1 = GroupMessageInput_V1.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { v1: isSet(object.v1) ? GroupMessageInput_V1.fromJSON(object.v1) : undefined };
    },
    toJSON(message) {
        const obj = {};
        if (message.v1 !== undefined) {
            obj.v1 = GroupMessageInput_V1.toJSON(message.v1);
        }
        return obj;
    },
    create(base) {
        return GroupMessageInput.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGroupMessageInput();
        message.v1 = (object.v1 !== undefined && object.v1 !== null)
            ? GroupMessageInput_V1.fromPartial(object.v1)
            : undefined;
        return message;
    },
};
function createBaseGroupMessageInput_V1() {
    return { data: new Uint8Array(0), senderHmac: new Uint8Array(0), shouldPush: false };
}
export const GroupMessageInput_V1 = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.data.length !== 0) {
            writer.uint32(10).bytes(message.data);
        }
        if (message.senderHmac.length !== 0) {
            writer.uint32(18).bytes(message.senderHmac);
        }
        if (message.shouldPush !== false) {
            writer.uint32(24).bool(message.shouldPush);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGroupMessageInput_V1();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.data = reader.bytes();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.senderHmac = reader.bytes();
                    continue;
                case 3:
                    if (tag !== 24) {
                        break;
                    }
                    message.shouldPush = reader.bool();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            data: isSet(object.data) ? bytesFromBase64(object.data) : new Uint8Array(0),
            senderHmac: isSet(object.senderHmac) ? bytesFromBase64(object.senderHmac) : new Uint8Array(0),
            shouldPush: isSet(object.shouldPush) ? globalThis.Boolean(object.shouldPush) : false,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.data.length !== 0) {
            obj.data = base64FromBytes(message.data);
        }
        if (message.senderHmac.length !== 0) {
            obj.senderHmac = base64FromBytes(message.senderHmac);
        }
        if (message.shouldPush !== false) {
            obj.shouldPush = message.shouldPush;
        }
        return obj;
    },
    create(base) {
        return GroupMessageInput_V1.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGroupMessageInput_V1();
        message.data = object.data ?? new Uint8Array(0);
        message.senderHmac = object.senderHmac ?? new Uint8Array(0);
        message.shouldPush = object.shouldPush ?? false;
        return message;
    },
};
function createBaseSendGroupMessagesRequest() {
    return { messages: [] };
}
export const SendGroupMessagesRequest = {
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.messages) {
            GroupMessageInput.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSendGroupMessagesRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.messages.push(GroupMessageInput.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            messages: globalThis.Array.isArray(object?.messages)
                ? object.messages.map((e) => GroupMessageInput.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.messages?.length) {
            obj.messages = message.messages.map((e) => GroupMessageInput.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return SendGroupMessagesRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseSendGroupMessagesRequest();
        message.messages = object.messages?.map((e) => GroupMessageInput.fromPartial(e)) || [];
        return message;
    },
};
function createBaseSendWelcomeMessagesRequest() {
    return { messages: [] };
}
export const SendWelcomeMessagesRequest = {
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.messages) {
            WelcomeMessageInput.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSendWelcomeMessagesRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.messages.push(WelcomeMessageInput.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            messages: globalThis.Array.isArray(object?.messages)
                ? object.messages.map((e) => WelcomeMessageInput.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.messages?.length) {
            obj.messages = message.messages.map((e) => WelcomeMessageInput.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return SendWelcomeMessagesRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseSendWelcomeMessagesRequest();
        message.messages = object.messages?.map((e) => WelcomeMessageInput.fromPartial(e)) || [];
        return message;
    },
};
function createBaseKeyPackageUpload() {
    return { keyPackageTlsSerialized: new Uint8Array(0) };
}
export const KeyPackageUpload = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.keyPackageTlsSerialized.length !== 0) {
            writer.uint32(10).bytes(message.keyPackageTlsSerialized);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseKeyPackageUpload();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.keyPackageTlsSerialized = reader.bytes();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            keyPackageTlsSerialized: isSet(object.keyPackageTlsSerialized)
                ? bytesFromBase64(object.keyPackageTlsSerialized)
                : new Uint8Array(0),
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.keyPackageTlsSerialized.length !== 0) {
            obj.keyPackageTlsSerialized = base64FromBytes(message.keyPackageTlsSerialized);
        }
        return obj;
    },
    create(base) {
        return KeyPackageUpload.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseKeyPackageUpload();
        message.keyPackageTlsSerialized = object.keyPackageTlsSerialized ?? new Uint8Array(0);
        return message;
    },
};
function createBaseRegisterInstallationRequest() {
    return { keyPackage: undefined, isInboxIdCredential: false };
}
export const RegisterInstallationRequest = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.keyPackage !== undefined) {
            KeyPackageUpload.encode(message.keyPackage, writer.uint32(10).fork()).ldelim();
        }
        if (message.isInboxIdCredential !== false) {
            writer.uint32(16).bool(message.isInboxIdCredential);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseRegisterInstallationRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.keyPackage = KeyPackageUpload.decode(reader, reader.uint32());
                    continue;
                case 2:
                    if (tag !== 16) {
                        break;
                    }
                    message.isInboxIdCredential = reader.bool();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            keyPackage: isSet(object.keyPackage) ? KeyPackageUpload.fromJSON(object.keyPackage) : undefined,
            isInboxIdCredential: isSet(object.isInboxIdCredential) ? globalThis.Boolean(object.isInboxIdCredential) : false,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.keyPackage !== undefined) {
            obj.keyPackage = KeyPackageUpload.toJSON(message.keyPackage);
        }
        if (message.isInboxIdCredential !== false) {
            obj.isInboxIdCredential = message.isInboxIdCredential;
        }
        return obj;
    },
    create(base) {
        return RegisterInstallationRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseRegisterInstallationRequest();
        message.keyPackage = (object.keyPackage !== undefined && object.keyPackage !== null)
            ? KeyPackageUpload.fromPartial(object.keyPackage)
            : undefined;
        message.isInboxIdCredential = object.isInboxIdCredential ?? false;
        return message;
    },
};
function createBaseRegisterInstallationResponse() {
    return { installationKey: new Uint8Array(0) };
}
export const RegisterInstallationResponse = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.installationKey.length !== 0) {
            writer.uint32(10).bytes(message.installationKey);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseRegisterInstallationResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.installationKey = reader.bytes();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            installationKey: isSet(object.installationKey) ? bytesFromBase64(object.installationKey) : new Uint8Array(0),
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.installationKey.length !== 0) {
            obj.installationKey = base64FromBytes(message.installationKey);
        }
        return obj;
    },
    create(base) {
        return RegisterInstallationResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseRegisterInstallationResponse();
        message.installationKey = object.installationKey ?? new Uint8Array(0);
        return message;
    },
};
function createBaseUploadKeyPackageRequest() {
    return { keyPackage: undefined, isInboxIdCredential: false };
}
export const UploadKeyPackageRequest = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.keyPackage !== undefined) {
            KeyPackageUpload.encode(message.keyPackage, writer.uint32(10).fork()).ldelim();
        }
        if (message.isInboxIdCredential !== false) {
            writer.uint32(16).bool(message.isInboxIdCredential);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseUploadKeyPackageRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.keyPackage = KeyPackageUpload.decode(reader, reader.uint32());
                    continue;
                case 2:
                    if (tag !== 16) {
                        break;
                    }
                    message.isInboxIdCredential = reader.bool();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            keyPackage: isSet(object.keyPackage) ? KeyPackageUpload.fromJSON(object.keyPackage) : undefined,
            isInboxIdCredential: isSet(object.isInboxIdCredential) ? globalThis.Boolean(object.isInboxIdCredential) : false,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.keyPackage !== undefined) {
            obj.keyPackage = KeyPackageUpload.toJSON(message.keyPackage);
        }
        if (message.isInboxIdCredential !== false) {
            obj.isInboxIdCredential = message.isInboxIdCredential;
        }
        return obj;
    },
    create(base) {
        return UploadKeyPackageRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseUploadKeyPackageRequest();
        message.keyPackage = (object.keyPackage !== undefined && object.keyPackage !== null)
            ? KeyPackageUpload.fromPartial(object.keyPackage)
            : undefined;
        message.isInboxIdCredential = object.isInboxIdCredential ?? false;
        return message;
    },
};
function createBaseFetchKeyPackagesRequest() {
    return { installationKeys: [] };
}
export const FetchKeyPackagesRequest = {
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.installationKeys) {
            writer.uint32(10).bytes(v);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseFetchKeyPackagesRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.installationKeys.push(reader.bytes());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            installationKeys: globalThis.Array.isArray(object?.installationKeys)
                ? object.installationKeys.map((e) => bytesFromBase64(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.installationKeys?.length) {
            obj.installationKeys = message.installationKeys.map((e) => base64FromBytes(e));
        }
        return obj;
    },
    create(base) {
        return FetchKeyPackagesRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseFetchKeyPackagesRequest();
        message.installationKeys = object.installationKeys?.map((e) => e) || [];
        return message;
    },
};
function createBaseFetchKeyPackagesResponse() {
    return { keyPackages: [] };
}
export const FetchKeyPackagesResponse = {
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.keyPackages) {
            FetchKeyPackagesResponse_KeyPackage.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseFetchKeyPackagesResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.keyPackages.push(FetchKeyPackagesResponse_KeyPackage.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            keyPackages: globalThis.Array.isArray(object?.keyPackages)
                ? object.keyPackages.map((e) => FetchKeyPackagesResponse_KeyPackage.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.keyPackages?.length) {
            obj.keyPackages = message.keyPackages.map((e) => FetchKeyPackagesResponse_KeyPackage.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return FetchKeyPackagesResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseFetchKeyPackagesResponse();
        message.keyPackages = object.keyPackages?.map((e) => FetchKeyPackagesResponse_KeyPackage.fromPartial(e)) || [];
        return message;
    },
};
function createBaseFetchKeyPackagesResponse_KeyPackage() {
    return { keyPackageTlsSerialized: new Uint8Array(0) };
}
export const FetchKeyPackagesResponse_KeyPackage = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.keyPackageTlsSerialized.length !== 0) {
            writer.uint32(10).bytes(message.keyPackageTlsSerialized);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseFetchKeyPackagesResponse_KeyPackage();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.keyPackageTlsSerialized = reader.bytes();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            keyPackageTlsSerialized: isSet(object.keyPackageTlsSerialized)
                ? bytesFromBase64(object.keyPackageTlsSerialized)
                : new Uint8Array(0),
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.keyPackageTlsSerialized.length !== 0) {
            obj.keyPackageTlsSerialized = base64FromBytes(message.keyPackageTlsSerialized);
        }
        return obj;
    },
    create(base) {
        return FetchKeyPackagesResponse_KeyPackage.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseFetchKeyPackagesResponse_KeyPackage();
        message.keyPackageTlsSerialized = object.keyPackageTlsSerialized ?? new Uint8Array(0);
        return message;
    },
};
function createBaseRevokeInstallationRequest() {
    return { installationKey: new Uint8Array(0), walletSignature: undefined };
}
export const RevokeInstallationRequest = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.installationKey.length !== 0) {
            writer.uint32(10).bytes(message.installationKey);
        }
        if (message.walletSignature !== undefined) {
            Signature.encode(message.walletSignature, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseRevokeInstallationRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.installationKey = reader.bytes();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.walletSignature = Signature.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            installationKey: isSet(object.installationKey) ? bytesFromBase64(object.installationKey) : new Uint8Array(0),
            walletSignature: isSet(object.walletSignature) ? Signature.fromJSON(object.walletSignature) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.installationKey.length !== 0) {
            obj.installationKey = base64FromBytes(message.installationKey);
        }
        if (message.walletSignature !== undefined) {
            obj.walletSignature = Signature.toJSON(message.walletSignature);
        }
        return obj;
    },
    create(base) {
        return RevokeInstallationRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseRevokeInstallationRequest();
        message.installationKey = object.installationKey ?? new Uint8Array(0);
        message.walletSignature = (object.walletSignature !== undefined && object.walletSignature !== null)
            ? Signature.fromPartial(object.walletSignature)
            : undefined;
        return message;
    },
};
function createBaseGetIdentityUpdatesRequest() {
    return { accountAddresses: [], startTimeNs: Long.UZERO };
}
export const GetIdentityUpdatesRequest = {
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.accountAddresses) {
            writer.uint32(10).string(v);
        }
        if (!message.startTimeNs.equals(Long.UZERO)) {
            writer.uint32(16).uint64(message.startTimeNs);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetIdentityUpdatesRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.accountAddresses.push(reader.string());
                    continue;
                case 2:
                    if (tag !== 16) {
                        break;
                    }
                    message.startTimeNs = reader.uint64();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            accountAddresses: globalThis.Array.isArray(object?.accountAddresses)
                ? object.accountAddresses.map((e) => globalThis.String(e))
                : [],
            startTimeNs: isSet(object.startTimeNs) ? Long.fromValue(object.startTimeNs) : Long.UZERO,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.accountAddresses?.length) {
            obj.accountAddresses = message.accountAddresses;
        }
        if (!message.startTimeNs.equals(Long.UZERO)) {
            obj.startTimeNs = (message.startTimeNs || Long.UZERO).toString();
        }
        return obj;
    },
    create(base) {
        return GetIdentityUpdatesRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGetIdentityUpdatesRequest();
        message.accountAddresses = object.accountAddresses?.map((e) => e) || [];
        message.startTimeNs = (object.startTimeNs !== undefined && object.startTimeNs !== null)
            ? Long.fromValue(object.startTimeNs)
            : Long.UZERO;
        return message;
    },
};
function createBaseGetIdentityUpdatesResponse() {
    return { updates: [] };
}
export const GetIdentityUpdatesResponse = {
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.updates) {
            GetIdentityUpdatesResponse_WalletUpdates.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetIdentityUpdatesResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.updates.push(GetIdentityUpdatesResponse_WalletUpdates.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            updates: globalThis.Array.isArray(object?.updates)
                ? object.updates.map((e) => GetIdentityUpdatesResponse_WalletUpdates.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.updates?.length) {
            obj.updates = message.updates.map((e) => GetIdentityUpdatesResponse_WalletUpdates.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return GetIdentityUpdatesResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGetIdentityUpdatesResponse();
        message.updates = object.updates?.map((e) => GetIdentityUpdatesResponse_WalletUpdates.fromPartial(e)) || [];
        return message;
    },
};
function createBaseGetIdentityUpdatesResponse_NewInstallationUpdate() {
    return { installationKey: new Uint8Array(0), credentialIdentity: new Uint8Array(0) };
}
export const GetIdentityUpdatesResponse_NewInstallationUpdate = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.installationKey.length !== 0) {
            writer.uint32(10).bytes(message.installationKey);
        }
        if (message.credentialIdentity.length !== 0) {
            writer.uint32(18).bytes(message.credentialIdentity);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetIdentityUpdatesResponse_NewInstallationUpdate();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.installationKey = reader.bytes();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.credentialIdentity = reader.bytes();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            installationKey: isSet(object.installationKey) ? bytesFromBase64(object.installationKey) : new Uint8Array(0),
            credentialIdentity: isSet(object.credentialIdentity)
                ? bytesFromBase64(object.credentialIdentity)
                : new Uint8Array(0),
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.installationKey.length !== 0) {
            obj.installationKey = base64FromBytes(message.installationKey);
        }
        if (message.credentialIdentity.length !== 0) {
            obj.credentialIdentity = base64FromBytes(message.credentialIdentity);
        }
        return obj;
    },
    create(base) {
        return GetIdentityUpdatesResponse_NewInstallationUpdate.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGetIdentityUpdatesResponse_NewInstallationUpdate();
        message.installationKey = object.installationKey ?? new Uint8Array(0);
        message.credentialIdentity = object.credentialIdentity ?? new Uint8Array(0);
        return message;
    },
};
function createBaseGetIdentityUpdatesResponse_RevokedInstallationUpdate() {
    return { installationKey: new Uint8Array(0) };
}
export const GetIdentityUpdatesResponse_RevokedInstallationUpdate = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.installationKey.length !== 0) {
            writer.uint32(10).bytes(message.installationKey);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetIdentityUpdatesResponse_RevokedInstallationUpdate();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.installationKey = reader.bytes();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            installationKey: isSet(object.installationKey) ? bytesFromBase64(object.installationKey) : new Uint8Array(0),
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.installationKey.length !== 0) {
            obj.installationKey = base64FromBytes(message.installationKey);
        }
        return obj;
    },
    create(base) {
        return GetIdentityUpdatesResponse_RevokedInstallationUpdate.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGetIdentityUpdatesResponse_RevokedInstallationUpdate();
        message.installationKey = object.installationKey ?? new Uint8Array(0);
        return message;
    },
};
function createBaseGetIdentityUpdatesResponse_Update() {
    return { timestampNs: Long.UZERO, newInstallation: undefined, revokedInstallation: undefined };
}
export const GetIdentityUpdatesResponse_Update = {
    encode(message, writer = _m0.Writer.create()) {
        if (!message.timestampNs.equals(Long.UZERO)) {
            writer.uint32(8).uint64(message.timestampNs);
        }
        if (message.newInstallation !== undefined) {
            GetIdentityUpdatesResponse_NewInstallationUpdate.encode(message.newInstallation, writer.uint32(18).fork())
                .ldelim();
        }
        if (message.revokedInstallation !== undefined) {
            GetIdentityUpdatesResponse_RevokedInstallationUpdate.encode(message.revokedInstallation, writer.uint32(26).fork())
                .ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetIdentityUpdatesResponse_Update();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 8) {
                        break;
                    }
                    message.timestampNs = reader.uint64();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.newInstallation = GetIdentityUpdatesResponse_NewInstallationUpdate.decode(reader, reader.uint32());
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.revokedInstallation = GetIdentityUpdatesResponse_RevokedInstallationUpdate.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            timestampNs: isSet(object.timestampNs) ? Long.fromValue(object.timestampNs) : Long.UZERO,
            newInstallation: isSet(object.newInstallation)
                ? GetIdentityUpdatesResponse_NewInstallationUpdate.fromJSON(object.newInstallation)
                : undefined,
            revokedInstallation: isSet(object.revokedInstallation)
                ? GetIdentityUpdatesResponse_RevokedInstallationUpdate.fromJSON(object.revokedInstallation)
                : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (!message.timestampNs.equals(Long.UZERO)) {
            obj.timestampNs = (message.timestampNs || Long.UZERO).toString();
        }
        if (message.newInstallation !== undefined) {
            obj.newInstallation = GetIdentityUpdatesResponse_NewInstallationUpdate.toJSON(message.newInstallation);
        }
        if (message.revokedInstallation !== undefined) {
            obj.revokedInstallation = GetIdentityUpdatesResponse_RevokedInstallationUpdate.toJSON(message.revokedInstallation);
        }
        return obj;
    },
    create(base) {
        return GetIdentityUpdatesResponse_Update.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGetIdentityUpdatesResponse_Update();
        message.timestampNs = (object.timestampNs !== undefined && object.timestampNs !== null)
            ? Long.fromValue(object.timestampNs)
            : Long.UZERO;
        message.newInstallation = (object.newInstallation !== undefined && object.newInstallation !== null)
            ? GetIdentityUpdatesResponse_NewInstallationUpdate.fromPartial(object.newInstallation)
            : undefined;
        message.revokedInstallation = (object.revokedInstallation !== undefined && object.revokedInstallation !== null)
            ? GetIdentityUpdatesResponse_RevokedInstallationUpdate.fromPartial(object.revokedInstallation)
            : undefined;
        return message;
    },
};
function createBaseGetIdentityUpdatesResponse_WalletUpdates() {
    return { updates: [] };
}
export const GetIdentityUpdatesResponse_WalletUpdates = {
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.updates) {
            GetIdentityUpdatesResponse_Update.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetIdentityUpdatesResponse_WalletUpdates();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.updates.push(GetIdentityUpdatesResponse_Update.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            updates: globalThis.Array.isArray(object?.updates)
                ? object.updates.map((e) => GetIdentityUpdatesResponse_Update.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.updates?.length) {
            obj.updates = message.updates.map((e) => GetIdentityUpdatesResponse_Update.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return GetIdentityUpdatesResponse_WalletUpdates.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGetIdentityUpdatesResponse_WalletUpdates();
        message.updates = object.updates?.map((e) => GetIdentityUpdatesResponse_Update.fromPartial(e)) || [];
        return message;
    },
};
function createBasePagingInfo() {
    return { direction: 0, limit: 0, idCursor: Long.UZERO };
}
export const PagingInfo = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.direction !== 0) {
            writer.uint32(8).int32(message.direction);
        }
        if (message.limit !== 0) {
            writer.uint32(16).uint32(message.limit);
        }
        if (!message.idCursor.equals(Long.UZERO)) {
            writer.uint32(24).uint64(message.idCursor);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBasePagingInfo();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 8) {
                        break;
                    }
                    message.direction = reader.int32();
                    continue;
                case 2:
                    if (tag !== 16) {
                        break;
                    }
                    message.limit = reader.uint32();
                    continue;
                case 3:
                    if (tag !== 24) {
                        break;
                    }
                    message.idCursor = reader.uint64();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            direction: isSet(object.direction) ? sortDirectionFromJSON(object.direction) : 0,
            limit: isSet(object.limit) ? globalThis.Number(object.limit) : 0,
            idCursor: isSet(object.idCursor) ? Long.fromValue(object.idCursor) : Long.UZERO,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.direction !== 0) {
            obj.direction = sortDirectionToJSON(message.direction);
        }
        if (message.limit !== 0) {
            obj.limit = Math.round(message.limit);
        }
        if (!message.idCursor.equals(Long.UZERO)) {
            obj.idCursor = (message.idCursor || Long.UZERO).toString();
        }
        return obj;
    },
    create(base) {
        return PagingInfo.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBasePagingInfo();
        message.direction = object.direction ?? 0;
        message.limit = object.limit ?? 0;
        message.idCursor = (object.idCursor !== undefined && object.idCursor !== null)
            ? Long.fromValue(object.idCursor)
            : Long.UZERO;
        return message;
    },
};
function createBaseQueryGroupMessagesRequest() {
    return { groupId: new Uint8Array(0), pagingInfo: undefined };
}
export const QueryGroupMessagesRequest = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.groupId.length !== 0) {
            writer.uint32(10).bytes(message.groupId);
        }
        if (message.pagingInfo !== undefined) {
            PagingInfo.encode(message.pagingInfo, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryGroupMessagesRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.groupId = reader.bytes();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.pagingInfo = PagingInfo.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            groupId: isSet(object.groupId) ? bytesFromBase64(object.groupId) : new Uint8Array(0),
            pagingInfo: isSet(object.pagingInfo) ? PagingInfo.fromJSON(object.pagingInfo) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.groupId.length !== 0) {
            obj.groupId = base64FromBytes(message.groupId);
        }
        if (message.pagingInfo !== undefined) {
            obj.pagingInfo = PagingInfo.toJSON(message.pagingInfo);
        }
        return obj;
    },
    create(base) {
        return QueryGroupMessagesRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseQueryGroupMessagesRequest();
        message.groupId = object.groupId ?? new Uint8Array(0);
        message.pagingInfo = (object.pagingInfo !== undefined && object.pagingInfo !== null)
            ? PagingInfo.fromPartial(object.pagingInfo)
            : undefined;
        return message;
    },
};
function createBaseQueryGroupMessagesResponse() {
    return { messages: [], pagingInfo: undefined };
}
export const QueryGroupMessagesResponse = {
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.messages) {
            GroupMessage.encode(v, writer.uint32(10).fork()).ldelim();
        }
        if (message.pagingInfo !== undefined) {
            PagingInfo.encode(message.pagingInfo, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryGroupMessagesResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.messages.push(GroupMessage.decode(reader, reader.uint32()));
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.pagingInfo = PagingInfo.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            messages: globalThis.Array.isArray(object?.messages)
                ? object.messages.map((e) => GroupMessage.fromJSON(e))
                : [],
            pagingInfo: isSet(object.pagingInfo) ? PagingInfo.fromJSON(object.pagingInfo) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.messages?.length) {
            obj.messages = message.messages.map((e) => GroupMessage.toJSON(e));
        }
        if (message.pagingInfo !== undefined) {
            obj.pagingInfo = PagingInfo.toJSON(message.pagingInfo);
        }
        return obj;
    },
    create(base) {
        return QueryGroupMessagesResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseQueryGroupMessagesResponse();
        message.messages = object.messages?.map((e) => GroupMessage.fromPartial(e)) || [];
        message.pagingInfo = (object.pagingInfo !== undefined && object.pagingInfo !== null)
            ? PagingInfo.fromPartial(object.pagingInfo)
            : undefined;
        return message;
    },
};
function createBaseQueryWelcomeMessagesRequest() {
    return { installationKey: new Uint8Array(0), pagingInfo: undefined };
}
export const QueryWelcomeMessagesRequest = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.installationKey.length !== 0) {
            writer.uint32(10).bytes(message.installationKey);
        }
        if (message.pagingInfo !== undefined) {
            PagingInfo.encode(message.pagingInfo, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryWelcomeMessagesRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.installationKey = reader.bytes();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.pagingInfo = PagingInfo.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            installationKey: isSet(object.installationKey) ? bytesFromBase64(object.installationKey) : new Uint8Array(0),
            pagingInfo: isSet(object.pagingInfo) ? PagingInfo.fromJSON(object.pagingInfo) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.installationKey.length !== 0) {
            obj.installationKey = base64FromBytes(message.installationKey);
        }
        if (message.pagingInfo !== undefined) {
            obj.pagingInfo = PagingInfo.toJSON(message.pagingInfo);
        }
        return obj;
    },
    create(base) {
        return QueryWelcomeMessagesRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseQueryWelcomeMessagesRequest();
        message.installationKey = object.installationKey ?? new Uint8Array(0);
        message.pagingInfo = (object.pagingInfo !== undefined && object.pagingInfo !== null)
            ? PagingInfo.fromPartial(object.pagingInfo)
            : undefined;
        return message;
    },
};
function createBaseQueryWelcomeMessagesResponse() {
    return { messages: [], pagingInfo: undefined };
}
export const QueryWelcomeMessagesResponse = {
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.messages) {
            WelcomeMessage.encode(v, writer.uint32(10).fork()).ldelim();
        }
        if (message.pagingInfo !== undefined) {
            PagingInfo.encode(message.pagingInfo, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryWelcomeMessagesResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.messages.push(WelcomeMessage.decode(reader, reader.uint32()));
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.pagingInfo = PagingInfo.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            messages: globalThis.Array.isArray(object?.messages)
                ? object.messages.map((e) => WelcomeMessage.fromJSON(e))
                : [],
            pagingInfo: isSet(object.pagingInfo) ? PagingInfo.fromJSON(object.pagingInfo) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.messages?.length) {
            obj.messages = message.messages.map((e) => WelcomeMessage.toJSON(e));
        }
        if (message.pagingInfo !== undefined) {
            obj.pagingInfo = PagingInfo.toJSON(message.pagingInfo);
        }
        return obj;
    },
    create(base) {
        return QueryWelcomeMessagesResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseQueryWelcomeMessagesResponse();
        message.messages = object.messages?.map((e) => WelcomeMessage.fromPartial(e)) || [];
        message.pagingInfo = (object.pagingInfo !== undefined && object.pagingInfo !== null)
            ? PagingInfo.fromPartial(object.pagingInfo)
            : undefined;
        return message;
    },
};
function createBaseSubscribeGroupMessagesRequest() {
    return { filters: [] };
}
export const SubscribeGroupMessagesRequest = {
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.filters) {
            SubscribeGroupMessagesRequest_Filter.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSubscribeGroupMessagesRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.filters.push(SubscribeGroupMessagesRequest_Filter.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            filters: globalThis.Array.isArray(object?.filters)
                ? object.filters.map((e) => SubscribeGroupMessagesRequest_Filter.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.filters?.length) {
            obj.filters = message.filters.map((e) => SubscribeGroupMessagesRequest_Filter.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return SubscribeGroupMessagesRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseSubscribeGroupMessagesRequest();
        message.filters = object.filters?.map((e) => SubscribeGroupMessagesRequest_Filter.fromPartial(e)) || [];
        return message;
    },
};
function createBaseSubscribeGroupMessagesRequest_Filter() {
    return { groupId: new Uint8Array(0), idCursor: Long.UZERO };
}
export const SubscribeGroupMessagesRequest_Filter = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.groupId.length !== 0) {
            writer.uint32(10).bytes(message.groupId);
        }
        if (!message.idCursor.equals(Long.UZERO)) {
            writer.uint32(16).uint64(message.idCursor);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSubscribeGroupMessagesRequest_Filter();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.groupId = reader.bytes();
                    continue;
                case 2:
                    if (tag !== 16) {
                        break;
                    }
                    message.idCursor = reader.uint64();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            groupId: isSet(object.groupId) ? bytesFromBase64(object.groupId) : new Uint8Array(0),
            idCursor: isSet(object.idCursor) ? Long.fromValue(object.idCursor) : Long.UZERO,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.groupId.length !== 0) {
            obj.groupId = base64FromBytes(message.groupId);
        }
        if (!message.idCursor.equals(Long.UZERO)) {
            obj.idCursor = (message.idCursor || Long.UZERO).toString();
        }
        return obj;
    },
    create(base) {
        return SubscribeGroupMessagesRequest_Filter.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseSubscribeGroupMessagesRequest_Filter();
        message.groupId = object.groupId ?? new Uint8Array(0);
        message.idCursor = (object.idCursor !== undefined && object.idCursor !== null)
            ? Long.fromValue(object.idCursor)
            : Long.UZERO;
        return message;
    },
};
function createBaseSubscribeWelcomeMessagesRequest() {
    return { filters: [] };
}
export const SubscribeWelcomeMessagesRequest = {
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.filters) {
            SubscribeWelcomeMessagesRequest_Filter.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSubscribeWelcomeMessagesRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.filters.push(SubscribeWelcomeMessagesRequest_Filter.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            filters: globalThis.Array.isArray(object?.filters)
                ? object.filters.map((e) => SubscribeWelcomeMessagesRequest_Filter.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.filters?.length) {
            obj.filters = message.filters.map((e) => SubscribeWelcomeMessagesRequest_Filter.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return SubscribeWelcomeMessagesRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseSubscribeWelcomeMessagesRequest();
        message.filters = object.filters?.map((e) => SubscribeWelcomeMessagesRequest_Filter.fromPartial(e)) || [];
        return message;
    },
};
function createBaseSubscribeWelcomeMessagesRequest_Filter() {
    return { installationKey: new Uint8Array(0), idCursor: Long.UZERO };
}
export const SubscribeWelcomeMessagesRequest_Filter = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.installationKey.length !== 0) {
            writer.uint32(10).bytes(message.installationKey);
        }
        if (!message.idCursor.equals(Long.UZERO)) {
            writer.uint32(16).uint64(message.idCursor);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSubscribeWelcomeMessagesRequest_Filter();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.installationKey = reader.bytes();
                    continue;
                case 2:
                    if (tag !== 16) {
                        break;
                    }
                    message.idCursor = reader.uint64();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            installationKey: isSet(object.installationKey) ? bytesFromBase64(object.installationKey) : new Uint8Array(0),
            idCursor: isSet(object.idCursor) ? Long.fromValue(object.idCursor) : Long.UZERO,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.installationKey.length !== 0) {
            obj.installationKey = base64FromBytes(message.installationKey);
        }
        if (!message.idCursor.equals(Long.UZERO)) {
            obj.idCursor = (message.idCursor || Long.UZERO).toString();
        }
        return obj;
    },
    create(base) {
        return SubscribeWelcomeMessagesRequest_Filter.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseSubscribeWelcomeMessagesRequest_Filter();
        message.installationKey = object.installationKey ?? new Uint8Array(0);
        message.idCursor = (object.idCursor !== undefined && object.idCursor !== null)
            ? Long.fromValue(object.idCursor)
            : Long.UZERO;
        return message;
    },
};
function createBaseBatchPublishCommitLogRequest() {
    return { requests: [] };
}
export const BatchPublishCommitLogRequest = {
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.requests) {
            PublishCommitLogRequest.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseBatchPublishCommitLogRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.requests.push(PublishCommitLogRequest.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            requests: globalThis.Array.isArray(object?.requests)
                ? object.requests.map((e) => PublishCommitLogRequest.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.requests?.length) {
            obj.requests = message.requests.map((e) => PublishCommitLogRequest.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return BatchPublishCommitLogRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseBatchPublishCommitLogRequest();
        message.requests = object.requests?.map((e) => PublishCommitLogRequest.fromPartial(e)) || [];
        return message;
    },
};
function createBasePublishCommitLogRequest() {
    return { groupId: new Uint8Array(0), serializedCommitLogEntry: new Uint8Array(0), signature: undefined };
}
export const PublishCommitLogRequest = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.groupId.length !== 0) {
            writer.uint32(10).bytes(message.groupId);
        }
        if (message.serializedCommitLogEntry.length !== 0) {
            writer.uint32(18).bytes(message.serializedCommitLogEntry);
        }
        if (message.signature !== undefined) {
            RecoverableEd25519Signature.encode(message.signature, writer.uint32(26).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBasePublishCommitLogRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.groupId = reader.bytes();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.serializedCommitLogEntry = reader.bytes();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.signature = RecoverableEd25519Signature.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            groupId: isSet(object.groupId) ? bytesFromBase64(object.groupId) : new Uint8Array(0),
            serializedCommitLogEntry: isSet(object.serializedCommitLogEntry)
                ? bytesFromBase64(object.serializedCommitLogEntry)
                : new Uint8Array(0),
            signature: isSet(object.signature) ? RecoverableEd25519Signature.fromJSON(object.signature) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.groupId.length !== 0) {
            obj.groupId = base64FromBytes(message.groupId);
        }
        if (message.serializedCommitLogEntry.length !== 0) {
            obj.serializedCommitLogEntry = base64FromBytes(message.serializedCommitLogEntry);
        }
        if (message.signature !== undefined) {
            obj.signature = RecoverableEd25519Signature.toJSON(message.signature);
        }
        return obj;
    },
    create(base) {
        return PublishCommitLogRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBasePublishCommitLogRequest();
        message.groupId = object.groupId ?? new Uint8Array(0);
        message.serializedCommitLogEntry = object.serializedCommitLogEntry ?? new Uint8Array(0);
        message.signature = (object.signature !== undefined && object.signature !== null)
            ? RecoverableEd25519Signature.fromPartial(object.signature)
            : undefined;
        return message;
    },
};
function createBaseQueryCommitLogRequest() {
    return { groupId: new Uint8Array(0), pagingInfo: undefined };
}
export const QueryCommitLogRequest = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.groupId.length !== 0) {
            writer.uint32(10).bytes(message.groupId);
        }
        if (message.pagingInfo !== undefined) {
            PagingInfo.encode(message.pagingInfo, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryCommitLogRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.groupId = reader.bytes();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.pagingInfo = PagingInfo.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            groupId: isSet(object.groupId) ? bytesFromBase64(object.groupId) : new Uint8Array(0),
            pagingInfo: isSet(object.pagingInfo) ? PagingInfo.fromJSON(object.pagingInfo) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.groupId.length !== 0) {
            obj.groupId = base64FromBytes(message.groupId);
        }
        if (message.pagingInfo !== undefined) {
            obj.pagingInfo = PagingInfo.toJSON(message.pagingInfo);
        }
        return obj;
    },
    create(base) {
        return QueryCommitLogRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseQueryCommitLogRequest();
        message.groupId = object.groupId ?? new Uint8Array(0);
        message.pagingInfo = (object.pagingInfo !== undefined && object.pagingInfo !== null)
            ? PagingInfo.fromPartial(object.pagingInfo)
            : undefined;
        return message;
    },
};
function createBaseQueryCommitLogResponse() {
    return { groupId: new Uint8Array(0), commitLogEntries: [], pagingInfo: undefined };
}
export const QueryCommitLogResponse = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.groupId.length !== 0) {
            writer.uint32(10).bytes(message.groupId);
        }
        for (const v of message.commitLogEntries) {
            CommitLogEntry.encode(v, writer.uint32(18).fork()).ldelim();
        }
        if (message.pagingInfo !== undefined) {
            PagingInfo.encode(message.pagingInfo, writer.uint32(26).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseQueryCommitLogResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.groupId = reader.bytes();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.commitLogEntries.push(CommitLogEntry.decode(reader, reader.uint32()));
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.pagingInfo = PagingInfo.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            groupId: isSet(object.groupId) ? bytesFromBase64(object.groupId) : new Uint8Array(0),
            commitLogEntries: globalThis.Array.isArray(object?.commitLogEntries)
                ? object.commitLogEntries.map((e) => CommitLogEntry.fromJSON(e))
                : [],
            pagingInfo: isSet(object.pagingInfo) ? PagingInfo.fromJSON(object.pagingInfo) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.groupId.length !== 0) {
            obj.groupId = base64FromBytes(message.groupId);
        }
        if (message.commitLogEntries?.length) {
            obj.commitLogEntries = message.commitLogEntries.map((e) => CommitLogEntry.toJSON(e));
        }
        if (message.pagingInfo !== undefined) {
            obj.pagingInfo = PagingInfo.toJSON(message.pagingInfo);
        }
        return obj;
    },
    create(base) {
        return QueryCommitLogResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseQueryCommitLogResponse();
        message.groupId = object.groupId ?? new Uint8Array(0);
        message.commitLogEntries = object.commitLogEntries?.map((e) => CommitLogEntry.fromPartial(e)) || [];
        message.pagingInfo = (object.pagingInfo !== undefined && object.pagingInfo !== null)
            ? PagingInfo.fromPartial(object.pagingInfo)
            : undefined;
        return message;
    },
};
function createBaseBatchQueryCommitLogRequest() {
    return { requests: [] };
}
export const BatchQueryCommitLogRequest = {
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.requests) {
            QueryCommitLogRequest.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseBatchQueryCommitLogRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.requests.push(QueryCommitLogRequest.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            requests: globalThis.Array.isArray(object?.requests)
                ? object.requests.map((e) => QueryCommitLogRequest.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.requests?.length) {
            obj.requests = message.requests.map((e) => QueryCommitLogRequest.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return BatchQueryCommitLogRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseBatchQueryCommitLogRequest();
        message.requests = object.requests?.map((e) => QueryCommitLogRequest.fromPartial(e)) || [];
        return message;
    },
};
function createBaseBatchQueryCommitLogResponse() {
    return { responses: [] };
}
export const BatchQueryCommitLogResponse = {
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.responses) {
            QueryCommitLogResponse.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseBatchQueryCommitLogResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.responses.push(QueryCommitLogResponse.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            responses: globalThis.Array.isArray(object?.responses)
                ? object.responses.map((e) => QueryCommitLogResponse.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.responses?.length) {
            obj.responses = message.responses.map((e) => QueryCommitLogResponse.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return BatchQueryCommitLogResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseBatchQueryCommitLogResponse();
        message.responses = object.responses?.map((e) => QueryCommitLogResponse.fromPartial(e)) || [];
        return message;
    },
};
function createBaseGetNewestGroupMessageRequest() {
    return { groupIds: [], includeContent: false };
}
export const GetNewestGroupMessageRequest = {
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.groupIds) {
            writer.uint32(10).bytes(v);
        }
        if (message.includeContent !== false) {
            writer.uint32(16).bool(message.includeContent);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetNewestGroupMessageRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.groupIds.push(reader.bytes());
                    continue;
                case 2:
                    if (tag !== 16) {
                        break;
                    }
                    message.includeContent = reader.bool();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            groupIds: globalThis.Array.isArray(object?.groupIds) ? object.groupIds.map((e) => bytesFromBase64(e)) : [],
            includeContent: isSet(object.includeContent) ? globalThis.Boolean(object.includeContent) : false,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.groupIds?.length) {
            obj.groupIds = message.groupIds.map((e) => base64FromBytes(e));
        }
        if (message.includeContent !== false) {
            obj.includeContent = message.includeContent;
        }
        return obj;
    },
    create(base) {
        return GetNewestGroupMessageRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGetNewestGroupMessageRequest();
        message.groupIds = object.groupIds?.map((e) => e) || [];
        message.includeContent = object.includeContent ?? false;
        return message;
    },
};
function createBaseGetNewestGroupMessageResponse() {
    return { responses: [] };
}
export const GetNewestGroupMessageResponse = {
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.responses) {
            GetNewestGroupMessageResponse_Response.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetNewestGroupMessageResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.responses.push(GetNewestGroupMessageResponse_Response.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            responses: globalThis.Array.isArray(object?.responses)
                ? object.responses.map((e) => GetNewestGroupMessageResponse_Response.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.responses?.length) {
            obj.responses = message.responses.map((e) => GetNewestGroupMessageResponse_Response.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return GetNewestGroupMessageResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGetNewestGroupMessageResponse();
        message.responses = object.responses?.map((e) => GetNewestGroupMessageResponse_Response.fromPartial(e)) || [];
        return message;
    },
};
function createBaseGetNewestGroupMessageResponse_Response() {
    return { groupMessage: undefined };
}
export const GetNewestGroupMessageResponse_Response = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.groupMessage !== undefined) {
            GroupMessage.encode(message.groupMessage, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetNewestGroupMessageResponse_Response();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.groupMessage = GroupMessage.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { groupMessage: isSet(object.groupMessage) ? GroupMessage.fromJSON(object.groupMessage) : undefined };
    },
    toJSON(message) {
        const obj = {};
        if (message.groupMessage !== undefined) {
            obj.groupMessage = GroupMessage.toJSON(message.groupMessage);
        }
        return obj;
    },
    create(base) {
        return GetNewestGroupMessageResponse_Response.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGetNewestGroupMessageResponse_Response();
        message.groupMessage = (object.groupMessage !== undefined && object.groupMessage !== null)
            ? GroupMessage.fromPartial(object.groupMessage)
            : undefined;
        return message;
    },
};
export const MlsApiServiceName = "xmtp.mls.api.v1.MlsApi";
export class MlsApiClientImpl {
    rpc;
    service;
    constructor(rpc, opts) {
        this.service = opts?.service || MlsApiServiceName;
        this.rpc = rpc;
        this.SendGroupMessages = this.SendGroupMessages.bind(this);
        this.SendWelcomeMessages = this.SendWelcomeMessages.bind(this);
        this.RegisterInstallation = this.RegisterInstallation.bind(this);
        this.UploadKeyPackage = this.UploadKeyPackage.bind(this);
        this.FetchKeyPackages = this.FetchKeyPackages.bind(this);
        this.RevokeInstallation = this.RevokeInstallation.bind(this);
        this.GetIdentityUpdates = this.GetIdentityUpdates.bind(this);
        this.QueryGroupMessages = this.QueryGroupMessages.bind(this);
        this.QueryWelcomeMessages = this.QueryWelcomeMessages.bind(this);
        this.SubscribeGroupMessages = this.SubscribeGroupMessages.bind(this);
        this.SubscribeWelcomeMessages = this.SubscribeWelcomeMessages.bind(this);
        this.BatchPublishCommitLog = this.BatchPublishCommitLog.bind(this);
        this.BatchQueryCommitLog = this.BatchQueryCommitLog.bind(this);
        this.GetNewestGroupMessage = this.GetNewestGroupMessage.bind(this);
    }
    SendGroupMessages(request) {
        const data = SendGroupMessagesRequest.encode(request).finish();
        const promise = this.rpc.request(this.service, "SendGroupMessages", data);
        return promise.then((data) => Empty.decode(_m0.Reader.create(data)));
    }
    SendWelcomeMessages(request) {
        const data = SendWelcomeMessagesRequest.encode(request).finish();
        const promise = this.rpc.request(this.service, "SendWelcomeMessages", data);
        return promise.then((data) => Empty.decode(_m0.Reader.create(data)));
    }
    RegisterInstallation(request) {
        const data = RegisterInstallationRequest.encode(request).finish();
        const promise = this.rpc.request(this.service, "RegisterInstallation", data);
        return promise.then((data) => RegisterInstallationResponse.decode(_m0.Reader.create(data)));
    }
    UploadKeyPackage(request) {
        const data = UploadKeyPackageRequest.encode(request).finish();
        const promise = this.rpc.request(this.service, "UploadKeyPackage", data);
        return promise.then((data) => Empty.decode(_m0.Reader.create(data)));
    }
    FetchKeyPackages(request) {
        const data = FetchKeyPackagesRequest.encode(request).finish();
        const promise = this.rpc.request(this.service, "FetchKeyPackages", data);
        return promise.then((data) => FetchKeyPackagesResponse.decode(_m0.Reader.create(data)));
    }
    RevokeInstallation(request) {
        const data = RevokeInstallationRequest.encode(request).finish();
        const promise = this.rpc.request(this.service, "RevokeInstallation", data);
        return promise.then((data) => Empty.decode(_m0.Reader.create(data)));
    }
    GetIdentityUpdates(request) {
        const data = GetIdentityUpdatesRequest.encode(request).finish();
        const promise = this.rpc.request(this.service, "GetIdentityUpdates", data);
        return promise.then((data) => GetIdentityUpdatesResponse.decode(_m0.Reader.create(data)));
    }
    QueryGroupMessages(request) {
        const data = QueryGroupMessagesRequest.encode(request).finish();
        const promise = this.rpc.request(this.service, "QueryGroupMessages", data);
        return promise.then((data) => QueryGroupMessagesResponse.decode(_m0.Reader.create(data)));
    }
    QueryWelcomeMessages(request) {
        const data = QueryWelcomeMessagesRequest.encode(request).finish();
        const promise = this.rpc.request(this.service, "QueryWelcomeMessages", data);
        return promise.then((data) => QueryWelcomeMessagesResponse.decode(_m0.Reader.create(data)));
    }
    SubscribeGroupMessages(request) {
        const data = SubscribeGroupMessagesRequest.encode(request).finish();
        const result = this.rpc.serverStreamingRequest(this.service, "SubscribeGroupMessages", data);
        return result.pipe(map((data) => GroupMessage.decode(_m0.Reader.create(data))));
    }
    SubscribeWelcomeMessages(request) {
        const data = SubscribeWelcomeMessagesRequest.encode(request).finish();
        const result = this.rpc.serverStreamingRequest(this.service, "SubscribeWelcomeMessages", data);
        return result.pipe(map((data) => WelcomeMessage.decode(_m0.Reader.create(data))));
    }
    BatchPublishCommitLog(request) {
        const data = BatchPublishCommitLogRequest.encode(request).finish();
        const promise = this.rpc.request(this.service, "BatchPublishCommitLog", data);
        return promise.then((data) => Empty.decode(_m0.Reader.create(data)));
    }
    BatchQueryCommitLog(request) {
        const data = BatchQueryCommitLogRequest.encode(request).finish();
        const promise = this.rpc.request(this.service, "BatchQueryCommitLog", data);
        return promise.then((data) => BatchQueryCommitLogResponse.decode(_m0.Reader.create(data)));
    }
    GetNewestGroupMessage(request) {
        const data = GetNewestGroupMessageRequest.encode(request).finish();
        const promise = this.rpc.request(this.service, "GetNewestGroupMessage", data);
        return promise.then((data) => GetNewestGroupMessageResponse.decode(_m0.Reader.create(data)));
    }
}
function bytesFromBase64(b64) {
    if (globalThis.Buffer) {
        return Uint8Array.from(globalThis.Buffer.from(b64, "base64"));
    }
    else {
        const bin = globalThis.atob(b64);
        const arr = new Uint8Array(bin.length);
        for (let i = 0; i < bin.length; ++i) {
            arr[i] = bin.charCodeAt(i);
        }
        return arr;
    }
}
function base64FromBytes(arr) {
    if (globalThis.Buffer) {
        return globalThis.Buffer.from(arr).toString("base64");
    }
    else {
        const bin = [];
        arr.forEach((byte) => {
            bin.push(globalThis.String.fromCharCode(byte));
        });
        return globalThis.btoa(bin.join(""));
    }
}
if (_m0.util.Long !== Long) {
    _m0.util.Long = Long;
    _m0.configure();
}
function isSet(value) {
    return value !== null && value !== undefined;
}
