// @xmtp/proto/ts/dist/cjs/device_sync/content.pb.js
"use strict";
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v1.181.2
//   protoc               v3.21.12
// source: device_sync/content.proto
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeviceSyncKeyType = exports.DeviceSyncReply = exports.DeviceSyncRequest = exports.HmacKeyUpdate = exports.V1UserPreferenceUpdate = exports.PreferenceUpdate = exports.PreferenceUpdates = exports.DeviceSyncAcknowledge = exports.DeviceSyncContent = exports.protobufPackage = void 0;
/* eslint-disable */
const long_1 = __importDefault(require("long"));
const minimal_1 = __importDefault(require("protobufjs/minimal"));
const consent_backup_pb_1 = require("./consent_backup.pb");
const device_sync_pb_1 = require("./device_sync.pb");
exports.protobufPackage = "xmtp.device_sync.content";
function createBaseDeviceSyncContent() {
    return { request: undefined, acknowledge: undefined, reply: undefined, preferenceUpdates: undefined };
}
exports.DeviceSyncContent = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.request !== undefined) {
            exports.DeviceSyncRequest.encode(message.request, writer.uint32(10).fork()).ldelim();
        }
        if (message.acknowledge !== undefined) {
            exports.DeviceSyncAcknowledge.encode(message.acknowledge, writer.uint32(18).fork()).ldelim();
        }
        if (message.reply !== undefined) {
            exports.DeviceSyncReply.encode(message.reply, writer.uint32(26).fork()).ldelim();
        }
        if (message.preferenceUpdates !== undefined) {
            exports.PreferenceUpdates.encode(message.preferenceUpdates, writer.uint32(34).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseDeviceSyncContent();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.request = exports.DeviceSyncRequest.decode(reader, reader.uint32());
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.acknowledge = exports.DeviceSyncAcknowledge.decode(reader, reader.uint32());
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.reply = exports.DeviceSyncReply.decode(reader, reader.uint32());
                    continue;
                case 4:
                    if (tag !== 34) {
                        break;
                    }
                    message.preferenceUpdates = exports.PreferenceUpdates.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            request: isSet(object.request) ? exports.DeviceSyncRequest.fromJSON(object.request) : undefined,
            acknowledge: isSet(object.acknowledge) ? exports.DeviceSyncAcknowledge.fromJSON(object.acknowledge) : undefined,
            reply: isSet(object.reply) ? exports.DeviceSyncReply.fromJSON(object.reply) : undefined,
            preferenceUpdates: isSet(object.preferenceUpdates)
                ? exports.PreferenceUpdates.fromJSON(object.preferenceUpdates)
                : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.request !== undefined) {
            obj.request = exports.DeviceSyncRequest.toJSON(message.request);
        }
        if (message.acknowledge !== undefined) {
            obj.acknowledge = exports.DeviceSyncAcknowledge.toJSON(message.acknowledge);
        }
        if (message.reply !== undefined) {
            obj.reply = exports.DeviceSyncReply.toJSON(message.reply);
        }
        if (message.preferenceUpdates !== undefined) {
            obj.preferenceUpdates = exports.PreferenceUpdates.toJSON(message.preferenceUpdates);
        }
        return obj;
    },
    create(base) {
        return exports.DeviceSyncContent.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseDeviceSyncContent();
        message.request = (object.request !== undefined && object.request !== null)
            ? exports.DeviceSyncRequest.fromPartial(object.request)
            : undefined;
        message.acknowledge = (object.acknowledge !== undefined && object.acknowledge !== null)
            ? exports.DeviceSyncAcknowledge.fromPartial(object.acknowledge)
            : undefined;
        message.reply = (object.reply !== undefined && object.reply !== null)
            ? exports.DeviceSyncReply.fromPartial(object.reply)
            : undefined;
        message.preferenceUpdates = (object.preferenceUpdates !== undefined && object.preferenceUpdates !== null)
            ? exports.PreferenceUpdates.fromPartial(object.preferenceUpdates)
            : undefined;
        return message;
    },
};
function createBaseDeviceSyncAcknowledge() {
    return { requestId: "" };
}
exports.DeviceSyncAcknowledge = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.requestId !== "") {
            writer.uint32(10).string(message.requestId);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseDeviceSyncAcknowledge();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.requestId = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { requestId: isSet(object.requestId) ? globalThis.String(object.requestId) : "" };
    },
    toJSON(message) {
        const obj = {};
        if (message.requestId !== "") {
            obj.requestId = message.requestId;
        }
        return obj;
    },
    create(base) {
        return exports.DeviceSyncAcknowledge.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseDeviceSyncAcknowledge();
        message.requestId = object.requestId ?? "";
        return message;
    },
};
function createBasePreferenceUpdates() {
    return { updates: [] };
}
exports.PreferenceUpdates = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        for (const v of message.updates) {
            exports.PreferenceUpdate.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBasePreferenceUpdates();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.updates.push(exports.PreferenceUpdate.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            updates: globalThis.Array.isArray(object?.updates)
                ? object.updates.map((e) => exports.PreferenceUpdate.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.updates?.length) {
            obj.updates = message.updates.map((e) => exports.PreferenceUpdate.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return exports.PreferenceUpdates.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBasePreferenceUpdates();
        message.updates = object.updates?.map((e) => exports.PreferenceUpdate.fromPartial(e)) || [];
        return message;
    },
};
function createBasePreferenceUpdate() {
    return { consent: undefined, hmac: undefined };
}
exports.PreferenceUpdate = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.consent !== undefined) {
            consent_backup_pb_1.ConsentSave.encode(message.consent, writer.uint32(10).fork()).ldelim();
        }
        if (message.hmac !== undefined) {
            exports.HmacKeyUpdate.encode(message.hmac, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBasePreferenceUpdate();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.consent = consent_backup_pb_1.ConsentSave.decode(reader, reader.uint32());
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.hmac = exports.HmacKeyUpdate.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            consent: isSet(object.consent) ? consent_backup_pb_1.ConsentSave.fromJSON(object.consent) : undefined,
            hmac: isSet(object.hmac) ? exports.HmacKeyUpdate.fromJSON(object.hmac) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.consent !== undefined) {
            obj.consent = consent_backup_pb_1.ConsentSave.toJSON(message.consent);
        }
        if (message.hmac !== undefined) {
            obj.hmac = exports.HmacKeyUpdate.toJSON(message.hmac);
        }
        return obj;
    },
    create(base) {
        return exports.PreferenceUpdate.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBasePreferenceUpdate();
        message.consent = (object.consent !== undefined && object.consent !== null)
            ? consent_backup_pb_1.ConsentSave.fromPartial(object.consent)
            : undefined;
        message.hmac = (object.hmac !== undefined && object.hmac !== null)
            ? exports.HmacKeyUpdate.fromPartial(object.hmac)
            : undefined;
        return message;
    },
};
function createBaseV1UserPreferenceUpdate() {
    return { contents: [] };
}
exports.V1UserPreferenceUpdate = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        for (const v of message.contents) {
            writer.uint32(10).bytes(v);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseV1UserPreferenceUpdate();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.contents.push(reader.bytes());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            contents: globalThis.Array.isArray(object?.contents) ? object.contents.map((e) => bytesFromBase64(e)) : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.contents?.length) {
            obj.contents = message.contents.map((e) => base64FromBytes(e));
        }
        return obj;
    },
    create(base) {
        return exports.V1UserPreferenceUpdate.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseV1UserPreferenceUpdate();
        message.contents = object.contents?.map((e) => e) || [];
        return message;
    },
};
function createBaseHmacKeyUpdate() {
    return { key: new Uint8Array(0), cycledAtNs: long_1.default.ZERO };
}
exports.HmacKeyUpdate = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.key.length !== 0) {
            writer.uint32(10).bytes(message.key);
        }
        if (!message.cycledAtNs.equals(long_1.default.ZERO)) {
            writer.uint32(16).int64(message.cycledAtNs);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseHmacKeyUpdate();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.key = reader.bytes();
                    continue;
                case 2:
                    if (tag !== 16) {
                        break;
                    }
                    message.cycledAtNs = reader.int64();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            key: isSet(object.key) ? bytesFromBase64(object.key) : new Uint8Array(0),
            cycledAtNs: isSet(object.cycledAtNs) ? long_1.default.fromValue(object.cycledAtNs) : long_1.default.ZERO,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.key.length !== 0) {
            obj.key = base64FromBytes(message.key);
        }
        if (!message.cycledAtNs.equals(long_1.default.ZERO)) {
            obj.cycledAtNs = (message.cycledAtNs || long_1.default.ZERO).toString();
        }
        return obj;
    },
    create(base) {
        return exports.HmacKeyUpdate.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseHmacKeyUpdate();
        message.key = object.key ?? new Uint8Array(0);
        message.cycledAtNs = (object.cycledAtNs !== undefined && object.cycledAtNs !== null)
            ? long_1.default.fromValue(object.cycledAtNs)
            : long_1.default.ZERO;
        return message;
    },
};
function createBaseDeviceSyncRequest() {
    return { pin: "", serverUrl: "", deprecatedKind: 0, options: undefined };
}
exports.DeviceSyncRequest = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.pin !== "") {
            writer.uint32(10).string(message.pin);
        }
        if (message.serverUrl !== "") {
            writer.uint32(18).string(message.serverUrl);
        }
        if (message.deprecatedKind !== 0) {
            writer.uint32(24).int32(message.deprecatedKind);
        }
        if (message.options !== undefined) {
            device_sync_pb_1.ArchiveOptions.encode(message.options, writer.uint32(34).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseDeviceSyncRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.pin = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.serverUrl = reader.string();
                    continue;
                case 3:
                    if (tag !== 24) {
                        break;
                    }
                    message.deprecatedKind = reader.int32();
                    continue;
                case 4:
                    if (tag !== 34) {
                        break;
                    }
                    message.options = device_sync_pb_1.ArchiveOptions.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            pin: isSet(object.pin) ? globalThis.String(object.pin) : "",
            serverUrl: isSet(object.serverUrl) ? globalThis.String(object.serverUrl) : "",
            deprecatedKind: isSet(object.deprecatedKind) ? (0, device_sync_pb_1.backupElementSelectionFromJSON)(object.deprecatedKind) : 0,
            options: isSet(object.options) ? device_sync_pb_1.ArchiveOptions.fromJSON(object.options) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.pin !== "") {
            obj.pin = message.pin;
        }
        if (message.serverUrl !== "") {
            obj.serverUrl = message.serverUrl;
        }
        if (message.deprecatedKind !== 0) {
            obj.deprecatedKind = (0, device_sync_pb_1.backupElementSelectionToJSON)(message.deprecatedKind);
        }
        if (message.options !== undefined) {
            obj.options = device_sync_pb_1.ArchiveOptions.toJSON(message.options);
        }
        return obj;
    },
    create(base) {
        return exports.DeviceSyncRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseDeviceSyncRequest();
        message.pin = object.pin ?? "";
        message.serverUrl = object.serverUrl ?? "";
        message.deprecatedKind = object.deprecatedKind ?? 0;
        message.options = (object.options !== undefined && object.options !== null)
            ? device_sync_pb_1.ArchiveOptions.fromPartial(object.options)
            : undefined;
        return message;
    },
};
function createBaseDeviceSyncReply() {
    return { requestId: "", url: "", encryptionKey: undefined, timestampNs: long_1.default.UZERO, kind: 0, metadata: undefined };
}
exports.DeviceSyncReply = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.requestId !== "") {
            writer.uint32(10).string(message.requestId);
        }
        if (message.url !== "") {
            writer.uint32(18).string(message.url);
        }
        if (message.encryptionKey !== undefined) {
            exports.DeviceSyncKeyType.encode(message.encryptionKey, writer.uint32(26).fork()).ldelim();
        }
        if (!message.timestampNs.equals(long_1.default.UZERO)) {
            writer.uint32(32).uint64(message.timestampNs);
        }
        if (message.kind !== 0) {
            writer.uint32(40).int32(message.kind);
        }
        if (message.metadata !== undefined) {
            device_sync_pb_1.BackupMetadataSave.encode(message.metadata, writer.uint32(50).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseDeviceSyncReply();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.requestId = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.url = reader.string();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.encryptionKey = exports.DeviceSyncKeyType.decode(reader, reader.uint32());
                    continue;
                case 4:
                    if (tag !== 32) {
                        break;
                    }
                    message.timestampNs = reader.uint64();
                    continue;
                case 5:
                    if (tag !== 40) {
                        break;
                    }
                    message.kind = reader.int32();
                    continue;
                case 6:
                    if (tag !== 50) {
                        break;
                    }
                    message.metadata = device_sync_pb_1.BackupMetadataSave.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            requestId: isSet(object.requestId) ? globalThis.String(object.requestId) : "",
            url: isSet(object.url) ? globalThis.String(object.url) : "",
            encryptionKey: isSet(object.encryptionKey) ? exports.DeviceSyncKeyType.fromJSON(object.encryptionKey) : undefined,
            timestampNs: isSet(object.timestampNs) ? long_1.default.fromValue(object.timestampNs) : long_1.default.UZERO,
            kind: isSet(object.kind) ? (0, device_sync_pb_1.backupElementSelectionFromJSON)(object.kind) : 0,
            metadata: isSet(object.metadata) ? device_sync_pb_1.BackupMetadataSave.fromJSON(object.metadata) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.requestId !== "") {
            obj.requestId = message.requestId;
        }
        if (message.url !== "") {
            obj.url = message.url;
        }
        if (message.encryptionKey !== undefined) {
            obj.encryptionKey = exports.DeviceSyncKeyType.toJSON(message.encryptionKey);
        }
        if (!message.timestampNs.equals(long_1.default.UZERO)) {
            obj.timestampNs = (message.timestampNs || long_1.default.UZERO).toString();
        }
        if (message.kind !== 0) {
            obj.kind = (0, device_sync_pb_1.backupElementSelectionToJSON)(message.kind);
        }
        if (message.metadata !== undefined) {
            obj.metadata = device_sync_pb_1.BackupMetadataSave.toJSON(message.metadata);
        }
        return obj;
    },
    create(base) {
        return exports.DeviceSyncReply.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseDeviceSyncReply();
        message.requestId = object.requestId ?? "";
        message.url = object.url ?? "";
        message.encryptionKey = (object.encryptionKey !== undefined && object.encryptionKey !== null)
            ? exports.DeviceSyncKeyType.fromPartial(object.encryptionKey)
            : undefined;
        message.timestampNs = (object.timestampNs !== undefined && object.timestampNs !== null)
            ? long_1.default.fromValue(object.timestampNs)
            : long_1.default.UZERO;
        message.kind = object.kind ?? 0;
        message.metadata = (object.metadata !== undefined && object.metadata !== null)
            ? device_sync_pb_1.BackupMetadataSave.fromPartial(object.metadata)
            : undefined;
        return message;
    },
};
function createBaseDeviceSyncKeyType() {
    return { aes256Gcm: undefined };
}
exports.DeviceSyncKeyType = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.aes256Gcm !== undefined) {
            writer.uint32(10).bytes(message.aes256Gcm);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseDeviceSyncKeyType();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.aes256Gcm = reader.bytes();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { aes256Gcm: isSet(object.aes256Gcm) ? bytesFromBase64(object.aes256Gcm) : undefined };
    },
    toJSON(message) {
        const obj = {};
        if (message.aes256Gcm !== undefined) {
            obj.aes256Gcm = base64FromBytes(message.aes256Gcm);
        }
        return obj;
    },
    create(base) {
        return exports.DeviceSyncKeyType.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseDeviceSyncKeyType();
        message.aes256Gcm = object.aes256Gcm ?? undefined;
        return message;
    },
};
function bytesFromBase64(b64) {
    if (globalThis.Buffer) {
        return Uint8Array.from(globalThis.Buffer.from(b64, "base64"));
    }
    else {
        const bin = globalThis.atob(b64);
        const arr = new Uint8Array(bin.length);
        for (let i = 0; i < bin.length; ++i) {
            arr[i] = bin.charCodeAt(i);
        }
        return arr;
    }
}
function base64FromBytes(arr) {
    if (globalThis.Buffer) {
        return globalThis.Buffer.from(arr).toString("base64");
    }
    else {
        const bin = [];
        arr.forEach((byte) => {
            bin.push(globalThis.String.fromCharCode(byte));
        });
        return globalThis.btoa(bin.join(""));
    }
}
if (minimal_1.default.util.Long !== long_1.default) {
    minimal_1.default.util.Long = long_1.default;
    minimal_1.default.configure();
}
function isSet(value) {
    return value !== null && value !== undefined;
}
