// @xmtp/proto/ts/dist/cjs/device_sync/event_backup.pb.js
"use strict";
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v1.181.2
//   protoc               v3.21.12
// source: device_sync/event_backup.proto
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.EventSave = exports.EventLevelSave = exports.protobufPackage = void 0;
exports.eventLevelSaveFromJSON = eventLevelSaveFromJSON;
exports.eventLevelSaveToJSON = eventLevelSaveToJSON;
/* eslint-disable */
const long_1 = __importDefault(require("long"));
const minimal_1 = __importDefault(require("protobufjs/minimal"));
exports.protobufPackage = "xmtp.device_sync.event_backup";
/** Definitions for backups */
/** @deprecated */
var EventLevelSave;
(function (EventLevelSave) {
    EventLevelSave[EventLevelSave["EVENT_LEVEL_SAVE_UNSPECIFIED"] = 0] = "EVENT_LEVEL_SAVE_UNSPECIFIED";
    EventLevelSave[EventLevelSave["EVENT_LEVEL_SAVE_NONE"] = 1] = "EVENT_LEVEL_SAVE_NONE";
    EventLevelSave[EventLevelSave["EVENT_LEVEL_SAVE_SUCCESS"] = 2] = "EVENT_LEVEL_SAVE_SUCCESS";
    EventLevelSave[EventLevelSave["EVENT_LEVEL_SAVE_WARN"] = 3] = "EVENT_LEVEL_SAVE_WARN";
    EventLevelSave[EventLevelSave["EVENT_LEVEL_SAVE_ERROR"] = 4] = "EVENT_LEVEL_SAVE_ERROR";
    EventLevelSave[EventLevelSave["EVENT_LEVEL_SAVE_FAULT"] = 5] = "EVENT_LEVEL_SAVE_FAULT";
    EventLevelSave[EventLevelSave["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(EventLevelSave || (exports.EventLevelSave = EventLevelSave = {}));
function eventLevelSaveFromJSON(object) {
    switch (object) {
        case 0:
        case "EVENT_LEVEL_SAVE_UNSPECIFIED":
            return EventLevelSave.EVENT_LEVEL_SAVE_UNSPECIFIED;
        case 1:
        case "EVENT_LEVEL_SAVE_NONE":
            return EventLevelSave.EVENT_LEVEL_SAVE_NONE;
        case 2:
        case "EVENT_LEVEL_SAVE_SUCCESS":
            return EventLevelSave.EVENT_LEVEL_SAVE_SUCCESS;
        case 3:
        case "EVENT_LEVEL_SAVE_WARN":
            return EventLevelSave.EVENT_LEVEL_SAVE_WARN;
        case 4:
        case "EVENT_LEVEL_SAVE_ERROR":
            return EventLevelSave.EVENT_LEVEL_SAVE_ERROR;
        case 5:
        case "EVENT_LEVEL_SAVE_FAULT":
            return EventLevelSave.EVENT_LEVEL_SAVE_FAULT;
        case -1:
        case "UNRECOGNIZED":
        default:
            return EventLevelSave.UNRECOGNIZED;
    }
}
function eventLevelSaveToJSON(object) {
    switch (object) {
        case EventLevelSave.EVENT_LEVEL_SAVE_UNSPECIFIED:
            return "EVENT_LEVEL_SAVE_UNSPECIFIED";
        case EventLevelSave.EVENT_LEVEL_SAVE_NONE:
            return "EVENT_LEVEL_SAVE_NONE";
        case EventLevelSave.EVENT_LEVEL_SAVE_SUCCESS:
            return "EVENT_LEVEL_SAVE_SUCCESS";
        case EventLevelSave.EVENT_LEVEL_SAVE_WARN:
            return "EVENT_LEVEL_SAVE_WARN";
        case EventLevelSave.EVENT_LEVEL_SAVE_ERROR:
            return "EVENT_LEVEL_SAVE_ERROR";
        case EventLevelSave.EVENT_LEVEL_SAVE_FAULT:
            return "EVENT_LEVEL_SAVE_FAULT";
        case EventLevelSave.UNRECOGNIZED:
        default:
            return "UNRECOGNIZED";
    }
}
function createBaseEventSave() {
    return {
        createdAtNs: long_1.default.ZERO,
        event: "",
        details: new Uint8Array(0),
        groupId: undefined,
        level: 0,
        icon: undefined,
    };
}
exports.EventSave = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (!message.createdAtNs.equals(long_1.default.ZERO)) {
            writer.uint32(8).int64(message.createdAtNs);
        }
        if (message.event !== "") {
            writer.uint32(18).string(message.event);
        }
        if (message.details.length !== 0) {
            writer.uint32(26).bytes(message.details);
        }
        if (message.groupId !== undefined) {
            writer.uint32(34).bytes(message.groupId);
        }
        if (message.level !== 0) {
            writer.uint32(40).int32(message.level);
        }
        if (message.icon !== undefined) {
            writer.uint32(50).string(message.icon);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseEventSave();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 8) {
                        break;
                    }
                    message.createdAtNs = reader.int64();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.event = reader.string();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.details = reader.bytes();
                    continue;
                case 4:
                    if (tag !== 34) {
                        break;
                    }
                    message.groupId = reader.bytes();
                    continue;
                case 5:
                    if (tag !== 40) {
                        break;
                    }
                    message.level = reader.int32();
                    continue;
                case 6:
                    if (tag !== 50) {
                        break;
                    }
                    message.icon = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            createdAtNs: isSet(object.createdAtNs) ? long_1.default.fromValue(object.createdAtNs) : long_1.default.ZERO,
            event: isSet(object.event) ? globalThis.String(object.event) : "",
            details: isSet(object.details) ? bytesFromBase64(object.details) : new Uint8Array(0),
            groupId: isSet(object.groupId) ? bytesFromBase64(object.groupId) : undefined,
            level: isSet(object.level) ? eventLevelSaveFromJSON(object.level) : 0,
            icon: isSet(object.icon) ? globalThis.String(object.icon) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (!message.createdAtNs.equals(long_1.default.ZERO)) {
            obj.createdAtNs = (message.createdAtNs || long_1.default.ZERO).toString();
        }
        if (message.event !== "") {
            obj.event = message.event;
        }
        if (message.details.length !== 0) {
            obj.details = base64FromBytes(message.details);
        }
        if (message.groupId !== undefined) {
            obj.groupId = base64FromBytes(message.groupId);
        }
        if (message.level !== 0) {
            obj.level = eventLevelSaveToJSON(message.level);
        }
        if (message.icon !== undefined) {
            obj.icon = message.icon;
        }
        return obj;
    },
    create(base) {
        return exports.EventSave.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseEventSave();
        message.createdAtNs = (object.createdAtNs !== undefined && object.createdAtNs !== null)
            ? long_1.default.fromValue(object.createdAtNs)
            : long_1.default.ZERO;
        message.event = object.event ?? "";
        message.details = object.details ?? new Uint8Array(0);
        message.groupId = object.groupId ?? undefined;
        message.level = object.level ?? 0;
        message.icon = object.icon ?? undefined;
        return message;
    },
};
function bytesFromBase64(b64) {
    if (globalThis.Buffer) {
        return Uint8Array.from(globalThis.Buffer.from(b64, "base64"));
    }
    else {
        const bin = globalThis.atob(b64);
        const arr = new Uint8Array(bin.length);
        for (let i = 0; i < bin.length; ++i) {
            arr[i] = bin.charCodeAt(i);
        }
        return arr;
    }
}
function base64FromBytes(arr) {
    if (globalThis.Buffer) {
        return globalThis.Buffer.from(arr).toString("base64");
    }
    else {
        const bin = [];
        arr.forEach((byte) => {
            bin.push(globalThis.String.fromCharCode(byte));
        });
        return globalThis.btoa(bin.join(""));
    }
}
if (minimal_1.default.util.Long !== long_1.default) {
    minimal_1.default.util.Long = long_1.default;
    minimal_1.default.configure();
}
function isSet(value) {
    return value !== null && value !== undefined;
}
