// @xmtp/proto/ts/dist/cjs/device_sync/group_backup.pb.js
"use strict";
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v1.181.2
//   protoc               v3.21.12
// source: device_sync/group_backup.proto
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ImmutableMetadataSave = exports.MutableMetadataSave_AttributesEntry = exports.MutableMetadataSave = exports.GroupSave = exports.ConversationTypeSave = exports.GroupMembershipStateSave = exports.protobufPackage = void 0;
exports.groupMembershipStateSaveFromJSON = groupMembershipStateSaveFromJSON;
exports.groupMembershipStateSaveToJSON = groupMembershipStateSaveToJSON;
exports.conversationTypeSaveFromJSON = conversationTypeSaveFromJSON;
exports.conversationTypeSaveToJSON = conversationTypeSaveToJSON;
/* eslint-disable */
const long_1 = __importDefault(require("long"));
const minimal_1 = __importDefault(require("protobufjs/minimal"));
exports.protobufPackage = "xmtp.device_sync.group_backup";
/** Definitions for backups */
/** Group membership state */
var GroupMembershipStateSave;
(function (GroupMembershipStateSave) {
    GroupMembershipStateSave[GroupMembershipStateSave["GROUP_MEMBERSHIP_STATE_SAVE_UNSPECIFIED"] = 0] = "GROUP_MEMBERSHIP_STATE_SAVE_UNSPECIFIED";
    GroupMembershipStateSave[GroupMembershipStateSave["GROUP_MEMBERSHIP_STATE_SAVE_ALLOWED"] = 1] = "GROUP_MEMBERSHIP_STATE_SAVE_ALLOWED";
    GroupMembershipStateSave[GroupMembershipStateSave["GROUP_MEMBERSHIP_STATE_SAVE_REJECTED"] = 2] = "GROUP_MEMBERSHIP_STATE_SAVE_REJECTED";
    GroupMembershipStateSave[GroupMembershipStateSave["GROUP_MEMBERSHIP_STATE_SAVE_PENDING"] = 3] = "GROUP_MEMBERSHIP_STATE_SAVE_PENDING";
    /**
     * GROUP_MEMBERSHIP_STATE_SAVE_RESTORED - A group is marked as this state when it is restored
     * from a backup. This is a non-functional archive state
     * that can be reactivated when the user is re-added to
     * the group.
     */
    GroupMembershipStateSave[GroupMembershipStateSave["GROUP_MEMBERSHIP_STATE_SAVE_RESTORED"] = 4] = "GROUP_MEMBERSHIP_STATE_SAVE_RESTORED";
    GroupMembershipStateSave[GroupMembershipStateSave["GROUP_MEMBERSHIP_STATE_SAVE_PENDING_REMOVE"] = 5] = "GROUP_MEMBERSHIP_STATE_SAVE_PENDING_REMOVE";
    GroupMembershipStateSave[GroupMembershipStateSave["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(GroupMembershipStateSave || (exports.GroupMembershipStateSave = GroupMembershipStateSave = {}));
function groupMembershipStateSaveFromJSON(object) {
    switch (object) {
        case 0:
        case "GROUP_MEMBERSHIP_STATE_SAVE_UNSPECIFIED":
            return GroupMembershipStateSave.GROUP_MEMBERSHIP_STATE_SAVE_UNSPECIFIED;
        case 1:
        case "GROUP_MEMBERSHIP_STATE_SAVE_ALLOWED":
            return GroupMembershipStateSave.GROUP_MEMBERSHIP_STATE_SAVE_ALLOWED;
        case 2:
        case "GROUP_MEMBERSHIP_STATE_SAVE_REJECTED":
            return GroupMembershipStateSave.GROUP_MEMBERSHIP_STATE_SAVE_REJECTED;
        case 3:
        case "GROUP_MEMBERSHIP_STATE_SAVE_PENDING":
            return GroupMembershipStateSave.GROUP_MEMBERSHIP_STATE_SAVE_PENDING;
        case 4:
        case "GROUP_MEMBERSHIP_STATE_SAVE_RESTORED":
            return GroupMembershipStateSave.GROUP_MEMBERSHIP_STATE_SAVE_RESTORED;
        case 5:
        case "GROUP_MEMBERSHIP_STATE_SAVE_PENDING_REMOVE":
            return GroupMembershipStateSave.GROUP_MEMBERSHIP_STATE_SAVE_PENDING_REMOVE;
        case -1:
        case "UNRECOGNIZED":
        default:
            return GroupMembershipStateSave.UNRECOGNIZED;
    }
}
function groupMembershipStateSaveToJSON(object) {
    switch (object) {
        case GroupMembershipStateSave.GROUP_MEMBERSHIP_STATE_SAVE_UNSPECIFIED:
            return "GROUP_MEMBERSHIP_STATE_SAVE_UNSPECIFIED";
        case GroupMembershipStateSave.GROUP_MEMBERSHIP_STATE_SAVE_ALLOWED:
            return "GROUP_MEMBERSHIP_STATE_SAVE_ALLOWED";
        case GroupMembershipStateSave.GROUP_MEMBERSHIP_STATE_SAVE_REJECTED:
            return "GROUP_MEMBERSHIP_STATE_SAVE_REJECTED";
        case GroupMembershipStateSave.GROUP_MEMBERSHIP_STATE_SAVE_PENDING:
            return "GROUP_MEMBERSHIP_STATE_SAVE_PENDING";
        case GroupMembershipStateSave.GROUP_MEMBERSHIP_STATE_SAVE_RESTORED:
            return "GROUP_MEMBERSHIP_STATE_SAVE_RESTORED";
        case GroupMembershipStateSave.GROUP_MEMBERSHIP_STATE_SAVE_PENDING_REMOVE:
            return "GROUP_MEMBERSHIP_STATE_SAVE_PENDING_REMOVE";
        case GroupMembershipStateSave.UNRECOGNIZED:
        default:
            return "UNRECOGNIZED";
    }
}
/** Conversation type */
var ConversationTypeSave;
(function (ConversationTypeSave) {
    ConversationTypeSave[ConversationTypeSave["CONVERSATION_TYPE_SAVE_UNSPECIFIED"] = 0] = "CONVERSATION_TYPE_SAVE_UNSPECIFIED";
    ConversationTypeSave[ConversationTypeSave["CONVERSATION_TYPE_SAVE_GROUP"] = 1] = "CONVERSATION_TYPE_SAVE_GROUP";
    ConversationTypeSave[ConversationTypeSave["CONVERSATION_TYPE_SAVE_DM"] = 2] = "CONVERSATION_TYPE_SAVE_DM";
    ConversationTypeSave[ConversationTypeSave["CONVERSATION_TYPE_SAVE_SYNC"] = 3] = "CONVERSATION_TYPE_SAVE_SYNC";
    ConversationTypeSave[ConversationTypeSave["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(ConversationTypeSave || (exports.ConversationTypeSave = ConversationTypeSave = {}));
function conversationTypeSaveFromJSON(object) {
    switch (object) {
        case 0:
        case "CONVERSATION_TYPE_SAVE_UNSPECIFIED":
            return ConversationTypeSave.CONVERSATION_TYPE_SAVE_UNSPECIFIED;
        case 1:
        case "CONVERSATION_TYPE_SAVE_GROUP":
            return ConversationTypeSave.CONVERSATION_TYPE_SAVE_GROUP;
        case 2:
        case "CONVERSATION_TYPE_SAVE_DM":
            return ConversationTypeSave.CONVERSATION_TYPE_SAVE_DM;
        case 3:
        case "CONVERSATION_TYPE_SAVE_SYNC":
            return ConversationTypeSave.CONVERSATION_TYPE_SAVE_SYNC;
        case -1:
        case "UNRECOGNIZED":
        default:
            return ConversationTypeSave.UNRECOGNIZED;
    }
}
function conversationTypeSaveToJSON(object) {
    switch (object) {
        case ConversationTypeSave.CONVERSATION_TYPE_SAVE_UNSPECIFIED:
            return "CONVERSATION_TYPE_SAVE_UNSPECIFIED";
        case ConversationTypeSave.CONVERSATION_TYPE_SAVE_GROUP:
            return "CONVERSATION_TYPE_SAVE_GROUP";
        case ConversationTypeSave.CONVERSATION_TYPE_SAVE_DM:
            return "CONVERSATION_TYPE_SAVE_DM";
        case ConversationTypeSave.CONVERSATION_TYPE_SAVE_SYNC:
            return "CONVERSATION_TYPE_SAVE_SYNC";
        case ConversationTypeSave.UNRECOGNIZED:
        default:
            return "UNRECOGNIZED";
    }
}
function createBaseGroupSave() {
    return {
        id: new Uint8Array(0),
        createdAtNs: long_1.default.ZERO,
        membershipState: 0,
        installationsLastChecked: long_1.default.ZERO,
        addedByInboxId: "",
        welcomeId: undefined,
        rotatedAtNs: long_1.default.ZERO,
        conversationType: 0,
        dmId: undefined,
        lastMessageNs: undefined,
        messageDisappearFromNs: undefined,
        messageDisappearInNs: undefined,
        metadata: undefined,
        mutableMetadata: undefined,
        pausedForVersion: undefined,
    };
}
exports.GroupSave = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.id.length !== 0) {
            writer.uint32(10).bytes(message.id);
        }
        if (!message.createdAtNs.equals(long_1.default.ZERO)) {
            writer.uint32(16).int64(message.createdAtNs);
        }
        if (message.membershipState !== 0) {
            writer.uint32(24).int32(message.membershipState);
        }
        if (!message.installationsLastChecked.equals(long_1.default.ZERO)) {
            writer.uint32(32).int64(message.installationsLastChecked);
        }
        if (message.addedByInboxId !== "") {
            writer.uint32(42).string(message.addedByInboxId);
        }
        if (message.welcomeId !== undefined) {
            writer.uint32(48).int64(message.welcomeId);
        }
        if (!message.rotatedAtNs.equals(long_1.default.ZERO)) {
            writer.uint32(56).int64(message.rotatedAtNs);
        }
        if (message.conversationType !== 0) {
            writer.uint32(64).int32(message.conversationType);
        }
        if (message.dmId !== undefined) {
            writer.uint32(74).string(message.dmId);
        }
        if (message.lastMessageNs !== undefined) {
            writer.uint32(80).int64(message.lastMessageNs);
        }
        if (message.messageDisappearFromNs !== undefined) {
            writer.uint32(88).int64(message.messageDisappearFromNs);
        }
        if (message.messageDisappearInNs !== undefined) {
            writer.uint32(96).int64(message.messageDisappearInNs);
        }
        if (message.metadata !== undefined) {
            exports.ImmutableMetadataSave.encode(message.metadata, writer.uint32(106).fork()).ldelim();
        }
        if (message.mutableMetadata !== undefined) {
            exports.MutableMetadataSave.encode(message.mutableMetadata, writer.uint32(114).fork()).ldelim();
        }
        if (message.pausedForVersion !== undefined) {
            writer.uint32(122).string(message.pausedForVersion);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGroupSave();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.id = reader.bytes();
                    continue;
                case 2:
                    if (tag !== 16) {
                        break;
                    }
                    message.createdAtNs = reader.int64();
                    continue;
                case 3:
                    if (tag !== 24) {
                        break;
                    }
                    message.membershipState = reader.int32();
                    continue;
                case 4:
                    if (tag !== 32) {
                        break;
                    }
                    message.installationsLastChecked = reader.int64();
                    continue;
                case 5:
                    if (tag !== 42) {
                        break;
                    }
                    message.addedByInboxId = reader.string();
                    continue;
                case 6:
                    if (tag !== 48) {
                        break;
                    }
                    message.welcomeId = reader.int64();
                    continue;
                case 7:
                    if (tag !== 56) {
                        break;
                    }
                    message.rotatedAtNs = reader.int64();
                    continue;
                case 8:
                    if (tag !== 64) {
                        break;
                    }
                    message.conversationType = reader.int32();
                    continue;
                case 9:
                    if (tag !== 74) {
                        break;
                    }
                    message.dmId = reader.string();
                    continue;
                case 10:
                    if (tag !== 80) {
                        break;
                    }
                    message.lastMessageNs = reader.int64();
                    continue;
                case 11:
                    if (tag !== 88) {
                        break;
                    }
                    message.messageDisappearFromNs = reader.int64();
                    continue;
                case 12:
                    if (tag !== 96) {
                        break;
                    }
                    message.messageDisappearInNs = reader.int64();
                    continue;
                case 13:
                    if (tag !== 106) {
                        break;
                    }
                    message.metadata = exports.ImmutableMetadataSave.decode(reader, reader.uint32());
                    continue;
                case 14:
                    if (tag !== 114) {
                        break;
                    }
                    message.mutableMetadata = exports.MutableMetadataSave.decode(reader, reader.uint32());
                    continue;
                case 15:
                    if (tag !== 122) {
                        break;
                    }
                    message.pausedForVersion = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            id: isSet(object.id) ? bytesFromBase64(object.id) : new Uint8Array(0),
            createdAtNs: isSet(object.createdAtNs) ? long_1.default.fromValue(object.createdAtNs) : long_1.default.ZERO,
            membershipState: isSet(object.membershipState) ? groupMembershipStateSaveFromJSON(object.membershipState) : 0,
            installationsLastChecked: isSet(object.installationsLastChecked)
                ? long_1.default.fromValue(object.installationsLastChecked)
                : long_1.default.ZERO,
            addedByInboxId: isSet(object.addedByInboxId) ? globalThis.String(object.addedByInboxId) : "",
            welcomeId: isSet(object.welcomeId) ? long_1.default.fromValue(object.welcomeId) : undefined,
            rotatedAtNs: isSet(object.rotatedAtNs) ? long_1.default.fromValue(object.rotatedAtNs) : long_1.default.ZERO,
            conversationType: isSet(object.conversationType) ? conversationTypeSaveFromJSON(object.conversationType) : 0,
            dmId: isSet(object.dmId) ? globalThis.String(object.dmId) : undefined,
            lastMessageNs: isSet(object.lastMessageNs) ? long_1.default.fromValue(object.lastMessageNs) : undefined,
            messageDisappearFromNs: isSet(object.messageDisappearFromNs)
                ? long_1.default.fromValue(object.messageDisappearFromNs)
                : undefined,
            messageDisappearInNs: isSet(object.messageDisappearInNs)
                ? long_1.default.fromValue(object.messageDisappearInNs)
                : undefined,
            metadata: isSet(object.metadata) ? exports.ImmutableMetadataSave.fromJSON(object.metadata) : undefined,
            mutableMetadata: isSet(object.mutableMetadata) ? exports.MutableMetadataSave.fromJSON(object.mutableMetadata) : undefined,
            pausedForVersion: isSet(object.pausedForVersion) ? globalThis.String(object.pausedForVersion) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.id.length !== 0) {
            obj.id = base64FromBytes(message.id);
        }
        if (!message.createdAtNs.equals(long_1.default.ZERO)) {
            obj.createdAtNs = (message.createdAtNs || long_1.default.ZERO).toString();
        }
        if (message.membershipState !== 0) {
            obj.membershipState = groupMembershipStateSaveToJSON(message.membershipState);
        }
        if (!message.installationsLastChecked.equals(long_1.default.ZERO)) {
            obj.installationsLastChecked = (message.installationsLastChecked || long_1.default.ZERO).toString();
        }
        if (message.addedByInboxId !== "") {
            obj.addedByInboxId = message.addedByInboxId;
        }
        if (message.welcomeId !== undefined) {
            obj.welcomeId = (message.welcomeId || long_1.default.ZERO).toString();
        }
        if (!message.rotatedAtNs.equals(long_1.default.ZERO)) {
            obj.rotatedAtNs = (message.rotatedAtNs || long_1.default.ZERO).toString();
        }
        if (message.conversationType !== 0) {
            obj.conversationType = conversationTypeSaveToJSON(message.conversationType);
        }
        if (message.dmId !== undefined) {
            obj.dmId = message.dmId;
        }
        if (message.lastMessageNs !== undefined) {
            obj.lastMessageNs = (message.lastMessageNs || long_1.default.ZERO).toString();
        }
        if (message.messageDisappearFromNs !== undefined) {
            obj.messageDisappearFromNs = (message.messageDisappearFromNs || long_1.default.ZERO).toString();
        }
        if (message.messageDisappearInNs !== undefined) {
            obj.messageDisappearInNs = (message.messageDisappearInNs || long_1.default.ZERO).toString();
        }
        if (message.metadata !== undefined) {
            obj.metadata = exports.ImmutableMetadataSave.toJSON(message.metadata);
        }
        if (message.mutableMetadata !== undefined) {
            obj.mutableMetadata = exports.MutableMetadataSave.toJSON(message.mutableMetadata);
        }
        if (message.pausedForVersion !== undefined) {
            obj.pausedForVersion = message.pausedForVersion;
        }
        return obj;
    },
    create(base) {
        return exports.GroupSave.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGroupSave();
        message.id = object.id ?? new Uint8Array(0);
        message.createdAtNs = (object.createdAtNs !== undefined && object.createdAtNs !== null)
            ? long_1.default.fromValue(object.createdAtNs)
            : long_1.default.ZERO;
        message.membershipState = object.membershipState ?? 0;
        message.installationsLastChecked =
            (object.installationsLastChecked !== undefined && object.installationsLastChecked !== null)
                ? long_1.default.fromValue(object.installationsLastChecked)
                : long_1.default.ZERO;
        message.addedByInboxId = object.addedByInboxId ?? "";
        message.welcomeId = (object.welcomeId !== undefined && object.welcomeId !== null)
            ? long_1.default.fromValue(object.welcomeId)
            : undefined;
        message.rotatedAtNs = (object.rotatedAtNs !== undefined && object.rotatedAtNs !== null)
            ? long_1.default.fromValue(object.rotatedAtNs)
            : long_1.default.ZERO;
        message.conversationType = object.conversationType ?? 0;
        message.dmId = object.dmId ?? undefined;
        message.lastMessageNs = (object.lastMessageNs !== undefined && object.lastMessageNs !== null)
            ? long_1.default.fromValue(object.lastMessageNs)
            : undefined;
        message.messageDisappearFromNs =
            (object.messageDisappearFromNs !== undefined && object.messageDisappearFromNs !== null)
                ? long_1.default.fromValue(object.messageDisappearFromNs)
                : undefined;
        message.messageDisappearInNs = (object.messageDisappearInNs !== undefined && object.messageDisappearInNs !== null)
            ? long_1.default.fromValue(object.messageDisappearInNs)
            : undefined;
        message.metadata = (object.metadata !== undefined && object.metadata !== null)
            ? exports.ImmutableMetadataSave.fromPartial(object.metadata)
            : undefined;
        message.mutableMetadata = (object.mutableMetadata !== undefined && object.mutableMetadata !== null)
            ? exports.MutableMetadataSave.fromPartial(object.mutableMetadata)
            : undefined;
        message.pausedForVersion = object.pausedForVersion ?? undefined;
        return message;
    },
};
function createBaseMutableMetadataSave() {
    return { attributes: {}, adminList: [], superAdminList: [] };
}
exports.MutableMetadataSave = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        Object.entries(message.attributes).forEach(([key, value]) => {
            exports.MutableMetadataSave_AttributesEntry.encode({ key: key, value }, writer.uint32(10).fork()).ldelim();
        });
        for (const v of message.adminList) {
            writer.uint32(18).string(v);
        }
        for (const v of message.superAdminList) {
            writer.uint32(26).string(v);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMutableMetadataSave();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    const entry1 = exports.MutableMetadataSave_AttributesEntry.decode(reader, reader.uint32());
                    if (entry1.value !== undefined) {
                        message.attributes[entry1.key] = entry1.value;
                    }
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.adminList.push(reader.string());
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.superAdminList.push(reader.string());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            attributes: isObject(object.attributes)
                ? Object.entries(object.attributes).reduce((acc, [key, value]) => {
                    acc[key] = String(value);
                    return acc;
                }, {})
                : {},
            adminList: globalThis.Array.isArray(object?.adminList)
                ? object.adminList.map((e) => globalThis.String(e))
                : [],
            superAdminList: globalThis.Array.isArray(object?.superAdminList)
                ? object.superAdminList.map((e) => globalThis.String(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.attributes) {
            const entries = Object.entries(message.attributes);
            if (entries.length > 0) {
                obj.attributes = {};
                entries.forEach(([k, v]) => {
                    obj.attributes[k] = v;
                });
            }
        }
        if (message.adminList?.length) {
            obj.adminList = message.adminList;
        }
        if (message.superAdminList?.length) {
            obj.superAdminList = message.superAdminList;
        }
        return obj;
    },
    create(base) {
        return exports.MutableMetadataSave.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseMutableMetadataSave();
        message.attributes = Object.entries(object.attributes ?? {}).reduce((acc, [key, value]) => {
            if (value !== undefined) {
                acc[key] = globalThis.String(value);
            }
            return acc;
        }, {});
        message.adminList = object.adminList?.map((e) => e) || [];
        message.superAdminList = object.superAdminList?.map((e) => e) || [];
        return message;
    },
};
function createBaseMutableMetadataSave_AttributesEntry() {
    return { key: "", value: "" };
}
exports.MutableMetadataSave_AttributesEntry = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.key !== "") {
            writer.uint32(10).string(message.key);
        }
        if (message.value !== "") {
            writer.uint32(18).string(message.value);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMutableMetadataSave_AttributesEntry();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.key = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.value = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            key: isSet(object.key) ? globalThis.String(object.key) : "",
            value: isSet(object.value) ? globalThis.String(object.value) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.key !== "") {
            obj.key = message.key;
        }
        if (message.value !== "") {
            obj.value = message.value;
        }
        return obj;
    },
    create(base) {
        return exports.MutableMetadataSave_AttributesEntry.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseMutableMetadataSave_AttributesEntry();
        message.key = object.key ?? "";
        message.value = object.value ?? "";
        return message;
    },
};
function createBaseImmutableMetadataSave() {
    return { creatorInboxId: "" };
}
exports.ImmutableMetadataSave = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.creatorInboxId !== "") {
            writer.uint32(10).string(message.creatorInboxId);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseImmutableMetadataSave();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.creatorInboxId = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { creatorInboxId: isSet(object.creatorInboxId) ? globalThis.String(object.creatorInboxId) : "" };
    },
    toJSON(message) {
        const obj = {};
        if (message.creatorInboxId !== "") {
            obj.creatorInboxId = message.creatorInboxId;
        }
        return obj;
    },
    create(base) {
        return exports.ImmutableMetadataSave.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseImmutableMetadataSave();
        message.creatorInboxId = object.creatorInboxId ?? "";
        return message;
    },
};
function bytesFromBase64(b64) {
    if (globalThis.Buffer) {
        return Uint8Array.from(globalThis.Buffer.from(b64, "base64"));
    }
    else {
        const bin = globalThis.atob(b64);
        const arr = new Uint8Array(bin.length);
        for (let i = 0; i < bin.length; ++i) {
            arr[i] = bin.charCodeAt(i);
        }
        return arr;
    }
}
function base64FromBytes(arr) {
    if (globalThis.Buffer) {
        return globalThis.Buffer.from(arr).toString("base64");
    }
    else {
        const bin = [];
        arr.forEach((byte) => {
            bin.push(globalThis.String.fromCharCode(byte));
        });
        return globalThis.btoa(bin.join(""));
    }
}
if (minimal_1.default.util.Long !== long_1.default) {
    minimal_1.default.util.Long = long_1.default;
    minimal_1.default.configure();
}
function isObject(value) {
    return typeof value === "object" && value !== null;
}
function isSet(value) {
    return value !== null && value !== undefined;
}
