// @xmtp/proto/ts/dist/cjs/device_sync/message_backup.pb.js
"use strict";
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v1.181.2
//   protoc               v3.21.12
// source: device_sync/message_backup.proto
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GroupMessageSave = exports.ContentTypeSave = exports.DeliveryStatusSave = exports.GroupMessageKindSave = exports.protobufPackage = void 0;
exports.groupMessageKindSaveFromJSON = groupMessageKindSaveFromJSON;
exports.groupMessageKindSaveToJSON = groupMessageKindSaveToJSON;
exports.deliveryStatusSaveFromJSON = deliveryStatusSaveFromJSON;
exports.deliveryStatusSaveToJSON = deliveryStatusSaveToJSON;
exports.contentTypeSaveFromJSON = contentTypeSaveFromJSON;
exports.contentTypeSaveToJSON = contentTypeSaveToJSON;
/* eslint-disable */
const long_1 = __importDefault(require("long"));
const minimal_1 = __importDefault(require("protobufjs/minimal"));
exports.protobufPackage = "xmtp.device_sync.message_backup";
/** Definitions for backups */
/** Group message kind */
var GroupMessageKindSave;
(function (GroupMessageKindSave) {
    GroupMessageKindSave[GroupMessageKindSave["GROUP_MESSAGE_KIND_SAVE_UNSPECIFIED"] = 0] = "GROUP_MESSAGE_KIND_SAVE_UNSPECIFIED";
    GroupMessageKindSave[GroupMessageKindSave["GROUP_MESSAGE_KIND_SAVE_APPLICATION"] = 1] = "GROUP_MESSAGE_KIND_SAVE_APPLICATION";
    GroupMessageKindSave[GroupMessageKindSave["GROUP_MESSAGE_KIND_SAVE_MEMBERSHIP_CHANGE"] = 2] = "GROUP_MESSAGE_KIND_SAVE_MEMBERSHIP_CHANGE";
    GroupMessageKindSave[GroupMessageKindSave["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(GroupMessageKindSave || (exports.GroupMessageKindSave = GroupMessageKindSave = {}));
function groupMessageKindSaveFromJSON(object) {
    switch (object) {
        case 0:
        case "GROUP_MESSAGE_KIND_SAVE_UNSPECIFIED":
            return GroupMessageKindSave.GROUP_MESSAGE_KIND_SAVE_UNSPECIFIED;
        case 1:
        case "GROUP_MESSAGE_KIND_SAVE_APPLICATION":
            return GroupMessageKindSave.GROUP_MESSAGE_KIND_SAVE_APPLICATION;
        case 2:
        case "GROUP_MESSAGE_KIND_SAVE_MEMBERSHIP_CHANGE":
            return GroupMessageKindSave.GROUP_MESSAGE_KIND_SAVE_MEMBERSHIP_CHANGE;
        case -1:
        case "UNRECOGNIZED":
        default:
            return GroupMessageKindSave.UNRECOGNIZED;
    }
}
function groupMessageKindSaveToJSON(object) {
    switch (object) {
        case GroupMessageKindSave.GROUP_MESSAGE_KIND_SAVE_UNSPECIFIED:
            return "GROUP_MESSAGE_KIND_SAVE_UNSPECIFIED";
        case GroupMessageKindSave.GROUP_MESSAGE_KIND_SAVE_APPLICATION:
            return "GROUP_MESSAGE_KIND_SAVE_APPLICATION";
        case GroupMessageKindSave.GROUP_MESSAGE_KIND_SAVE_MEMBERSHIP_CHANGE:
            return "GROUP_MESSAGE_KIND_SAVE_MEMBERSHIP_CHANGE";
        case GroupMessageKindSave.UNRECOGNIZED:
        default:
            return "UNRECOGNIZED";
    }
}
/** Group message delivery status */
var DeliveryStatusSave;
(function (DeliveryStatusSave) {
    DeliveryStatusSave[DeliveryStatusSave["DELIVERY_STATUS_SAVE_UNSPECIFIED"] = 0] = "DELIVERY_STATUS_SAVE_UNSPECIFIED";
    DeliveryStatusSave[DeliveryStatusSave["DELIVERY_STATUS_SAVE_UNPUBLISHED"] = 1] = "DELIVERY_STATUS_SAVE_UNPUBLISHED";
    DeliveryStatusSave[DeliveryStatusSave["DELIVERY_STATUS_SAVE_PUBLISHED"] = 2] = "DELIVERY_STATUS_SAVE_PUBLISHED";
    DeliveryStatusSave[DeliveryStatusSave["DELIVERY_STATUS_SAVE_FAILED"] = 3] = "DELIVERY_STATUS_SAVE_FAILED";
    DeliveryStatusSave[DeliveryStatusSave["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(DeliveryStatusSave || (exports.DeliveryStatusSave = DeliveryStatusSave = {}));
function deliveryStatusSaveFromJSON(object) {
    switch (object) {
        case 0:
        case "DELIVERY_STATUS_SAVE_UNSPECIFIED":
            return DeliveryStatusSave.DELIVERY_STATUS_SAVE_UNSPECIFIED;
        case 1:
        case "DELIVERY_STATUS_SAVE_UNPUBLISHED":
            return DeliveryStatusSave.DELIVERY_STATUS_SAVE_UNPUBLISHED;
        case 2:
        case "DELIVERY_STATUS_SAVE_PUBLISHED":
            return DeliveryStatusSave.DELIVERY_STATUS_SAVE_PUBLISHED;
        case 3:
        case "DELIVERY_STATUS_SAVE_FAILED":
            return DeliveryStatusSave.DELIVERY_STATUS_SAVE_FAILED;
        case -1:
        case "UNRECOGNIZED":
        default:
            return DeliveryStatusSave.UNRECOGNIZED;
    }
}
function deliveryStatusSaveToJSON(object) {
    switch (object) {
        case DeliveryStatusSave.DELIVERY_STATUS_SAVE_UNSPECIFIED:
            return "DELIVERY_STATUS_SAVE_UNSPECIFIED";
        case DeliveryStatusSave.DELIVERY_STATUS_SAVE_UNPUBLISHED:
            return "DELIVERY_STATUS_SAVE_UNPUBLISHED";
        case DeliveryStatusSave.DELIVERY_STATUS_SAVE_PUBLISHED:
            return "DELIVERY_STATUS_SAVE_PUBLISHED";
        case DeliveryStatusSave.DELIVERY_STATUS_SAVE_FAILED:
            return "DELIVERY_STATUS_SAVE_FAILED";
        case DeliveryStatusSave.UNRECOGNIZED:
        default:
            return "UNRECOGNIZED";
    }
}
/**
 * Group message content type
 *
 * @deprecated
 */
var ContentTypeSave;
(function (ContentTypeSave) {
    ContentTypeSave[ContentTypeSave["CONTENT_TYPE_SAVE_UNSPECIFIED"] = 0] = "CONTENT_TYPE_SAVE_UNSPECIFIED";
    ContentTypeSave[ContentTypeSave["CONTENT_TYPE_SAVE_UNKNOWN"] = 1] = "CONTENT_TYPE_SAVE_UNKNOWN";
    ContentTypeSave[ContentTypeSave["CONTENT_TYPE_SAVE_TEXT"] = 2] = "CONTENT_TYPE_SAVE_TEXT";
    ContentTypeSave[ContentTypeSave["CONTENT_TYPE_SAVE_GROUP_MEMBERSHIP_CHANGE"] = 3] = "CONTENT_TYPE_SAVE_GROUP_MEMBERSHIP_CHANGE";
    ContentTypeSave[ContentTypeSave["CONTENT_TYPE_SAVE_GROUP_UPDATED"] = 4] = "CONTENT_TYPE_SAVE_GROUP_UPDATED";
    ContentTypeSave[ContentTypeSave["CONTENT_TYPE_SAVE_REACTION"] = 5] = "CONTENT_TYPE_SAVE_REACTION";
    ContentTypeSave[ContentTypeSave["CONTENT_TYPE_SAVE_READ_RECEIPT"] = 6] = "CONTENT_TYPE_SAVE_READ_RECEIPT";
    ContentTypeSave[ContentTypeSave["CONTENT_TYPE_SAVE_REPLY"] = 7] = "CONTENT_TYPE_SAVE_REPLY";
    ContentTypeSave[ContentTypeSave["CONTENT_TYPE_SAVE_ATTACHMENT"] = 8] = "CONTENT_TYPE_SAVE_ATTACHMENT";
    ContentTypeSave[ContentTypeSave["CONTENT_TYPE_SAVE_REMOTE_ATTACHMENT"] = 9] = "CONTENT_TYPE_SAVE_REMOTE_ATTACHMENT";
    ContentTypeSave[ContentTypeSave["CONTENT_TYPE_SAVE_TRANSACTION_REFERENCE"] = 10] = "CONTENT_TYPE_SAVE_TRANSACTION_REFERENCE";
    ContentTypeSave[ContentTypeSave["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(ContentTypeSave || (exports.ContentTypeSave = ContentTypeSave = {}));
function contentTypeSaveFromJSON(object) {
    switch (object) {
        case 0:
        case "CONTENT_TYPE_SAVE_UNSPECIFIED":
            return ContentTypeSave.CONTENT_TYPE_SAVE_UNSPECIFIED;
        case 1:
        case "CONTENT_TYPE_SAVE_UNKNOWN":
            return ContentTypeSave.CONTENT_TYPE_SAVE_UNKNOWN;
        case 2:
        case "CONTENT_TYPE_SAVE_TEXT":
            return ContentTypeSave.CONTENT_TYPE_SAVE_TEXT;
        case 3:
        case "CONTENT_TYPE_SAVE_GROUP_MEMBERSHIP_CHANGE":
            return ContentTypeSave.CONTENT_TYPE_SAVE_GROUP_MEMBERSHIP_CHANGE;
        case 4:
        case "CONTENT_TYPE_SAVE_GROUP_UPDATED":
            return ContentTypeSave.CONTENT_TYPE_SAVE_GROUP_UPDATED;
        case 5:
        case "CONTENT_TYPE_SAVE_REACTION":
            return ContentTypeSave.CONTENT_TYPE_SAVE_REACTION;
        case 6:
        case "CONTENT_TYPE_SAVE_READ_RECEIPT":
            return ContentTypeSave.CONTENT_TYPE_SAVE_READ_RECEIPT;
        case 7:
        case "CONTENT_TYPE_SAVE_REPLY":
            return ContentTypeSave.CONTENT_TYPE_SAVE_REPLY;
        case 8:
        case "CONTENT_TYPE_SAVE_ATTACHMENT":
            return ContentTypeSave.CONTENT_TYPE_SAVE_ATTACHMENT;
        case 9:
        case "CONTENT_TYPE_SAVE_REMOTE_ATTACHMENT":
            return ContentTypeSave.CONTENT_TYPE_SAVE_REMOTE_ATTACHMENT;
        case 10:
        case "CONTENT_TYPE_SAVE_TRANSACTION_REFERENCE":
            return ContentTypeSave.CONTENT_TYPE_SAVE_TRANSACTION_REFERENCE;
        case -1:
        case "UNRECOGNIZED":
        default:
            return ContentTypeSave.UNRECOGNIZED;
    }
}
function contentTypeSaveToJSON(object) {
    switch (object) {
        case ContentTypeSave.CONTENT_TYPE_SAVE_UNSPECIFIED:
            return "CONTENT_TYPE_SAVE_UNSPECIFIED";
        case ContentTypeSave.CONTENT_TYPE_SAVE_UNKNOWN:
            return "CONTENT_TYPE_SAVE_UNKNOWN";
        case ContentTypeSave.CONTENT_TYPE_SAVE_TEXT:
            return "CONTENT_TYPE_SAVE_TEXT";
        case ContentTypeSave.CONTENT_TYPE_SAVE_GROUP_MEMBERSHIP_CHANGE:
            return "CONTENT_TYPE_SAVE_GROUP_MEMBERSHIP_CHANGE";
        case ContentTypeSave.CONTENT_TYPE_SAVE_GROUP_UPDATED:
            return "CONTENT_TYPE_SAVE_GROUP_UPDATED";
        case ContentTypeSave.CONTENT_TYPE_SAVE_REACTION:
            return "CONTENT_TYPE_SAVE_REACTION";
        case ContentTypeSave.CONTENT_TYPE_SAVE_READ_RECEIPT:
            return "CONTENT_TYPE_SAVE_READ_RECEIPT";
        case ContentTypeSave.CONTENT_TYPE_SAVE_REPLY:
            return "CONTENT_TYPE_SAVE_REPLY";
        case ContentTypeSave.CONTENT_TYPE_SAVE_ATTACHMENT:
            return "CONTENT_TYPE_SAVE_ATTACHMENT";
        case ContentTypeSave.CONTENT_TYPE_SAVE_REMOTE_ATTACHMENT:
            return "CONTENT_TYPE_SAVE_REMOTE_ATTACHMENT";
        case ContentTypeSave.CONTENT_TYPE_SAVE_TRANSACTION_REFERENCE:
            return "CONTENT_TYPE_SAVE_TRANSACTION_REFERENCE";
        case ContentTypeSave.UNRECOGNIZED:
        default:
            return "UNRECOGNIZED";
    }
}
function createBaseGroupMessageSave() {
    return {
        id: new Uint8Array(0),
        groupId: new Uint8Array(0),
        decryptedMessageBytes: new Uint8Array(0),
        sentAtNs: long_1.default.ZERO,
        kind: 0,
        senderInstallationId: new Uint8Array(0),
        senderInboxId: "",
        deliveryStatus: 0,
        contentTypeSave: 0,
        versionMajor: 0,
        versionMinor: 0,
        authorityId: "",
        referenceId: undefined,
        sequenceId: undefined,
        originatorId: undefined,
        contentType: "",
    };
}
exports.GroupMessageSave = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.id.length !== 0) {
            writer.uint32(10).bytes(message.id);
        }
        if (message.groupId.length !== 0) {
            writer.uint32(18).bytes(message.groupId);
        }
        if (message.decryptedMessageBytes.length !== 0) {
            writer.uint32(26).bytes(message.decryptedMessageBytes);
        }
        if (!message.sentAtNs.equals(long_1.default.ZERO)) {
            writer.uint32(32).int64(message.sentAtNs);
        }
        if (message.kind !== 0) {
            writer.uint32(40).int32(message.kind);
        }
        if (message.senderInstallationId.length !== 0) {
            writer.uint32(50).bytes(message.senderInstallationId);
        }
        if (message.senderInboxId !== "") {
            writer.uint32(58).string(message.senderInboxId);
        }
        if (message.deliveryStatus !== 0) {
            writer.uint32(64).int32(message.deliveryStatus);
        }
        if (message.contentTypeSave !== 0) {
            writer.uint32(72).int32(message.contentTypeSave);
        }
        if (message.versionMajor !== 0) {
            writer.uint32(80).int32(message.versionMajor);
        }
        if (message.versionMinor !== 0) {
            writer.uint32(88).int32(message.versionMinor);
        }
        if (message.authorityId !== "") {
            writer.uint32(98).string(message.authorityId);
        }
        if (message.referenceId !== undefined) {
            writer.uint32(106).bytes(message.referenceId);
        }
        if (message.sequenceId !== undefined) {
            writer.uint32(112).int64(message.sequenceId);
        }
        if (message.originatorId !== undefined) {
            writer.uint32(120).int64(message.originatorId);
        }
        if (message.contentType !== "") {
            writer.uint32(130).string(message.contentType);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGroupMessageSave();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.id = reader.bytes();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.groupId = reader.bytes();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.decryptedMessageBytes = reader.bytes();
                    continue;
                case 4:
                    if (tag !== 32) {
                        break;
                    }
                    message.sentAtNs = reader.int64();
                    continue;
                case 5:
                    if (tag !== 40) {
                        break;
                    }
                    message.kind = reader.int32();
                    continue;
                case 6:
                    if (tag !== 50) {
                        break;
                    }
                    message.senderInstallationId = reader.bytes();
                    continue;
                case 7:
                    if (tag !== 58) {
                        break;
                    }
                    message.senderInboxId = reader.string();
                    continue;
                case 8:
                    if (tag !== 64) {
                        break;
                    }
                    message.deliveryStatus = reader.int32();
                    continue;
                case 9:
                    if (tag !== 72) {
                        break;
                    }
                    message.contentTypeSave = reader.int32();
                    continue;
                case 10:
                    if (tag !== 80) {
                        break;
                    }
                    message.versionMajor = reader.int32();
                    continue;
                case 11:
                    if (tag !== 88) {
                        break;
                    }
                    message.versionMinor = reader.int32();
                    continue;
                case 12:
                    if (tag !== 98) {
                        break;
                    }
                    message.authorityId = reader.string();
                    continue;
                case 13:
                    if (tag !== 106) {
                        break;
                    }
                    message.referenceId = reader.bytes();
                    continue;
                case 14:
                    if (tag !== 112) {
                        break;
                    }
                    message.sequenceId = reader.int64();
                    continue;
                case 15:
                    if (tag !== 120) {
                        break;
                    }
                    message.originatorId = reader.int64();
                    continue;
                case 16:
                    if (tag !== 130) {
                        break;
                    }
                    message.contentType = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            id: isSet(object.id) ? bytesFromBase64(object.id) : new Uint8Array(0),
            groupId: isSet(object.groupId) ? bytesFromBase64(object.groupId) : new Uint8Array(0),
            decryptedMessageBytes: isSet(object.decryptedMessageBytes)
                ? bytesFromBase64(object.decryptedMessageBytes)
                : new Uint8Array(0),
            sentAtNs: isSet(object.sentAtNs) ? long_1.default.fromValue(object.sentAtNs) : long_1.default.ZERO,
            kind: isSet(object.kind) ? groupMessageKindSaveFromJSON(object.kind) : 0,
            senderInstallationId: isSet(object.senderInstallationId)
                ? bytesFromBase64(object.senderInstallationId)
                : new Uint8Array(0),
            senderInboxId: isSet(object.senderInboxId) ? globalThis.String(object.senderInboxId) : "",
            deliveryStatus: isSet(object.deliveryStatus) ? deliveryStatusSaveFromJSON(object.deliveryStatus) : 0,
            contentTypeSave: isSet(object.contentTypeSave) ? contentTypeSaveFromJSON(object.contentTypeSave) : 0,
            versionMajor: isSet(object.versionMajor) ? globalThis.Number(object.versionMajor) : 0,
            versionMinor: isSet(object.versionMinor) ? globalThis.Number(object.versionMinor) : 0,
            authorityId: isSet(object.authorityId) ? globalThis.String(object.authorityId) : "",
            referenceId: isSet(object.referenceId) ? bytesFromBase64(object.referenceId) : undefined,
            sequenceId: isSet(object.sequenceId) ? long_1.default.fromValue(object.sequenceId) : undefined,
            originatorId: isSet(object.originatorId) ? long_1.default.fromValue(object.originatorId) : undefined,
            contentType: isSet(object.contentType) ? globalThis.String(object.contentType) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.id.length !== 0) {
            obj.id = base64FromBytes(message.id);
        }
        if (message.groupId.length !== 0) {
            obj.groupId = base64FromBytes(message.groupId);
        }
        if (message.decryptedMessageBytes.length !== 0) {
            obj.decryptedMessageBytes = base64FromBytes(message.decryptedMessageBytes);
        }
        if (!message.sentAtNs.equals(long_1.default.ZERO)) {
            obj.sentAtNs = (message.sentAtNs || long_1.default.ZERO).toString();
        }
        if (message.kind !== 0) {
            obj.kind = groupMessageKindSaveToJSON(message.kind);
        }
        if (message.senderInstallationId.length !== 0) {
            obj.senderInstallationId = base64FromBytes(message.senderInstallationId);
        }
        if (message.senderInboxId !== "") {
            obj.senderInboxId = message.senderInboxId;
        }
        if (message.deliveryStatus !== 0) {
            obj.deliveryStatus = deliveryStatusSaveToJSON(message.deliveryStatus);
        }
        if (message.contentTypeSave !== 0) {
            obj.contentTypeSave = contentTypeSaveToJSON(message.contentTypeSave);
        }
        if (message.versionMajor !== 0) {
            obj.versionMajor = Math.round(message.versionMajor);
        }
        if (message.versionMinor !== 0) {
            obj.versionMinor = Math.round(message.versionMinor);
        }
        if (message.authorityId !== "") {
            obj.authorityId = message.authorityId;
        }
        if (message.referenceId !== undefined) {
            obj.referenceId = base64FromBytes(message.referenceId);
        }
        if (message.sequenceId !== undefined) {
            obj.sequenceId = (message.sequenceId || long_1.default.ZERO).toString();
        }
        if (message.originatorId !== undefined) {
            obj.originatorId = (message.originatorId || long_1.default.ZERO).toString();
        }
        if (message.contentType !== "") {
            obj.contentType = message.contentType;
        }
        return obj;
    },
    create(base) {
        return exports.GroupMessageSave.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGroupMessageSave();
        message.id = object.id ?? new Uint8Array(0);
        message.groupId = object.groupId ?? new Uint8Array(0);
        message.decryptedMessageBytes = object.decryptedMessageBytes ?? new Uint8Array(0);
        message.sentAtNs = (object.sentAtNs !== undefined && object.sentAtNs !== null)
            ? long_1.default.fromValue(object.sentAtNs)
            : long_1.default.ZERO;
        message.kind = object.kind ?? 0;
        message.senderInstallationId = object.senderInstallationId ?? new Uint8Array(0);
        message.senderInboxId = object.senderInboxId ?? "";
        message.deliveryStatus = object.deliveryStatus ?? 0;
        message.contentTypeSave = object.contentTypeSave ?? 0;
        message.versionMajor = object.versionMajor ?? 0;
        message.versionMinor = object.versionMinor ?? 0;
        message.authorityId = object.authorityId ?? "";
        message.referenceId = object.referenceId ?? undefined;
        message.sequenceId = (object.sequenceId !== undefined && object.sequenceId !== null)
            ? long_1.default.fromValue(object.sequenceId)
            : undefined;
        message.originatorId = (object.originatorId !== undefined && object.originatorId !== null)
            ? long_1.default.fromValue(object.originatorId)
            : undefined;
        message.contentType = object.contentType ?? "";
        return message;
    },
};
function bytesFromBase64(b64) {
    if (globalThis.Buffer) {
        return Uint8Array.from(globalThis.Buffer.from(b64, "base64"));
    }
    else {
        const bin = globalThis.atob(b64);
        const arr = new Uint8Array(bin.length);
        for (let i = 0; i < bin.length; ++i) {
            arr[i] = bin.charCodeAt(i);
        }
        return arr;
    }
}
function base64FromBytes(arr) {
    if (globalThis.Buffer) {
        return globalThis.Buffer.from(arr).toString("base64");
    }
    else {
        const bin = [];
        arr.forEach((byte) => {
            bin.push(globalThis.String.fromCharCode(byte));
        });
        return globalThis.btoa(bin.join(""));
    }
}
if (minimal_1.default.util.Long !== long_1.default) {
    minimal_1.default.util.Long = long_1.default;
    minimal_1.default.configure();
}
function isSet(value) {
    return value !== null && value !== undefined;
}
