// @xmtp/proto/ts/dist/esm/mls/message_contents/group_metadata.pb.js
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v1.181.2
//   protoc               v3.21.12
// source: mls/message_contents/group_metadata.proto
/* eslint-disable */
import Long from "long";
import _m0 from "protobufjs/minimal";
import { OneshotMessage } from "./oneshot.pb";
export const protobufPackage = "xmtp.mls.message_contents";
/** Group immutable metadata */
/** Defines the type of conversation */
export var ConversationType;
(function (ConversationType) {
    ConversationType[ConversationType["CONVERSATION_TYPE_UNSPECIFIED"] = 0] = "CONVERSATION_TYPE_UNSPECIFIED";
    ConversationType[ConversationType["CONVERSATION_TYPE_GROUP"] = 1] = "CONVERSATION_TYPE_GROUP";
    ConversationType[ConversationType["CONVERSATION_TYPE_DM"] = 2] = "CONVERSATION_TYPE_DM";
    ConversationType[ConversationType["CONVERSATION_TYPE_SYNC"] = 3] = "CONVERSATION_TYPE_SYNC";
    ConversationType[ConversationType["CONVERSATION_TYPE_ONESHOT"] = 4] = "CONVERSATION_TYPE_ONESHOT";
    ConversationType[ConversationType["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(ConversationType || (ConversationType = {}));
export function conversationTypeFromJSON(object) {
    switch (object) {
        case 0:
        case "CONVERSATION_TYPE_UNSPECIFIED":
            return ConversationType.CONVERSATION_TYPE_UNSPECIFIED;
        case 1:
        case "CONVERSATION_TYPE_GROUP":
            return ConversationType.CONVERSATION_TYPE_GROUP;
        case 2:
        case "CONVERSATION_TYPE_DM":
            return ConversationType.CONVERSATION_TYPE_DM;
        case 3:
        case "CONVERSATION_TYPE_SYNC":
            return ConversationType.CONVERSATION_TYPE_SYNC;
        case 4:
        case "CONVERSATION_TYPE_ONESHOT":
            return ConversationType.CONVERSATION_TYPE_ONESHOT;
        case -1:
        case "UNRECOGNIZED":
        default:
            return ConversationType.UNRECOGNIZED;
    }
}
export function conversationTypeToJSON(object) {
    switch (object) {
        case ConversationType.CONVERSATION_TYPE_UNSPECIFIED:
            return "CONVERSATION_TYPE_UNSPECIFIED";
        case ConversationType.CONVERSATION_TYPE_GROUP:
            return "CONVERSATION_TYPE_GROUP";
        case ConversationType.CONVERSATION_TYPE_DM:
            return "CONVERSATION_TYPE_DM";
        case ConversationType.CONVERSATION_TYPE_SYNC:
            return "CONVERSATION_TYPE_SYNC";
        case ConversationType.CONVERSATION_TYPE_ONESHOT:
            return "CONVERSATION_TYPE_ONESHOT";
        case ConversationType.UNRECOGNIZED:
        default:
            return "UNRECOGNIZED";
    }
}
function createBaseGroupMetadataV1() {
    return {
        conversationType: 0,
        creatorAccountAddress: "",
        creatorInboxId: "",
        dmMembers: undefined,
        oneshotMessage: undefined,
    };
}
export const GroupMetadataV1 = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.conversationType !== 0) {
            writer.uint32(8).int32(message.conversationType);
        }
        if (message.creatorAccountAddress !== "") {
            writer.uint32(18).string(message.creatorAccountAddress);
        }
        if (message.creatorInboxId !== "") {
            writer.uint32(26).string(message.creatorInboxId);
        }
        if (message.dmMembers !== undefined) {
            DmMembers.encode(message.dmMembers, writer.uint32(34).fork()).ldelim();
        }
        if (message.oneshotMessage !== undefined) {
            OneshotMessage.encode(message.oneshotMessage, writer.uint32(42).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGroupMetadataV1();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 8) {
                        break;
                    }
                    message.conversationType = reader.int32();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.creatorAccountAddress = reader.string();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.creatorInboxId = reader.string();
                    continue;
                case 4:
                    if (tag !== 34) {
                        break;
                    }
                    message.dmMembers = DmMembers.decode(reader, reader.uint32());
                    continue;
                case 5:
                    if (tag !== 42) {
                        break;
                    }
                    message.oneshotMessage = OneshotMessage.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            conversationType: isSet(object.conversationType) ? conversationTypeFromJSON(object.conversationType) : 0,
            creatorAccountAddress: isSet(object.creatorAccountAddress) ? globalThis.String(object.creatorAccountAddress) : "",
            creatorInboxId: isSet(object.creatorInboxId) ? globalThis.String(object.creatorInboxId) : "",
            dmMembers: isSet(object.dmMembers) ? DmMembers.fromJSON(object.dmMembers) : undefined,
            oneshotMessage: isSet(object.oneshotMessage) ? OneshotMessage.fromJSON(object.oneshotMessage) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.conversationType !== 0) {
            obj.conversationType = conversationTypeToJSON(message.conversationType);
        }
        if (message.creatorAccountAddress !== "") {
            obj.creatorAccountAddress = message.creatorAccountAddress;
        }
        if (message.creatorInboxId !== "") {
            obj.creatorInboxId = message.creatorInboxId;
        }
        if (message.dmMembers !== undefined) {
            obj.dmMembers = DmMembers.toJSON(message.dmMembers);
        }
        if (message.oneshotMessage !== undefined) {
            obj.oneshotMessage = OneshotMessage.toJSON(message.oneshotMessage);
        }
        return obj;
    },
    create(base) {
        return GroupMetadataV1.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGroupMetadataV1();
        message.conversationType = object.conversationType ?? 0;
        message.creatorAccountAddress = object.creatorAccountAddress ?? "";
        message.creatorInboxId = object.creatorInboxId ?? "";
        message.dmMembers = (object.dmMembers !== undefined && object.dmMembers !== null)
            ? DmMembers.fromPartial(object.dmMembers)
            : undefined;
        message.oneshotMessage = (object.oneshotMessage !== undefined && object.oneshotMessage !== null)
            ? OneshotMessage.fromPartial(object.oneshotMessage)
            : undefined;
        return message;
    },
};
function createBaseInbox() {
    return { inboxId: "" };
}
export const Inbox = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.inboxId !== "") {
            writer.uint32(10).string(message.inboxId);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseInbox();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.inboxId = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { inboxId: isSet(object.inboxId) ? globalThis.String(object.inboxId) : "" };
    },
    toJSON(message) {
        const obj = {};
        if (message.inboxId !== "") {
            obj.inboxId = message.inboxId;
        }
        return obj;
    },
    create(base) {
        return Inbox.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseInbox();
        message.inboxId = object.inboxId ?? "";
        return message;
    },
};
function createBaseDmMembers() {
    return { dmMemberOne: undefined, dmMemberTwo: undefined };
}
export const DmMembers = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.dmMemberOne !== undefined) {
            Inbox.encode(message.dmMemberOne, writer.uint32(10).fork()).ldelim();
        }
        if (message.dmMemberTwo !== undefined) {
            Inbox.encode(message.dmMemberTwo, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseDmMembers();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.dmMemberOne = Inbox.decode(reader, reader.uint32());
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.dmMemberTwo = Inbox.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            dmMemberOne: isSet(object.dmMemberOne) ? Inbox.fromJSON(object.dmMemberOne) : undefined,
            dmMemberTwo: isSet(object.dmMemberTwo) ? Inbox.fromJSON(object.dmMemberTwo) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.dmMemberOne !== undefined) {
            obj.dmMemberOne = Inbox.toJSON(message.dmMemberOne);
        }
        if (message.dmMemberTwo !== undefined) {
            obj.dmMemberTwo = Inbox.toJSON(message.dmMemberTwo);
        }
        return obj;
    },
    create(base) {
        return DmMembers.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseDmMembers();
        message.dmMemberOne = (object.dmMemberOne !== undefined && object.dmMemberOne !== null)
            ? Inbox.fromPartial(object.dmMemberOne)
            : undefined;
        message.dmMemberTwo = (object.dmMemberTwo !== undefined && object.dmMemberTwo !== null)
            ? Inbox.fromPartial(object.dmMemberTwo)
            : undefined;
        return message;
    },
};
if (_m0.util.Long !== Long) {
    _m0.util.Long = Long;
    _m0.configure();
}
function isSet(value) {
    return value !== null && value !== undefined;
}
