// @xmtp/proto/ts/dist/esm/mls/message_contents/group_membership.pb.js
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v1.181.2
//   protoc               v3.21.12
// source: mls/message_contents/group_membership.proto
/* eslint-disable */
import Long from "long";
import _m0 from "protobufjs/minimal";
export const protobufPackage = "xmtp.mls.message_contents";
function createBaseGroupMembership() {
    return { members: {}, failedInstallations: [] };
}
export const GroupMembership = {
    encode(message, writer = _m0.Writer.create()) {
        Object.entries(message.members).forEach(([key, value]) => {
            GroupMembership_MembersEntry.encode({ key: key, value }, writer.uint32(10).fork()).ldelim();
        });
        for (const v of message.failedInstallations) {
            writer.uint32(18).bytes(v);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGroupMembership();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    const entry1 = GroupMembership_MembersEntry.decode(reader, reader.uint32());
                    if (entry1.value !== undefined) {
                        message.members[entry1.key] = entry1.value;
                    }
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.failedInstallations.push(reader.bytes());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            members: isObject(object.members)
                ? Object.entries(object.members).reduce((acc, [key, value]) => {
                    acc[key] = Long.fromValue(value);
                    return acc;
                }, {})
                : {},
            failedInstallations: globalThis.Array.isArray(object?.failedInstallations)
                ? object.failedInstallations.map((e) => bytesFromBase64(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.members) {
            const entries = Object.entries(message.members);
            if (entries.length > 0) {
                obj.members = {};
                entries.forEach(([k, v]) => {
                    obj.members[k] = v.toString();
                });
            }
        }
        if (message.failedInstallations?.length) {
            obj.failedInstallations = message.failedInstallations.map((e) => base64FromBytes(e));
        }
        return obj;
    },
    create(base) {
        return GroupMembership.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGroupMembership();
        message.members = Object.entries(object.members ?? {}).reduce((acc, [key, value]) => {
            if (value !== undefined) {
                acc[key] = Long.fromValue(value);
            }
            return acc;
        }, {});
        message.failedInstallations = object.failedInstallations?.map((e) => e) || [];
        return message;
    },
};
function createBaseGroupMembership_MembersEntry() {
    return { key: "", value: Long.UZERO };
}
export const GroupMembership_MembersEntry = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.key !== "") {
            writer.uint32(10).string(message.key);
        }
        if (!message.value.equals(Long.UZERO)) {
            writer.uint32(16).uint64(message.value);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGroupMembership_MembersEntry();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.key = reader.string();
                    continue;
                case 2:
                    if (tag !== 16) {
                        break;
                    }
                    message.value = reader.uint64();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            key: isSet(object.key) ? globalThis.String(object.key) : "",
            value: isSet(object.value) ? Long.fromValue(object.value) : Long.UZERO,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.key !== "") {
            obj.key = message.key;
        }
        if (!message.value.equals(Long.UZERO)) {
            obj.value = (message.value || Long.UZERO).toString();
        }
        return obj;
    },
    create(base) {
        return GroupMembership_MembersEntry.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGroupMembership_MembersEntry();
        message.key = object.key ?? "";
        message.value = (object.value !== undefined && object.value !== null) ? Long.fromValue(object.value) : Long.UZERO;
        return message;
    },
};
function bytesFromBase64(b64) {
    if (globalThis.Buffer) {
        return Uint8Array.from(globalThis.Buffer.from(b64, "base64"));
    }
    else {
        const bin = globalThis.atob(b64);
        const arr = new Uint8Array(bin.length);
        for (let i = 0; i < bin.length; ++i) {
            arr[i] = bin.charCodeAt(i);
        }
        return arr;
    }
}
function base64FromBytes(arr) {
    if (globalThis.Buffer) {
        return globalThis.Buffer.from(arr).toString("base64");
    }
    else {
        const bin = [];
        arr.forEach((byte) => {
            bin.push(globalThis.String.fromCharCode(byte));
        });
        return globalThis.btoa(bin.join(""));
    }
}
if (_m0.util.Long !== Long) {
    _m0.util.Long = Long;
    _m0.configure();
}
function isObject(value) {
    return typeof value === "object" && value !== null;
}
function isSet(value) {
    return value !== null && value !== undefined;
}
