// @xmtp/proto/ts/dist/esm/mls/message_contents/welcome_pointer.pb.js
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v1.181.2
//   protoc               v3.21.12
// source: mls/message_contents/welcome_pointer.proto
/* eslint-disable */
import Long from "long";
import _m0 from "protobufjs/minimal";
export const protobufPackage = "xmtp.mls.message_contents";
/** WelcomePointer is used to point to the welcome message for several installations at once to save overhead */
export var WelcomePointeeEncryptionAeadType;
(function (WelcomePointeeEncryptionAeadType) {
    WelcomePointeeEncryptionAeadType[WelcomePointeeEncryptionAeadType["WELCOME_POINTEE_ENCRYPTION_AEAD_TYPE_UNSPECIFIED"] = 0] = "WELCOME_POINTEE_ENCRYPTION_AEAD_TYPE_UNSPECIFIED";
    /** WELCOME_POINTEE_ENCRYPTION_AEAD_TYPE_CHACHA20_POLY1305 - Use same encoding as openmls::AeadType */
    WelcomePointeeEncryptionAeadType[WelcomePointeeEncryptionAeadType["WELCOME_POINTEE_ENCRYPTION_AEAD_TYPE_CHACHA20_POLY1305"] = 3] = "WELCOME_POINTEE_ENCRYPTION_AEAD_TYPE_CHACHA20_POLY1305";
    WelcomePointeeEncryptionAeadType[WelcomePointeeEncryptionAeadType["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(WelcomePointeeEncryptionAeadType || (WelcomePointeeEncryptionAeadType = {}));
export function welcomePointeeEncryptionAeadTypeFromJSON(object) {
    switch (object) {
        case 0:
        case "WELCOME_POINTEE_ENCRYPTION_AEAD_TYPE_UNSPECIFIED":
            return WelcomePointeeEncryptionAeadType.WELCOME_POINTEE_ENCRYPTION_AEAD_TYPE_UNSPECIFIED;
        case 3:
        case "WELCOME_POINTEE_ENCRYPTION_AEAD_TYPE_CHACHA20_POLY1305":
            return WelcomePointeeEncryptionAeadType.WELCOME_POINTEE_ENCRYPTION_AEAD_TYPE_CHACHA20_POLY1305;
        case -1:
        case "UNRECOGNIZED":
        default:
            return WelcomePointeeEncryptionAeadType.UNRECOGNIZED;
    }
}
export function welcomePointeeEncryptionAeadTypeToJSON(object) {
    switch (object) {
        case WelcomePointeeEncryptionAeadType.WELCOME_POINTEE_ENCRYPTION_AEAD_TYPE_UNSPECIFIED:
            return "WELCOME_POINTEE_ENCRYPTION_AEAD_TYPE_UNSPECIFIED";
        case WelcomePointeeEncryptionAeadType.WELCOME_POINTEE_ENCRYPTION_AEAD_TYPE_CHACHA20_POLY1305:
            return "WELCOME_POINTEE_ENCRYPTION_AEAD_TYPE_CHACHA20_POLY1305";
        case WelcomePointeeEncryptionAeadType.UNRECOGNIZED:
        default:
            return "UNRECOGNIZED";
    }
}
/** MUST match the WelcomeWrapperAlgorithm enum values without 25519 so that the i32 transformations are compatible */
export var WelcomePointerWrapperAlgorithm;
(function (WelcomePointerWrapperAlgorithm) {
    WelcomePointerWrapperAlgorithm[WelcomePointerWrapperAlgorithm["WELCOME_POINTER_WRAPPER_ALGORITHM_UNSPECIFIED"] = 0] = "WELCOME_POINTER_WRAPPER_ALGORITHM_UNSPECIFIED";
    WelcomePointerWrapperAlgorithm[WelcomePointerWrapperAlgorithm["WELCOME_POINTER_WRAPPER_ALGORITHM_XWING_MLKEM_768_DRAFT_6"] = 2] = "WELCOME_POINTER_WRAPPER_ALGORITHM_XWING_MLKEM_768_DRAFT_6";
    WelcomePointerWrapperAlgorithm[WelcomePointerWrapperAlgorithm["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(WelcomePointerWrapperAlgorithm || (WelcomePointerWrapperAlgorithm = {}));
export function welcomePointerWrapperAlgorithmFromJSON(object) {
    switch (object) {
        case 0:
        case "WELCOME_POINTER_WRAPPER_ALGORITHM_UNSPECIFIED":
            return WelcomePointerWrapperAlgorithm.WELCOME_POINTER_WRAPPER_ALGORITHM_UNSPECIFIED;
        case 2:
        case "WELCOME_POINTER_WRAPPER_ALGORITHM_XWING_MLKEM_768_DRAFT_6":
            return WelcomePointerWrapperAlgorithm.WELCOME_POINTER_WRAPPER_ALGORITHM_XWING_MLKEM_768_DRAFT_6;
        case -1:
        case "UNRECOGNIZED":
        default:
            return WelcomePointerWrapperAlgorithm.UNRECOGNIZED;
    }
}
export function welcomePointerWrapperAlgorithmToJSON(object) {
    switch (object) {
        case WelcomePointerWrapperAlgorithm.WELCOME_POINTER_WRAPPER_ALGORITHM_UNSPECIFIED:
            return "WELCOME_POINTER_WRAPPER_ALGORITHM_UNSPECIFIED";
        case WelcomePointerWrapperAlgorithm.WELCOME_POINTER_WRAPPER_ALGORITHM_XWING_MLKEM_768_DRAFT_6:
            return "WELCOME_POINTER_WRAPPER_ALGORITHM_XWING_MLKEM_768_DRAFT_6";
        case WelcomePointerWrapperAlgorithm.UNRECOGNIZED:
        default:
            return "UNRECOGNIZED";
    }
}
function createBaseWelcomePointer() {
    return { welcomeV1Pointer: undefined };
}
export const WelcomePointer = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.welcomeV1Pointer !== undefined) {
            WelcomePointer_WelcomeV1Pointer.encode(message.welcomeV1Pointer, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseWelcomePointer();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.welcomeV1Pointer = WelcomePointer_WelcomeV1Pointer.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            welcomeV1Pointer: isSet(object.welcomeV1Pointer)
                ? WelcomePointer_WelcomeV1Pointer.fromJSON(object.welcomeV1Pointer)
                : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.welcomeV1Pointer !== undefined) {
            obj.welcomeV1Pointer = WelcomePointer_WelcomeV1Pointer.toJSON(message.welcomeV1Pointer);
        }
        return obj;
    },
    create(base) {
        return WelcomePointer.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseWelcomePointer();
        message.welcomeV1Pointer = (object.welcomeV1Pointer !== undefined && object.welcomeV1Pointer !== null)
            ? WelcomePointer_WelcomeV1Pointer.fromPartial(object.welcomeV1Pointer)
            : undefined;
        return message;
    },
};
function createBaseWelcomePointer_WelcomeV1Pointer() {
    return {
        destination: new Uint8Array(0),
        aeadType: 0,
        encryptionKey: new Uint8Array(0),
        dataNonce: new Uint8Array(0),
        welcomeMetadataNonce: new Uint8Array(0),
    };
}
export const WelcomePointer_WelcomeV1Pointer = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.destination.length !== 0) {
            writer.uint32(10).bytes(message.destination);
        }
        if (message.aeadType !== 0) {
            writer.uint32(16).int32(message.aeadType);
        }
        if (message.encryptionKey.length !== 0) {
            writer.uint32(26).bytes(message.encryptionKey);
        }
        if (message.dataNonce.length !== 0) {
            writer.uint32(34).bytes(message.dataNonce);
        }
        if (message.welcomeMetadataNonce.length !== 0) {
            writer.uint32(42).bytes(message.welcomeMetadataNonce);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseWelcomePointer_WelcomeV1Pointer();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.destination = reader.bytes();
                    continue;
                case 2:
                    if (tag !== 16) {
                        break;
                    }
                    message.aeadType = reader.int32();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.encryptionKey = reader.bytes();
                    continue;
                case 4:
                    if (tag !== 34) {
                        break;
                    }
                    message.dataNonce = reader.bytes();
                    continue;
                case 5:
                    if (tag !== 42) {
                        break;
                    }
                    message.welcomeMetadataNonce = reader.bytes();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            destination: isSet(object.destination) ? bytesFromBase64(object.destination) : new Uint8Array(0),
            aeadType: isSet(object.aeadType) ? welcomePointeeEncryptionAeadTypeFromJSON(object.aeadType) : 0,
            encryptionKey: isSet(object.encryptionKey) ? bytesFromBase64(object.encryptionKey) : new Uint8Array(0),
            dataNonce: isSet(object.dataNonce) ? bytesFromBase64(object.dataNonce) : new Uint8Array(0),
            welcomeMetadataNonce: isSet(object.welcomeMetadataNonce)
                ? bytesFromBase64(object.welcomeMetadataNonce)
                : new Uint8Array(0),
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.destination.length !== 0) {
            obj.destination = base64FromBytes(message.destination);
        }
        if (message.aeadType !== 0) {
            obj.aeadType = welcomePointeeEncryptionAeadTypeToJSON(message.aeadType);
        }
        if (message.encryptionKey.length !== 0) {
            obj.encryptionKey = base64FromBytes(message.encryptionKey);
        }
        if (message.dataNonce.length !== 0) {
            obj.dataNonce = base64FromBytes(message.dataNonce);
        }
        if (message.welcomeMetadataNonce.length !== 0) {
            obj.welcomeMetadataNonce = base64FromBytes(message.welcomeMetadataNonce);
        }
        return obj;
    },
    create(base) {
        return WelcomePointer_WelcomeV1Pointer.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseWelcomePointer_WelcomeV1Pointer();
        message.destination = object.destination ?? new Uint8Array(0);
        message.aeadType = object.aeadType ?? 0;
        message.encryptionKey = object.encryptionKey ?? new Uint8Array(0);
        message.dataNonce = object.dataNonce ?? new Uint8Array(0);
        message.welcomeMetadataNonce = object.welcomeMetadataNonce ?? new Uint8Array(0);
        return message;
    },
};
function createBaseWelcomePointeeEncryptionAeadTypesExtension() {
    return { supportedAeadTypes: [] };
}
export const WelcomePointeeEncryptionAeadTypesExtension = {
    encode(message, writer = _m0.Writer.create()) {
        writer.uint32(10).fork();
        for (const v of message.supportedAeadTypes) {
            writer.int32(v);
        }
        writer.ldelim();
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseWelcomePointeeEncryptionAeadTypesExtension();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag === 8) {
                        message.supportedAeadTypes.push(reader.int32());
                        continue;
                    }
                    if (tag === 10) {
                        const end2 = reader.uint32() + reader.pos;
                        while (reader.pos < end2) {
                            message.supportedAeadTypes.push(reader.int32());
                        }
                        continue;
                    }
                    break;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            supportedAeadTypes: globalThis.Array.isArray(object?.supportedAeadTypes)
                ? object.supportedAeadTypes.map((e) => welcomePointeeEncryptionAeadTypeFromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.supportedAeadTypes?.length) {
            obj.supportedAeadTypes = message.supportedAeadTypes.map((e) => welcomePointeeEncryptionAeadTypeToJSON(e));
        }
        return obj;
    },
    create(base) {
        return WelcomePointeeEncryptionAeadTypesExtension.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseWelcomePointeeEncryptionAeadTypesExtension();
        message.supportedAeadTypes = object.supportedAeadTypes?.map((e) => e) || [];
        return message;
    },
};
function bytesFromBase64(b64) {
    if (globalThis.Buffer) {
        return Uint8Array.from(globalThis.Buffer.from(b64, "base64"));
    }
    else {
        const bin = globalThis.atob(b64);
        const arr = new Uint8Array(bin.length);
        for (let i = 0; i < bin.length; ++i) {
            arr[i] = bin.charCodeAt(i);
        }
        return arr;
    }
}
function base64FromBytes(arr) {
    if (globalThis.Buffer) {
        return globalThis.Buffer.from(arr).toString("base64");
    }
    else {
        const bin = [];
        arr.forEach((byte) => {
            bin.push(globalThis.String.fromCharCode(byte));
        });
        return globalThis.btoa(bin.join(""));
    }
}
if (_m0.util.Long !== Long) {
    _m0.util.Long = Long;
    _m0.configure();
}
function isSet(value) {
    return value !== null && value !== undefined;
}
