// @xmtp/proto/ts/dist/esm/mls_validation/v1/service.pb.js
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v1.181.2
//   protoc               v3.21.12
// source: mls_validation/v1/service.proto
/* eslint-disable */
import Long from "long";
import _m0 from "protobufjs/minimal";
import { VerifySmartContractWalletSignaturesRequest, VerifySmartContractWalletSignaturesResponse, } from "../../identity/api/v1/identity.pb";
import { AssociationState, AssociationStateDiff, IdentityUpdate } from "../../identity/associations/association.pb";
import { MlsCredential } from "../../identity/credential.pb";
export const protobufPackage = "xmtp.mls_validation.v1";
function createBaseValidateInboxIdKeyPackagesRequest() {
    return { keyPackages: [] };
}
export const ValidateInboxIdKeyPackagesRequest = {
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.keyPackages) {
            ValidateInboxIdKeyPackagesRequest_KeyPackage.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseValidateInboxIdKeyPackagesRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.keyPackages.push(ValidateInboxIdKeyPackagesRequest_KeyPackage.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            keyPackages: globalThis.Array.isArray(object?.keyPackages)
                ? object.keyPackages.map((e) => ValidateInboxIdKeyPackagesRequest_KeyPackage.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.keyPackages?.length) {
            obj.keyPackages = message.keyPackages.map((e) => ValidateInboxIdKeyPackagesRequest_KeyPackage.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return ValidateInboxIdKeyPackagesRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseValidateInboxIdKeyPackagesRequest();
        message.keyPackages = object.keyPackages?.map((e) => ValidateInboxIdKeyPackagesRequest_KeyPackage.fromPartial(e)) ||
            [];
        return message;
    },
};
function createBaseValidateInboxIdKeyPackagesRequest_KeyPackage() {
    return { keyPackageBytesTlsSerialized: new Uint8Array(0), isInboxIdCredential: false };
}
export const ValidateInboxIdKeyPackagesRequest_KeyPackage = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.keyPackageBytesTlsSerialized.length !== 0) {
            writer.uint32(10).bytes(message.keyPackageBytesTlsSerialized);
        }
        if (message.isInboxIdCredential !== false) {
            writer.uint32(16).bool(message.isInboxIdCredential);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseValidateInboxIdKeyPackagesRequest_KeyPackage();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.keyPackageBytesTlsSerialized = reader.bytes();
                    continue;
                case 2:
                    if (tag !== 16) {
                        break;
                    }
                    message.isInboxIdCredential = reader.bool();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            keyPackageBytesTlsSerialized: isSet(object.keyPackageBytesTlsSerialized)
                ? bytesFromBase64(object.keyPackageBytesTlsSerialized)
                : new Uint8Array(0),
            isInboxIdCredential: isSet(object.isInboxIdCredential) ? globalThis.Boolean(object.isInboxIdCredential) : false,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.keyPackageBytesTlsSerialized.length !== 0) {
            obj.keyPackageBytesTlsSerialized = base64FromBytes(message.keyPackageBytesTlsSerialized);
        }
        if (message.isInboxIdCredential !== false) {
            obj.isInboxIdCredential = message.isInboxIdCredential;
        }
        return obj;
    },
    create(base) {
        return ValidateInboxIdKeyPackagesRequest_KeyPackage.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseValidateInboxIdKeyPackagesRequest_KeyPackage();
        message.keyPackageBytesTlsSerialized = object.keyPackageBytesTlsSerialized ?? new Uint8Array(0);
        message.isInboxIdCredential = object.isInboxIdCredential ?? false;
        return message;
    },
};
function createBaseValidateInboxIdKeyPackagesResponse() {
    return { responses: [] };
}
export const ValidateInboxIdKeyPackagesResponse = {
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.responses) {
            ValidateInboxIdKeyPackagesResponse_Response.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseValidateInboxIdKeyPackagesResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.responses.push(ValidateInboxIdKeyPackagesResponse_Response.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            responses: globalThis.Array.isArray(object?.responses)
                ? object.responses.map((e) => ValidateInboxIdKeyPackagesResponse_Response.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.responses?.length) {
            obj.responses = message.responses.map((e) => ValidateInboxIdKeyPackagesResponse_Response.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return ValidateInboxIdKeyPackagesResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseValidateInboxIdKeyPackagesResponse();
        message.responses = object.responses?.map((e) => ValidateInboxIdKeyPackagesResponse_Response.fromPartial(e)) || [];
        return message;
    },
};
function createBaseValidateInboxIdKeyPackagesResponse_Response() {
    return {
        isOk: false,
        errorMessage: "",
        credential: undefined,
        installationPublicKey: new Uint8Array(0),
        expiration: Long.UZERO,
    };
}
export const ValidateInboxIdKeyPackagesResponse_Response = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.isOk !== false) {
            writer.uint32(8).bool(message.isOk);
        }
        if (message.errorMessage !== "") {
            writer.uint32(18).string(message.errorMessage);
        }
        if (message.credential !== undefined) {
            MlsCredential.encode(message.credential, writer.uint32(26).fork()).ldelim();
        }
        if (message.installationPublicKey.length !== 0) {
            writer.uint32(34).bytes(message.installationPublicKey);
        }
        if (!message.expiration.equals(Long.UZERO)) {
            writer.uint32(40).uint64(message.expiration);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseValidateInboxIdKeyPackagesResponse_Response();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 8) {
                        break;
                    }
                    message.isOk = reader.bool();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.errorMessage = reader.string();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.credential = MlsCredential.decode(reader, reader.uint32());
                    continue;
                case 4:
                    if (tag !== 34) {
                        break;
                    }
                    message.installationPublicKey = reader.bytes();
                    continue;
                case 5:
                    if (tag !== 40) {
                        break;
                    }
                    message.expiration = reader.uint64();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            isOk: isSet(object.isOk) ? globalThis.Boolean(object.isOk) : false,
            errorMessage: isSet(object.errorMessage) ? globalThis.String(object.errorMessage) : "",
            credential: isSet(object.credential) ? MlsCredential.fromJSON(object.credential) : undefined,
            installationPublicKey: isSet(object.installationPublicKey)
                ? bytesFromBase64(object.installationPublicKey)
                : new Uint8Array(0),
            expiration: isSet(object.expiration) ? Long.fromValue(object.expiration) : Long.UZERO,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.isOk !== false) {
            obj.isOk = message.isOk;
        }
        if (message.errorMessage !== "") {
            obj.errorMessage = message.errorMessage;
        }
        if (message.credential !== undefined) {
            obj.credential = MlsCredential.toJSON(message.credential);
        }
        if (message.installationPublicKey.length !== 0) {
            obj.installationPublicKey = base64FromBytes(message.installationPublicKey);
        }
        if (!message.expiration.equals(Long.UZERO)) {
            obj.expiration = (message.expiration || Long.UZERO).toString();
        }
        return obj;
    },
    create(base) {
        return ValidateInboxIdKeyPackagesResponse_Response.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseValidateInboxIdKeyPackagesResponse_Response();
        message.isOk = object.isOk ?? false;
        message.errorMessage = object.errorMessage ?? "";
        message.credential = (object.credential !== undefined && object.credential !== null)
            ? MlsCredential.fromPartial(object.credential)
            : undefined;
        message.installationPublicKey = object.installationPublicKey ?? new Uint8Array(0);
        message.expiration = (object.expiration !== undefined && object.expiration !== null)
            ? Long.fromValue(object.expiration)
            : Long.UZERO;
        return message;
    },
};
function createBaseValidateKeyPackagesRequest() {
    return { keyPackages: [] };
}
export const ValidateKeyPackagesRequest = {
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.keyPackages) {
            ValidateKeyPackagesRequest_KeyPackage.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseValidateKeyPackagesRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.keyPackages.push(ValidateKeyPackagesRequest_KeyPackage.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            keyPackages: globalThis.Array.isArray(object?.keyPackages)
                ? object.keyPackages.map((e) => ValidateKeyPackagesRequest_KeyPackage.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.keyPackages?.length) {
            obj.keyPackages = message.keyPackages.map((e) => ValidateKeyPackagesRequest_KeyPackage.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return ValidateKeyPackagesRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseValidateKeyPackagesRequest();
        message.keyPackages = object.keyPackages?.map((e) => ValidateKeyPackagesRequest_KeyPackage.fromPartial(e)) || [];
        return message;
    },
};
function createBaseValidateKeyPackagesRequest_KeyPackage() {
    return { keyPackageBytesTlsSerialized: new Uint8Array(0), isInboxIdCredential: false };
}
export const ValidateKeyPackagesRequest_KeyPackage = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.keyPackageBytesTlsSerialized.length !== 0) {
            writer.uint32(10).bytes(message.keyPackageBytesTlsSerialized);
        }
        if (message.isInboxIdCredential !== false) {
            writer.uint32(16).bool(message.isInboxIdCredential);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseValidateKeyPackagesRequest_KeyPackage();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.keyPackageBytesTlsSerialized = reader.bytes();
                    continue;
                case 2:
                    if (tag !== 16) {
                        break;
                    }
                    message.isInboxIdCredential = reader.bool();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            keyPackageBytesTlsSerialized: isSet(object.keyPackageBytesTlsSerialized)
                ? bytesFromBase64(object.keyPackageBytesTlsSerialized)
                : new Uint8Array(0),
            isInboxIdCredential: isSet(object.isInboxIdCredential) ? globalThis.Boolean(object.isInboxIdCredential) : false,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.keyPackageBytesTlsSerialized.length !== 0) {
            obj.keyPackageBytesTlsSerialized = base64FromBytes(message.keyPackageBytesTlsSerialized);
        }
        if (message.isInboxIdCredential !== false) {
            obj.isInboxIdCredential = message.isInboxIdCredential;
        }
        return obj;
    },
    create(base) {
        return ValidateKeyPackagesRequest_KeyPackage.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseValidateKeyPackagesRequest_KeyPackage();
        message.keyPackageBytesTlsSerialized = object.keyPackageBytesTlsSerialized ?? new Uint8Array(0);
        message.isInboxIdCredential = object.isInboxIdCredential ?? false;
        return message;
    },
};
function createBaseValidateKeyPackagesResponse() {
    return { responses: [] };
}
export const ValidateKeyPackagesResponse = {
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.responses) {
            ValidateKeyPackagesResponse_ValidationResponse.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseValidateKeyPackagesResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.responses.push(ValidateKeyPackagesResponse_ValidationResponse.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            responses: globalThis.Array.isArray(object?.responses)
                ? object.responses.map((e) => ValidateKeyPackagesResponse_ValidationResponse.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.responses?.length) {
            obj.responses = message.responses.map((e) => ValidateKeyPackagesResponse_ValidationResponse.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return ValidateKeyPackagesResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseValidateKeyPackagesResponse();
        message.responses = object.responses?.map((e) => ValidateKeyPackagesResponse_ValidationResponse.fromPartial(e)) ||
            [];
        return message;
    },
};
function createBaseValidateKeyPackagesResponse_ValidationResponse() {
    return {
        isOk: false,
        errorMessage: "",
        installationId: new Uint8Array(0),
        accountAddress: "",
        credentialIdentityBytes: new Uint8Array(0),
        expiration: Long.UZERO,
    };
}
export const ValidateKeyPackagesResponse_ValidationResponse = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.isOk !== false) {
            writer.uint32(8).bool(message.isOk);
        }
        if (message.errorMessage !== "") {
            writer.uint32(18).string(message.errorMessage);
        }
        if (message.installationId.length !== 0) {
            writer.uint32(26).bytes(message.installationId);
        }
        if (message.accountAddress !== "") {
            writer.uint32(34).string(message.accountAddress);
        }
        if (message.credentialIdentityBytes.length !== 0) {
            writer.uint32(42).bytes(message.credentialIdentityBytes);
        }
        if (!message.expiration.equals(Long.UZERO)) {
            writer.uint32(48).uint64(message.expiration);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseValidateKeyPackagesResponse_ValidationResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 8) {
                        break;
                    }
                    message.isOk = reader.bool();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.errorMessage = reader.string();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.installationId = reader.bytes();
                    continue;
                case 4:
                    if (tag !== 34) {
                        break;
                    }
                    message.accountAddress = reader.string();
                    continue;
                case 5:
                    if (tag !== 42) {
                        break;
                    }
                    message.credentialIdentityBytes = reader.bytes();
                    continue;
                case 6:
                    if (tag !== 48) {
                        break;
                    }
                    message.expiration = reader.uint64();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            isOk: isSet(object.isOk) ? globalThis.Boolean(object.isOk) : false,
            errorMessage: isSet(object.errorMessage) ? globalThis.String(object.errorMessage) : "",
            installationId: isSet(object.installationId) ? bytesFromBase64(object.installationId) : new Uint8Array(0),
            accountAddress: isSet(object.accountAddress) ? globalThis.String(object.accountAddress) : "",
            credentialIdentityBytes: isSet(object.credentialIdentityBytes)
                ? bytesFromBase64(object.credentialIdentityBytes)
                : new Uint8Array(0),
            expiration: isSet(object.expiration) ? Long.fromValue(object.expiration) : Long.UZERO,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.isOk !== false) {
            obj.isOk = message.isOk;
        }
        if (message.errorMessage !== "") {
            obj.errorMessage = message.errorMessage;
        }
        if (message.installationId.length !== 0) {
            obj.installationId = base64FromBytes(message.installationId);
        }
        if (message.accountAddress !== "") {
            obj.accountAddress = message.accountAddress;
        }
        if (message.credentialIdentityBytes.length !== 0) {
            obj.credentialIdentityBytes = base64FromBytes(message.credentialIdentityBytes);
        }
        if (!message.expiration.equals(Long.UZERO)) {
            obj.expiration = (message.expiration || Long.UZERO).toString();
        }
        return obj;
    },
    create(base) {
        return ValidateKeyPackagesResponse_ValidationResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseValidateKeyPackagesResponse_ValidationResponse();
        message.isOk = object.isOk ?? false;
        message.errorMessage = object.errorMessage ?? "";
        message.installationId = object.installationId ?? new Uint8Array(0);
        message.accountAddress = object.accountAddress ?? "";
        message.credentialIdentityBytes = object.credentialIdentityBytes ?? new Uint8Array(0);
        message.expiration = (object.expiration !== undefined && object.expiration !== null)
            ? Long.fromValue(object.expiration)
            : Long.UZERO;
        return message;
    },
};
function createBaseValidateGroupMessagesRequest() {
    return { groupMessages: [] };
}
export const ValidateGroupMessagesRequest = {
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.groupMessages) {
            ValidateGroupMessagesRequest_GroupMessage.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseValidateGroupMessagesRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.groupMessages.push(ValidateGroupMessagesRequest_GroupMessage.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            groupMessages: globalThis.Array.isArray(object?.groupMessages)
                ? object.groupMessages.map((e) => ValidateGroupMessagesRequest_GroupMessage.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.groupMessages?.length) {
            obj.groupMessages = message.groupMessages.map((e) => ValidateGroupMessagesRequest_GroupMessage.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return ValidateGroupMessagesRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseValidateGroupMessagesRequest();
        message.groupMessages =
            object.groupMessages?.map((e) => ValidateGroupMessagesRequest_GroupMessage.fromPartial(e)) || [];
        return message;
    },
};
function createBaseValidateGroupMessagesRequest_GroupMessage() {
    return { groupMessageBytesTlsSerialized: new Uint8Array(0) };
}
export const ValidateGroupMessagesRequest_GroupMessage = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.groupMessageBytesTlsSerialized.length !== 0) {
            writer.uint32(10).bytes(message.groupMessageBytesTlsSerialized);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseValidateGroupMessagesRequest_GroupMessage();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.groupMessageBytesTlsSerialized = reader.bytes();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            groupMessageBytesTlsSerialized: isSet(object.groupMessageBytesTlsSerialized)
                ? bytesFromBase64(object.groupMessageBytesTlsSerialized)
                : new Uint8Array(0),
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.groupMessageBytesTlsSerialized.length !== 0) {
            obj.groupMessageBytesTlsSerialized = base64FromBytes(message.groupMessageBytesTlsSerialized);
        }
        return obj;
    },
    create(base) {
        return ValidateGroupMessagesRequest_GroupMessage.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseValidateGroupMessagesRequest_GroupMessage();
        message.groupMessageBytesTlsSerialized = object.groupMessageBytesTlsSerialized ?? new Uint8Array(0);
        return message;
    },
};
function createBaseValidateGroupMessagesResponse() {
    return { responses: [] };
}
export const ValidateGroupMessagesResponse = {
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.responses) {
            ValidateGroupMessagesResponse_ValidationResponse.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseValidateGroupMessagesResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.responses.push(ValidateGroupMessagesResponse_ValidationResponse.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            responses: globalThis.Array.isArray(object?.responses)
                ? object.responses.map((e) => ValidateGroupMessagesResponse_ValidationResponse.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.responses?.length) {
            obj.responses = message.responses.map((e) => ValidateGroupMessagesResponse_ValidationResponse.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return ValidateGroupMessagesResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseValidateGroupMessagesResponse();
        message.responses = object.responses?.map((e) => ValidateGroupMessagesResponse_ValidationResponse.fromPartial(e)) ||
            [];
        return message;
    },
};
function createBaseValidateGroupMessagesResponse_ValidationResponse() {
    return { isOk: false, errorMessage: "", groupId: "", isCommit: false };
}
export const ValidateGroupMessagesResponse_ValidationResponse = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.isOk !== false) {
            writer.uint32(8).bool(message.isOk);
        }
        if (message.errorMessage !== "") {
            writer.uint32(18).string(message.errorMessage);
        }
        if (message.groupId !== "") {
            writer.uint32(26).string(message.groupId);
        }
        if (message.isCommit !== false) {
            writer.uint32(32).bool(message.isCommit);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseValidateGroupMessagesResponse_ValidationResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 8) {
                        break;
                    }
                    message.isOk = reader.bool();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.errorMessage = reader.string();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.groupId = reader.string();
                    continue;
                case 4:
                    if (tag !== 32) {
                        break;
                    }
                    message.isCommit = reader.bool();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            isOk: isSet(object.isOk) ? globalThis.Boolean(object.isOk) : false,
            errorMessage: isSet(object.errorMessage) ? globalThis.String(object.errorMessage) : "",
            groupId: isSet(object.groupId) ? globalThis.String(object.groupId) : "",
            isCommit: isSet(object.isCommit) ? globalThis.Boolean(object.isCommit) : false,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.isOk !== false) {
            obj.isOk = message.isOk;
        }
        if (message.errorMessage !== "") {
            obj.errorMessage = message.errorMessage;
        }
        if (message.groupId !== "") {
            obj.groupId = message.groupId;
        }
        if (message.isCommit !== false) {
            obj.isCommit = message.isCommit;
        }
        return obj;
    },
    create(base) {
        return ValidateGroupMessagesResponse_ValidationResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseValidateGroupMessagesResponse_ValidationResponse();
        message.isOk = object.isOk ?? false;
        message.errorMessage = object.errorMessage ?? "";
        message.groupId = object.groupId ?? "";
        message.isCommit = object.isCommit ?? false;
        return message;
    },
};
function createBaseGetAssociationStateRequest() {
    return { oldUpdates: [], newUpdates: [] };
}
export const GetAssociationStateRequest = {
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.oldUpdates) {
            IdentityUpdate.encode(v, writer.uint32(10).fork()).ldelim();
        }
        for (const v of message.newUpdates) {
            IdentityUpdate.encode(v, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetAssociationStateRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.oldUpdates.push(IdentityUpdate.decode(reader, reader.uint32()));
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.newUpdates.push(IdentityUpdate.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            oldUpdates: globalThis.Array.isArray(object?.oldUpdates)
                ? object.oldUpdates.map((e) => IdentityUpdate.fromJSON(e))
                : [],
            newUpdates: globalThis.Array.isArray(object?.newUpdates)
                ? object.newUpdates.map((e) => IdentityUpdate.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.oldUpdates?.length) {
            obj.oldUpdates = message.oldUpdates.map((e) => IdentityUpdate.toJSON(e));
        }
        if (message.newUpdates?.length) {
            obj.newUpdates = message.newUpdates.map((e) => IdentityUpdate.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return GetAssociationStateRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGetAssociationStateRequest();
        message.oldUpdates = object.oldUpdates?.map((e) => IdentityUpdate.fromPartial(e)) || [];
        message.newUpdates = object.newUpdates?.map((e) => IdentityUpdate.fromPartial(e)) || [];
        return message;
    },
};
function createBaseGetAssociationStateResponse() {
    return { associationState: undefined, stateDiff: undefined };
}
export const GetAssociationStateResponse = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.associationState !== undefined) {
            AssociationState.encode(message.associationState, writer.uint32(10).fork()).ldelim();
        }
        if (message.stateDiff !== undefined) {
            AssociationStateDiff.encode(message.stateDiff, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetAssociationStateResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.associationState = AssociationState.decode(reader, reader.uint32());
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.stateDiff = AssociationStateDiff.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            associationState: isSet(object.associationState) ? AssociationState.fromJSON(object.associationState) : undefined,
            stateDiff: isSet(object.stateDiff) ? AssociationStateDiff.fromJSON(object.stateDiff) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.associationState !== undefined) {
            obj.associationState = AssociationState.toJSON(message.associationState);
        }
        if (message.stateDiff !== undefined) {
            obj.stateDiff = AssociationStateDiff.toJSON(message.stateDiff);
        }
        return obj;
    },
    create(base) {
        return GetAssociationStateResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGetAssociationStateResponse();
        message.associationState = (object.associationState !== undefined && object.associationState !== null)
            ? AssociationState.fromPartial(object.associationState)
            : undefined;
        message.stateDiff = (object.stateDiff !== undefined && object.stateDiff !== null)
            ? AssociationStateDiff.fromPartial(object.stateDiff)
            : undefined;
        return message;
    },
};
export const ValidationApiServiceName = "xmtp.mls_validation.v1.ValidationApi";
export class ValidationApiClientImpl {
    rpc;
    service;
    constructor(rpc, opts) {
        this.service = opts?.service || ValidationApiServiceName;
        this.rpc = rpc;
        this.ValidateGroupMessages = this.ValidateGroupMessages.bind(this);
        this.GetAssociationState = this.GetAssociationState.bind(this);
        this.ValidateInboxIdKeyPackages = this.ValidateInboxIdKeyPackages.bind(this);
        this.VerifySmartContractWalletSignatures = this.VerifySmartContractWalletSignatures.bind(this);
    }
    ValidateGroupMessages(request) {
        const data = ValidateGroupMessagesRequest.encode(request).finish();
        const promise = this.rpc.request(this.service, "ValidateGroupMessages", data);
        return promise.then((data) => ValidateGroupMessagesResponse.decode(_m0.Reader.create(data)));
    }
    GetAssociationState(request) {
        const data = GetAssociationStateRequest.encode(request).finish();
        const promise = this.rpc.request(this.service, "GetAssociationState", data);
        return promise.then((data) => GetAssociationStateResponse.decode(_m0.Reader.create(data)));
    }
    ValidateInboxIdKeyPackages(request) {
        const data = ValidateKeyPackagesRequest.encode(request).finish();
        const promise = this.rpc.request(this.service, "ValidateInboxIdKeyPackages", data);
        return promise.then((data) => ValidateInboxIdKeyPackagesResponse.decode(_m0.Reader.create(data)));
    }
    VerifySmartContractWalletSignatures(request) {
        const data = VerifySmartContractWalletSignaturesRequest.encode(request).finish();
        const promise = this.rpc.request(this.service, "VerifySmartContractWalletSignatures", data);
        return promise.then((data) => VerifySmartContractWalletSignaturesResponse.decode(_m0.Reader.create(data)));
    }
}
function bytesFromBase64(b64) {
    if (globalThis.Buffer) {
        return Uint8Array.from(globalThis.Buffer.from(b64, "base64"));
    }
    else {
        const bin = globalThis.atob(b64);
        const arr = new Uint8Array(bin.length);
        for (let i = 0; i < bin.length; ++i) {
            arr[i] = bin.charCodeAt(i);
        }
        return arr;
    }
}
function base64FromBytes(arr) {
    if (globalThis.Buffer) {
        return globalThis.Buffer.from(arr).toString("base64");
    }
    else {
        const bin = [];
        arr.forEach((byte) => {
            bin.push(globalThis.String.fromCharCode(byte));
        });
        return globalThis.btoa(bin.join(""));
    }
}
if (_m0.util.Long !== Long) {
    _m0.util.Long = Long;
    _m0.configure();
}
function isSet(value) {
    return value !== null && value !== undefined;
}
