// @xmtp/proto/ts/dist/esm/mls/message_contents/content_types/multi_remote_attachment.pb.js
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v1.181.2
//   protoc               v3.21.12
// source: mls/message_contents/content_types/multi_remote_attachment.proto
/* eslint-disable */
import Long from "long";
import _m0 from "protobufjs/minimal";
export const protobufPackage = "xmtp.mls.message_contents.content_types";
function createBaseMultiRemoteAttachment() {
    return { attachments: [] };
}
export const MultiRemoteAttachment = {
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.attachments) {
            RemoteAttachmentInfo.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMultiRemoteAttachment();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.attachments.push(RemoteAttachmentInfo.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            attachments: globalThis.Array.isArray(object?.attachments)
                ? object.attachments.map((e) => RemoteAttachmentInfo.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.attachments?.length) {
            obj.attachments = message.attachments.map((e) => RemoteAttachmentInfo.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return MultiRemoteAttachment.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseMultiRemoteAttachment();
        message.attachments = object.attachments?.map((e) => RemoteAttachmentInfo.fromPartial(e)) || [];
        return message;
    },
};
function createBaseRemoteAttachmentInfo() {
    return {
        contentDigest: "",
        secret: new Uint8Array(0),
        nonce: new Uint8Array(0),
        salt: new Uint8Array(0),
        scheme: "",
        url: "",
        contentLength: undefined,
        filename: undefined,
    };
}
export const RemoteAttachmentInfo = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.contentDigest !== "") {
            writer.uint32(10).string(message.contentDigest);
        }
        if (message.secret.length !== 0) {
            writer.uint32(18).bytes(message.secret);
        }
        if (message.nonce.length !== 0) {
            writer.uint32(26).bytes(message.nonce);
        }
        if (message.salt.length !== 0) {
            writer.uint32(34).bytes(message.salt);
        }
        if (message.scheme !== "") {
            writer.uint32(42).string(message.scheme);
        }
        if (message.url !== "") {
            writer.uint32(50).string(message.url);
        }
        if (message.contentLength !== undefined) {
            writer.uint32(56).uint32(message.contentLength);
        }
        if (message.filename !== undefined) {
            writer.uint32(66).string(message.filename);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseRemoteAttachmentInfo();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.contentDigest = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.secret = reader.bytes();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.nonce = reader.bytes();
                    continue;
                case 4:
                    if (tag !== 34) {
                        break;
                    }
                    message.salt = reader.bytes();
                    continue;
                case 5:
                    if (tag !== 42) {
                        break;
                    }
                    message.scheme = reader.string();
                    continue;
                case 6:
                    if (tag !== 50) {
                        break;
                    }
                    message.url = reader.string();
                    continue;
                case 7:
                    if (tag !== 56) {
                        break;
                    }
                    message.contentLength = reader.uint32();
                    continue;
                case 8:
                    if (tag !== 66) {
                        break;
                    }
                    message.filename = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            contentDigest: isSet(object.contentDigest) ? globalThis.String(object.contentDigest) : "",
            secret: isSet(object.secret) ? bytesFromBase64(object.secret) : new Uint8Array(0),
            nonce: isSet(object.nonce) ? bytesFromBase64(object.nonce) : new Uint8Array(0),
            salt: isSet(object.salt) ? bytesFromBase64(object.salt) : new Uint8Array(0),
            scheme: isSet(object.scheme) ? globalThis.String(object.scheme) : "",
            url: isSet(object.url) ? globalThis.String(object.url) : "",
            contentLength: isSet(object.contentLength) ? globalThis.Number(object.contentLength) : undefined,
            filename: isSet(object.filename) ? globalThis.String(object.filename) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.contentDigest !== "") {
            obj.contentDigest = message.contentDigest;
        }
        if (message.secret.length !== 0) {
            obj.secret = base64FromBytes(message.secret);
        }
        if (message.nonce.length !== 0) {
            obj.nonce = base64FromBytes(message.nonce);
        }
        if (message.salt.length !== 0) {
            obj.salt = base64FromBytes(message.salt);
        }
        if (message.scheme !== "") {
            obj.scheme = message.scheme;
        }
        if (message.url !== "") {
            obj.url = message.url;
        }
        if (message.contentLength !== undefined) {
            obj.contentLength = Math.round(message.contentLength);
        }
        if (message.filename !== undefined) {
            obj.filename = message.filename;
        }
        return obj;
    },
    create(base) {
        return RemoteAttachmentInfo.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseRemoteAttachmentInfo();
        message.contentDigest = object.contentDigest ?? "";
        message.secret = object.secret ?? new Uint8Array(0);
        message.nonce = object.nonce ?? new Uint8Array(0);
        message.salt = object.salt ?? new Uint8Array(0);
        message.scheme = object.scheme ?? "";
        message.url = object.url ?? "";
        message.contentLength = object.contentLength ?? undefined;
        message.filename = object.filename ?? undefined;
        return message;
    },
};
function bytesFromBase64(b64) {
    if (globalThis.Buffer) {
        return Uint8Array.from(globalThis.Buffer.from(b64, "base64"));
    }
    else {
        const bin = globalThis.atob(b64);
        const arr = new Uint8Array(bin.length);
        for (let i = 0; i < bin.length; ++i) {
            arr[i] = bin.charCodeAt(i);
        }
        return arr;
    }
}
function base64FromBytes(arr) {
    if (globalThis.Buffer) {
        return globalThis.Buffer.from(arr).toString("base64");
    }
    else {
        const bin = [];
        arr.forEach((byte) => {
            bin.push(globalThis.String.fromCharCode(byte));
        });
        return globalThis.btoa(bin.join(""));
    }
}
if (_m0.util.Long !== Long) {
    _m0.util.Long = Long;
    _m0.configure();
}
function isSet(value) {
    return value !== null && value !== undefined;
}
