// @xmtp/proto/ts/dist/esm/mls/message_contents/content_types/wallet_send_calls.pb.js
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v1.181.2
//   protoc               v3.21.12
// source: mls/message_contents/content_types/wallet_send_calls.proto
/* eslint-disable */
import Long from "long";
import _m0 from "protobufjs/minimal";
export const protobufPackage = "xmtp.mls.message_contents.content_types";
function createBaseWalletSendCalls() {
    return { version: "", chainId: "", from: "", calls: [], capabilities: {} };
}
export const WalletSendCalls = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.version !== "") {
            writer.uint32(10).string(message.version);
        }
        if (message.chainId !== "") {
            writer.uint32(18).string(message.chainId);
        }
        if (message.from !== "") {
            writer.uint32(26).string(message.from);
        }
        for (const v of message.calls) {
            Call.encode(v, writer.uint32(34).fork()).ldelim();
        }
        Object.entries(message.capabilities).forEach(([key, value]) => {
            WalletSendCalls_CapabilitiesEntry.encode({ key: key, value }, writer.uint32(42).fork()).ldelim();
        });
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseWalletSendCalls();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.version = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.chainId = reader.string();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.from = reader.string();
                    continue;
                case 4:
                    if (tag !== 34) {
                        break;
                    }
                    message.calls.push(Call.decode(reader, reader.uint32()));
                    continue;
                case 5:
                    if (tag !== 42) {
                        break;
                    }
                    const entry5 = WalletSendCalls_CapabilitiesEntry.decode(reader, reader.uint32());
                    if (entry5.value !== undefined) {
                        message.capabilities[entry5.key] = entry5.value;
                    }
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            version: isSet(object.version) ? globalThis.String(object.version) : "",
            chainId: isSet(object.chainId) ? globalThis.String(object.chainId) : "",
            from: isSet(object.from) ? globalThis.String(object.from) : "",
            calls: globalThis.Array.isArray(object?.calls) ? object.calls.map((e) => Call.fromJSON(e)) : [],
            capabilities: isObject(object.capabilities)
                ? Object.entries(object.capabilities).reduce((acc, [key, value]) => {
                    acc[key] = String(value);
                    return acc;
                }, {})
                : {},
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.version !== "") {
            obj.version = message.version;
        }
        if (message.chainId !== "") {
            obj.chainId = message.chainId;
        }
        if (message.from !== "") {
            obj.from = message.from;
        }
        if (message.calls?.length) {
            obj.calls = message.calls.map((e) => Call.toJSON(e));
        }
        if (message.capabilities) {
            const entries = Object.entries(message.capabilities);
            if (entries.length > 0) {
                obj.capabilities = {};
                entries.forEach(([k, v]) => {
                    obj.capabilities[k] = v;
                });
            }
        }
        return obj;
    },
    create(base) {
        return WalletSendCalls.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseWalletSendCalls();
        message.version = object.version ?? "";
        message.chainId = object.chainId ?? "";
        message.from = object.from ?? "";
        message.calls = object.calls?.map((e) => Call.fromPartial(e)) || [];
        message.capabilities = Object.entries(object.capabilities ?? {}).reduce((acc, [key, value]) => {
            if (value !== undefined) {
                acc[key] = globalThis.String(value);
            }
            return acc;
        }, {});
        return message;
    },
};
function createBaseWalletSendCalls_CapabilitiesEntry() {
    return { key: "", value: "" };
}
export const WalletSendCalls_CapabilitiesEntry = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.key !== "") {
            writer.uint32(10).string(message.key);
        }
        if (message.value !== "") {
            writer.uint32(18).string(message.value);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseWalletSendCalls_CapabilitiesEntry();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.key = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.value = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            key: isSet(object.key) ? globalThis.String(object.key) : "",
            value: isSet(object.value) ? globalThis.String(object.value) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.key !== "") {
            obj.key = message.key;
        }
        if (message.value !== "") {
            obj.value = message.value;
        }
        return obj;
    },
    create(base) {
        return WalletSendCalls_CapabilitiesEntry.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseWalletSendCalls_CapabilitiesEntry();
        message.key = object.key ?? "";
        message.value = object.value ?? "";
        return message;
    },
};
function createBaseCall() {
    return { to: "", data: "", value: "", gas: "", metadata: {} };
}
export const Call = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.to !== "") {
            writer.uint32(10).string(message.to);
        }
        if (message.data !== "") {
            writer.uint32(18).string(message.data);
        }
        if (message.value !== "") {
            writer.uint32(26).string(message.value);
        }
        if (message.gas !== "") {
            writer.uint32(34).string(message.gas);
        }
        Object.entries(message.metadata).forEach(([key, value]) => {
            Call_MetadataEntry.encode({ key: key, value }, writer.uint32(42).fork()).ldelim();
        });
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseCall();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.to = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.data = reader.string();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.value = reader.string();
                    continue;
                case 4:
                    if (tag !== 34) {
                        break;
                    }
                    message.gas = reader.string();
                    continue;
                case 5:
                    if (tag !== 42) {
                        break;
                    }
                    const entry5 = Call_MetadataEntry.decode(reader, reader.uint32());
                    if (entry5.value !== undefined) {
                        message.metadata[entry5.key] = entry5.value;
                    }
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            to: isSet(object.to) ? globalThis.String(object.to) : "",
            data: isSet(object.data) ? globalThis.String(object.data) : "",
            value: isSet(object.value) ? globalThis.String(object.value) : "",
            gas: isSet(object.gas) ? globalThis.String(object.gas) : "",
            metadata: isObject(object.metadata)
                ? Object.entries(object.metadata).reduce((acc, [key, value]) => {
                    acc[key] = String(value);
                    return acc;
                }, {})
                : {},
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.to !== "") {
            obj.to = message.to;
        }
        if (message.data !== "") {
            obj.data = message.data;
        }
        if (message.value !== "") {
            obj.value = message.value;
        }
        if (message.gas !== "") {
            obj.gas = message.gas;
        }
        if (message.metadata) {
            const entries = Object.entries(message.metadata);
            if (entries.length > 0) {
                obj.metadata = {};
                entries.forEach(([k, v]) => {
                    obj.metadata[k] = v;
                });
            }
        }
        return obj;
    },
    create(base) {
        return Call.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseCall();
        message.to = object.to ?? "";
        message.data = object.data ?? "";
        message.value = object.value ?? "";
        message.gas = object.gas ?? "";
        message.metadata = Object.entries(object.metadata ?? {}).reduce((acc, [key, value]) => {
            if (value !== undefined) {
                acc[key] = globalThis.String(value);
            }
            return acc;
        }, {});
        return message;
    },
};
function createBaseCall_MetadataEntry() {
    return { key: "", value: "" };
}
export const Call_MetadataEntry = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.key !== "") {
            writer.uint32(10).string(message.key);
        }
        if (message.value !== "") {
            writer.uint32(18).string(message.value);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseCall_MetadataEntry();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.key = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.value = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            key: isSet(object.key) ? globalThis.String(object.key) : "",
            value: isSet(object.value) ? globalThis.String(object.value) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.key !== "") {
            obj.key = message.key;
        }
        if (message.value !== "") {
            obj.value = message.value;
        }
        return obj;
    },
    create(base) {
        return Call_MetadataEntry.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseCall_MetadataEntry();
        message.key = object.key ?? "";
        message.value = object.value ?? "";
        return message;
    },
};
if (_m0.util.Long !== Long) {
    _m0.util.Long = Long;
    _m0.configure();
}
function isObject(value) {
    return typeof value === "object" && value !== null;
}
function isSet(value) {
    return value !== null && value !== undefined;
}
