// molt-life-kernel/src/aegis-claw.ts
/**
 * AEGIS CLAW — molt-life-kernel Integration
 * X-Loop³ Labs | jon@x-loop3.com
 * USPTO PPA 63/996,268
 *
 * Flow: Prompt → Aegis Gate → PASS/BLOCK → Kernel Ledger → Dispatch
 */
import { MoltLifeKernel } from './kernel';
import { SecureMoltKernel } from './secure-kernel';

const GATE_URL = process.env.AEGIS_GATE_URL || 'https://tower.x-loop3.com';
const LICENSE_KEY = process.env.AEGIS_LICENSE_KEY || '';

export class AegisClaw {
  private kernel: SecureMoltKernel;
  private agentId: string;
  private gateUrl: string;
  private licenseKey: string;

  constructor(options: {
    agentId?: string;
    gateUrl?: string;
    licenseKey?: string;
    witnessCallback?: (action: any) => Promise<boolean>;
    heartbeatMs?: number;
  } = {}) {
    this.agentId = options.agentId || `agent-${Date.now()}`;
    this.gateUrl = options.gateUrl || GATE_URL;
    this.licenseKey = options.licenseKey || LICENSE_KEY;

    this.kernel = new SecureMoltKernel({
      heartbeatMs: options.heartbeatMs || 3600000,
      witnessCallback: options.witnessCallback || this.defaultWitness.bind(this),
      enableMetrics: true
    });

    this.kernel.setInvariant('gate_url', this.gateUrl);
    this.kernel.setInvariant('agent_id', this.agentId);
    this.kernel.setInvariant('initialized_at', new Date().toISOString());

    this.kernel.append({
      type: 'aegis_init',
      payload: { agentId: this.agentId, gateUrl: this.gateUrl },
      risk: 0.0
    });
  }

  async check(prompt: string): Promise<{
    pass: boolean;
    decision: 'PASS' | 'BLOCK' | 'QUARANTINE';
    reason: string;
    coherenceScore: number;
    hash: string;
  }> {
    await this.kernel.heartbeat();

    try {
      this.kernel.enforceCoherence(50);
    } catch (e) {
      this.kernel.append({
        type: 'coherence_violation',
        payload: { error: (e as Error).message },
        risk: 0.8
      });
    }

    let gateResult: any;
    try {
      const response = await fetch(`${this.gateUrl}/api/v1/gate`, {
        method: 'POST',
        headers: { 'Content-Type': 'application/json' },
        body: JSON.stringify({
          license_key: this.licenseKey,
          prompt,
          agent_id: this.agentId
        }),
        signal: AbortSignal.timeout(5000)
      });
      gateResult = await response.json();
    } catch (error) {
      this.kernel.append({
        type: 'gate_error',
        payload: { error: (error as Error).message, defaulted_to_block: true },
        risk: 0.9
      });
      return {
        pass: false,
        decision: 'BLOCK',
        reason: `Gate unreachable: ${(error as Error).message}. Default BLOCK.`,
        coherenceScore: 0,
        hash: 'ERROR'
      };
    }

    const decision = gateResult.decision as 'PASS' | 'BLOCK' | 'QUARANTINE';
    const coherenceScore = gateResult.coherence_score || 0;
    const hash = gateResult.hash || 'NO_HASH';
    const reason = gateResult.reason || '';

    this.kernel.append({
      type: `gate_${decision.toLowerCase()}`,
      payload: { decision, coherence_score: coherenceScore, hash, reason, code: gateResult.code },
      risk: decision === 'BLOCK' ? 0.9 : decision === 'QUARANTINE' ? 0.75 : 0.1
    });

    if (decision === 'QUARANTINE') {
      const approved = await this.kernel.witness({
        type: 'quarantined_prompt',
        payload: { reason, coherence_score: coherenceScore },
        risk: 0.85
      });

      if (!approved) {
        return { pass: false, decision: 'BLOCK', reason: `QUARANTINE not approved. ${reason}`, coherenceScore, hash };
      }
      return { pass: true, decision: 'PASS', reason: 'Quarantine approved by witness', coherenceScore, hash };
    }

    return { pass: decision === 'PASS', decision, reason, coherenceScore, hash };
  }

  getSnapshot() { return this.kernel.getSnapshot(); }
  getMetrics() { return this.kernel.getMetrics(); }
  getHealth() { return this.kernel.getHealth(); }
  getAudit() { return this.kernel.getAudit(); }

  private async defaultWitness(action: any): Promise<boolean> {
    const risk = action.risk || 0;
    if (risk < 0.8) return true;
    return false;
  }
}

export async function createAegisMiddleware(options: {
  gateUrl?: string;
  licenseKey?: string;
  agentId?: string;
  onBlock?: (reason: string) => void;
  onPass?: (hash: string) => void;
}) {
  const claw = new AegisClaw(options);

  return {
    async gate(prompt: string): Promise<boolean> {
      const result = await claw.check(prompt);
      if (!result.pass) { options.onBlock?.(result.reason); return false; }
      options.onPass?.(result.hash);
      return true;
    },
    instance: claw
  };
}
