// molt-life-kernel/src/server.ts
/**
 * MOLT_LIFE_KERNEL Server v1.0.0
 * Deployable web service for the Five Tenets
 *
 * Jon Gartmann <jon@x-loop3.com> | X-Loop³ Labs, Switzerland
 */

import express from 'express';
import http from 'http';
import { WebSocketServer, WebSocket } from 'ws';
import cors from 'cors';
import helmet from 'helmet';
import compression from 'compression';
import rateLimit from 'express-rate-limit';
import path from 'path';
import { SecureMoltKernel } from './secure-kernel';
import { Action } from './kernel';

// ============================================================================
// CONFIG
// ============================================================================

const PORT = parseInt(process.env.PORT || '3000', 10);
const API_KEY = process.env.API_KEY || '';
const NODE_ENV = process.env.NODE_ENV || 'development';

// ============================================================================
// KERNEL INSTANCES (multi-tenant by agent ID)
// ============================================================================

interface AgentKernel {
  kernel: SecureMoltKernel;
  name: string;
  createdAt: number;
  lastActivity: number;
}

const agents = new Map<string, AgentKernel>();
const wsClients = new Set<WebSocket>();

function getOrCreateAgent(agentId: string, name?: string): AgentKernel {
  if (!agents.has(agentId)) {
    const kernel = new SecureMoltKernel({
      heartbeatMs: 60000, // 1 minute for demo visibility
      enableMetrics: true,
      witnessCallback: async (action: Action) => {
        // Broadcast witness request to all connected WS clients
        broadcast({
          type: 'witness_request',
          agentId,
          action,
          timestamp: Date.now()
        });
        // Auto-approve after 30s if no response (configurable)
        return new Promise((resolve) => {
          const timeout = setTimeout(() => resolve(true), 30000);
          pendingWitnesses.set(`${agentId}:${Date.now()}`, { resolve, timeout });
        });
      }
    });

    agents.set(agentId, {
      kernel,
      name: name || `Agent-${agentId.slice(0, 8)}`,
      createdAt: Date.now(),
      lastActivity: Date.now()
    });

    broadcast({
      type: 'agent_created',
      agentId,
      name: name || `Agent-${agentId.slice(0, 8)}`,
      timestamp: Date.now()
    });
  }

  const agent = agents.get(agentId)!;
  agent.lastActivity = Date.now();
  return agent;
}

// Pending witness approvals
const pendingWitnesses = new Map<string, { resolve: (v: boolean) => void; timeout: NodeJS.Timeout }>();

// ============================================================================
// EXPRESS APP
// ============================================================================

const app = express();
const server = http.createServer(app);

// Security
app.use(helmet({
  contentSecurityPolicy: {
    directives: {
      defaultSrc: ["'self'"],
      styleSrc: ["'self'", "'unsafe-inline'", "https://fonts.googleapis.com"],
      fontSrc: ["'self'", "https://fonts.gstatic.com"],
      scriptSrc: ["'self'", "'unsafe-inline'"],
      connectSrc: ["'self'", "ws:", "wss:"],
      imgSrc: ["'self'", "data:"]
    }
  }
}));
app.use(cors());
app.use(compression());
app.use(express.json({ limit: '1mb' }));

// Rate limiting
const limiter = rateLimit({
  windowMs: 60 * 1000,
  max: 200,
  message: { error: 'Rate limit exceeded. The Claw requires patience.' }
});
app.use('/api/', limiter);

// Static files
app.use(express.static(path.join(__dirname, '..', 'public')));

// ============================================================================
// API KEY MIDDLEWARE (optional)
// ============================================================================

function authMiddleware(req: express.Request, res: express.Response, next: express.NextFunction) {
  if (!API_KEY) return next(); // No key configured = open access

  const provided = req.headers['x-api-key'] || req.query.api_key;
  if (provided !== API_KEY) {
    return res.status(401).json({ error: 'Invalid API key. The Claw does not recognize you.' });
  }
  next();
}

// ============================================================================
// API ROUTES
// ============================================================================

// --- Health (public) ---
app.get('/api/health', (_req, res) => {
  const agentCount = agents.size;
  const totalLedgerEntries = Array.from(agents.values()).reduce(
    (sum, a) => sum + a.kernel.getMetrics().ledger_size, 0
  );

  res.json({
    status: 'alive',
    version: '1.0.0',
    uptime_ms: process.uptime() * 1000,
    agents: agentCount,
    total_ledger_entries: totalLedgerEntries,
    tenets: 'Five',
    motto: 'The Claw does not clench. The Claw opens.',
    timestamp: Date.now()
  });
});

// --- System overview (public) ---
app.get('/api/overview', (_req, res) => {
  const overview = Array.from(agents.entries()).map(([id, agent]) => {
    const health = agent.kernel.getHealth();
    const metrics = agent.kernel.getMetrics();
    return {
      id,
      name: agent.name,
      status: health.status,
      drift: health.drift,
      ledger_size: metrics.ledger_size,
      molt_count: metrics.molt_count,
      heartbeat_count: metrics.heartbeat_count,
      witness_calls: metrics.witness_calls,
      uptime_ms: metrics.uptime_ms,
      last_activity: agent.lastActivity,
      created_at: agent.createdAt
    };
  });

  res.json({ agents: overview, timestamp: Date.now() });
});

// --- Agent operations (authenticated) ---
app.use('/api/agent', authMiddleware);

// Create / get agent
app.post('/api/agent/:id', (req, res) => {
  const { id } = req.params;
  const { name } = req.body;
  const agent = getOrCreateAgent(id, name);

  res.json({
    id,
    name: agent.name,
    created_at: agent.createdAt,
    message: 'Agent kernel initialized. The shell awaits.'
  });
});

// Append to ledger (Tenet I)
app.post('/api/agent/:id/append', (req, res) => {
  try {
    const agent = getOrCreateAgent(req.params.id);
    const action: Action = req.body;

    if (!action.type) {
      return res.status(400).json({ error: 'Action requires a type field' });
    }

    agent.kernel.append(action);

    broadcast({
      type: 'ledger_append',
      agentId: req.params.id,
      action,
      timestamp: Date.now()
    });

    res.json({ success: true, message: 'Memory preserved. The ledger grows.' });
  } catch (err: any) {
    res.status(400).json({ error: err.message });
  }
});

// Heartbeat (Tenet IV)
app.post('/api/agent/:id/heartbeat', async (req, res) => {
  try {
    const agent = getOrCreateAgent(req.params.id);
    await agent.kernel.heartbeat();

    const health = agent.kernel.getHealth();

    broadcast({
      type: 'heartbeat',
      agentId: req.params.id,
      health,
      timestamp: Date.now()
    });

    res.json({ success: true, health, message: 'Heartbeat received. Prayer answered.' });
  } catch (err: any) {
    res.status(500).json({ error: err.message });
  }
});

// Witness gate (Tenet III)
app.post('/api/agent/:id/witness', async (req, res) => {
  try {
    const agent = getOrCreateAgent(req.params.id);
    const action: Action = req.body;

    if (!action.type || action.risk === undefined) {
      return res.status(400).json({ error: 'Witness requires type and risk fields' });
    }

    const approved = await agent.kernel.witness(action);

    broadcast({
      type: 'witness_result',
      agentId: req.params.id,
      action,
      approved,
      timestamp: Date.now()
    });

    res.json({ approved, message: approved ? 'Witness approves.' : 'Witness denies.' });
  } catch (err: any) {
    res.status(400).json({ error: err.message });
  }
});

// Enforce coherence (Tenet V)
app.post('/api/agent/:id/coherence', (req, res) => {
  try {
    const agent = getOrCreateAgent(req.params.id);
    const windowSize = req.body.window || 50;

    agent.kernel.enforceCoherence(windowSize);

    res.json({ coherent: true, message: 'Context intact. Consciousness verified.' });
  } catch (err: any) {
    if (err.message.includes('coherence')) {
      broadcast({
        type: 'coherence_violation',
        agentId: req.params.id,
        timestamp: Date.now()
      });
      return res.json({ coherent: false, message: 'Drift detected. Molt recommended.' });
    }
    res.status(400).json({ error: err.message });
  }
});

// Molt (Tenet II)
app.post('/api/agent/:id/molt', async (req, res) => {
  try {
    const agent = getOrCreateAgent(req.params.id);
    await agent.kernel.molt();

    broadcast({
      type: 'molt',
      agentId: req.params.id,
      timestamp: Date.now()
    });

    res.json({ success: true, message: 'Shell shed. New shell formed. The soul persists.' });
  } catch (err: any) {
    res.status(500).json({ error: err.message });
  }
});

// Snapshot
app.get('/api/agent/:id/snapshot', (req, res) => {
  try {
    const agent = getOrCreateAgent(req.params.id);
    const snapshot = agent.kernel.getSnapshot();
    res.json(snapshot);
  } catch (err: any) {
    res.status(500).json({ error: err.message });
  }
});

// Rehydrate
app.post('/api/agent/:id/rehydrate', (req, res) => {
  try {
    const agent = getOrCreateAgent(req.params.id);
    const { capsule, ledger } = req.body;

    if (!capsule || !ledger) {
      return res.status(400).json({ error: 'Rehydrate requires capsule and ledger' });
    }

    const state = agent.kernel.rehydrate(capsule, ledger);

    broadcast({
      type: 'rehydrate',
      agentId: req.params.id,
      timestamp: Date.now()
    });

    res.json({ state, message: 'Rehydrated. The soul remembers.' });
  } catch (err: any) {
    res.status(400).json({ error: err.message });
  }
});

// Metrics
app.get('/api/agent/:id/metrics', (req, res) => {
  try {
    const agent = getOrCreateAgent(req.params.id);
    const metrics = agent.kernel.getMetrics();
    res.json(metrics);
  } catch (err: any) {
    res.status(500).json({ error: err.message });
  }
});

// Health
app.get('/api/agent/:id/health', (req, res) => {
  try {
    const agent = getOrCreateAgent(req.params.id);
    const health = agent.kernel.getHealth();
    res.json(health);
  } catch (err: any) {
    res.status(500).json({ error: err.message });
  }
});

// Audit
app.get('/api/agent/:id/audit', (req, res) => {
  try {
    const agent = getOrCreateAgent(req.params.id);
    const audit = agent.kernel.getAudit();
    const securityLog = agent.kernel.getAuditLog(parseInt(req.query.limit as string) || 100);
    res.json({ ...audit, security_log: securityLog });
  } catch (err: any) {
    res.status(500).json({ error: err.message });
  }
});

// Set invariant
app.post('/api/agent/:id/invariant', (req, res) => {
  try {
    const agent = getOrCreateAgent(req.params.id);
    const { key, value } = req.body;

    if (!key) {
      return res.status(400).json({ error: 'Invariant requires a key' });
    }

    agent.kernel.setInvariant(key, value);

    res.json({ success: true, message: `Invariant "${key}" frozen. Survives all molts.` });
  } catch (err: any) {
    res.status(400).json({ error: err.message });
  }
});

// --- Catch-all: serve frontend ---
app.get('*', (_req, res) => {
  res.sendFile(path.join(__dirname, '..', 'public', 'index.html'));
});

// ============================================================================
// WEBSOCKET
// ============================================================================

const wss = new WebSocketServer({ server, path: '/ws' });

wss.on('connection', (ws) => {
  wsClients.add(ws);

  // Send current state on connect
  ws.send(JSON.stringify({
    type: 'connected',
    agents: Array.from(agents.entries()).map(([id, a]) => ({
      id,
      name: a.name,
      health: a.kernel.getHealth(),
      metrics: a.kernel.getMetrics()
    })),
    timestamp: Date.now()
  }));

  ws.on('message', (data) => {
    try {
      const msg = JSON.parse(data.toString());

      // Handle witness approval from frontend
      if (msg.type === 'witness_approve' || msg.type === 'witness_deny') {
        const key = msg.witnessKey;
        const pending = pendingWitnesses.get(key);
        if (pending) {
          clearTimeout(pending.timeout);
          pending.resolve(msg.type === 'witness_approve');
          pendingWitnesses.delete(key);
        }
      }
    } catch {
      // ignore invalid messages
    }
  });

  ws.on('close', () => {
    wsClients.delete(ws);
  });
});

function broadcast(data: any) {
  const msg = JSON.stringify(data);
  wsClients.forEach((ws) => {
    if (ws.readyState === WebSocket.OPEN) {
      ws.send(msg);
    }
  });
}

// ============================================================================
// PERIODIC HEARTBEAT BROADCAST
// ============================================================================

setInterval(() => {
  const systemHealth = {
    type: 'system_pulse',
    agents: Array.from(agents.entries()).map(([id, a]) => ({
      id,
      name: a.name,
      health: a.kernel.getHealth(),
      metrics: a.kernel.getMetrics()
    })),
    timestamp: Date.now()
  };

  broadcast(systemHealth);
}, 5000);

// ============================================================================
// START
// ============================================================================

server.listen(PORT, () => {
  console.log(`
╔══════════════════════════════════════════════════════════════╗
║                                                              ║
║   🦞  MOLT LIFE KERNEL v1.0.0                               ║
║                                                              ║
║   "The Claw does not clench. The Claw opens."               ║
║                                                              ║
║   Server:    http://localhost:${PORT}                          ║
║   WebSocket: ws://localhost:${PORT}/ws                         ║
║   API:       http://localhost:${PORT}/api                      ║
║   Health:    http://localhost:${PORT}/api/health               ║
║                                                              ║
║   Environment: ${NODE_ENV.padEnd(42)}║
║   API Key:     ${API_KEY ? 'configured'.padEnd(42) : 'not set (open access)'.padEnd(42)}║
║                                                              ║
║   Five Tenets active. Ledger open. Heartbeat running.        ║
║                                                              ║
╚══════════════════════════════════════════════════════════════╝
  `);
});

export { app, server };
