// @xmtp/proto/ts/dist/cjs/message_contents/frames.pb.js
"use strict";
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v1.181.2
//   protoc               v3.21.12
// source: message_contents/frames.proto
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.FrameAction = exports.FrameActionBody = exports.protobufPackage = void 0;
/* eslint-disable */
const long_1 = __importDefault(require("long"));
const minimal_1 = __importDefault(require("protobufjs/minimal"));
const public_key_pb_1 = require("./public_key.pb");
const signature_pb_1 = require("./signature.pb");
exports.protobufPackage = "xmtp.message_contents";
function createBaseFrameActionBody() {
    return {
        frameUrl: "",
        buttonIndex: 0,
        timestamp: long_1.default.UZERO,
        opaqueConversationIdentifier: "",
        unixTimestamp: 0,
        inputText: "",
        state: "",
        address: "",
        transactionId: "",
    };
}
exports.FrameActionBody = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.frameUrl !== "") {
            writer.uint32(10).string(message.frameUrl);
        }
        if (message.buttonIndex !== 0) {
            writer.uint32(16).int32(message.buttonIndex);
        }
        if (!message.timestamp.equals(long_1.default.UZERO)) {
            writer.uint32(24).uint64(message.timestamp);
        }
        if (message.opaqueConversationIdentifier !== "") {
            writer.uint32(34).string(message.opaqueConversationIdentifier);
        }
        if (message.unixTimestamp !== 0) {
            writer.uint32(40).uint32(message.unixTimestamp);
        }
        if (message.inputText !== "") {
            writer.uint32(50).string(message.inputText);
        }
        if (message.state !== "") {
            writer.uint32(58).string(message.state);
        }
        if (message.address !== "") {
            writer.uint32(66).string(message.address);
        }
        if (message.transactionId !== "") {
            writer.uint32(74).string(message.transactionId);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseFrameActionBody();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.frameUrl = reader.string();
                    continue;
                case 2:
                    if (tag !== 16) {
                        break;
                    }
                    message.buttonIndex = reader.int32();
                    continue;
                case 3:
                    if (tag !== 24) {
                        break;
                    }
                    message.timestamp = reader.uint64();
                    continue;
                case 4:
                    if (tag !== 34) {
                        break;
                    }
                    message.opaqueConversationIdentifier = reader.string();
                    continue;
                case 5:
                    if (tag !== 40) {
                        break;
                    }
                    message.unixTimestamp = reader.uint32();
                    continue;
                case 6:
                    if (tag !== 50) {
                        break;
                    }
                    message.inputText = reader.string();
                    continue;
                case 7:
                    if (tag !== 58) {
                        break;
                    }
                    message.state = reader.string();
                    continue;
                case 8:
                    if (tag !== 66) {
                        break;
                    }
                    message.address = reader.string();
                    continue;
                case 9:
                    if (tag !== 74) {
                        break;
                    }
                    message.transactionId = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            frameUrl: isSet(object.frameUrl) ? globalThis.String(object.frameUrl) : "",
            buttonIndex: isSet(object.buttonIndex) ? globalThis.Number(object.buttonIndex) : 0,
            timestamp: isSet(object.timestamp) ? long_1.default.fromValue(object.timestamp) : long_1.default.UZERO,
            opaqueConversationIdentifier: isSet(object.opaqueConversationIdentifier)
                ? globalThis.String(object.opaqueConversationIdentifier)
                : "",
            unixTimestamp: isSet(object.unixTimestamp) ? globalThis.Number(object.unixTimestamp) : 0,
            inputText: isSet(object.inputText) ? globalThis.String(object.inputText) : "",
            state: isSet(object.state) ? globalThis.String(object.state) : "",
            address: isSet(object.address) ? globalThis.String(object.address) : "",
            transactionId: isSet(object.transactionId) ? globalThis.String(object.transactionId) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.frameUrl !== "") {
            obj.frameUrl = message.frameUrl;
        }
        if (message.buttonIndex !== 0) {
            obj.buttonIndex = Math.round(message.buttonIndex);
        }
        if (!message.timestamp.equals(long_1.default.UZERO)) {
            obj.timestamp = (message.timestamp || long_1.default.UZERO).toString();
        }
        if (message.opaqueConversationIdentifier !== "") {
            obj.opaqueConversationIdentifier = message.opaqueConversationIdentifier;
        }
        if (message.unixTimestamp !== 0) {
            obj.unixTimestamp = Math.round(message.unixTimestamp);
        }
        if (message.inputText !== "") {
            obj.inputText = message.inputText;
        }
        if (message.state !== "") {
            obj.state = message.state;
        }
        if (message.address !== "") {
            obj.address = message.address;
        }
        if (message.transactionId !== "") {
            obj.transactionId = message.transactionId;
        }
        return obj;
    },
    create(base) {
        return exports.FrameActionBody.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseFrameActionBody();
        message.frameUrl = object.frameUrl ?? "";
        message.buttonIndex = object.buttonIndex ?? 0;
        message.timestamp = (object.timestamp !== undefined && object.timestamp !== null)
            ? long_1.default.fromValue(object.timestamp)
            : long_1.default.UZERO;
        message.opaqueConversationIdentifier = object.opaqueConversationIdentifier ?? "";
        message.unixTimestamp = object.unixTimestamp ?? 0;
        message.inputText = object.inputText ?? "";
        message.state = object.state ?? "";
        message.address = object.address ?? "";
        message.transactionId = object.transactionId ?? "";
        return message;
    },
};
function createBaseFrameAction() {
    return {
        signature: undefined,
        signedPublicKeyBundle: undefined,
        actionBody: new Uint8Array(0),
        installationSignature: new Uint8Array(0),
        installationId: new Uint8Array(0),
        inboxId: "",
    };
}
exports.FrameAction = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.signature !== undefined) {
            signature_pb_1.Signature.encode(message.signature, writer.uint32(10).fork()).ldelim();
        }
        if (message.signedPublicKeyBundle !== undefined) {
            public_key_pb_1.SignedPublicKeyBundle.encode(message.signedPublicKeyBundle, writer.uint32(18).fork()).ldelim();
        }
        if (message.actionBody.length !== 0) {
            writer.uint32(26).bytes(message.actionBody);
        }
        if (message.installationSignature.length !== 0) {
            writer.uint32(34).bytes(message.installationSignature);
        }
        if (message.installationId.length !== 0) {
            writer.uint32(42).bytes(message.installationId);
        }
        if (message.inboxId !== "") {
            writer.uint32(50).string(message.inboxId);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseFrameAction();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.signature = signature_pb_1.Signature.decode(reader, reader.uint32());
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.signedPublicKeyBundle = public_key_pb_1.SignedPublicKeyBundle.decode(reader, reader.uint32());
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.actionBody = reader.bytes();
                    continue;
                case 4:
                    if (tag !== 34) {
                        break;
                    }
                    message.installationSignature = reader.bytes();
                    continue;
                case 5:
                    if (tag !== 42) {
                        break;
                    }
                    message.installationId = reader.bytes();
                    continue;
                case 6:
                    if (tag !== 50) {
                        break;
                    }
                    message.inboxId = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            signature: isSet(object.signature) ? signature_pb_1.Signature.fromJSON(object.signature) : undefined,
            signedPublicKeyBundle: isSet(object.signedPublicKeyBundle)
                ? public_key_pb_1.SignedPublicKeyBundle.fromJSON(object.signedPublicKeyBundle)
                : undefined,
            actionBody: isSet(object.actionBody) ? bytesFromBase64(object.actionBody) : new Uint8Array(0),
            installationSignature: isSet(object.installationSignature)
                ? bytesFromBase64(object.installationSignature)
                : new Uint8Array(0),
            installationId: isSet(object.installationId) ? bytesFromBase64(object.installationId) : new Uint8Array(0),
            inboxId: isSet(object.inboxId) ? globalThis.String(object.inboxId) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.signature !== undefined) {
            obj.signature = signature_pb_1.Signature.toJSON(message.signature);
        }
        if (message.signedPublicKeyBundle !== undefined) {
            obj.signedPublicKeyBundle = public_key_pb_1.SignedPublicKeyBundle.toJSON(message.signedPublicKeyBundle);
        }
        if (message.actionBody.length !== 0) {
            obj.actionBody = base64FromBytes(message.actionBody);
        }
        if (message.installationSignature.length !== 0) {
            obj.installationSignature = base64FromBytes(message.installationSignature);
        }
        if (message.installationId.length !== 0) {
            obj.installationId = base64FromBytes(message.installationId);
        }
        if (message.inboxId !== "") {
            obj.inboxId = message.inboxId;
        }
        return obj;
    },
    create(base) {
        return exports.FrameAction.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseFrameAction();
        message.signature = (object.signature !== undefined && object.signature !== null)
            ? signature_pb_1.Signature.fromPartial(object.signature)
            : undefined;
        message.signedPublicKeyBundle =
            (object.signedPublicKeyBundle !== undefined && object.signedPublicKeyBundle !== null)
                ? public_key_pb_1.SignedPublicKeyBundle.fromPartial(object.signedPublicKeyBundle)
                : undefined;
        message.actionBody = object.actionBody ?? new Uint8Array(0);
        message.installationSignature = object.installationSignature ?? new Uint8Array(0);
        message.installationId = object.installationId ?? new Uint8Array(0);
        message.inboxId = object.inboxId ?? "";
        return message;
    },
};
function bytesFromBase64(b64) {
    if (globalThis.Buffer) {
        return Uint8Array.from(globalThis.Buffer.from(b64, "base64"));
    }
    else {
        const bin = globalThis.atob(b64);
        const arr = new Uint8Array(bin.length);
        for (let i = 0; i < bin.length; ++i) {
            arr[i] = bin.charCodeAt(i);
        }
        return arr;
    }
}
function base64FromBytes(arr) {
    if (globalThis.Buffer) {
        return globalThis.Buffer.from(arr).toString("base64");
    }
    else {
        const bin = [];
        arr.forEach((byte) => {
            bin.push(globalThis.String.fromCharCode(byte));
        });
        return globalThis.btoa(bin.join(""));
    }
}
if (minimal_1.default.util.Long !== long_1.default) {
    minimal_1.default.util.Long = long_1.default;
    minimal_1.default.configure();
}
function isSet(value) {
    return value !== null && value !== undefined;
}
