// @xmtp/proto/ts/dist/cjs/message_contents/private_preferences.pb.js
"use strict";
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v1.181.2
//   protoc               v3.21.12
// source: message_contents/private_preferences.proto
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PrivatePreferencesPayload = exports.PrivatePreferencesAction_DenyGroup = exports.PrivatePreferencesAction_AllowGroup = exports.PrivatePreferencesAction_DenyInboxId = exports.PrivatePreferencesAction_AllowInboxId = exports.PrivatePreferencesAction_DenyAddress = exports.PrivatePreferencesAction_AllowAddress = exports.PrivatePreferencesAction = exports.protobufPackage = void 0;
/* eslint-disable */
const long_1 = __importDefault(require("long"));
const minimal_1 = __importDefault(require("protobufjs/minimal"));
const ciphertext_pb_1 = require("./ciphertext.pb");
exports.protobufPackage = "xmtp.message_contents";
function createBasePrivatePreferencesAction() {
    return {
        allowAddress: undefined,
        denyAddress: undefined,
        allowGroup: undefined,
        denyGroup: undefined,
        allowInboxId: undefined,
        denyInboxId: undefined,
    };
}
exports.PrivatePreferencesAction = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.allowAddress !== undefined) {
            exports.PrivatePreferencesAction_AllowAddress.encode(message.allowAddress, writer.uint32(10).fork()).ldelim();
        }
        if (message.denyAddress !== undefined) {
            exports.PrivatePreferencesAction_DenyAddress.encode(message.denyAddress, writer.uint32(18).fork()).ldelim();
        }
        if (message.allowGroup !== undefined) {
            exports.PrivatePreferencesAction_AllowGroup.encode(message.allowGroup, writer.uint32(26).fork()).ldelim();
        }
        if (message.denyGroup !== undefined) {
            exports.PrivatePreferencesAction_DenyGroup.encode(message.denyGroup, writer.uint32(34).fork()).ldelim();
        }
        if (message.allowInboxId !== undefined) {
            exports.PrivatePreferencesAction_AllowInboxId.encode(message.allowInboxId, writer.uint32(42).fork()).ldelim();
        }
        if (message.denyInboxId !== undefined) {
            exports.PrivatePreferencesAction_DenyInboxId.encode(message.denyInboxId, writer.uint32(50).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBasePrivatePreferencesAction();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.allowAddress = exports.PrivatePreferencesAction_AllowAddress.decode(reader, reader.uint32());
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.denyAddress = exports.PrivatePreferencesAction_DenyAddress.decode(reader, reader.uint32());
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.allowGroup = exports.PrivatePreferencesAction_AllowGroup.decode(reader, reader.uint32());
                    continue;
                case 4:
                    if (tag !== 34) {
                        break;
                    }
                    message.denyGroup = exports.PrivatePreferencesAction_DenyGroup.decode(reader, reader.uint32());
                    continue;
                case 5:
                    if (tag !== 42) {
                        break;
                    }
                    message.allowInboxId = exports.PrivatePreferencesAction_AllowInboxId.decode(reader, reader.uint32());
                    continue;
                case 6:
                    if (tag !== 50) {
                        break;
                    }
                    message.denyInboxId = exports.PrivatePreferencesAction_DenyInboxId.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            allowAddress: isSet(object.allowAddress)
                ? exports.PrivatePreferencesAction_AllowAddress.fromJSON(object.allowAddress)
                : undefined,
            denyAddress: isSet(object.denyAddress)
                ? exports.PrivatePreferencesAction_DenyAddress.fromJSON(object.denyAddress)
                : undefined,
            allowGroup: isSet(object.allowGroup)
                ? exports.PrivatePreferencesAction_AllowGroup.fromJSON(object.allowGroup)
                : undefined,
            denyGroup: isSet(object.denyGroup) ? exports.PrivatePreferencesAction_DenyGroup.fromJSON(object.denyGroup) : undefined,
            allowInboxId: isSet(object.allowInboxId)
                ? exports.PrivatePreferencesAction_AllowInboxId.fromJSON(object.allowInboxId)
                : undefined,
            denyInboxId: isSet(object.denyInboxId)
                ? exports.PrivatePreferencesAction_DenyInboxId.fromJSON(object.denyInboxId)
                : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.allowAddress !== undefined) {
            obj.allowAddress = exports.PrivatePreferencesAction_AllowAddress.toJSON(message.allowAddress);
        }
        if (message.denyAddress !== undefined) {
            obj.denyAddress = exports.PrivatePreferencesAction_DenyAddress.toJSON(message.denyAddress);
        }
        if (message.allowGroup !== undefined) {
            obj.allowGroup = exports.PrivatePreferencesAction_AllowGroup.toJSON(message.allowGroup);
        }
        if (message.denyGroup !== undefined) {
            obj.denyGroup = exports.PrivatePreferencesAction_DenyGroup.toJSON(message.denyGroup);
        }
        if (message.allowInboxId !== undefined) {
            obj.allowInboxId = exports.PrivatePreferencesAction_AllowInboxId.toJSON(message.allowInboxId);
        }
        if (message.denyInboxId !== undefined) {
            obj.denyInboxId = exports.PrivatePreferencesAction_DenyInboxId.toJSON(message.denyInboxId);
        }
        return obj;
    },
    create(base) {
        return exports.PrivatePreferencesAction.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBasePrivatePreferencesAction();
        message.allowAddress = (object.allowAddress !== undefined && object.allowAddress !== null)
            ? exports.PrivatePreferencesAction_AllowAddress.fromPartial(object.allowAddress)
            : undefined;
        message.denyAddress = (object.denyAddress !== undefined && object.denyAddress !== null)
            ? exports.PrivatePreferencesAction_DenyAddress.fromPartial(object.denyAddress)
            : undefined;
        message.allowGroup = (object.allowGroup !== undefined && object.allowGroup !== null)
            ? exports.PrivatePreferencesAction_AllowGroup.fromPartial(object.allowGroup)
            : undefined;
        message.denyGroup = (object.denyGroup !== undefined && object.denyGroup !== null)
            ? exports.PrivatePreferencesAction_DenyGroup.fromPartial(object.denyGroup)
            : undefined;
        message.allowInboxId = (object.allowInboxId !== undefined && object.allowInboxId !== null)
            ? exports.PrivatePreferencesAction_AllowInboxId.fromPartial(object.allowInboxId)
            : undefined;
        message.denyInboxId = (object.denyInboxId !== undefined && object.denyInboxId !== null)
            ? exports.PrivatePreferencesAction_DenyInboxId.fromPartial(object.denyInboxId)
            : undefined;
        return message;
    },
};
function createBasePrivatePreferencesAction_AllowAddress() {
    return { walletAddresses: [] };
}
exports.PrivatePreferencesAction_AllowAddress = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        for (const v of message.walletAddresses) {
            writer.uint32(10).string(v);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBasePrivatePreferencesAction_AllowAddress();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.walletAddresses.push(reader.string());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            walletAddresses: globalThis.Array.isArray(object?.walletAddresses)
                ? object.walletAddresses.map((e) => globalThis.String(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.walletAddresses?.length) {
            obj.walletAddresses = message.walletAddresses;
        }
        return obj;
    },
    create(base) {
        return exports.PrivatePreferencesAction_AllowAddress.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBasePrivatePreferencesAction_AllowAddress();
        message.walletAddresses = object.walletAddresses?.map((e) => e) || [];
        return message;
    },
};
function createBasePrivatePreferencesAction_DenyAddress() {
    return { walletAddresses: [] };
}
exports.PrivatePreferencesAction_DenyAddress = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        for (const v of message.walletAddresses) {
            writer.uint32(10).string(v);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBasePrivatePreferencesAction_DenyAddress();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.walletAddresses.push(reader.string());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            walletAddresses: globalThis.Array.isArray(object?.walletAddresses)
                ? object.walletAddresses.map((e) => globalThis.String(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.walletAddresses?.length) {
            obj.walletAddresses = message.walletAddresses;
        }
        return obj;
    },
    create(base) {
        return exports.PrivatePreferencesAction_DenyAddress.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBasePrivatePreferencesAction_DenyAddress();
        message.walletAddresses = object.walletAddresses?.map((e) => e) || [];
        return message;
    },
};
function createBasePrivatePreferencesAction_AllowInboxId() {
    return { inboxIds: [] };
}
exports.PrivatePreferencesAction_AllowInboxId = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        for (const v of message.inboxIds) {
            writer.uint32(10).string(v);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBasePrivatePreferencesAction_AllowInboxId();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.inboxIds.push(reader.string());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            inboxIds: globalThis.Array.isArray(object?.inboxIds) ? object.inboxIds.map((e) => globalThis.String(e)) : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.inboxIds?.length) {
            obj.inboxIds = message.inboxIds;
        }
        return obj;
    },
    create(base) {
        return exports.PrivatePreferencesAction_AllowInboxId.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBasePrivatePreferencesAction_AllowInboxId();
        message.inboxIds = object.inboxIds?.map((e) => e) || [];
        return message;
    },
};
function createBasePrivatePreferencesAction_DenyInboxId() {
    return { inboxIds: [] };
}
exports.PrivatePreferencesAction_DenyInboxId = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        for (const v of message.inboxIds) {
            writer.uint32(10).string(v);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBasePrivatePreferencesAction_DenyInboxId();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.inboxIds.push(reader.string());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            inboxIds: globalThis.Array.isArray(object?.inboxIds) ? object.inboxIds.map((e) => globalThis.String(e)) : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.inboxIds?.length) {
            obj.inboxIds = message.inboxIds;
        }
        return obj;
    },
    create(base) {
        return exports.PrivatePreferencesAction_DenyInboxId.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBasePrivatePreferencesAction_DenyInboxId();
        message.inboxIds = object.inboxIds?.map((e) => e) || [];
        return message;
    },
};
function createBasePrivatePreferencesAction_AllowGroup() {
    return { groupIds: [] };
}
exports.PrivatePreferencesAction_AllowGroup = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        for (const v of message.groupIds) {
            writer.uint32(10).string(v);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBasePrivatePreferencesAction_AllowGroup();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.groupIds.push(reader.string());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            groupIds: globalThis.Array.isArray(object?.groupIds) ? object.groupIds.map((e) => globalThis.String(e)) : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.groupIds?.length) {
            obj.groupIds = message.groupIds;
        }
        return obj;
    },
    create(base) {
        return exports.PrivatePreferencesAction_AllowGroup.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBasePrivatePreferencesAction_AllowGroup();
        message.groupIds = object.groupIds?.map((e) => e) || [];
        return message;
    },
};
function createBasePrivatePreferencesAction_DenyGroup() {
    return { groupIds: [] };
}
exports.PrivatePreferencesAction_DenyGroup = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        for (const v of message.groupIds) {
            writer.uint32(10).string(v);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBasePrivatePreferencesAction_DenyGroup();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.groupIds.push(reader.string());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            groupIds: globalThis.Array.isArray(object?.groupIds) ? object.groupIds.map((e) => globalThis.String(e)) : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.groupIds?.length) {
            obj.groupIds = message.groupIds;
        }
        return obj;
    },
    create(base) {
        return exports.PrivatePreferencesAction_DenyGroup.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBasePrivatePreferencesAction_DenyGroup();
        message.groupIds = object.groupIds?.map((e) => e) || [];
        return message;
    },
};
function createBasePrivatePreferencesPayload() {
    return { v1: undefined };
}
exports.PrivatePreferencesPayload = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.v1 !== undefined) {
            ciphertext_pb_1.Ciphertext.encode(message.v1, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBasePrivatePreferencesPayload();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.v1 = ciphertext_pb_1.Ciphertext.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { v1: isSet(object.v1) ? ciphertext_pb_1.Ciphertext.fromJSON(object.v1) : undefined };
    },
    toJSON(message) {
        const obj = {};
        if (message.v1 !== undefined) {
            obj.v1 = ciphertext_pb_1.Ciphertext.toJSON(message.v1);
        }
        return obj;
    },
    create(base) {
        return exports.PrivatePreferencesPayload.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBasePrivatePreferencesPayload();
        message.v1 = (object.v1 !== undefined && object.v1 !== null) ? ciphertext_pb_1.Ciphertext.fromPartial(object.v1) : undefined;
        return message;
    },
};
if (minimal_1.default.util.Long !== long_1.default) {
    minimal_1.default.util.Long = long_1.default;
    minimal_1.default.configure();
}
function isSet(value) {
    return value !== null && value !== undefined;
}
