// @xmtp/proto/ts/dist/cjs/message_contents/public_key.pb.js
"use strict";
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v1.181.2
//   protoc               v3.21.12
// source: message_contents/public_key.proto
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PublicKeyBundle = exports.PublicKey_Secp256k1Uncompressed = exports.PublicKey = exports.SignedPublicKeyBundle = exports.SignedPublicKey = exports.UnsignedPublicKey_Secp256k1Uncompressed = exports.UnsignedPublicKey = exports.protobufPackage = void 0;
/* eslint-disable */
const long_1 = __importDefault(require("long"));
const minimal_1 = __importDefault(require("protobufjs/minimal"));
const signature_pb_1 = require("./signature.pb");
exports.protobufPackage = "xmtp.message_contents";
function createBaseUnsignedPublicKey() {
    return { createdNs: long_1.default.UZERO, secp256k1Uncompressed: undefined };
}
exports.UnsignedPublicKey = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (!message.createdNs.equals(long_1.default.UZERO)) {
            writer.uint32(8).uint64(message.createdNs);
        }
        if (message.secp256k1Uncompressed !== undefined) {
            exports.UnsignedPublicKey_Secp256k1Uncompressed.encode(message.secp256k1Uncompressed, writer.uint32(26).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseUnsignedPublicKey();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 8) {
                        break;
                    }
                    message.createdNs = reader.uint64();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.secp256k1Uncompressed = exports.UnsignedPublicKey_Secp256k1Uncompressed.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            createdNs: isSet(object.createdNs) ? long_1.default.fromValue(object.createdNs) : long_1.default.UZERO,
            secp256k1Uncompressed: isSet(object.secp256k1Uncompressed)
                ? exports.UnsignedPublicKey_Secp256k1Uncompressed.fromJSON(object.secp256k1Uncompressed)
                : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (!message.createdNs.equals(long_1.default.UZERO)) {
            obj.createdNs = (message.createdNs || long_1.default.UZERO).toString();
        }
        if (message.secp256k1Uncompressed !== undefined) {
            obj.secp256k1Uncompressed = exports.UnsignedPublicKey_Secp256k1Uncompressed.toJSON(message.secp256k1Uncompressed);
        }
        return obj;
    },
    create(base) {
        return exports.UnsignedPublicKey.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseUnsignedPublicKey();
        message.createdNs = (object.createdNs !== undefined && object.createdNs !== null)
            ? long_1.default.fromValue(object.createdNs)
            : long_1.default.UZERO;
        message.secp256k1Uncompressed =
            (object.secp256k1Uncompressed !== undefined && object.secp256k1Uncompressed !== null)
                ? exports.UnsignedPublicKey_Secp256k1Uncompressed.fromPartial(object.secp256k1Uncompressed)
                : undefined;
        return message;
    },
};
function createBaseUnsignedPublicKey_Secp256k1Uncompressed() {
    return { bytes: new Uint8Array(0) };
}
exports.UnsignedPublicKey_Secp256k1Uncompressed = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.bytes.length !== 0) {
            writer.uint32(10).bytes(message.bytes);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseUnsignedPublicKey_Secp256k1Uncompressed();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.bytes = reader.bytes();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { bytes: isSet(object.bytes) ? bytesFromBase64(object.bytes) : new Uint8Array(0) };
    },
    toJSON(message) {
        const obj = {};
        if (message.bytes.length !== 0) {
            obj.bytes = base64FromBytes(message.bytes);
        }
        return obj;
    },
    create(base) {
        return exports.UnsignedPublicKey_Secp256k1Uncompressed.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseUnsignedPublicKey_Secp256k1Uncompressed();
        message.bytes = object.bytes ?? new Uint8Array(0);
        return message;
    },
};
function createBaseSignedPublicKey() {
    return { keyBytes: new Uint8Array(0), signature: undefined };
}
exports.SignedPublicKey = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.keyBytes.length !== 0) {
            writer.uint32(10).bytes(message.keyBytes);
        }
        if (message.signature !== undefined) {
            signature_pb_1.Signature.encode(message.signature, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSignedPublicKey();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.keyBytes = reader.bytes();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.signature = signature_pb_1.Signature.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            keyBytes: isSet(object.keyBytes) ? bytesFromBase64(object.keyBytes) : new Uint8Array(0),
            signature: isSet(object.signature) ? signature_pb_1.Signature.fromJSON(object.signature) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.keyBytes.length !== 0) {
            obj.keyBytes = base64FromBytes(message.keyBytes);
        }
        if (message.signature !== undefined) {
            obj.signature = signature_pb_1.Signature.toJSON(message.signature);
        }
        return obj;
    },
    create(base) {
        return exports.SignedPublicKey.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseSignedPublicKey();
        message.keyBytes = object.keyBytes ?? new Uint8Array(0);
        message.signature = (object.signature !== undefined && object.signature !== null)
            ? signature_pb_1.Signature.fromPartial(object.signature)
            : undefined;
        return message;
    },
};
function createBaseSignedPublicKeyBundle() {
    return { identityKey: undefined, preKey: undefined };
}
exports.SignedPublicKeyBundle = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.identityKey !== undefined) {
            exports.SignedPublicKey.encode(message.identityKey, writer.uint32(10).fork()).ldelim();
        }
        if (message.preKey !== undefined) {
            exports.SignedPublicKey.encode(message.preKey, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSignedPublicKeyBundle();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.identityKey = exports.SignedPublicKey.decode(reader, reader.uint32());
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.preKey = exports.SignedPublicKey.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            identityKey: isSet(object.identityKey) ? exports.SignedPublicKey.fromJSON(object.identityKey) : undefined,
            preKey: isSet(object.preKey) ? exports.SignedPublicKey.fromJSON(object.preKey) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.identityKey !== undefined) {
            obj.identityKey = exports.SignedPublicKey.toJSON(message.identityKey);
        }
        if (message.preKey !== undefined) {
            obj.preKey = exports.SignedPublicKey.toJSON(message.preKey);
        }
        return obj;
    },
    create(base) {
        return exports.SignedPublicKeyBundle.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseSignedPublicKeyBundle();
        message.identityKey = (object.identityKey !== undefined && object.identityKey !== null)
            ? exports.SignedPublicKey.fromPartial(object.identityKey)
            : undefined;
        message.preKey = (object.preKey !== undefined && object.preKey !== null)
            ? exports.SignedPublicKey.fromPartial(object.preKey)
            : undefined;
        return message;
    },
};
function createBasePublicKey() {
    return { timestamp: long_1.default.UZERO, signature: undefined, secp256k1Uncompressed: undefined };
}
exports.PublicKey = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (!message.timestamp.equals(long_1.default.UZERO)) {
            writer.uint32(8).uint64(message.timestamp);
        }
        if (message.signature !== undefined) {
            signature_pb_1.Signature.encode(message.signature, writer.uint32(18).fork()).ldelim();
        }
        if (message.secp256k1Uncompressed !== undefined) {
            exports.PublicKey_Secp256k1Uncompressed.encode(message.secp256k1Uncompressed, writer.uint32(26).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBasePublicKey();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 8) {
                        break;
                    }
                    message.timestamp = reader.uint64();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.signature = signature_pb_1.Signature.decode(reader, reader.uint32());
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.secp256k1Uncompressed = exports.PublicKey_Secp256k1Uncompressed.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            timestamp: isSet(object.timestamp) ? long_1.default.fromValue(object.timestamp) : long_1.default.UZERO,
            signature: isSet(object.signature) ? signature_pb_1.Signature.fromJSON(object.signature) : undefined,
            secp256k1Uncompressed: isSet(object.secp256k1Uncompressed)
                ? exports.PublicKey_Secp256k1Uncompressed.fromJSON(object.secp256k1Uncompressed)
                : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (!message.timestamp.equals(long_1.default.UZERO)) {
            obj.timestamp = (message.timestamp || long_1.default.UZERO).toString();
        }
        if (message.signature !== undefined) {
            obj.signature = signature_pb_1.Signature.toJSON(message.signature);
        }
        if (message.secp256k1Uncompressed !== undefined) {
            obj.secp256k1Uncompressed = exports.PublicKey_Secp256k1Uncompressed.toJSON(message.secp256k1Uncompressed);
        }
        return obj;
    },
    create(base) {
        return exports.PublicKey.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBasePublicKey();
        message.timestamp = (object.timestamp !== undefined && object.timestamp !== null)
            ? long_1.default.fromValue(object.timestamp)
            : long_1.default.UZERO;
        message.signature = (object.signature !== undefined && object.signature !== null)
            ? signature_pb_1.Signature.fromPartial(object.signature)
            : undefined;
        message.secp256k1Uncompressed =
            (object.secp256k1Uncompressed !== undefined && object.secp256k1Uncompressed !== null)
                ? exports.PublicKey_Secp256k1Uncompressed.fromPartial(object.secp256k1Uncompressed)
                : undefined;
        return message;
    },
};
function createBasePublicKey_Secp256k1Uncompressed() {
    return { bytes: new Uint8Array(0) };
}
exports.PublicKey_Secp256k1Uncompressed = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.bytes.length !== 0) {
            writer.uint32(10).bytes(message.bytes);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBasePublicKey_Secp256k1Uncompressed();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.bytes = reader.bytes();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { bytes: isSet(object.bytes) ? bytesFromBase64(object.bytes) : new Uint8Array(0) };
    },
    toJSON(message) {
        const obj = {};
        if (message.bytes.length !== 0) {
            obj.bytes = base64FromBytes(message.bytes);
        }
        return obj;
    },
    create(base) {
        return exports.PublicKey_Secp256k1Uncompressed.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBasePublicKey_Secp256k1Uncompressed();
        message.bytes = object.bytes ?? new Uint8Array(0);
        return message;
    },
};
function createBasePublicKeyBundle() {
    return { identityKey: undefined, preKey: undefined };
}
exports.PublicKeyBundle = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.identityKey !== undefined) {
            exports.PublicKey.encode(message.identityKey, writer.uint32(10).fork()).ldelim();
        }
        if (message.preKey !== undefined) {
            exports.PublicKey.encode(message.preKey, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBasePublicKeyBundle();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.identityKey = exports.PublicKey.decode(reader, reader.uint32());
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.preKey = exports.PublicKey.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            identityKey: isSet(object.identityKey) ? exports.PublicKey.fromJSON(object.identityKey) : undefined,
            preKey: isSet(object.preKey) ? exports.PublicKey.fromJSON(object.preKey) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.identityKey !== undefined) {
            obj.identityKey = exports.PublicKey.toJSON(message.identityKey);
        }
        if (message.preKey !== undefined) {
            obj.preKey = exports.PublicKey.toJSON(message.preKey);
        }
        return obj;
    },
    create(base) {
        return exports.PublicKeyBundle.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBasePublicKeyBundle();
        message.identityKey = (object.identityKey !== undefined && object.identityKey !== null)
            ? exports.PublicKey.fromPartial(object.identityKey)
            : undefined;
        message.preKey = (object.preKey !== undefined && object.preKey !== null)
            ? exports.PublicKey.fromPartial(object.preKey)
            : undefined;
        return message;
    },
};
function bytesFromBase64(b64) {
    if (globalThis.Buffer) {
        return Uint8Array.from(globalThis.Buffer.from(b64, "base64"));
    }
    else {
        const bin = globalThis.atob(b64);
        const arr = new Uint8Array(bin.length);
        for (let i = 0; i < bin.length; ++i) {
            arr[i] = bin.charCodeAt(i);
        }
        return arr;
    }
}
function base64FromBytes(arr) {
    if (globalThis.Buffer) {
        return globalThis.Buffer.from(arr).toString("base64");
    }
    else {
        const bin = [];
        arr.forEach((byte) => {
            bin.push(globalThis.String.fromCharCode(byte));
        });
        return globalThis.btoa(bin.join(""));
    }
}
if (minimal_1.default.util.Long !== long_1.default) {
    minimal_1.default.util.Long = long_1.default;
    minimal_1.default.configure();
}
function isSet(value) {
    return value !== null && value !== undefined;
}
