// @xmtp/proto/ts/dist/cjs/message_contents/signature.pb.js
"use strict";
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v1.181.2
//   protoc               v3.21.12
// source: message_contents/signature.proto
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Signature_WalletECDSACompact = exports.Signature_ECDSACompact = exports.Signature = exports.protobufPackage = void 0;
/* eslint-disable */
const long_1 = __importDefault(require("long"));
const minimal_1 = __importDefault(require("protobufjs/minimal"));
exports.protobufPackage = "xmtp.message_contents";
function createBaseSignature() {
    return { ecdsaCompact: undefined, walletEcdsaCompact: undefined };
}
exports.Signature = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.ecdsaCompact !== undefined) {
            exports.Signature_ECDSACompact.encode(message.ecdsaCompact, writer.uint32(10).fork()).ldelim();
        }
        if (message.walletEcdsaCompact !== undefined) {
            exports.Signature_WalletECDSACompact.encode(message.walletEcdsaCompact, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSignature();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.ecdsaCompact = exports.Signature_ECDSACompact.decode(reader, reader.uint32());
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.walletEcdsaCompact = exports.Signature_WalletECDSACompact.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            ecdsaCompact: isSet(object.ecdsaCompact) ? exports.Signature_ECDSACompact.fromJSON(object.ecdsaCompact) : undefined,
            walletEcdsaCompact: isSet(object.walletEcdsaCompact)
                ? exports.Signature_WalletECDSACompact.fromJSON(object.walletEcdsaCompact)
                : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.ecdsaCompact !== undefined) {
            obj.ecdsaCompact = exports.Signature_ECDSACompact.toJSON(message.ecdsaCompact);
        }
        if (message.walletEcdsaCompact !== undefined) {
            obj.walletEcdsaCompact = exports.Signature_WalletECDSACompact.toJSON(message.walletEcdsaCompact);
        }
        return obj;
    },
    create(base) {
        return exports.Signature.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseSignature();
        message.ecdsaCompact = (object.ecdsaCompact !== undefined && object.ecdsaCompact !== null)
            ? exports.Signature_ECDSACompact.fromPartial(object.ecdsaCompact)
            : undefined;
        message.walletEcdsaCompact = (object.walletEcdsaCompact !== undefined && object.walletEcdsaCompact !== null)
            ? exports.Signature_WalletECDSACompact.fromPartial(object.walletEcdsaCompact)
            : undefined;
        return message;
    },
};
function createBaseSignature_ECDSACompact() {
    return { bytes: new Uint8Array(0), recovery: 0 };
}
exports.Signature_ECDSACompact = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.bytes.length !== 0) {
            writer.uint32(10).bytes(message.bytes);
        }
        if (message.recovery !== 0) {
            writer.uint32(16).uint32(message.recovery);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSignature_ECDSACompact();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.bytes = reader.bytes();
                    continue;
                case 2:
                    if (tag !== 16) {
                        break;
                    }
                    message.recovery = reader.uint32();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            bytes: isSet(object.bytes) ? bytesFromBase64(object.bytes) : new Uint8Array(0),
            recovery: isSet(object.recovery) ? globalThis.Number(object.recovery) : 0,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.bytes.length !== 0) {
            obj.bytes = base64FromBytes(message.bytes);
        }
        if (message.recovery !== 0) {
            obj.recovery = Math.round(message.recovery);
        }
        return obj;
    },
    create(base) {
        return exports.Signature_ECDSACompact.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseSignature_ECDSACompact();
        message.bytes = object.bytes ?? new Uint8Array(0);
        message.recovery = object.recovery ?? 0;
        return message;
    },
};
function createBaseSignature_WalletECDSACompact() {
    return { bytes: new Uint8Array(0), recovery: 0 };
}
exports.Signature_WalletECDSACompact = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.bytes.length !== 0) {
            writer.uint32(10).bytes(message.bytes);
        }
        if (message.recovery !== 0) {
            writer.uint32(16).uint32(message.recovery);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSignature_WalletECDSACompact();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.bytes = reader.bytes();
                    continue;
                case 2:
                    if (tag !== 16) {
                        break;
                    }
                    message.recovery = reader.uint32();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            bytes: isSet(object.bytes) ? bytesFromBase64(object.bytes) : new Uint8Array(0),
            recovery: isSet(object.recovery) ? globalThis.Number(object.recovery) : 0,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.bytes.length !== 0) {
            obj.bytes = base64FromBytes(message.bytes);
        }
        if (message.recovery !== 0) {
            obj.recovery = Math.round(message.recovery);
        }
        return obj;
    },
    create(base) {
        return exports.Signature_WalletECDSACompact.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseSignature_WalletECDSACompact();
        message.bytes = object.bytes ?? new Uint8Array(0);
        message.recovery = object.recovery ?? 0;
        return message;
    },
};
function bytesFromBase64(b64) {
    if (globalThis.Buffer) {
        return Uint8Array.from(globalThis.Buffer.from(b64, "base64"));
    }
    else {
        const bin = globalThis.atob(b64);
        const arr = new Uint8Array(bin.length);
        for (let i = 0; i < bin.length; ++i) {
            arr[i] = bin.charCodeAt(i);
        }
        return arr;
    }
}
function base64FromBytes(arr) {
    if (globalThis.Buffer) {
        return globalThis.Buffer.from(arr).toString("base64");
    }
    else {
        const bin = [];
        arr.forEach((byte) => {
            bin.push(globalThis.String.fromCharCode(byte));
        });
        return globalThis.btoa(bin.join(""));
    }
}
if (minimal_1.default.util.Long !== long_1.default) {
    minimal_1.default.util.Long = long_1.default;
    minimal_1.default.configure();
}
function isSet(value) {
    return value !== null && value !== undefined;
}
