// @xmtp/proto/ts/dist/cjs/identity/api/v1/identity.pb.js
"use strict";
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v1.181.2
//   protoc               v3.21.12
// source: identity/api/v1/identity.proto
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.IdentityApiClientImpl = exports.IdentityApiServiceName = exports.GetInboxIdsResponse_Response = exports.GetInboxIdsResponse = exports.GetInboxIdsRequest_Request = exports.GetInboxIdsRequest = exports.GetIdentityUpdatesResponse_Response = exports.GetIdentityUpdatesResponse_IdentityUpdateLog = exports.GetIdentityUpdatesResponse = exports.GetIdentityUpdatesRequest_Request = exports.GetIdentityUpdatesRequest = exports.PublishIdentityUpdateResponse = exports.PublishIdentityUpdateRequest = exports.VerifySmartContractWalletSignaturesResponse_ValidationResponse = exports.VerifySmartContractWalletSignaturesResponse = exports.VerifySmartContractWalletSignatureRequestSignature = exports.VerifySmartContractWalletSignaturesRequest = exports.protobufPackage = void 0;
/* eslint-disable */
const long_1 = __importDefault(require("long"));
const minimal_1 = __importDefault(require("protobufjs/minimal"));
const association_pb_1 = require("../../associations/association.pb");
exports.protobufPackage = "xmtp.identity.api.v1";
function createBaseVerifySmartContractWalletSignaturesRequest() {
    return { signatures: [] };
}
exports.VerifySmartContractWalletSignaturesRequest = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        for (const v of message.signatures) {
            exports.VerifySmartContractWalletSignatureRequestSignature.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseVerifySmartContractWalletSignaturesRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.signatures.push(exports.VerifySmartContractWalletSignatureRequestSignature.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            signatures: globalThis.Array.isArray(object?.signatures)
                ? object.signatures.map((e) => exports.VerifySmartContractWalletSignatureRequestSignature.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.signatures?.length) {
            obj.signatures = message.signatures.map((e) => exports.VerifySmartContractWalletSignatureRequestSignature.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return exports.VerifySmartContractWalletSignaturesRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseVerifySmartContractWalletSignaturesRequest();
        message.signatures =
            object.signatures?.map((e) => exports.VerifySmartContractWalletSignatureRequestSignature.fromPartial(e)) || [];
        return message;
    },
};
function createBaseVerifySmartContractWalletSignatureRequestSignature() {
    return { accountId: "", blockNumber: undefined, signature: new Uint8Array(0), hash: new Uint8Array(0) };
}
exports.VerifySmartContractWalletSignatureRequestSignature = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.accountId !== "") {
            writer.uint32(10).string(message.accountId);
        }
        if (message.blockNumber !== undefined) {
            writer.uint32(16).uint64(message.blockNumber);
        }
        if (message.signature.length !== 0) {
            writer.uint32(26).bytes(message.signature);
        }
        if (message.hash.length !== 0) {
            writer.uint32(34).bytes(message.hash);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseVerifySmartContractWalletSignatureRequestSignature();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.accountId = reader.string();
                    continue;
                case 2:
                    if (tag !== 16) {
                        break;
                    }
                    message.blockNumber = reader.uint64();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.signature = reader.bytes();
                    continue;
                case 4:
                    if (tag !== 34) {
                        break;
                    }
                    message.hash = reader.bytes();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            accountId: isSet(object.accountId) ? globalThis.String(object.accountId) : "",
            blockNumber: isSet(object.blockNumber) ? long_1.default.fromValue(object.blockNumber) : undefined,
            signature: isSet(object.signature) ? bytesFromBase64(object.signature) : new Uint8Array(0),
            hash: isSet(object.hash) ? bytesFromBase64(object.hash) : new Uint8Array(0),
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.accountId !== "") {
            obj.accountId = message.accountId;
        }
        if (message.blockNumber !== undefined) {
            obj.blockNumber = (message.blockNumber || long_1.default.UZERO).toString();
        }
        if (message.signature.length !== 0) {
            obj.signature = base64FromBytes(message.signature);
        }
        if (message.hash.length !== 0) {
            obj.hash = base64FromBytes(message.hash);
        }
        return obj;
    },
    create(base) {
        return exports.VerifySmartContractWalletSignatureRequestSignature.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseVerifySmartContractWalletSignatureRequestSignature();
        message.accountId = object.accountId ?? "";
        message.blockNumber = (object.blockNumber !== undefined && object.blockNumber !== null)
            ? long_1.default.fromValue(object.blockNumber)
            : undefined;
        message.signature = object.signature ?? new Uint8Array(0);
        message.hash = object.hash ?? new Uint8Array(0);
        return message;
    },
};
function createBaseVerifySmartContractWalletSignaturesResponse() {
    return { responses: [] };
}
exports.VerifySmartContractWalletSignaturesResponse = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        for (const v of message.responses) {
            exports.VerifySmartContractWalletSignaturesResponse_ValidationResponse.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseVerifySmartContractWalletSignaturesResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.responses.push(exports.VerifySmartContractWalletSignaturesResponse_ValidationResponse.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            responses: globalThis.Array.isArray(object?.responses)
                ? object.responses.map((e) => exports.VerifySmartContractWalletSignaturesResponse_ValidationResponse.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.responses?.length) {
            obj.responses = message.responses.map((e) => exports.VerifySmartContractWalletSignaturesResponse_ValidationResponse.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return exports.VerifySmartContractWalletSignaturesResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseVerifySmartContractWalletSignaturesResponse();
        message.responses =
            object.responses?.map((e) => exports.VerifySmartContractWalletSignaturesResponse_ValidationResponse.fromPartial(e)) || [];
        return message;
    },
};
function createBaseVerifySmartContractWalletSignaturesResponse_ValidationResponse() {
    return { isValid: false, blockNumber: undefined, error: undefined };
}
exports.VerifySmartContractWalletSignaturesResponse_ValidationResponse = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.isValid !== false) {
            writer.uint32(8).bool(message.isValid);
        }
        if (message.blockNumber !== undefined) {
            writer.uint32(16).uint64(message.blockNumber);
        }
        if (message.error !== undefined) {
            writer.uint32(26).string(message.error);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseVerifySmartContractWalletSignaturesResponse_ValidationResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 8) {
                        break;
                    }
                    message.isValid = reader.bool();
                    continue;
                case 2:
                    if (tag !== 16) {
                        break;
                    }
                    message.blockNumber = reader.uint64();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.error = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            isValid: isSet(object.isValid) ? globalThis.Boolean(object.isValid) : false,
            blockNumber: isSet(object.blockNumber) ? long_1.default.fromValue(object.blockNumber) : undefined,
            error: isSet(object.error) ? globalThis.String(object.error) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.isValid !== false) {
            obj.isValid = message.isValid;
        }
        if (message.blockNumber !== undefined) {
            obj.blockNumber = (message.blockNumber || long_1.default.UZERO).toString();
        }
        if (message.error !== undefined) {
            obj.error = message.error;
        }
        return obj;
    },
    create(base) {
        return exports.VerifySmartContractWalletSignaturesResponse_ValidationResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseVerifySmartContractWalletSignaturesResponse_ValidationResponse();
        message.isValid = object.isValid ?? false;
        message.blockNumber = (object.blockNumber !== undefined && object.blockNumber !== null)
            ? long_1.default.fromValue(object.blockNumber)
            : undefined;
        message.error = object.error ?? undefined;
        return message;
    },
};
function createBasePublishIdentityUpdateRequest() {
    return { identityUpdate: undefined };
}
exports.PublishIdentityUpdateRequest = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.identityUpdate !== undefined) {
            association_pb_1.IdentityUpdate.encode(message.identityUpdate, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBasePublishIdentityUpdateRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.identityUpdate = association_pb_1.IdentityUpdate.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            identityUpdate: isSet(object.identityUpdate) ? association_pb_1.IdentityUpdate.fromJSON(object.identityUpdate) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.identityUpdate !== undefined) {
            obj.identityUpdate = association_pb_1.IdentityUpdate.toJSON(message.identityUpdate);
        }
        return obj;
    },
    create(base) {
        return exports.PublishIdentityUpdateRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBasePublishIdentityUpdateRequest();
        message.identityUpdate = (object.identityUpdate !== undefined && object.identityUpdate !== null)
            ? association_pb_1.IdentityUpdate.fromPartial(object.identityUpdate)
            : undefined;
        return message;
    },
};
function createBasePublishIdentityUpdateResponse() {
    return {};
}
exports.PublishIdentityUpdateResponse = {
    encode(_, writer = minimal_1.default.Writer.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBasePublishIdentityUpdateResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return exports.PublishIdentityUpdateResponse.fromPartial(base ?? {});
    },
    fromPartial(_) {
        const message = createBasePublishIdentityUpdateResponse();
        return message;
    },
};
function createBaseGetIdentityUpdatesRequest() {
    return { requests: [] };
}
exports.GetIdentityUpdatesRequest = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        for (const v of message.requests) {
            exports.GetIdentityUpdatesRequest_Request.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetIdentityUpdatesRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.requests.push(exports.GetIdentityUpdatesRequest_Request.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            requests: globalThis.Array.isArray(object?.requests)
                ? object.requests.map((e) => exports.GetIdentityUpdatesRequest_Request.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.requests?.length) {
            obj.requests = message.requests.map((e) => exports.GetIdentityUpdatesRequest_Request.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return exports.GetIdentityUpdatesRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGetIdentityUpdatesRequest();
        message.requests = object.requests?.map((e) => exports.GetIdentityUpdatesRequest_Request.fromPartial(e)) || [];
        return message;
    },
};
function createBaseGetIdentityUpdatesRequest_Request() {
    return { inboxId: "", sequenceId: long_1.default.UZERO };
}
exports.GetIdentityUpdatesRequest_Request = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.inboxId !== "") {
            writer.uint32(10).string(message.inboxId);
        }
        if (!message.sequenceId.equals(long_1.default.UZERO)) {
            writer.uint32(16).uint64(message.sequenceId);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetIdentityUpdatesRequest_Request();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.inboxId = reader.string();
                    continue;
                case 2:
                    if (tag !== 16) {
                        break;
                    }
                    message.sequenceId = reader.uint64();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            inboxId: isSet(object.inboxId) ? globalThis.String(object.inboxId) : "",
            sequenceId: isSet(object.sequenceId) ? long_1.default.fromValue(object.sequenceId) : long_1.default.UZERO,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.inboxId !== "") {
            obj.inboxId = message.inboxId;
        }
        if (!message.sequenceId.equals(long_1.default.UZERO)) {
            obj.sequenceId = (message.sequenceId || long_1.default.UZERO).toString();
        }
        return obj;
    },
    create(base) {
        return exports.GetIdentityUpdatesRequest_Request.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGetIdentityUpdatesRequest_Request();
        message.inboxId = object.inboxId ?? "";
        message.sequenceId = (object.sequenceId !== undefined && object.sequenceId !== null)
            ? long_1.default.fromValue(object.sequenceId)
            : long_1.default.UZERO;
        return message;
    },
};
function createBaseGetIdentityUpdatesResponse() {
    return { responses: [] };
}
exports.GetIdentityUpdatesResponse = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        for (const v of message.responses) {
            exports.GetIdentityUpdatesResponse_Response.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetIdentityUpdatesResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.responses.push(exports.GetIdentityUpdatesResponse_Response.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            responses: globalThis.Array.isArray(object?.responses)
                ? object.responses.map((e) => exports.GetIdentityUpdatesResponse_Response.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.responses?.length) {
            obj.responses = message.responses.map((e) => exports.GetIdentityUpdatesResponse_Response.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return exports.GetIdentityUpdatesResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGetIdentityUpdatesResponse();
        message.responses = object.responses?.map((e) => exports.GetIdentityUpdatesResponse_Response.fromPartial(e)) || [];
        return message;
    },
};
function createBaseGetIdentityUpdatesResponse_IdentityUpdateLog() {
    return { sequenceId: long_1.default.UZERO, serverTimestampNs: long_1.default.UZERO, update: undefined };
}
exports.GetIdentityUpdatesResponse_IdentityUpdateLog = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (!message.sequenceId.equals(long_1.default.UZERO)) {
            writer.uint32(8).uint64(message.sequenceId);
        }
        if (!message.serverTimestampNs.equals(long_1.default.UZERO)) {
            writer.uint32(16).uint64(message.serverTimestampNs);
        }
        if (message.update !== undefined) {
            association_pb_1.IdentityUpdate.encode(message.update, writer.uint32(26).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetIdentityUpdatesResponse_IdentityUpdateLog();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 8) {
                        break;
                    }
                    message.sequenceId = reader.uint64();
                    continue;
                case 2:
                    if (tag !== 16) {
                        break;
                    }
                    message.serverTimestampNs = reader.uint64();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.update = association_pb_1.IdentityUpdate.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            sequenceId: isSet(object.sequenceId) ? long_1.default.fromValue(object.sequenceId) : long_1.default.UZERO,
            serverTimestampNs: isSet(object.serverTimestampNs) ? long_1.default.fromValue(object.serverTimestampNs) : long_1.default.UZERO,
            update: isSet(object.update) ? association_pb_1.IdentityUpdate.fromJSON(object.update) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (!message.sequenceId.equals(long_1.default.UZERO)) {
            obj.sequenceId = (message.sequenceId || long_1.default.UZERO).toString();
        }
        if (!message.serverTimestampNs.equals(long_1.default.UZERO)) {
            obj.serverTimestampNs = (message.serverTimestampNs || long_1.default.UZERO).toString();
        }
        if (message.update !== undefined) {
            obj.update = association_pb_1.IdentityUpdate.toJSON(message.update);
        }
        return obj;
    },
    create(base) {
        return exports.GetIdentityUpdatesResponse_IdentityUpdateLog.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGetIdentityUpdatesResponse_IdentityUpdateLog();
        message.sequenceId = (object.sequenceId !== undefined && object.sequenceId !== null)
            ? long_1.default.fromValue(object.sequenceId)
            : long_1.default.UZERO;
        message.serverTimestampNs = (object.serverTimestampNs !== undefined && object.serverTimestampNs !== null)
            ? long_1.default.fromValue(object.serverTimestampNs)
            : long_1.default.UZERO;
        message.update = (object.update !== undefined && object.update !== null)
            ? association_pb_1.IdentityUpdate.fromPartial(object.update)
            : undefined;
        return message;
    },
};
function createBaseGetIdentityUpdatesResponse_Response() {
    return { inboxId: "", updates: [] };
}
exports.GetIdentityUpdatesResponse_Response = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.inboxId !== "") {
            writer.uint32(10).string(message.inboxId);
        }
        for (const v of message.updates) {
            exports.GetIdentityUpdatesResponse_IdentityUpdateLog.encode(v, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetIdentityUpdatesResponse_Response();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.inboxId = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.updates.push(exports.GetIdentityUpdatesResponse_IdentityUpdateLog.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            inboxId: isSet(object.inboxId) ? globalThis.String(object.inboxId) : "",
            updates: globalThis.Array.isArray(object?.updates)
                ? object.updates.map((e) => exports.GetIdentityUpdatesResponse_IdentityUpdateLog.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.inboxId !== "") {
            obj.inboxId = message.inboxId;
        }
        if (message.updates?.length) {
            obj.updates = message.updates.map((e) => exports.GetIdentityUpdatesResponse_IdentityUpdateLog.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return exports.GetIdentityUpdatesResponse_Response.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGetIdentityUpdatesResponse_Response();
        message.inboxId = object.inboxId ?? "";
        message.updates = object.updates?.map((e) => exports.GetIdentityUpdatesResponse_IdentityUpdateLog.fromPartial(e)) || [];
        return message;
    },
};
function createBaseGetInboxIdsRequest() {
    return { requests: [] };
}
exports.GetInboxIdsRequest = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        for (const v of message.requests) {
            exports.GetInboxIdsRequest_Request.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetInboxIdsRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.requests.push(exports.GetInboxIdsRequest_Request.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            requests: globalThis.Array.isArray(object?.requests)
                ? object.requests.map((e) => exports.GetInboxIdsRequest_Request.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.requests?.length) {
            obj.requests = message.requests.map((e) => exports.GetInboxIdsRequest_Request.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return exports.GetInboxIdsRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGetInboxIdsRequest();
        message.requests = object.requests?.map((e) => exports.GetInboxIdsRequest_Request.fromPartial(e)) || [];
        return message;
    },
};
function createBaseGetInboxIdsRequest_Request() {
    return { identifier: "", identifierKind: 0 };
}
exports.GetInboxIdsRequest_Request = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.identifier !== "") {
            writer.uint32(10).string(message.identifier);
        }
        if (message.identifierKind !== 0) {
            writer.uint32(16).int32(message.identifierKind);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetInboxIdsRequest_Request();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.identifier = reader.string();
                    continue;
                case 2:
                    if (tag !== 16) {
                        break;
                    }
                    message.identifierKind = reader.int32();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            identifier: isSet(object.identifier) ? globalThis.String(object.identifier) : "",
            identifierKind: isSet(object.identifierKind) ? (0, association_pb_1.identifierKindFromJSON)(object.identifierKind) : 0,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.identifier !== "") {
            obj.identifier = message.identifier;
        }
        if (message.identifierKind !== 0) {
            obj.identifierKind = (0, association_pb_1.identifierKindToJSON)(message.identifierKind);
        }
        return obj;
    },
    create(base) {
        return exports.GetInboxIdsRequest_Request.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGetInboxIdsRequest_Request();
        message.identifier = object.identifier ?? "";
        message.identifierKind = object.identifierKind ?? 0;
        return message;
    },
};
function createBaseGetInboxIdsResponse() {
    return { responses: [] };
}
exports.GetInboxIdsResponse = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        for (const v of message.responses) {
            exports.GetInboxIdsResponse_Response.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetInboxIdsResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.responses.push(exports.GetInboxIdsResponse_Response.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            responses: globalThis.Array.isArray(object?.responses)
                ? object.responses.map((e) => exports.GetInboxIdsResponse_Response.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.responses?.length) {
            obj.responses = message.responses.map((e) => exports.GetInboxIdsResponse_Response.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return exports.GetInboxIdsResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGetInboxIdsResponse();
        message.responses = object.responses?.map((e) => exports.GetInboxIdsResponse_Response.fromPartial(e)) || [];
        return message;
    },
};
function createBaseGetInboxIdsResponse_Response() {
    return { identifier: "", inboxId: undefined, identifierKind: 0 };
}
exports.GetInboxIdsResponse_Response = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.identifier !== "") {
            writer.uint32(10).string(message.identifier);
        }
        if (message.inboxId !== undefined) {
            writer.uint32(18).string(message.inboxId);
        }
        if (message.identifierKind !== 0) {
            writer.uint32(24).int32(message.identifierKind);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetInboxIdsResponse_Response();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.identifier = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.inboxId = reader.string();
                    continue;
                case 3:
                    if (tag !== 24) {
                        break;
                    }
                    message.identifierKind = reader.int32();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            identifier: isSet(object.identifier) ? globalThis.String(object.identifier) : "",
            inboxId: isSet(object.inboxId) ? globalThis.String(object.inboxId) : undefined,
            identifierKind: isSet(object.identifierKind) ? (0, association_pb_1.identifierKindFromJSON)(object.identifierKind) : 0,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.identifier !== "") {
            obj.identifier = message.identifier;
        }
        if (message.inboxId !== undefined) {
            obj.inboxId = message.inboxId;
        }
        if (message.identifierKind !== 0) {
            obj.identifierKind = (0, association_pb_1.identifierKindToJSON)(message.identifierKind);
        }
        return obj;
    },
    create(base) {
        return exports.GetInboxIdsResponse_Response.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGetInboxIdsResponse_Response();
        message.identifier = object.identifier ?? "";
        message.inboxId = object.inboxId ?? undefined;
        message.identifierKind = object.identifierKind ?? 0;
        return message;
    },
};
exports.IdentityApiServiceName = "xmtp.identity.api.v1.IdentityApi";
class IdentityApiClientImpl {
    rpc;
    service;
    constructor(rpc, opts) {
        this.service = opts?.service || exports.IdentityApiServiceName;
        this.rpc = rpc;
        this.PublishIdentityUpdate = this.PublishIdentityUpdate.bind(this);
        this.GetIdentityUpdates = this.GetIdentityUpdates.bind(this);
        this.GetInboxIds = this.GetInboxIds.bind(this);
        this.VerifySmartContractWalletSignatures = this.VerifySmartContractWalletSignatures.bind(this);
    }
    PublishIdentityUpdate(request) {
        const data = exports.PublishIdentityUpdateRequest.encode(request).finish();
        const promise = this.rpc.request(this.service, "PublishIdentityUpdate", data);
        return promise.then((data) => exports.PublishIdentityUpdateResponse.decode(minimal_1.default.Reader.create(data)));
    }
    GetIdentityUpdates(request) {
        const data = exports.GetIdentityUpdatesRequest.encode(request).finish();
        const promise = this.rpc.request(this.service, "GetIdentityUpdates", data);
        return promise.then((data) => exports.GetIdentityUpdatesResponse.decode(minimal_1.default.Reader.create(data)));
    }
    GetInboxIds(request) {
        const data = exports.GetInboxIdsRequest.encode(request).finish();
        const promise = this.rpc.request(this.service, "GetInboxIds", data);
        return promise.then((data) => exports.GetInboxIdsResponse.decode(minimal_1.default.Reader.create(data)));
    }
    VerifySmartContractWalletSignatures(request) {
        const data = exports.VerifySmartContractWalletSignaturesRequest.encode(request).finish();
        const promise = this.rpc.request(this.service, "VerifySmartContractWalletSignatures", data);
        return promise.then((data) => exports.VerifySmartContractWalletSignaturesResponse.decode(minimal_1.default.Reader.create(data)));
    }
}
exports.IdentityApiClientImpl = IdentityApiClientImpl;
function bytesFromBase64(b64) {
    if (globalThis.Buffer) {
        return Uint8Array.from(globalThis.Buffer.from(b64, "base64"));
    }
    else {
        const bin = globalThis.atob(b64);
        const arr = new Uint8Array(bin.length);
        for (let i = 0; i < bin.length; ++i) {
            arr[i] = bin.charCodeAt(i);
        }
        return arr;
    }
}
function base64FromBytes(arr) {
    if (globalThis.Buffer) {
        return globalThis.Buffer.from(arr).toString("base64");
    }
    else {
        const bin = [];
        arr.forEach((byte) => {
            bin.push(globalThis.String.fromCharCode(byte));
        });
        return globalThis.btoa(bin.join(""));
    }
}
if (minimal_1.default.util.Long !== long_1.default) {
    minimal_1.default.util.Long = long_1.default;
    minimal_1.default.configure();
}
function isSet(value) {
    return value !== null && value !== undefined;
}
