// @xmtp/proto/ts/dist/esm/identity/associations/signature.pb.js
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v1.181.2
//   protoc               v3.21.12
// source: identity/associations/signature.proto
/* eslint-disable */
import Long from "long";
import _m0 from "protobufjs/minimal";
import { SignedPublicKey } from "../../message_contents/public_key.pb";
export const protobufPackage = "xmtp.identity.associations";
function createBaseRecoverableEcdsaSignature() {
    return { bytes: new Uint8Array(0) };
}
export const RecoverableEcdsaSignature = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.bytes.length !== 0) {
            writer.uint32(10).bytes(message.bytes);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseRecoverableEcdsaSignature();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.bytes = reader.bytes();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { bytes: isSet(object.bytes) ? bytesFromBase64(object.bytes) : new Uint8Array(0) };
    },
    toJSON(message) {
        const obj = {};
        if (message.bytes.length !== 0) {
            obj.bytes = base64FromBytes(message.bytes);
        }
        return obj;
    },
    create(base) {
        return RecoverableEcdsaSignature.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseRecoverableEcdsaSignature();
        message.bytes = object.bytes ?? new Uint8Array(0);
        return message;
    },
};
function createBaseRecoverableEd25519Signature() {
    return { bytes: new Uint8Array(0), publicKey: new Uint8Array(0) };
}
export const RecoverableEd25519Signature = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.bytes.length !== 0) {
            writer.uint32(10).bytes(message.bytes);
        }
        if (message.publicKey.length !== 0) {
            writer.uint32(18).bytes(message.publicKey);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseRecoverableEd25519Signature();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.bytes = reader.bytes();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.publicKey = reader.bytes();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            bytes: isSet(object.bytes) ? bytesFromBase64(object.bytes) : new Uint8Array(0),
            publicKey: isSet(object.publicKey) ? bytesFromBase64(object.publicKey) : new Uint8Array(0),
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.bytes.length !== 0) {
            obj.bytes = base64FromBytes(message.bytes);
        }
        if (message.publicKey.length !== 0) {
            obj.publicKey = base64FromBytes(message.publicKey);
        }
        return obj;
    },
    create(base) {
        return RecoverableEd25519Signature.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseRecoverableEd25519Signature();
        message.bytes = object.bytes ?? new Uint8Array(0);
        message.publicKey = object.publicKey ?? new Uint8Array(0);
        return message;
    },
};
function createBaseSmartContractWalletSignature() {
    return { accountId: "", blockNumber: Long.UZERO, signature: new Uint8Array(0) };
}
export const SmartContractWalletSignature = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.accountId !== "") {
            writer.uint32(10).string(message.accountId);
        }
        if (!message.blockNumber.equals(Long.UZERO)) {
            writer.uint32(16).uint64(message.blockNumber);
        }
        if (message.signature.length !== 0) {
            writer.uint32(26).bytes(message.signature);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSmartContractWalletSignature();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.accountId = reader.string();
                    continue;
                case 2:
                    if (tag !== 16) {
                        break;
                    }
                    message.blockNumber = reader.uint64();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.signature = reader.bytes();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            accountId: isSet(object.accountId) ? globalThis.String(object.accountId) : "",
            blockNumber: isSet(object.blockNumber) ? Long.fromValue(object.blockNumber) : Long.UZERO,
            signature: isSet(object.signature) ? bytesFromBase64(object.signature) : new Uint8Array(0),
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.accountId !== "") {
            obj.accountId = message.accountId;
        }
        if (!message.blockNumber.equals(Long.UZERO)) {
            obj.blockNumber = (message.blockNumber || Long.UZERO).toString();
        }
        if (message.signature.length !== 0) {
            obj.signature = base64FromBytes(message.signature);
        }
        return obj;
    },
    create(base) {
        return SmartContractWalletSignature.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseSmartContractWalletSignature();
        message.accountId = object.accountId ?? "";
        message.blockNumber = (object.blockNumber !== undefined && object.blockNumber !== null)
            ? Long.fromValue(object.blockNumber)
            : Long.UZERO;
        message.signature = object.signature ?? new Uint8Array(0);
        return message;
    },
};
function createBaseRecoverablePasskeySignature() {
    return {
        publicKey: new Uint8Array(0),
        signature: new Uint8Array(0),
        authenticatorData: new Uint8Array(0),
        clientDataJson: new Uint8Array(0),
    };
}
export const RecoverablePasskeySignature = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.publicKey.length !== 0) {
            writer.uint32(10).bytes(message.publicKey);
        }
        if (message.signature.length !== 0) {
            writer.uint32(18).bytes(message.signature);
        }
        if (message.authenticatorData.length !== 0) {
            writer.uint32(26).bytes(message.authenticatorData);
        }
        if (message.clientDataJson.length !== 0) {
            writer.uint32(34).bytes(message.clientDataJson);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseRecoverablePasskeySignature();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.publicKey = reader.bytes();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.signature = reader.bytes();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.authenticatorData = reader.bytes();
                    continue;
                case 4:
                    if (tag !== 34) {
                        break;
                    }
                    message.clientDataJson = reader.bytes();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            publicKey: isSet(object.publicKey) ? bytesFromBase64(object.publicKey) : new Uint8Array(0),
            signature: isSet(object.signature) ? bytesFromBase64(object.signature) : new Uint8Array(0),
            authenticatorData: isSet(object.authenticatorData)
                ? bytesFromBase64(object.authenticatorData)
                : new Uint8Array(0),
            clientDataJson: isSet(object.clientDataJson) ? bytesFromBase64(object.clientDataJson) : new Uint8Array(0),
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.publicKey.length !== 0) {
            obj.publicKey = base64FromBytes(message.publicKey);
        }
        if (message.signature.length !== 0) {
            obj.signature = base64FromBytes(message.signature);
        }
        if (message.authenticatorData.length !== 0) {
            obj.authenticatorData = base64FromBytes(message.authenticatorData);
        }
        if (message.clientDataJson.length !== 0) {
            obj.clientDataJson = base64FromBytes(message.clientDataJson);
        }
        return obj;
    },
    create(base) {
        return RecoverablePasskeySignature.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseRecoverablePasskeySignature();
        message.publicKey = object.publicKey ?? new Uint8Array(0);
        message.signature = object.signature ?? new Uint8Array(0);
        message.authenticatorData = object.authenticatorData ?? new Uint8Array(0);
        message.clientDataJson = object.clientDataJson ?? new Uint8Array(0);
        return message;
    },
};
function createBaseLegacyDelegatedSignature() {
    return { delegatedKey: undefined, signature: undefined };
}
export const LegacyDelegatedSignature = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.delegatedKey !== undefined) {
            SignedPublicKey.encode(message.delegatedKey, writer.uint32(10).fork()).ldelim();
        }
        if (message.signature !== undefined) {
            RecoverableEcdsaSignature.encode(message.signature, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseLegacyDelegatedSignature();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.delegatedKey = SignedPublicKey.decode(reader, reader.uint32());
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.signature = RecoverableEcdsaSignature.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            delegatedKey: isSet(object.delegatedKey) ? SignedPublicKey.fromJSON(object.delegatedKey) : undefined,
            signature: isSet(object.signature) ? RecoverableEcdsaSignature.fromJSON(object.signature) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.delegatedKey !== undefined) {
            obj.delegatedKey = SignedPublicKey.toJSON(message.delegatedKey);
        }
        if (message.signature !== undefined) {
            obj.signature = RecoverableEcdsaSignature.toJSON(message.signature);
        }
        return obj;
    },
    create(base) {
        return LegacyDelegatedSignature.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseLegacyDelegatedSignature();
        message.delegatedKey = (object.delegatedKey !== undefined && object.delegatedKey !== null)
            ? SignedPublicKey.fromPartial(object.delegatedKey)
            : undefined;
        message.signature = (object.signature !== undefined && object.signature !== null)
            ? RecoverableEcdsaSignature.fromPartial(object.signature)
            : undefined;
        return message;
    },
};
function createBaseSignature() {
    return {
        erc191: undefined,
        erc6492: undefined,
        installationKey: undefined,
        delegatedErc191: undefined,
        passkey: undefined,
    };
}
export const Signature = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.erc191 !== undefined) {
            RecoverableEcdsaSignature.encode(message.erc191, writer.uint32(10).fork()).ldelim();
        }
        if (message.erc6492 !== undefined) {
            SmartContractWalletSignature.encode(message.erc6492, writer.uint32(18).fork()).ldelim();
        }
        if (message.installationKey !== undefined) {
            RecoverableEd25519Signature.encode(message.installationKey, writer.uint32(26).fork()).ldelim();
        }
        if (message.delegatedErc191 !== undefined) {
            LegacyDelegatedSignature.encode(message.delegatedErc191, writer.uint32(34).fork()).ldelim();
        }
        if (message.passkey !== undefined) {
            RecoverablePasskeySignature.encode(message.passkey, writer.uint32(42).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSignature();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.erc191 = RecoverableEcdsaSignature.decode(reader, reader.uint32());
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.erc6492 = SmartContractWalletSignature.decode(reader, reader.uint32());
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.installationKey = RecoverableEd25519Signature.decode(reader, reader.uint32());
                    continue;
                case 4:
                    if (tag !== 34) {
                        break;
                    }
                    message.delegatedErc191 = LegacyDelegatedSignature.decode(reader, reader.uint32());
                    continue;
                case 5:
                    if (tag !== 42) {
                        break;
                    }
                    message.passkey = RecoverablePasskeySignature.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            erc191: isSet(object.erc191) ? RecoverableEcdsaSignature.fromJSON(object.erc191) : undefined,
            erc6492: isSet(object.erc6492) ? SmartContractWalletSignature.fromJSON(object.erc6492) : undefined,
            installationKey: isSet(object.installationKey)
                ? RecoverableEd25519Signature.fromJSON(object.installationKey)
                : undefined,
            delegatedErc191: isSet(object.delegatedErc191)
                ? LegacyDelegatedSignature.fromJSON(object.delegatedErc191)
                : undefined,
            passkey: isSet(object.passkey) ? RecoverablePasskeySignature.fromJSON(object.passkey) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.erc191 !== undefined) {
            obj.erc191 = RecoverableEcdsaSignature.toJSON(message.erc191);
        }
        if (message.erc6492 !== undefined) {
            obj.erc6492 = SmartContractWalletSignature.toJSON(message.erc6492);
        }
        if (message.installationKey !== undefined) {
            obj.installationKey = RecoverableEd25519Signature.toJSON(message.installationKey);
        }
        if (message.delegatedErc191 !== undefined) {
            obj.delegatedErc191 = LegacyDelegatedSignature.toJSON(message.delegatedErc191);
        }
        if (message.passkey !== undefined) {
            obj.passkey = RecoverablePasskeySignature.toJSON(message.passkey);
        }
        return obj;
    },
    create(base) {
        return Signature.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseSignature();
        message.erc191 = (object.erc191 !== undefined && object.erc191 !== null)
            ? RecoverableEcdsaSignature.fromPartial(object.erc191)
            : undefined;
        message.erc6492 = (object.erc6492 !== undefined && object.erc6492 !== null)
            ? SmartContractWalletSignature.fromPartial(object.erc6492)
            : undefined;
        message.installationKey = (object.installationKey !== undefined && object.installationKey !== null)
            ? RecoverableEd25519Signature.fromPartial(object.installationKey)
            : undefined;
        message.delegatedErc191 = (object.delegatedErc191 !== undefined && object.delegatedErc191 !== null)
            ? LegacyDelegatedSignature.fromPartial(object.delegatedErc191)
            : undefined;
        message.passkey = (object.passkey !== undefined && object.passkey !== null)
            ? RecoverablePasskeySignature.fromPartial(object.passkey)
            : undefined;
        return message;
    },
};
function bytesFromBase64(b64) {
    if (globalThis.Buffer) {
        return Uint8Array.from(globalThis.Buffer.from(b64, "base64"));
    }
    else {
        const bin = globalThis.atob(b64);
        const arr = new Uint8Array(bin.length);
        for (let i = 0; i < bin.length; ++i) {
            arr[i] = bin.charCodeAt(i);
        }
        return arr;
    }
}
function base64FromBytes(arr) {
    if (globalThis.Buffer) {
        return globalThis.Buffer.from(arr).toString("base64");
    }
    else {
        const bin = [];
        arr.forEach((byte) => {
            bin.push(globalThis.String.fromCharCode(byte));
        });
        return globalThis.btoa(bin.join(""));
    }
}
if (_m0.util.Long !== Long) {
    _m0.util.Long = Long;
    _m0.configure();
}
function isSet(value) {
    return value !== null && value !== undefined;
}
