// @xmtp/proto/ts/dist/cjs/mls/message_contents/content_types/reaction.pb.js
"use strict";
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v1.181.2
//   protoc               v3.21.12
// source: mls/message_contents/content_types/reaction.proto
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReactionV2 = exports.ReactionSchema = exports.ReactionAction = exports.protobufPackage = void 0;
exports.reactionActionFromJSON = reactionActionFromJSON;
exports.reactionActionToJSON = reactionActionToJSON;
exports.reactionSchemaFromJSON = reactionSchemaFromJSON;
exports.reactionSchemaToJSON = reactionSchemaToJSON;
/* eslint-disable */
const long_1 = __importDefault(require("long"));
const minimal_1 = __importDefault(require("protobufjs/minimal"));
exports.protobufPackage = "xmtp.mls.message_contents.content_types";
/**
 * reaction.proto
 * This file defines the ReactionV2 message type and is associated with the following ContentTypeId:
 *
 * ContentTypeId {
 *     authority_id: "xmtp.org",
 *     type_id:      "reaction",
 *     version_major: 2,
 *     version_minor: 0,
 * }
 */
/** Action enum to represent reaction states */
var ReactionAction;
(function (ReactionAction) {
    ReactionAction[ReactionAction["REACTION_ACTION_UNSPECIFIED"] = 0] = "REACTION_ACTION_UNSPECIFIED";
    ReactionAction[ReactionAction["REACTION_ACTION_ADDED"] = 1] = "REACTION_ACTION_ADDED";
    ReactionAction[ReactionAction["REACTION_ACTION_REMOVED"] = 2] = "REACTION_ACTION_REMOVED";
    ReactionAction[ReactionAction["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(ReactionAction || (exports.ReactionAction = ReactionAction = {}));
function reactionActionFromJSON(object) {
    switch (object) {
        case 0:
        case "REACTION_ACTION_UNSPECIFIED":
            return ReactionAction.REACTION_ACTION_UNSPECIFIED;
        case 1:
        case "REACTION_ACTION_ADDED":
            return ReactionAction.REACTION_ACTION_ADDED;
        case 2:
        case "REACTION_ACTION_REMOVED":
            return ReactionAction.REACTION_ACTION_REMOVED;
        case -1:
        case "UNRECOGNIZED":
        default:
            return ReactionAction.UNRECOGNIZED;
    }
}
function reactionActionToJSON(object) {
    switch (object) {
        case ReactionAction.REACTION_ACTION_UNSPECIFIED:
            return "REACTION_ACTION_UNSPECIFIED";
        case ReactionAction.REACTION_ACTION_ADDED:
            return "REACTION_ACTION_ADDED";
        case ReactionAction.REACTION_ACTION_REMOVED:
            return "REACTION_ACTION_REMOVED";
        case ReactionAction.UNRECOGNIZED:
        default:
            return "UNRECOGNIZED";
    }
}
/** Schema enum to represent reaction content types */
var ReactionSchema;
(function (ReactionSchema) {
    ReactionSchema[ReactionSchema["REACTION_SCHEMA_UNSPECIFIED"] = 0] = "REACTION_SCHEMA_UNSPECIFIED";
    ReactionSchema[ReactionSchema["REACTION_SCHEMA_UNICODE"] = 1] = "REACTION_SCHEMA_UNICODE";
    ReactionSchema[ReactionSchema["REACTION_SCHEMA_SHORTCODE"] = 2] = "REACTION_SCHEMA_SHORTCODE";
    ReactionSchema[ReactionSchema["REACTION_SCHEMA_CUSTOM"] = 3] = "REACTION_SCHEMA_CUSTOM";
    ReactionSchema[ReactionSchema["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(ReactionSchema || (exports.ReactionSchema = ReactionSchema = {}));
function reactionSchemaFromJSON(object) {
    switch (object) {
        case 0:
        case "REACTION_SCHEMA_UNSPECIFIED":
            return ReactionSchema.REACTION_SCHEMA_UNSPECIFIED;
        case 1:
        case "REACTION_SCHEMA_UNICODE":
            return ReactionSchema.REACTION_SCHEMA_UNICODE;
        case 2:
        case "REACTION_SCHEMA_SHORTCODE":
            return ReactionSchema.REACTION_SCHEMA_SHORTCODE;
        case 3:
        case "REACTION_SCHEMA_CUSTOM":
            return ReactionSchema.REACTION_SCHEMA_CUSTOM;
        case -1:
        case "UNRECOGNIZED":
        default:
            return ReactionSchema.UNRECOGNIZED;
    }
}
function reactionSchemaToJSON(object) {
    switch (object) {
        case ReactionSchema.REACTION_SCHEMA_UNSPECIFIED:
            return "REACTION_SCHEMA_UNSPECIFIED";
        case ReactionSchema.REACTION_SCHEMA_UNICODE:
            return "REACTION_SCHEMA_UNICODE";
        case ReactionSchema.REACTION_SCHEMA_SHORTCODE:
            return "REACTION_SCHEMA_SHORTCODE";
        case ReactionSchema.REACTION_SCHEMA_CUSTOM:
            return "REACTION_SCHEMA_CUSTOM";
        case ReactionSchema.UNRECOGNIZED:
        default:
            return "UNRECOGNIZED";
    }
}
function createBaseReactionV2() {
    return { reference: "", referenceInboxId: "", action: 0, content: "", schema: 0 };
}
exports.ReactionV2 = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.reference !== "") {
            writer.uint32(10).string(message.reference);
        }
        if (message.referenceInboxId !== "") {
            writer.uint32(18).string(message.referenceInboxId);
        }
        if (message.action !== 0) {
            writer.uint32(24).int32(message.action);
        }
        if (message.content !== "") {
            writer.uint32(34).string(message.content);
        }
        if (message.schema !== 0) {
            writer.uint32(40).int32(message.schema);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseReactionV2();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.reference = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.referenceInboxId = reader.string();
                    continue;
                case 3:
                    if (tag !== 24) {
                        break;
                    }
                    message.action = reader.int32();
                    continue;
                case 4:
                    if (tag !== 34) {
                        break;
                    }
                    message.content = reader.string();
                    continue;
                case 5:
                    if (tag !== 40) {
                        break;
                    }
                    message.schema = reader.int32();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            reference: isSet(object.reference) ? globalThis.String(object.reference) : "",
            referenceInboxId: isSet(object.referenceInboxId) ? globalThis.String(object.referenceInboxId) : "",
            action: isSet(object.action) ? reactionActionFromJSON(object.action) : 0,
            content: isSet(object.content) ? globalThis.String(object.content) : "",
            schema: isSet(object.schema) ? reactionSchemaFromJSON(object.schema) : 0,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.reference !== "") {
            obj.reference = message.reference;
        }
        if (message.referenceInboxId !== "") {
            obj.referenceInboxId = message.referenceInboxId;
        }
        if (message.action !== 0) {
            obj.action = reactionActionToJSON(message.action);
        }
        if (message.content !== "") {
            obj.content = message.content;
        }
        if (message.schema !== 0) {
            obj.schema = reactionSchemaToJSON(message.schema);
        }
        return obj;
    },
    create(base) {
        return exports.ReactionV2.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseReactionV2();
        message.reference = object.reference ?? "";
        message.referenceInboxId = object.referenceInboxId ?? "";
        message.action = object.action ?? 0;
        message.content = object.content ?? "";
        message.schema = object.schema ?? 0;
        return message;
    },
};
if (minimal_1.default.util.Long !== long_1.default) {
    minimal_1.default.util.Long = long_1.default;
    minimal_1.default.configure();
}
function isSet(value) {
    return value !== null && value !== undefined;
}
