// @xmtp/proto/ts/dist/cjs/mls/message_contents/content.pb.js
"use strict";
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v1.181.2
//   protoc               v3.21.12
// source: mls/message_contents/content.proto
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PlaintextEnvelope_V2 = exports.PlaintextEnvelope_V1 = exports.PlaintextEnvelope = exports.EncodedContent_ParametersEntry = exports.EncodedContent = exports.ContentTypeId = exports.Compression = exports.protobufPackage = void 0;
exports.compressionFromJSON = compressionFromJSON;
exports.compressionToJSON = compressionToJSON;
/* eslint-disable */
const long_1 = __importDefault(require("long"));
const minimal_1 = __importDefault(require("protobufjs/minimal"));
const content_pb_1 = require("../../device_sync/content.pb");
exports.protobufPackage = "xmtp.mls.message_contents";
/**
 * Message content encoding structures
 * Copied from V2 code so that we can eventually retire all V2 message content
 */
/**
 * Recognized compression algorithms
 * protolint:disable ENUM_FIELD_NAMES_ZERO_VALUE_END_WITH
 */
var Compression;
(function (Compression) {
    Compression[Compression["COMPRESSION_DEFLATE"] = 0] = "COMPRESSION_DEFLATE";
    Compression[Compression["COMPRESSION_GZIP"] = 1] = "COMPRESSION_GZIP";
    Compression[Compression["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(Compression || (exports.Compression = Compression = {}));
function compressionFromJSON(object) {
    switch (object) {
        case 0:
        case "COMPRESSION_DEFLATE":
            return Compression.COMPRESSION_DEFLATE;
        case 1:
        case "COMPRESSION_GZIP":
            return Compression.COMPRESSION_GZIP;
        case -1:
        case "UNRECOGNIZED":
        default:
            return Compression.UNRECOGNIZED;
    }
}
function compressionToJSON(object) {
    switch (object) {
        case Compression.COMPRESSION_DEFLATE:
            return "COMPRESSION_DEFLATE";
        case Compression.COMPRESSION_GZIP:
            return "COMPRESSION_GZIP";
        case Compression.UNRECOGNIZED:
        default:
            return "UNRECOGNIZED";
    }
}
function createBaseContentTypeId() {
    return { authorityId: "", typeId: "", versionMajor: 0, versionMinor: 0 };
}
exports.ContentTypeId = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.authorityId !== "") {
            writer.uint32(10).string(message.authorityId);
        }
        if (message.typeId !== "") {
            writer.uint32(18).string(message.typeId);
        }
        if (message.versionMajor !== 0) {
            writer.uint32(24).uint32(message.versionMajor);
        }
        if (message.versionMinor !== 0) {
            writer.uint32(32).uint32(message.versionMinor);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseContentTypeId();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.authorityId = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.typeId = reader.string();
                    continue;
                case 3:
                    if (tag !== 24) {
                        break;
                    }
                    message.versionMajor = reader.uint32();
                    continue;
                case 4:
                    if (tag !== 32) {
                        break;
                    }
                    message.versionMinor = reader.uint32();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            authorityId: isSet(object.authorityId) ? globalThis.String(object.authorityId) : "",
            typeId: isSet(object.typeId) ? globalThis.String(object.typeId) : "",
            versionMajor: isSet(object.versionMajor) ? globalThis.Number(object.versionMajor) : 0,
            versionMinor: isSet(object.versionMinor) ? globalThis.Number(object.versionMinor) : 0,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.authorityId !== "") {
            obj.authorityId = message.authorityId;
        }
        if (message.typeId !== "") {
            obj.typeId = message.typeId;
        }
        if (message.versionMajor !== 0) {
            obj.versionMajor = Math.round(message.versionMajor);
        }
        if (message.versionMinor !== 0) {
            obj.versionMinor = Math.round(message.versionMinor);
        }
        return obj;
    },
    create(base) {
        return exports.ContentTypeId.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseContentTypeId();
        message.authorityId = object.authorityId ?? "";
        message.typeId = object.typeId ?? "";
        message.versionMajor = object.versionMajor ?? 0;
        message.versionMinor = object.versionMinor ?? 0;
        return message;
    },
};
function createBaseEncodedContent() {
    return { type: undefined, parameters: {}, fallback: undefined, compression: undefined, content: new Uint8Array(0) };
}
exports.EncodedContent = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.type !== undefined) {
            exports.ContentTypeId.encode(message.type, writer.uint32(10).fork()).ldelim();
        }
        Object.entries(message.parameters).forEach(([key, value]) => {
            exports.EncodedContent_ParametersEntry.encode({ key: key, value }, writer.uint32(18).fork()).ldelim();
        });
        if (message.fallback !== undefined) {
            writer.uint32(26).string(message.fallback);
        }
        if (message.compression !== undefined) {
            writer.uint32(40).int32(message.compression);
        }
        if (message.content.length !== 0) {
            writer.uint32(34).bytes(message.content);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseEncodedContent();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.type = exports.ContentTypeId.decode(reader, reader.uint32());
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    const entry2 = exports.EncodedContent_ParametersEntry.decode(reader, reader.uint32());
                    if (entry2.value !== undefined) {
                        message.parameters[entry2.key] = entry2.value;
                    }
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.fallback = reader.string();
                    continue;
                case 5:
                    if (tag !== 40) {
                        break;
                    }
                    message.compression = reader.int32();
                    continue;
                case 4:
                    if (tag !== 34) {
                        break;
                    }
                    message.content = reader.bytes();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            type: isSet(object.type) ? exports.ContentTypeId.fromJSON(object.type) : undefined,
            parameters: isObject(object.parameters)
                ? Object.entries(object.parameters).reduce((acc, [key, value]) => {
                    acc[key] = String(value);
                    return acc;
                }, {})
                : {},
            fallback: isSet(object.fallback) ? globalThis.String(object.fallback) : undefined,
            compression: isSet(object.compression) ? compressionFromJSON(object.compression) : undefined,
            content: isSet(object.content) ? bytesFromBase64(object.content) : new Uint8Array(0),
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.type !== undefined) {
            obj.type = exports.ContentTypeId.toJSON(message.type);
        }
        if (message.parameters) {
            const entries = Object.entries(message.parameters);
            if (entries.length > 0) {
                obj.parameters = {};
                entries.forEach(([k, v]) => {
                    obj.parameters[k] = v;
                });
            }
        }
        if (message.fallback !== undefined) {
            obj.fallback = message.fallback;
        }
        if (message.compression !== undefined) {
            obj.compression = compressionToJSON(message.compression);
        }
        if (message.content.length !== 0) {
            obj.content = base64FromBytes(message.content);
        }
        return obj;
    },
    create(base) {
        return exports.EncodedContent.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseEncodedContent();
        message.type = (object.type !== undefined && object.type !== null)
            ? exports.ContentTypeId.fromPartial(object.type)
            : undefined;
        message.parameters = Object.entries(object.parameters ?? {}).reduce((acc, [key, value]) => {
            if (value !== undefined) {
                acc[key] = globalThis.String(value);
            }
            return acc;
        }, {});
        message.fallback = object.fallback ?? undefined;
        message.compression = object.compression ?? undefined;
        message.content = object.content ?? new Uint8Array(0);
        return message;
    },
};
function createBaseEncodedContent_ParametersEntry() {
    return { key: "", value: "" };
}
exports.EncodedContent_ParametersEntry = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.key !== "") {
            writer.uint32(10).string(message.key);
        }
        if (message.value !== "") {
            writer.uint32(18).string(message.value);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseEncodedContent_ParametersEntry();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.key = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.value = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            key: isSet(object.key) ? globalThis.String(object.key) : "",
            value: isSet(object.value) ? globalThis.String(object.value) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.key !== "") {
            obj.key = message.key;
        }
        if (message.value !== "") {
            obj.value = message.value;
        }
        return obj;
    },
    create(base) {
        return exports.EncodedContent_ParametersEntry.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseEncodedContent_ParametersEntry();
        message.key = object.key ?? "";
        message.value = object.value ?? "";
        return message;
    },
};
function createBasePlaintextEnvelope() {
    return { v1: undefined, v2: undefined };
}
exports.PlaintextEnvelope = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.v1 !== undefined) {
            exports.PlaintextEnvelope_V1.encode(message.v1, writer.uint32(10).fork()).ldelim();
        }
        if (message.v2 !== undefined) {
            exports.PlaintextEnvelope_V2.encode(message.v2, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBasePlaintextEnvelope();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.v1 = exports.PlaintextEnvelope_V1.decode(reader, reader.uint32());
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.v2 = exports.PlaintextEnvelope_V2.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            v1: isSet(object.v1) ? exports.PlaintextEnvelope_V1.fromJSON(object.v1) : undefined,
            v2: isSet(object.v2) ? exports.PlaintextEnvelope_V2.fromJSON(object.v2) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.v1 !== undefined) {
            obj.v1 = exports.PlaintextEnvelope_V1.toJSON(message.v1);
        }
        if (message.v2 !== undefined) {
            obj.v2 = exports.PlaintextEnvelope_V2.toJSON(message.v2);
        }
        return obj;
    },
    create(base) {
        return exports.PlaintextEnvelope.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBasePlaintextEnvelope();
        message.v1 = (object.v1 !== undefined && object.v1 !== null)
            ? exports.PlaintextEnvelope_V1.fromPartial(object.v1)
            : undefined;
        message.v2 = (object.v2 !== undefined && object.v2 !== null)
            ? exports.PlaintextEnvelope_V2.fromPartial(object.v2)
            : undefined;
        return message;
    },
};
function createBasePlaintextEnvelope_V1() {
    return { content: new Uint8Array(0), idempotencyKey: "" };
}
exports.PlaintextEnvelope_V1 = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.content.length !== 0) {
            writer.uint32(10).bytes(message.content);
        }
        if (message.idempotencyKey !== "") {
            writer.uint32(18).string(message.idempotencyKey);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBasePlaintextEnvelope_V1();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.content = reader.bytes();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.idempotencyKey = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            content: isSet(object.content) ? bytesFromBase64(object.content) : new Uint8Array(0),
            idempotencyKey: isSet(object.idempotencyKey) ? globalThis.String(object.idempotencyKey) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.content.length !== 0) {
            obj.content = base64FromBytes(message.content);
        }
        if (message.idempotencyKey !== "") {
            obj.idempotencyKey = message.idempotencyKey;
        }
        return obj;
    },
    create(base) {
        return exports.PlaintextEnvelope_V1.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBasePlaintextEnvelope_V1();
        message.content = object.content ?? new Uint8Array(0);
        message.idempotencyKey = object.idempotencyKey ?? "";
        return message;
    },
};
function createBasePlaintextEnvelope_V2() {
    return {
        idempotencyKey: "",
        content: undefined,
        deviceSyncRequest: undefined,
        deviceSyncReply: undefined,
        userPreferenceUpdate: undefined,
    };
}
exports.PlaintextEnvelope_V2 = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.idempotencyKey !== "") {
            writer.uint32(10).string(message.idempotencyKey);
        }
        if (message.content !== undefined) {
            writer.uint32(18).bytes(message.content);
        }
        if (message.deviceSyncRequest !== undefined) {
            content_pb_1.DeviceSyncRequest.encode(message.deviceSyncRequest, writer.uint32(26).fork()).ldelim();
        }
        if (message.deviceSyncReply !== undefined) {
            content_pb_1.DeviceSyncReply.encode(message.deviceSyncReply, writer.uint32(34).fork()).ldelim();
        }
        if (message.userPreferenceUpdate !== undefined) {
            content_pb_1.V1UserPreferenceUpdate.encode(message.userPreferenceUpdate, writer.uint32(42).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBasePlaintextEnvelope_V2();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.idempotencyKey = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.content = reader.bytes();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.deviceSyncRequest = content_pb_1.DeviceSyncRequest.decode(reader, reader.uint32());
                    continue;
                case 4:
                    if (tag !== 34) {
                        break;
                    }
                    message.deviceSyncReply = content_pb_1.DeviceSyncReply.decode(reader, reader.uint32());
                    continue;
                case 5:
                    if (tag !== 42) {
                        break;
                    }
                    message.userPreferenceUpdate = content_pb_1.V1UserPreferenceUpdate.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            idempotencyKey: isSet(object.idempotencyKey) ? globalThis.String(object.idempotencyKey) : "",
            content: isSet(object.content) ? bytesFromBase64(object.content) : undefined,
            deviceSyncRequest: isSet(object.deviceSyncRequest)
                ? content_pb_1.DeviceSyncRequest.fromJSON(object.deviceSyncRequest)
                : undefined,
            deviceSyncReply: isSet(object.deviceSyncReply) ? content_pb_1.DeviceSyncReply.fromJSON(object.deviceSyncReply) : undefined,
            userPreferenceUpdate: isSet(object.userPreferenceUpdate)
                ? content_pb_1.V1UserPreferenceUpdate.fromJSON(object.userPreferenceUpdate)
                : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.idempotencyKey !== "") {
            obj.idempotencyKey = message.idempotencyKey;
        }
        if (message.content !== undefined) {
            obj.content = base64FromBytes(message.content);
        }
        if (message.deviceSyncRequest !== undefined) {
            obj.deviceSyncRequest = content_pb_1.DeviceSyncRequest.toJSON(message.deviceSyncRequest);
        }
        if (message.deviceSyncReply !== undefined) {
            obj.deviceSyncReply = content_pb_1.DeviceSyncReply.toJSON(message.deviceSyncReply);
        }
        if (message.userPreferenceUpdate !== undefined) {
            obj.userPreferenceUpdate = content_pb_1.V1UserPreferenceUpdate.toJSON(message.userPreferenceUpdate);
        }
        return obj;
    },
    create(base) {
        return exports.PlaintextEnvelope_V2.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBasePlaintextEnvelope_V2();
        message.idempotencyKey = object.idempotencyKey ?? "";
        message.content = object.content ?? undefined;
        message.deviceSyncRequest = (object.deviceSyncRequest !== undefined && object.deviceSyncRequest !== null)
            ? content_pb_1.DeviceSyncRequest.fromPartial(object.deviceSyncRequest)
            : undefined;
        message.deviceSyncReply = (object.deviceSyncReply !== undefined && object.deviceSyncReply !== null)
            ? content_pb_1.DeviceSyncReply.fromPartial(object.deviceSyncReply)
            : undefined;
        message.userPreferenceUpdate = (object.userPreferenceUpdate !== undefined && object.userPreferenceUpdate !== null)
            ? content_pb_1.V1UserPreferenceUpdate.fromPartial(object.userPreferenceUpdate)
            : undefined;
        return message;
    },
};
function bytesFromBase64(b64) {
    if (globalThis.Buffer) {
        return Uint8Array.from(globalThis.Buffer.from(b64, "base64"));
    }
    else {
        const bin = globalThis.atob(b64);
        const arr = new Uint8Array(bin.length);
        for (let i = 0; i < bin.length; ++i) {
            arr[i] = bin.charCodeAt(i);
        }
        return arr;
    }
}
function base64FromBytes(arr) {
    if (globalThis.Buffer) {
        return globalThis.Buffer.from(arr).toString("base64");
    }
    else {
        const bin = [];
        arr.forEach((byte) => {
            bin.push(globalThis.String.fromCharCode(byte));
        });
        return globalThis.btoa(bin.join(""));
    }
}
if (minimal_1.default.util.Long !== long_1.default) {
    minimal_1.default.util.Long = long_1.default;
    minimal_1.default.configure();
}
function isObject(value) {
    return typeof value === "object" && value !== null;
}
function isSet(value) {
    return value !== null && value !== undefined;
}
