// @xmtp/proto/ts/dist/cjs/mls/message_contents/group_mutable_metadata.pb.js
"use strict";
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v1.181.2
//   protoc               v3.21.12
// source: mls/message_contents/group_mutable_metadata.proto
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Inboxes = exports.GroupMutableMetadataV1_AttributesEntry = exports.GroupMutableMetadataV1 = exports.protobufPackage = void 0;
/* eslint-disable */
const long_1 = __importDefault(require("long"));
const minimal_1 = __importDefault(require("protobufjs/minimal"));
exports.protobufPackage = "xmtp.mls.message_contents";
function createBaseGroupMutableMetadataV1() {
    return { attributes: {}, adminList: undefined, superAdminList: undefined };
}
exports.GroupMutableMetadataV1 = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        Object.entries(message.attributes).forEach(([key, value]) => {
            exports.GroupMutableMetadataV1_AttributesEntry.encode({ key: key, value }, writer.uint32(10).fork()).ldelim();
        });
        if (message.adminList !== undefined) {
            exports.Inboxes.encode(message.adminList, writer.uint32(18).fork()).ldelim();
        }
        if (message.superAdminList !== undefined) {
            exports.Inboxes.encode(message.superAdminList, writer.uint32(26).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGroupMutableMetadataV1();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    const entry1 = exports.GroupMutableMetadataV1_AttributesEntry.decode(reader, reader.uint32());
                    if (entry1.value !== undefined) {
                        message.attributes[entry1.key] = entry1.value;
                    }
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.adminList = exports.Inboxes.decode(reader, reader.uint32());
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.superAdminList = exports.Inboxes.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            attributes: isObject(object.attributes)
                ? Object.entries(object.attributes).reduce((acc, [key, value]) => {
                    acc[key] = String(value);
                    return acc;
                }, {})
                : {},
            adminList: isSet(object.adminList) ? exports.Inboxes.fromJSON(object.adminList) : undefined,
            superAdminList: isSet(object.superAdminList) ? exports.Inboxes.fromJSON(object.superAdminList) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.attributes) {
            const entries = Object.entries(message.attributes);
            if (entries.length > 0) {
                obj.attributes = {};
                entries.forEach(([k, v]) => {
                    obj.attributes[k] = v;
                });
            }
        }
        if (message.adminList !== undefined) {
            obj.adminList = exports.Inboxes.toJSON(message.adminList);
        }
        if (message.superAdminList !== undefined) {
            obj.superAdminList = exports.Inboxes.toJSON(message.superAdminList);
        }
        return obj;
    },
    create(base) {
        return exports.GroupMutableMetadataV1.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGroupMutableMetadataV1();
        message.attributes = Object.entries(object.attributes ?? {}).reduce((acc, [key, value]) => {
            if (value !== undefined) {
                acc[key] = globalThis.String(value);
            }
            return acc;
        }, {});
        message.adminList = (object.adminList !== undefined && object.adminList !== null)
            ? exports.Inboxes.fromPartial(object.adminList)
            : undefined;
        message.superAdminList = (object.superAdminList !== undefined && object.superAdminList !== null)
            ? exports.Inboxes.fromPartial(object.superAdminList)
            : undefined;
        return message;
    },
};
function createBaseGroupMutableMetadataV1_AttributesEntry() {
    return { key: "", value: "" };
}
exports.GroupMutableMetadataV1_AttributesEntry = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.key !== "") {
            writer.uint32(10).string(message.key);
        }
        if (message.value !== "") {
            writer.uint32(18).string(message.value);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGroupMutableMetadataV1_AttributesEntry();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.key = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.value = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            key: isSet(object.key) ? globalThis.String(object.key) : "",
            value: isSet(object.value) ? globalThis.String(object.value) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.key !== "") {
            obj.key = message.key;
        }
        if (message.value !== "") {
            obj.value = message.value;
        }
        return obj;
    },
    create(base) {
        return exports.GroupMutableMetadataV1_AttributesEntry.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGroupMutableMetadataV1_AttributesEntry();
        message.key = object.key ?? "";
        message.value = object.value ?? "";
        return message;
    },
};
function createBaseInboxes() {
    return { inboxIds: [] };
}
exports.Inboxes = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        for (const v of message.inboxIds) {
            writer.uint32(10).string(v);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseInboxes();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.inboxIds.push(reader.string());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            inboxIds: globalThis.Array.isArray(object?.inboxIds) ? object.inboxIds.map((e) => globalThis.String(e)) : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.inboxIds?.length) {
            obj.inboxIds = message.inboxIds;
        }
        return obj;
    },
    create(base) {
        return exports.Inboxes.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseInboxes();
        message.inboxIds = object.inboxIds?.map((e) => e) || [];
        return message;
    },
};
if (minimal_1.default.util.Long !== long_1.default) {
    minimal_1.default.util.Long = long_1.default;
    minimal_1.default.configure();
}
function isObject(value) {
    return typeof value === "object" && value !== null;
}
function isSet(value) {
    return value !== null && value !== undefined;
}
