// @xmtp/proto/ts/dist/cjs/mls/message_contents/oneshot.pb.js
"use strict";
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v1.181.2
//   protoc               v3.21.12
// source: mls/message_contents/oneshot.proto
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReaddRequest = exports.OneshotMessage = exports.protobufPackage = void 0;
/* eslint-disable */
const long_1 = __importDefault(require("long"));
const minimal_1 = __importDefault(require("protobufjs/minimal"));
exports.protobufPackage = "xmtp.mls.message_contents";
function createBaseOneshotMessage() {
    return { readdRequest: undefined };
}
exports.OneshotMessage = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.readdRequest !== undefined) {
            exports.ReaddRequest.encode(message.readdRequest, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseOneshotMessage();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.readdRequest = exports.ReaddRequest.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { readdRequest: isSet(object.readdRequest) ? exports.ReaddRequest.fromJSON(object.readdRequest) : undefined };
    },
    toJSON(message) {
        const obj = {};
        if (message.readdRequest !== undefined) {
            obj.readdRequest = exports.ReaddRequest.toJSON(message.readdRequest);
        }
        return obj;
    },
    create(base) {
        return exports.OneshotMessage.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseOneshotMessage();
        message.readdRequest = (object.readdRequest !== undefined && object.readdRequest !== null)
            ? exports.ReaddRequest.fromPartial(object.readdRequest)
            : undefined;
        return message;
    },
};
function createBaseReaddRequest() {
    return { groupId: new Uint8Array(0), latestCommitSequenceId: long_1.default.UZERO };
}
exports.ReaddRequest = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.groupId.length !== 0) {
            writer.uint32(10).bytes(message.groupId);
        }
        if (!message.latestCommitSequenceId.equals(long_1.default.UZERO)) {
            writer.uint32(16).uint64(message.latestCommitSequenceId);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseReaddRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.groupId = reader.bytes();
                    continue;
                case 2:
                    if (tag !== 16) {
                        break;
                    }
                    message.latestCommitSequenceId = reader.uint64();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            groupId: isSet(object.groupId) ? bytesFromBase64(object.groupId) : new Uint8Array(0),
            latestCommitSequenceId: isSet(object.latestCommitSequenceId)
                ? long_1.default.fromValue(object.latestCommitSequenceId)
                : long_1.default.UZERO,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.groupId.length !== 0) {
            obj.groupId = base64FromBytes(message.groupId);
        }
        if (!message.latestCommitSequenceId.equals(long_1.default.UZERO)) {
            obj.latestCommitSequenceId = (message.latestCommitSequenceId || long_1.default.UZERO).toString();
        }
        return obj;
    },
    create(base) {
        return exports.ReaddRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseReaddRequest();
        message.groupId = object.groupId ?? new Uint8Array(0);
        message.latestCommitSequenceId =
            (object.latestCommitSequenceId !== undefined && object.latestCommitSequenceId !== null)
                ? long_1.default.fromValue(object.latestCommitSequenceId)
                : long_1.default.UZERO;
        return message;
    },
};
function bytesFromBase64(b64) {
    if (globalThis.Buffer) {
        return Uint8Array.from(globalThis.Buffer.from(b64, "base64"));
    }
    else {
        const bin = globalThis.atob(b64);
        const arr = new Uint8Array(bin.length);
        for (let i = 0; i < bin.length; ++i) {
            arr[i] = bin.charCodeAt(i);
        }
        return arr;
    }
}
function base64FromBytes(arr) {
    if (globalThis.Buffer) {
        return globalThis.Buffer.from(arr).toString("base64");
    }
    else {
        const bin = [];
        arr.forEach((byte) => {
            bin.push(globalThis.String.fromCharCode(byte));
        });
        return globalThis.btoa(bin.join(""));
    }
}
if (minimal_1.default.util.Long !== long_1.default) {
    minimal_1.default.util.Long = long_1.default;
    minimal_1.default.configure();
}
function isSet(value) {
    return value !== null && value !== undefined;
}
