// @xmtp/proto/ts/dist/cjs/mls/message_contents/transcript_messages.pb.js
"use strict";
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v1.181.2
//   protoc               v3.21.12
// source: mls/message_contents/transcript_messages.proto
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GroupUpdated_MetadataFieldChange = exports.GroupUpdated_Inbox = exports.GroupUpdated = exports.GroupMembershipChanges = exports.MembershipChange = exports.protobufPackage = void 0;
/* eslint-disable */
const long_1 = __importDefault(require("long"));
const minimal_1 = __importDefault(require("protobufjs/minimal"));
exports.protobufPackage = "xmtp.mls.message_contents";
function createBaseMembershipChange() {
    return { installationIds: [], accountAddress: "", initiatedByAccountAddress: "" };
}
exports.MembershipChange = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        for (const v of message.installationIds) {
            writer.uint32(10).bytes(v);
        }
        if (message.accountAddress !== "") {
            writer.uint32(18).string(message.accountAddress);
        }
        if (message.initiatedByAccountAddress !== "") {
            writer.uint32(26).string(message.initiatedByAccountAddress);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseMembershipChange();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.installationIds.push(reader.bytes());
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.accountAddress = reader.string();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.initiatedByAccountAddress = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            installationIds: globalThis.Array.isArray(object?.installationIds)
                ? object.installationIds.map((e) => bytesFromBase64(e))
                : [],
            accountAddress: isSet(object.accountAddress) ? globalThis.String(object.accountAddress) : "",
            initiatedByAccountAddress: isSet(object.initiatedByAccountAddress)
                ? globalThis.String(object.initiatedByAccountAddress)
                : "",
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.installationIds?.length) {
            obj.installationIds = message.installationIds.map((e) => base64FromBytes(e));
        }
        if (message.accountAddress !== "") {
            obj.accountAddress = message.accountAddress;
        }
        if (message.initiatedByAccountAddress !== "") {
            obj.initiatedByAccountAddress = message.initiatedByAccountAddress;
        }
        return obj;
    },
    create(base) {
        return exports.MembershipChange.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseMembershipChange();
        message.installationIds = object.installationIds?.map((e) => e) || [];
        message.accountAddress = object.accountAddress ?? "";
        message.initiatedByAccountAddress = object.initiatedByAccountAddress ?? "";
        return message;
    },
};
function createBaseGroupMembershipChanges() {
    return { membersAdded: [], membersRemoved: [], installationsAdded: [], installationsRemoved: [] };
}
exports.GroupMembershipChanges = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        for (const v of message.membersAdded) {
            exports.MembershipChange.encode(v, writer.uint32(10).fork()).ldelim();
        }
        for (const v of message.membersRemoved) {
            exports.MembershipChange.encode(v, writer.uint32(18).fork()).ldelim();
        }
        for (const v of message.installationsAdded) {
            exports.MembershipChange.encode(v, writer.uint32(26).fork()).ldelim();
        }
        for (const v of message.installationsRemoved) {
            exports.MembershipChange.encode(v, writer.uint32(34).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGroupMembershipChanges();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.membersAdded.push(exports.MembershipChange.decode(reader, reader.uint32()));
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.membersRemoved.push(exports.MembershipChange.decode(reader, reader.uint32()));
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.installationsAdded.push(exports.MembershipChange.decode(reader, reader.uint32()));
                    continue;
                case 4:
                    if (tag !== 34) {
                        break;
                    }
                    message.installationsRemoved.push(exports.MembershipChange.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            membersAdded: globalThis.Array.isArray(object?.membersAdded)
                ? object.membersAdded.map((e) => exports.MembershipChange.fromJSON(e))
                : [],
            membersRemoved: globalThis.Array.isArray(object?.membersRemoved)
                ? object.membersRemoved.map((e) => exports.MembershipChange.fromJSON(e))
                : [],
            installationsAdded: globalThis.Array.isArray(object?.installationsAdded)
                ? object.installationsAdded.map((e) => exports.MembershipChange.fromJSON(e))
                : [],
            installationsRemoved: globalThis.Array.isArray(object?.installationsRemoved)
                ? object.installationsRemoved.map((e) => exports.MembershipChange.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.membersAdded?.length) {
            obj.membersAdded = message.membersAdded.map((e) => exports.MembershipChange.toJSON(e));
        }
        if (message.membersRemoved?.length) {
            obj.membersRemoved = message.membersRemoved.map((e) => exports.MembershipChange.toJSON(e));
        }
        if (message.installationsAdded?.length) {
            obj.installationsAdded = message.installationsAdded.map((e) => exports.MembershipChange.toJSON(e));
        }
        if (message.installationsRemoved?.length) {
            obj.installationsRemoved = message.installationsRemoved.map((e) => exports.MembershipChange.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return exports.GroupMembershipChanges.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGroupMembershipChanges();
        message.membersAdded = object.membersAdded?.map((e) => exports.MembershipChange.fromPartial(e)) || [];
        message.membersRemoved = object.membersRemoved?.map((e) => exports.MembershipChange.fromPartial(e)) || [];
        message.installationsAdded = object.installationsAdded?.map((e) => exports.MembershipChange.fromPartial(e)) || [];
        message.installationsRemoved = object.installationsRemoved?.map((e) => exports.MembershipChange.fromPartial(e)) || [];
        return message;
    },
};
function createBaseGroupUpdated() {
    return {
        initiatedByInboxId: "",
        addedInboxes: [],
        removedInboxes: [],
        metadataFieldChanges: [],
        leftInboxes: [],
        addedAdminInboxes: [],
        removedAdminInboxes: [],
        addedSuperAdminInboxes: [],
        removedSuperAdminInboxes: [],
    };
}
exports.GroupUpdated = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.initiatedByInboxId !== "") {
            writer.uint32(10).string(message.initiatedByInboxId);
        }
        for (const v of message.addedInboxes) {
            exports.GroupUpdated_Inbox.encode(v, writer.uint32(18).fork()).ldelim();
        }
        for (const v of message.removedInboxes) {
            exports.GroupUpdated_Inbox.encode(v, writer.uint32(26).fork()).ldelim();
        }
        for (const v of message.metadataFieldChanges) {
            exports.GroupUpdated_MetadataFieldChange.encode(v, writer.uint32(34).fork()).ldelim();
        }
        for (const v of message.leftInboxes) {
            exports.GroupUpdated_Inbox.encode(v, writer.uint32(42).fork()).ldelim();
        }
        for (const v of message.addedAdminInboxes) {
            exports.GroupUpdated_Inbox.encode(v, writer.uint32(50).fork()).ldelim();
        }
        for (const v of message.removedAdminInboxes) {
            exports.GroupUpdated_Inbox.encode(v, writer.uint32(58).fork()).ldelim();
        }
        for (const v of message.addedSuperAdminInboxes) {
            exports.GroupUpdated_Inbox.encode(v, writer.uint32(66).fork()).ldelim();
        }
        for (const v of message.removedSuperAdminInboxes) {
            exports.GroupUpdated_Inbox.encode(v, writer.uint32(74).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGroupUpdated();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.initiatedByInboxId = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.addedInboxes.push(exports.GroupUpdated_Inbox.decode(reader, reader.uint32()));
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.removedInboxes.push(exports.GroupUpdated_Inbox.decode(reader, reader.uint32()));
                    continue;
                case 4:
                    if (tag !== 34) {
                        break;
                    }
                    message.metadataFieldChanges.push(exports.GroupUpdated_MetadataFieldChange.decode(reader, reader.uint32()));
                    continue;
                case 5:
                    if (tag !== 42) {
                        break;
                    }
                    message.leftInboxes.push(exports.GroupUpdated_Inbox.decode(reader, reader.uint32()));
                    continue;
                case 6:
                    if (tag !== 50) {
                        break;
                    }
                    message.addedAdminInboxes.push(exports.GroupUpdated_Inbox.decode(reader, reader.uint32()));
                    continue;
                case 7:
                    if (tag !== 58) {
                        break;
                    }
                    message.removedAdminInboxes.push(exports.GroupUpdated_Inbox.decode(reader, reader.uint32()));
                    continue;
                case 8:
                    if (tag !== 66) {
                        break;
                    }
                    message.addedSuperAdminInboxes.push(exports.GroupUpdated_Inbox.decode(reader, reader.uint32()));
                    continue;
                case 9:
                    if (tag !== 74) {
                        break;
                    }
                    message.removedSuperAdminInboxes.push(exports.GroupUpdated_Inbox.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            initiatedByInboxId: isSet(object.initiatedByInboxId) ? globalThis.String(object.initiatedByInboxId) : "",
            addedInboxes: globalThis.Array.isArray(object?.addedInboxes)
                ? object.addedInboxes.map((e) => exports.GroupUpdated_Inbox.fromJSON(e))
                : [],
            removedInboxes: globalThis.Array.isArray(object?.removedInboxes)
                ? object.removedInboxes.map((e) => exports.GroupUpdated_Inbox.fromJSON(e))
                : [],
            metadataFieldChanges: globalThis.Array.isArray(object?.metadataFieldChanges)
                ? object.metadataFieldChanges.map((e) => exports.GroupUpdated_MetadataFieldChange.fromJSON(e))
                : [],
            leftInboxes: globalThis.Array.isArray(object?.leftInboxes)
                ? object.leftInboxes.map((e) => exports.GroupUpdated_Inbox.fromJSON(e))
                : [],
            addedAdminInboxes: globalThis.Array.isArray(object?.addedAdminInboxes)
                ? object.addedAdminInboxes.map((e) => exports.GroupUpdated_Inbox.fromJSON(e))
                : [],
            removedAdminInboxes: globalThis.Array.isArray(object?.removedAdminInboxes)
                ? object.removedAdminInboxes.map((e) => exports.GroupUpdated_Inbox.fromJSON(e))
                : [],
            addedSuperAdminInboxes: globalThis.Array.isArray(object?.addedSuperAdminInboxes)
                ? object.addedSuperAdminInboxes.map((e) => exports.GroupUpdated_Inbox.fromJSON(e))
                : [],
            removedSuperAdminInboxes: globalThis.Array.isArray(object?.removedSuperAdminInboxes)
                ? object.removedSuperAdminInboxes.map((e) => exports.GroupUpdated_Inbox.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.initiatedByInboxId !== "") {
            obj.initiatedByInboxId = message.initiatedByInboxId;
        }
        if (message.addedInboxes?.length) {
            obj.addedInboxes = message.addedInboxes.map((e) => exports.GroupUpdated_Inbox.toJSON(e));
        }
        if (message.removedInboxes?.length) {
            obj.removedInboxes = message.removedInboxes.map((e) => exports.GroupUpdated_Inbox.toJSON(e));
        }
        if (message.metadataFieldChanges?.length) {
            obj.metadataFieldChanges = message.metadataFieldChanges.map((e) => exports.GroupUpdated_MetadataFieldChange.toJSON(e));
        }
        if (message.leftInboxes?.length) {
            obj.leftInboxes = message.leftInboxes.map((e) => exports.GroupUpdated_Inbox.toJSON(e));
        }
        if (message.addedAdminInboxes?.length) {
            obj.addedAdminInboxes = message.addedAdminInboxes.map((e) => exports.GroupUpdated_Inbox.toJSON(e));
        }
        if (message.removedAdminInboxes?.length) {
            obj.removedAdminInboxes = message.removedAdminInboxes.map((e) => exports.GroupUpdated_Inbox.toJSON(e));
        }
        if (message.addedSuperAdminInboxes?.length) {
            obj.addedSuperAdminInboxes = message.addedSuperAdminInboxes.map((e) => exports.GroupUpdated_Inbox.toJSON(e));
        }
        if (message.removedSuperAdminInboxes?.length) {
            obj.removedSuperAdminInboxes = message.removedSuperAdminInboxes.map((e) => exports.GroupUpdated_Inbox.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return exports.GroupUpdated.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGroupUpdated();
        message.initiatedByInboxId = object.initiatedByInboxId ?? "";
        message.addedInboxes = object.addedInboxes?.map((e) => exports.GroupUpdated_Inbox.fromPartial(e)) || [];
        message.removedInboxes = object.removedInboxes?.map((e) => exports.GroupUpdated_Inbox.fromPartial(e)) || [];
        message.metadataFieldChanges =
            object.metadataFieldChanges?.map((e) => exports.GroupUpdated_MetadataFieldChange.fromPartial(e)) || [];
        message.leftInboxes = object.leftInboxes?.map((e) => exports.GroupUpdated_Inbox.fromPartial(e)) || [];
        message.addedAdminInboxes = object.addedAdminInboxes?.map((e) => exports.GroupUpdated_Inbox.fromPartial(e)) || [];
        message.removedAdminInboxes = object.removedAdminInboxes?.map((e) => exports.GroupUpdated_Inbox.fromPartial(e)) || [];
        message.addedSuperAdminInboxes = object.addedSuperAdminInboxes?.map((e) => exports.GroupUpdated_Inbox.fromPartial(e)) || [];
        message.removedSuperAdminInboxes = object.removedSuperAdminInboxes?.map((e) => exports.GroupUpdated_Inbox.fromPartial(e)) ||
            [];
        return message;
    },
};
function createBaseGroupUpdated_Inbox() {
    return { inboxId: "" };
}
exports.GroupUpdated_Inbox = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.inboxId !== "") {
            writer.uint32(10).string(message.inboxId);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGroupUpdated_Inbox();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.inboxId = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { inboxId: isSet(object.inboxId) ? globalThis.String(object.inboxId) : "" };
    },
    toJSON(message) {
        const obj = {};
        if (message.inboxId !== "") {
            obj.inboxId = message.inboxId;
        }
        return obj;
    },
    create(base) {
        return exports.GroupUpdated_Inbox.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGroupUpdated_Inbox();
        message.inboxId = object.inboxId ?? "";
        return message;
    },
};
function createBaseGroupUpdated_MetadataFieldChange() {
    return { fieldName: "", oldValue: undefined, newValue: undefined };
}
exports.GroupUpdated_MetadataFieldChange = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.fieldName !== "") {
            writer.uint32(10).string(message.fieldName);
        }
        if (message.oldValue !== undefined) {
            writer.uint32(18).string(message.oldValue);
        }
        if (message.newValue !== undefined) {
            writer.uint32(26).string(message.newValue);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : minimal_1.default.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGroupUpdated_MetadataFieldChange();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.fieldName = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.oldValue = reader.string();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.newValue = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            fieldName: isSet(object.fieldName) ? globalThis.String(object.fieldName) : "",
            oldValue: isSet(object.oldValue) ? globalThis.String(object.oldValue) : undefined,
            newValue: isSet(object.newValue) ? globalThis.String(object.newValue) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.fieldName !== "") {
            obj.fieldName = message.fieldName;
        }
        if (message.oldValue !== undefined) {
            obj.oldValue = message.oldValue;
        }
        if (message.newValue !== undefined) {
            obj.newValue = message.newValue;
        }
        return obj;
    },
    create(base) {
        return exports.GroupUpdated_MetadataFieldChange.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGroupUpdated_MetadataFieldChange();
        message.fieldName = object.fieldName ?? "";
        message.oldValue = object.oldValue ?? undefined;
        message.newValue = object.newValue ?? undefined;
        return message;
    },
};
function bytesFromBase64(b64) {
    if (globalThis.Buffer) {
        return Uint8Array.from(globalThis.Buffer.from(b64, "base64"));
    }
    else {
        const bin = globalThis.atob(b64);
        const arr = new Uint8Array(bin.length);
        for (let i = 0; i < bin.length; ++i) {
            arr[i] = bin.charCodeAt(i);
        }
        return arr;
    }
}
function base64FromBytes(arr) {
    if (globalThis.Buffer) {
        return globalThis.Buffer.from(arr).toString("base64");
    }
    else {
        const bin = [];
        arr.forEach((byte) => {
            bin.push(globalThis.String.fromCharCode(byte));
        });
        return globalThis.btoa(bin.join(""));
    }
}
if (minimal_1.default.util.Long !== long_1.default) {
    minimal_1.default.util.Long = long_1.default;
    minimal_1.default.configure();
}
function isSet(value) {
    return value !== null && value !== undefined;
}
