// @xmtp/proto/ts/dist/esm/mls/database/intents.pb.js
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v1.181.2
//   protoc               v3.21.12
// source: mls/database/intents.proto
/* eslint-disable */
import Long from "long";
import _m0 from "protobufjs/minimal";
import { WelcomePointeeEncryptionAeadTypesExtension } from "../message_contents/welcome_pointer.pb";
import { welcomeWrapperAlgorithmFromJSON, welcomeWrapperAlgorithmToJSON, } from "../message_contents/wrapper_encryption.pb";
export const protobufPackage = "xmtp.mls.database";
/** Type of update to admin lists */
export var AdminListUpdateType;
(function (AdminListUpdateType) {
    AdminListUpdateType[AdminListUpdateType["ADMIN_LIST_UPDATE_TYPE_UNSPECIFIED"] = 0] = "ADMIN_LIST_UPDATE_TYPE_UNSPECIFIED";
    AdminListUpdateType[AdminListUpdateType["ADMIN_LIST_UPDATE_TYPE_ADD_ADMIN"] = 1] = "ADMIN_LIST_UPDATE_TYPE_ADD_ADMIN";
    AdminListUpdateType[AdminListUpdateType["ADMIN_LIST_UPDATE_TYPE_REMOVE_ADMIN"] = 2] = "ADMIN_LIST_UPDATE_TYPE_REMOVE_ADMIN";
    AdminListUpdateType[AdminListUpdateType["ADMIN_LIST_UPDATE_TYPE_ADD_SUPER_ADMIN"] = 3] = "ADMIN_LIST_UPDATE_TYPE_ADD_SUPER_ADMIN";
    AdminListUpdateType[AdminListUpdateType["ADMIN_LIST_UPDATE_TYPE_REMOVE_SUPER_ADMIN"] = 4] = "ADMIN_LIST_UPDATE_TYPE_REMOVE_SUPER_ADMIN";
    AdminListUpdateType[AdminListUpdateType["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(AdminListUpdateType || (AdminListUpdateType = {}));
export function adminListUpdateTypeFromJSON(object) {
    switch (object) {
        case 0:
        case "ADMIN_LIST_UPDATE_TYPE_UNSPECIFIED":
            return AdminListUpdateType.ADMIN_LIST_UPDATE_TYPE_UNSPECIFIED;
        case 1:
        case "ADMIN_LIST_UPDATE_TYPE_ADD_ADMIN":
            return AdminListUpdateType.ADMIN_LIST_UPDATE_TYPE_ADD_ADMIN;
        case 2:
        case "ADMIN_LIST_UPDATE_TYPE_REMOVE_ADMIN":
            return AdminListUpdateType.ADMIN_LIST_UPDATE_TYPE_REMOVE_ADMIN;
        case 3:
        case "ADMIN_LIST_UPDATE_TYPE_ADD_SUPER_ADMIN":
            return AdminListUpdateType.ADMIN_LIST_UPDATE_TYPE_ADD_SUPER_ADMIN;
        case 4:
        case "ADMIN_LIST_UPDATE_TYPE_REMOVE_SUPER_ADMIN":
            return AdminListUpdateType.ADMIN_LIST_UPDATE_TYPE_REMOVE_SUPER_ADMIN;
        case -1:
        case "UNRECOGNIZED":
        default:
            return AdminListUpdateType.UNRECOGNIZED;
    }
}
export function adminListUpdateTypeToJSON(object) {
    switch (object) {
        case AdminListUpdateType.ADMIN_LIST_UPDATE_TYPE_UNSPECIFIED:
            return "ADMIN_LIST_UPDATE_TYPE_UNSPECIFIED";
        case AdminListUpdateType.ADMIN_LIST_UPDATE_TYPE_ADD_ADMIN:
            return "ADMIN_LIST_UPDATE_TYPE_ADD_ADMIN";
        case AdminListUpdateType.ADMIN_LIST_UPDATE_TYPE_REMOVE_ADMIN:
            return "ADMIN_LIST_UPDATE_TYPE_REMOVE_ADMIN";
        case AdminListUpdateType.ADMIN_LIST_UPDATE_TYPE_ADD_SUPER_ADMIN:
            return "ADMIN_LIST_UPDATE_TYPE_ADD_SUPER_ADMIN";
        case AdminListUpdateType.ADMIN_LIST_UPDATE_TYPE_REMOVE_SUPER_ADMIN:
            return "ADMIN_LIST_UPDATE_TYPE_REMOVE_SUPER_ADMIN";
        case AdminListUpdateType.UNRECOGNIZED:
        default:
            return "UNRECOGNIZED";
    }
}
/** Type of Permission to Update */
export var PermissionUpdateType;
(function (PermissionUpdateType) {
    PermissionUpdateType[PermissionUpdateType["PERMISSION_UPDATE_TYPE_UNSPECIFIED"] = 0] = "PERMISSION_UPDATE_TYPE_UNSPECIFIED";
    PermissionUpdateType[PermissionUpdateType["PERMISSION_UPDATE_TYPE_ADD_MEMBER"] = 1] = "PERMISSION_UPDATE_TYPE_ADD_MEMBER";
    PermissionUpdateType[PermissionUpdateType["PERMISSION_UPDATE_TYPE_REMOVE_MEMBER"] = 2] = "PERMISSION_UPDATE_TYPE_REMOVE_MEMBER";
    PermissionUpdateType[PermissionUpdateType["PERMISSION_UPDATE_TYPE_ADD_ADMIN"] = 3] = "PERMISSION_UPDATE_TYPE_ADD_ADMIN";
    PermissionUpdateType[PermissionUpdateType["PERMISSION_UPDATE_TYPE_REMOVE_ADMIN"] = 4] = "PERMISSION_UPDATE_TYPE_REMOVE_ADMIN";
    PermissionUpdateType[PermissionUpdateType["PERMISSION_UPDATE_TYPE_UPDATE_METADATA"] = 5] = "PERMISSION_UPDATE_TYPE_UPDATE_METADATA";
    PermissionUpdateType[PermissionUpdateType["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(PermissionUpdateType || (PermissionUpdateType = {}));
export function permissionUpdateTypeFromJSON(object) {
    switch (object) {
        case 0:
        case "PERMISSION_UPDATE_TYPE_UNSPECIFIED":
            return PermissionUpdateType.PERMISSION_UPDATE_TYPE_UNSPECIFIED;
        case 1:
        case "PERMISSION_UPDATE_TYPE_ADD_MEMBER":
            return PermissionUpdateType.PERMISSION_UPDATE_TYPE_ADD_MEMBER;
        case 2:
        case "PERMISSION_UPDATE_TYPE_REMOVE_MEMBER":
            return PermissionUpdateType.PERMISSION_UPDATE_TYPE_REMOVE_MEMBER;
        case 3:
        case "PERMISSION_UPDATE_TYPE_ADD_ADMIN":
            return PermissionUpdateType.PERMISSION_UPDATE_TYPE_ADD_ADMIN;
        case 4:
        case "PERMISSION_UPDATE_TYPE_REMOVE_ADMIN":
            return PermissionUpdateType.PERMISSION_UPDATE_TYPE_REMOVE_ADMIN;
        case 5:
        case "PERMISSION_UPDATE_TYPE_UPDATE_METADATA":
            return PermissionUpdateType.PERMISSION_UPDATE_TYPE_UPDATE_METADATA;
        case -1:
        case "UNRECOGNIZED":
        default:
            return PermissionUpdateType.UNRECOGNIZED;
    }
}
export function permissionUpdateTypeToJSON(object) {
    switch (object) {
        case PermissionUpdateType.PERMISSION_UPDATE_TYPE_UNSPECIFIED:
            return "PERMISSION_UPDATE_TYPE_UNSPECIFIED";
        case PermissionUpdateType.PERMISSION_UPDATE_TYPE_ADD_MEMBER:
            return "PERMISSION_UPDATE_TYPE_ADD_MEMBER";
        case PermissionUpdateType.PERMISSION_UPDATE_TYPE_REMOVE_MEMBER:
            return "PERMISSION_UPDATE_TYPE_REMOVE_MEMBER";
        case PermissionUpdateType.PERMISSION_UPDATE_TYPE_ADD_ADMIN:
            return "PERMISSION_UPDATE_TYPE_ADD_ADMIN";
        case PermissionUpdateType.PERMISSION_UPDATE_TYPE_REMOVE_ADMIN:
            return "PERMISSION_UPDATE_TYPE_REMOVE_ADMIN";
        case PermissionUpdateType.PERMISSION_UPDATE_TYPE_UPDATE_METADATA:
            return "PERMISSION_UPDATE_TYPE_UPDATE_METADATA";
        case PermissionUpdateType.UNRECOGNIZED:
        default:
            return "UNRECOGNIZED";
    }
}
/** Permission Policy */
export var PermissionPolicyOption;
(function (PermissionPolicyOption) {
    PermissionPolicyOption[PermissionPolicyOption["PERMISSION_POLICY_OPTION_UNSPECIFIED"] = 0] = "PERMISSION_POLICY_OPTION_UNSPECIFIED";
    PermissionPolicyOption[PermissionPolicyOption["PERMISSION_POLICY_OPTION_ALLOW"] = 1] = "PERMISSION_POLICY_OPTION_ALLOW";
    PermissionPolicyOption[PermissionPolicyOption["PERMISSION_POLICY_OPTION_DENY"] = 2] = "PERMISSION_POLICY_OPTION_DENY";
    PermissionPolicyOption[PermissionPolicyOption["PERMISSION_POLICY_OPTION_ADMIN_ONLY"] = 3] = "PERMISSION_POLICY_OPTION_ADMIN_ONLY";
    PermissionPolicyOption[PermissionPolicyOption["PERMISSION_POLICY_OPTION_SUPER_ADMIN_ONLY"] = 4] = "PERMISSION_POLICY_OPTION_SUPER_ADMIN_ONLY";
    PermissionPolicyOption[PermissionPolicyOption["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(PermissionPolicyOption || (PermissionPolicyOption = {}));
export function permissionPolicyOptionFromJSON(object) {
    switch (object) {
        case 0:
        case "PERMISSION_POLICY_OPTION_UNSPECIFIED":
            return PermissionPolicyOption.PERMISSION_POLICY_OPTION_UNSPECIFIED;
        case 1:
        case "PERMISSION_POLICY_OPTION_ALLOW":
            return PermissionPolicyOption.PERMISSION_POLICY_OPTION_ALLOW;
        case 2:
        case "PERMISSION_POLICY_OPTION_DENY":
            return PermissionPolicyOption.PERMISSION_POLICY_OPTION_DENY;
        case 3:
        case "PERMISSION_POLICY_OPTION_ADMIN_ONLY":
            return PermissionPolicyOption.PERMISSION_POLICY_OPTION_ADMIN_ONLY;
        case 4:
        case "PERMISSION_POLICY_OPTION_SUPER_ADMIN_ONLY":
            return PermissionPolicyOption.PERMISSION_POLICY_OPTION_SUPER_ADMIN_ONLY;
        case -1:
        case "UNRECOGNIZED":
        default:
            return PermissionPolicyOption.UNRECOGNIZED;
    }
}
export function permissionPolicyOptionToJSON(object) {
    switch (object) {
        case PermissionPolicyOption.PERMISSION_POLICY_OPTION_UNSPECIFIED:
            return "PERMISSION_POLICY_OPTION_UNSPECIFIED";
        case PermissionPolicyOption.PERMISSION_POLICY_OPTION_ALLOW:
            return "PERMISSION_POLICY_OPTION_ALLOW";
        case PermissionPolicyOption.PERMISSION_POLICY_OPTION_DENY:
            return "PERMISSION_POLICY_OPTION_DENY";
        case PermissionPolicyOption.PERMISSION_POLICY_OPTION_ADMIN_ONLY:
            return "PERMISSION_POLICY_OPTION_ADMIN_ONLY";
        case PermissionPolicyOption.PERMISSION_POLICY_OPTION_SUPER_ADMIN_ONLY:
            return "PERMISSION_POLICY_OPTION_SUPER_ADMIN_ONLY";
        case PermissionPolicyOption.UNRECOGNIZED:
        default:
            return "UNRECOGNIZED";
    }
}
function createBaseSendMessageData() {
    return { v1: undefined };
}
export const SendMessageData = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.v1 !== undefined) {
            SendMessageData_V1.encode(message.v1, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSendMessageData();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.v1 = SendMessageData_V1.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { v1: isSet(object.v1) ? SendMessageData_V1.fromJSON(object.v1) : undefined };
    },
    toJSON(message) {
        const obj = {};
        if (message.v1 !== undefined) {
            obj.v1 = SendMessageData_V1.toJSON(message.v1);
        }
        return obj;
    },
    create(base) {
        return SendMessageData.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseSendMessageData();
        message.v1 = (object.v1 !== undefined && object.v1 !== null)
            ? SendMessageData_V1.fromPartial(object.v1)
            : undefined;
        return message;
    },
};
function createBaseSendMessageData_V1() {
    return { payloadBytes: new Uint8Array(0) };
}
export const SendMessageData_V1 = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.payloadBytes.length !== 0) {
            writer.uint32(10).bytes(message.payloadBytes);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSendMessageData_V1();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.payloadBytes = reader.bytes();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { payloadBytes: isSet(object.payloadBytes) ? bytesFromBase64(object.payloadBytes) : new Uint8Array(0) };
    },
    toJSON(message) {
        const obj = {};
        if (message.payloadBytes.length !== 0) {
            obj.payloadBytes = base64FromBytes(message.payloadBytes);
        }
        return obj;
    },
    create(base) {
        return SendMessageData_V1.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseSendMessageData_V1();
        message.payloadBytes = object.payloadBytes ?? new Uint8Array(0);
        return message;
    },
};
function createBaseAccountAddresses() {
    return { accountAddresses: [] };
}
export const AccountAddresses = {
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.accountAddresses) {
            writer.uint32(10).string(v);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseAccountAddresses();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.accountAddresses.push(reader.string());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            accountAddresses: globalThis.Array.isArray(object?.accountAddresses)
                ? object.accountAddresses.map((e) => globalThis.String(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.accountAddresses?.length) {
            obj.accountAddresses = message.accountAddresses;
        }
        return obj;
    },
    create(base) {
        return AccountAddresses.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseAccountAddresses();
        message.accountAddresses = object.accountAddresses?.map((e) => e) || [];
        return message;
    },
};
function createBaseInstallationIds() {
    return { installationIds: [] };
}
export const InstallationIds = {
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.installationIds) {
            writer.uint32(10).bytes(v);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseInstallationIds();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.installationIds.push(reader.bytes());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            installationIds: globalThis.Array.isArray(object?.installationIds)
                ? object.installationIds.map((e) => bytesFromBase64(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.installationIds?.length) {
            obj.installationIds = message.installationIds.map((e) => base64FromBytes(e));
        }
        return obj;
    },
    create(base) {
        return InstallationIds.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseInstallationIds();
        message.installationIds = object.installationIds?.map((e) => e) || [];
        return message;
    },
};
function createBaseAddressesOrInstallationIds() {
    return { accountAddresses: undefined, installationIds: undefined };
}
export const AddressesOrInstallationIds = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.accountAddresses !== undefined) {
            AccountAddresses.encode(message.accountAddresses, writer.uint32(10).fork()).ldelim();
        }
        if (message.installationIds !== undefined) {
            InstallationIds.encode(message.installationIds, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseAddressesOrInstallationIds();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.accountAddresses = AccountAddresses.decode(reader, reader.uint32());
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.installationIds = InstallationIds.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            accountAddresses: isSet(object.accountAddresses) ? AccountAddresses.fromJSON(object.accountAddresses) : undefined,
            installationIds: isSet(object.installationIds) ? InstallationIds.fromJSON(object.installationIds) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.accountAddresses !== undefined) {
            obj.accountAddresses = AccountAddresses.toJSON(message.accountAddresses);
        }
        if (message.installationIds !== undefined) {
            obj.installationIds = InstallationIds.toJSON(message.installationIds);
        }
        return obj;
    },
    create(base) {
        return AddressesOrInstallationIds.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseAddressesOrInstallationIds();
        message.accountAddresses = (object.accountAddresses !== undefined && object.accountAddresses !== null)
            ? AccountAddresses.fromPartial(object.accountAddresses)
            : undefined;
        message.installationIds = (object.installationIds !== undefined && object.installationIds !== null)
            ? InstallationIds.fromPartial(object.installationIds)
            : undefined;
        return message;
    },
};
function createBaseAddMembersData() {
    return { v1: undefined };
}
export const AddMembersData = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.v1 !== undefined) {
            AddMembersData_V1.encode(message.v1, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseAddMembersData();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.v1 = AddMembersData_V1.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { v1: isSet(object.v1) ? AddMembersData_V1.fromJSON(object.v1) : undefined };
    },
    toJSON(message) {
        const obj = {};
        if (message.v1 !== undefined) {
            obj.v1 = AddMembersData_V1.toJSON(message.v1);
        }
        return obj;
    },
    create(base) {
        return AddMembersData.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseAddMembersData();
        message.v1 = (object.v1 !== undefined && object.v1 !== null) ? AddMembersData_V1.fromPartial(object.v1) : undefined;
        return message;
    },
};
function createBaseAddMembersData_V1() {
    return { addressesOrInstallationIds: undefined };
}
export const AddMembersData_V1 = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.addressesOrInstallationIds !== undefined) {
            AddressesOrInstallationIds.encode(message.addressesOrInstallationIds, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseAddMembersData_V1();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.addressesOrInstallationIds = AddressesOrInstallationIds.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            addressesOrInstallationIds: isSet(object.addressesOrInstallationIds)
                ? AddressesOrInstallationIds.fromJSON(object.addressesOrInstallationIds)
                : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.addressesOrInstallationIds !== undefined) {
            obj.addressesOrInstallationIds = AddressesOrInstallationIds.toJSON(message.addressesOrInstallationIds);
        }
        return obj;
    },
    create(base) {
        return AddMembersData_V1.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseAddMembersData_V1();
        message.addressesOrInstallationIds =
            (object.addressesOrInstallationIds !== undefined && object.addressesOrInstallationIds !== null)
                ? AddressesOrInstallationIds.fromPartial(object.addressesOrInstallationIds)
                : undefined;
        return message;
    },
};
function createBaseRemoveMembersData() {
    return { v1: undefined };
}
export const RemoveMembersData = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.v1 !== undefined) {
            RemoveMembersData_V1.encode(message.v1, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseRemoveMembersData();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.v1 = RemoveMembersData_V1.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { v1: isSet(object.v1) ? RemoveMembersData_V1.fromJSON(object.v1) : undefined };
    },
    toJSON(message) {
        const obj = {};
        if (message.v1 !== undefined) {
            obj.v1 = RemoveMembersData_V1.toJSON(message.v1);
        }
        return obj;
    },
    create(base) {
        return RemoveMembersData.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseRemoveMembersData();
        message.v1 = (object.v1 !== undefined && object.v1 !== null)
            ? RemoveMembersData_V1.fromPartial(object.v1)
            : undefined;
        return message;
    },
};
function createBaseRemoveMembersData_V1() {
    return { addressesOrInstallationIds: undefined };
}
export const RemoveMembersData_V1 = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.addressesOrInstallationIds !== undefined) {
            AddressesOrInstallationIds.encode(message.addressesOrInstallationIds, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseRemoveMembersData_V1();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.addressesOrInstallationIds = AddressesOrInstallationIds.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            addressesOrInstallationIds: isSet(object.addressesOrInstallationIds)
                ? AddressesOrInstallationIds.fromJSON(object.addressesOrInstallationIds)
                : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.addressesOrInstallationIds !== undefined) {
            obj.addressesOrInstallationIds = AddressesOrInstallationIds.toJSON(message.addressesOrInstallationIds);
        }
        return obj;
    },
    create(base) {
        return RemoveMembersData_V1.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseRemoveMembersData_V1();
        message.addressesOrInstallationIds =
            (object.addressesOrInstallationIds !== undefined && object.addressesOrInstallationIds !== null)
                ? AddressesOrInstallationIds.fromPartial(object.addressesOrInstallationIds)
                : undefined;
        return message;
    },
};
function createBaseUpdateGroupMembershipData() {
    return { v1: undefined };
}
export const UpdateGroupMembershipData = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.v1 !== undefined) {
            UpdateGroupMembershipData_V1.encode(message.v1, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseUpdateGroupMembershipData();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.v1 = UpdateGroupMembershipData_V1.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { v1: isSet(object.v1) ? UpdateGroupMembershipData_V1.fromJSON(object.v1) : undefined };
    },
    toJSON(message) {
        const obj = {};
        if (message.v1 !== undefined) {
            obj.v1 = UpdateGroupMembershipData_V1.toJSON(message.v1);
        }
        return obj;
    },
    create(base) {
        return UpdateGroupMembershipData.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseUpdateGroupMembershipData();
        message.v1 = (object.v1 !== undefined && object.v1 !== null)
            ? UpdateGroupMembershipData_V1.fromPartial(object.v1)
            : undefined;
        return message;
    },
};
function createBaseUpdateGroupMembershipData_V1() {
    return { membershipUpdates: {}, removedMembers: [], failedInstallations: [] };
}
export const UpdateGroupMembershipData_V1 = {
    encode(message, writer = _m0.Writer.create()) {
        Object.entries(message.membershipUpdates).forEach(([key, value]) => {
            UpdateGroupMembershipData_V1_MembershipUpdatesEntry.encode({ key: key, value }, writer.uint32(10).fork())
                .ldelim();
        });
        for (const v of message.removedMembers) {
            writer.uint32(18).string(v);
        }
        for (const v of message.failedInstallations) {
            writer.uint32(26).bytes(v);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseUpdateGroupMembershipData_V1();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    const entry1 = UpdateGroupMembershipData_V1_MembershipUpdatesEntry.decode(reader, reader.uint32());
                    if (entry1.value !== undefined) {
                        message.membershipUpdates[entry1.key] = entry1.value;
                    }
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.removedMembers.push(reader.string());
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.failedInstallations.push(reader.bytes());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            membershipUpdates: isObject(object.membershipUpdates)
                ? Object.entries(object.membershipUpdates).reduce((acc, [key, value]) => {
                    acc[key] = Long.fromValue(value);
                    return acc;
                }, {})
                : {},
            removedMembers: globalThis.Array.isArray(object?.removedMembers)
                ? object.removedMembers.map((e) => globalThis.String(e))
                : [],
            failedInstallations: globalThis.Array.isArray(object?.failedInstallations)
                ? object.failedInstallations.map((e) => bytesFromBase64(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.membershipUpdates) {
            const entries = Object.entries(message.membershipUpdates);
            if (entries.length > 0) {
                obj.membershipUpdates = {};
                entries.forEach(([k, v]) => {
                    obj.membershipUpdates[k] = v.toString();
                });
            }
        }
        if (message.removedMembers?.length) {
            obj.removedMembers = message.removedMembers;
        }
        if (message.failedInstallations?.length) {
            obj.failedInstallations = message.failedInstallations.map((e) => base64FromBytes(e));
        }
        return obj;
    },
    create(base) {
        return UpdateGroupMembershipData_V1.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseUpdateGroupMembershipData_V1();
        message.membershipUpdates = Object.entries(object.membershipUpdates ?? {}).reduce((acc, [key, value]) => {
            if (value !== undefined) {
                acc[key] = Long.fromValue(value);
            }
            return acc;
        }, {});
        message.removedMembers = object.removedMembers?.map((e) => e) || [];
        message.failedInstallations = object.failedInstallations?.map((e) => e) || [];
        return message;
    },
};
function createBaseUpdateGroupMembershipData_V1_MembershipUpdatesEntry() {
    return { key: "", value: Long.UZERO };
}
export const UpdateGroupMembershipData_V1_MembershipUpdatesEntry = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.key !== "") {
            writer.uint32(10).string(message.key);
        }
        if (!message.value.equals(Long.UZERO)) {
            writer.uint32(16).uint64(message.value);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseUpdateGroupMembershipData_V1_MembershipUpdatesEntry();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.key = reader.string();
                    continue;
                case 2:
                    if (tag !== 16) {
                        break;
                    }
                    message.value = reader.uint64();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            key: isSet(object.key) ? globalThis.String(object.key) : "",
            value: isSet(object.value) ? Long.fromValue(object.value) : Long.UZERO,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.key !== "") {
            obj.key = message.key;
        }
        if (!message.value.equals(Long.UZERO)) {
            obj.value = (message.value || Long.UZERO).toString();
        }
        return obj;
    },
    create(base) {
        return UpdateGroupMembershipData_V1_MembershipUpdatesEntry.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseUpdateGroupMembershipData_V1_MembershipUpdatesEntry();
        message.key = object.key ?? "";
        message.value = (object.value !== undefined && object.value !== null) ? Long.fromValue(object.value) : Long.UZERO;
        return message;
    },
};
function createBaseReaddInstallationsData() {
    return { v1: undefined };
}
export const ReaddInstallationsData = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.v1 !== undefined) {
            ReaddInstallationsData_V1.encode(message.v1, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseReaddInstallationsData();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.v1 = ReaddInstallationsData_V1.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { v1: isSet(object.v1) ? ReaddInstallationsData_V1.fromJSON(object.v1) : undefined };
    },
    toJSON(message) {
        const obj = {};
        if (message.v1 !== undefined) {
            obj.v1 = ReaddInstallationsData_V1.toJSON(message.v1);
        }
        return obj;
    },
    create(base) {
        return ReaddInstallationsData.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseReaddInstallationsData();
        message.v1 = (object.v1 !== undefined && object.v1 !== null)
            ? ReaddInstallationsData_V1.fromPartial(object.v1)
            : undefined;
        return message;
    },
};
function createBaseReaddInstallationsData_V1() {
    return { readdedInstallations: [] };
}
export const ReaddInstallationsData_V1 = {
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.readdedInstallations) {
            writer.uint32(10).bytes(v);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseReaddInstallationsData_V1();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.readdedInstallations.push(reader.bytes());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            readdedInstallations: globalThis.Array.isArray(object?.readdedInstallations)
                ? object.readdedInstallations.map((e) => bytesFromBase64(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.readdedInstallations?.length) {
            obj.readdedInstallations = message.readdedInstallations.map((e) => base64FromBytes(e));
        }
        return obj;
    },
    create(base) {
        return ReaddInstallationsData_V1.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseReaddInstallationsData_V1();
        message.readdedInstallations = object.readdedInstallations?.map((e) => e) || [];
        return message;
    },
};
function createBaseUpdateMetadataData() {
    return { v1: undefined };
}
export const UpdateMetadataData = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.v1 !== undefined) {
            UpdateMetadataData_V1.encode(message.v1, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseUpdateMetadataData();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.v1 = UpdateMetadataData_V1.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { v1: isSet(object.v1) ? UpdateMetadataData_V1.fromJSON(object.v1) : undefined };
    },
    toJSON(message) {
        const obj = {};
        if (message.v1 !== undefined) {
            obj.v1 = UpdateMetadataData_V1.toJSON(message.v1);
        }
        return obj;
    },
    create(base) {
        return UpdateMetadataData.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseUpdateMetadataData();
        message.v1 = (object.v1 !== undefined && object.v1 !== null)
            ? UpdateMetadataData_V1.fromPartial(object.v1)
            : undefined;
        return message;
    },
};
function createBaseUpdateMetadataData_V1() {
    return { fieldName: "", fieldValue: "" };
}
export const UpdateMetadataData_V1 = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.fieldName !== "") {
            writer.uint32(10).string(message.fieldName);
        }
        if (message.fieldValue !== "") {
            writer.uint32(18).string(message.fieldValue);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseUpdateMetadataData_V1();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.fieldName = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.fieldValue = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            fieldName: isSet(object.fieldName) ? globalThis.String(object.fieldName) : "",
            fieldValue: isSet(object.fieldValue) ? globalThis.String(object.fieldValue) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.fieldName !== "") {
            obj.fieldName = message.fieldName;
        }
        if (message.fieldValue !== "") {
            obj.fieldValue = message.fieldValue;
        }
        return obj;
    },
    create(base) {
        return UpdateMetadataData_V1.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseUpdateMetadataData_V1();
        message.fieldName = object.fieldName ?? "";
        message.fieldValue = object.fieldValue ?? "";
        return message;
    },
};
function createBaseUpdateAdminListsData() {
    return { v1: undefined };
}
export const UpdateAdminListsData = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.v1 !== undefined) {
            UpdateAdminListsData_V1.encode(message.v1, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseUpdateAdminListsData();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.v1 = UpdateAdminListsData_V1.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { v1: isSet(object.v1) ? UpdateAdminListsData_V1.fromJSON(object.v1) : undefined };
    },
    toJSON(message) {
        const obj = {};
        if (message.v1 !== undefined) {
            obj.v1 = UpdateAdminListsData_V1.toJSON(message.v1);
        }
        return obj;
    },
    create(base) {
        return UpdateAdminListsData.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseUpdateAdminListsData();
        message.v1 = (object.v1 !== undefined && object.v1 !== null)
            ? UpdateAdminListsData_V1.fromPartial(object.v1)
            : undefined;
        return message;
    },
};
function createBaseUpdateAdminListsData_V1() {
    return { adminListUpdateType: 0, inboxId: "" };
}
export const UpdateAdminListsData_V1 = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.adminListUpdateType !== 0) {
            writer.uint32(8).int32(message.adminListUpdateType);
        }
        if (message.inboxId !== "") {
            writer.uint32(18).string(message.inboxId);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseUpdateAdminListsData_V1();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 8) {
                        break;
                    }
                    message.adminListUpdateType = reader.int32();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.inboxId = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            adminListUpdateType: isSet(object.adminListUpdateType)
                ? adminListUpdateTypeFromJSON(object.adminListUpdateType)
                : 0,
            inboxId: isSet(object.inboxId) ? globalThis.String(object.inboxId) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.adminListUpdateType !== 0) {
            obj.adminListUpdateType = adminListUpdateTypeToJSON(message.adminListUpdateType);
        }
        if (message.inboxId !== "") {
            obj.inboxId = message.inboxId;
        }
        return obj;
    },
    create(base) {
        return UpdateAdminListsData_V1.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseUpdateAdminListsData_V1();
        message.adminListUpdateType = object.adminListUpdateType ?? 0;
        message.inboxId = object.inboxId ?? "";
        return message;
    },
};
function createBaseUpdatePermissionData() {
    return { v1: undefined };
}
export const UpdatePermissionData = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.v1 !== undefined) {
            UpdatePermissionData_V1.encode(message.v1, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseUpdatePermissionData();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.v1 = UpdatePermissionData_V1.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { v1: isSet(object.v1) ? UpdatePermissionData_V1.fromJSON(object.v1) : undefined };
    },
    toJSON(message) {
        const obj = {};
        if (message.v1 !== undefined) {
            obj.v1 = UpdatePermissionData_V1.toJSON(message.v1);
        }
        return obj;
    },
    create(base) {
        return UpdatePermissionData.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseUpdatePermissionData();
        message.v1 = (object.v1 !== undefined && object.v1 !== null)
            ? UpdatePermissionData_V1.fromPartial(object.v1)
            : undefined;
        return message;
    },
};
function createBaseUpdatePermissionData_V1() {
    return { permissionUpdateType: 0, permissionPolicyOption: 0, metadataFieldName: undefined };
}
export const UpdatePermissionData_V1 = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.permissionUpdateType !== 0) {
            writer.uint32(8).int32(message.permissionUpdateType);
        }
        if (message.permissionPolicyOption !== 0) {
            writer.uint32(16).int32(message.permissionPolicyOption);
        }
        if (message.metadataFieldName !== undefined) {
            writer.uint32(26).string(message.metadataFieldName);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseUpdatePermissionData_V1();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 8) {
                        break;
                    }
                    message.permissionUpdateType = reader.int32();
                    continue;
                case 2:
                    if (tag !== 16) {
                        break;
                    }
                    message.permissionPolicyOption = reader.int32();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.metadataFieldName = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            permissionUpdateType: isSet(object.permissionUpdateType)
                ? permissionUpdateTypeFromJSON(object.permissionUpdateType)
                : 0,
            permissionPolicyOption: isSet(object.permissionPolicyOption)
                ? permissionPolicyOptionFromJSON(object.permissionPolicyOption)
                : 0,
            metadataFieldName: isSet(object.metadataFieldName) ? globalThis.String(object.metadataFieldName) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.permissionUpdateType !== 0) {
            obj.permissionUpdateType = permissionUpdateTypeToJSON(message.permissionUpdateType);
        }
        if (message.permissionPolicyOption !== 0) {
            obj.permissionPolicyOption = permissionPolicyOptionToJSON(message.permissionPolicyOption);
        }
        if (message.metadataFieldName !== undefined) {
            obj.metadataFieldName = message.metadataFieldName;
        }
        return obj;
    },
    create(base) {
        return UpdatePermissionData_V1.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseUpdatePermissionData_V1();
        message.permissionUpdateType = object.permissionUpdateType ?? 0;
        message.permissionPolicyOption = object.permissionPolicyOption ?? 0;
        message.metadataFieldName = object.metadataFieldName ?? undefined;
        return message;
    },
};
function createBasePostCommitAction() {
    return { sendWelcomes: undefined };
}
export const PostCommitAction = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.sendWelcomes !== undefined) {
            PostCommitAction_SendWelcomes.encode(message.sendWelcomes, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBasePostCommitAction();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.sendWelcomes = PostCommitAction_SendWelcomes.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            sendWelcomes: isSet(object.sendWelcomes)
                ? PostCommitAction_SendWelcomes.fromJSON(object.sendWelcomes)
                : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.sendWelcomes !== undefined) {
            obj.sendWelcomes = PostCommitAction_SendWelcomes.toJSON(message.sendWelcomes);
        }
        return obj;
    },
    create(base) {
        return PostCommitAction.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBasePostCommitAction();
        message.sendWelcomes = (object.sendWelcomes !== undefined && object.sendWelcomes !== null)
            ? PostCommitAction_SendWelcomes.fromPartial(object.sendWelcomes)
            : undefined;
        return message;
    },
};
function createBasePostCommitAction_Installation() {
    return {
        installationKey: new Uint8Array(0),
        hpkePublicKey: new Uint8Array(0),
        welcomeWrapperAlgorithm: 0,
        welcomePointeeEncryptionAeadTypes: undefined,
    };
}
export const PostCommitAction_Installation = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.installationKey.length !== 0) {
            writer.uint32(10).bytes(message.installationKey);
        }
        if (message.hpkePublicKey.length !== 0) {
            writer.uint32(18).bytes(message.hpkePublicKey);
        }
        if (message.welcomeWrapperAlgorithm !== 0) {
            writer.uint32(24).int32(message.welcomeWrapperAlgorithm);
        }
        if (message.welcomePointeeEncryptionAeadTypes !== undefined) {
            WelcomePointeeEncryptionAeadTypesExtension.encode(message.welcomePointeeEncryptionAeadTypes, writer.uint32(34).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBasePostCommitAction_Installation();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.installationKey = reader.bytes();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.hpkePublicKey = reader.bytes();
                    continue;
                case 3:
                    if (tag !== 24) {
                        break;
                    }
                    message.welcomeWrapperAlgorithm = reader.int32();
                    continue;
                case 4:
                    if (tag !== 34) {
                        break;
                    }
                    message.welcomePointeeEncryptionAeadTypes = WelcomePointeeEncryptionAeadTypesExtension.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            installationKey: isSet(object.installationKey) ? bytesFromBase64(object.installationKey) : new Uint8Array(0),
            hpkePublicKey: isSet(object.hpkePublicKey) ? bytesFromBase64(object.hpkePublicKey) : new Uint8Array(0),
            welcomeWrapperAlgorithm: isSet(object.welcomeWrapperAlgorithm)
                ? welcomeWrapperAlgorithmFromJSON(object.welcomeWrapperAlgorithm)
                : 0,
            welcomePointeeEncryptionAeadTypes: isSet(object.welcomePointeeEncryptionAeadTypes)
                ? WelcomePointeeEncryptionAeadTypesExtension.fromJSON(object.welcomePointeeEncryptionAeadTypes)
                : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.installationKey.length !== 0) {
            obj.installationKey = base64FromBytes(message.installationKey);
        }
        if (message.hpkePublicKey.length !== 0) {
            obj.hpkePublicKey = base64FromBytes(message.hpkePublicKey);
        }
        if (message.welcomeWrapperAlgorithm !== 0) {
            obj.welcomeWrapperAlgorithm = welcomeWrapperAlgorithmToJSON(message.welcomeWrapperAlgorithm);
        }
        if (message.welcomePointeeEncryptionAeadTypes !== undefined) {
            obj.welcomePointeeEncryptionAeadTypes = WelcomePointeeEncryptionAeadTypesExtension.toJSON(message.welcomePointeeEncryptionAeadTypes);
        }
        return obj;
    },
    create(base) {
        return PostCommitAction_Installation.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBasePostCommitAction_Installation();
        message.installationKey = object.installationKey ?? new Uint8Array(0);
        message.hpkePublicKey = object.hpkePublicKey ?? new Uint8Array(0);
        message.welcomeWrapperAlgorithm = object.welcomeWrapperAlgorithm ?? 0;
        message.welcomePointeeEncryptionAeadTypes =
            (object.welcomePointeeEncryptionAeadTypes !== undefined && object.welcomePointeeEncryptionAeadTypes !== null)
                ? WelcomePointeeEncryptionAeadTypesExtension.fromPartial(object.welcomePointeeEncryptionAeadTypes)
                : undefined;
        return message;
    },
};
function createBasePostCommitAction_SendWelcomes() {
    return { installations: [], welcomeMessage: new Uint8Array(0) };
}
export const PostCommitAction_SendWelcomes = {
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.installations) {
            PostCommitAction_Installation.encode(v, writer.uint32(10).fork()).ldelim();
        }
        if (message.welcomeMessage.length !== 0) {
            writer.uint32(18).bytes(message.welcomeMessage);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBasePostCommitAction_SendWelcomes();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.installations.push(PostCommitAction_Installation.decode(reader, reader.uint32()));
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.welcomeMessage = reader.bytes();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            installations: globalThis.Array.isArray(object?.installations)
                ? object.installations.map((e) => PostCommitAction_Installation.fromJSON(e))
                : [],
            welcomeMessage: isSet(object.welcomeMessage) ? bytesFromBase64(object.welcomeMessage) : new Uint8Array(0),
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.installations?.length) {
            obj.installations = message.installations.map((e) => PostCommitAction_Installation.toJSON(e));
        }
        if (message.welcomeMessage.length !== 0) {
            obj.welcomeMessage = base64FromBytes(message.welcomeMessage);
        }
        return obj;
    },
    create(base) {
        return PostCommitAction_SendWelcomes.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBasePostCommitAction_SendWelcomes();
        message.installations = object.installations?.map((e) => PostCommitAction_Installation.fromPartial(e)) || [];
        message.welcomeMessage = object.welcomeMessage ?? new Uint8Array(0);
        return message;
    },
};
function createBaseProposeMemberUpdateData() {
    return { v1: undefined };
}
export const ProposeMemberUpdateData = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.v1 !== undefined) {
            ProposeMemberUpdateData_V1.encode(message.v1, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseProposeMemberUpdateData();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.v1 = ProposeMemberUpdateData_V1.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { v1: isSet(object.v1) ? ProposeMemberUpdateData_V1.fromJSON(object.v1) : undefined };
    },
    toJSON(message) {
        const obj = {};
        if (message.v1 !== undefined) {
            obj.v1 = ProposeMemberUpdateData_V1.toJSON(message.v1);
        }
        return obj;
    },
    create(base) {
        return ProposeMemberUpdateData.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseProposeMemberUpdateData();
        message.v1 = (object.v1 !== undefined && object.v1 !== null)
            ? ProposeMemberUpdateData_V1.fromPartial(object.v1)
            : undefined;
        return message;
    },
};
function createBaseProposeMemberUpdateData_V1() {
    return { addInboxIds: [], removeInboxIds: [] };
}
export const ProposeMemberUpdateData_V1 = {
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.addInboxIds) {
            writer.uint32(10).bytes(v);
        }
        for (const v of message.removeInboxIds) {
            writer.uint32(18).bytes(v);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseProposeMemberUpdateData_V1();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.addInboxIds.push(reader.bytes());
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.removeInboxIds.push(reader.bytes());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            addInboxIds: globalThis.Array.isArray(object?.addInboxIds)
                ? object.addInboxIds.map((e) => bytesFromBase64(e))
                : [],
            removeInboxIds: globalThis.Array.isArray(object?.removeInboxIds)
                ? object.removeInboxIds.map((e) => bytesFromBase64(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.addInboxIds?.length) {
            obj.addInboxIds = message.addInboxIds.map((e) => base64FromBytes(e));
        }
        if (message.removeInboxIds?.length) {
            obj.removeInboxIds = message.removeInboxIds.map((e) => base64FromBytes(e));
        }
        return obj;
    },
    create(base) {
        return ProposeMemberUpdateData_V1.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseProposeMemberUpdateData_V1();
        message.addInboxIds = object.addInboxIds?.map((e) => e) || [];
        message.removeInboxIds = object.removeInboxIds?.map((e) => e) || [];
        return message;
    },
};
function createBaseProposeGroupContextExtensionData() {
    return { v1: undefined };
}
export const ProposeGroupContextExtensionData = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.v1 !== undefined) {
            ProposeGroupContextExtensionData_V1.encode(message.v1, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseProposeGroupContextExtensionData();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.v1 = ProposeGroupContextExtensionData_V1.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { v1: isSet(object.v1) ? ProposeGroupContextExtensionData_V1.fromJSON(object.v1) : undefined };
    },
    toJSON(message) {
        const obj = {};
        if (message.v1 !== undefined) {
            obj.v1 = ProposeGroupContextExtensionData_V1.toJSON(message.v1);
        }
        return obj;
    },
    create(base) {
        return ProposeGroupContextExtensionData.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseProposeGroupContextExtensionData();
        message.v1 = (object.v1 !== undefined && object.v1 !== null)
            ? ProposeGroupContextExtensionData_V1.fromPartial(object.v1)
            : undefined;
        return message;
    },
};
function createBaseProposeGroupContextExtensionData_V1() {
    return { groupContextExtension: new Uint8Array(0) };
}
export const ProposeGroupContextExtensionData_V1 = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.groupContextExtension.length !== 0) {
            writer.uint32(10).bytes(message.groupContextExtension);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseProposeGroupContextExtensionData_V1();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.groupContextExtension = reader.bytes();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            groupContextExtension: isSet(object.groupContextExtension)
                ? bytesFromBase64(object.groupContextExtension)
                : new Uint8Array(0),
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.groupContextExtension.length !== 0) {
            obj.groupContextExtension = base64FromBytes(message.groupContextExtension);
        }
        return obj;
    },
    create(base) {
        return ProposeGroupContextExtensionData_V1.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseProposeGroupContextExtensionData_V1();
        message.groupContextExtension = object.groupContextExtension ?? new Uint8Array(0);
        return message;
    },
};
function createBaseCommitPendingProposalsData() {
    return { v1: undefined };
}
export const CommitPendingProposalsData = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.v1 !== undefined) {
            CommitPendingProposalsData_V1.encode(message.v1, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseCommitPendingProposalsData();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.v1 = CommitPendingProposalsData_V1.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { v1: isSet(object.v1) ? CommitPendingProposalsData_V1.fromJSON(object.v1) : undefined };
    },
    toJSON(message) {
        const obj = {};
        if (message.v1 !== undefined) {
            obj.v1 = CommitPendingProposalsData_V1.toJSON(message.v1);
        }
        return obj;
    },
    create(base) {
        return CommitPendingProposalsData.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseCommitPendingProposalsData();
        message.v1 = (object.v1 !== undefined && object.v1 !== null)
            ? CommitPendingProposalsData_V1.fromPartial(object.v1)
            : undefined;
        return message;
    },
};
function createBaseCommitPendingProposalsData_V1() {
    return { proposalHashes: [] };
}
export const CommitPendingProposalsData_V1 = {
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.proposalHashes) {
            writer.uint32(10).bytes(v);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseCommitPendingProposalsData_V1();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.proposalHashes.push(reader.bytes());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            proposalHashes: globalThis.Array.isArray(object?.proposalHashes)
                ? object.proposalHashes.map((e) => bytesFromBase64(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.proposalHashes?.length) {
            obj.proposalHashes = message.proposalHashes.map((e) => base64FromBytes(e));
        }
        return obj;
    },
    create(base) {
        return CommitPendingProposalsData_V1.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseCommitPendingProposalsData_V1();
        message.proposalHashes = object.proposalHashes?.map((e) => e) || [];
        return message;
    },
};
function bytesFromBase64(b64) {
    if (globalThis.Buffer) {
        return Uint8Array.from(globalThis.Buffer.from(b64, "base64"));
    }
    else {
        const bin = globalThis.atob(b64);
        const arr = new Uint8Array(bin.length);
        for (let i = 0; i < bin.length; ++i) {
            arr[i] = bin.charCodeAt(i);
        }
        return arr;
    }
}
function base64FromBytes(arr) {
    if (globalThis.Buffer) {
        return globalThis.Buffer.from(arr).toString("base64");
    }
    else {
        const bin = [];
        arr.forEach((byte) => {
            bin.push(globalThis.String.fromCharCode(byte));
        });
        return globalThis.btoa(bin.join(""));
    }
}
if (_m0.util.Long !== Long) {
    _m0.util.Long = Long;
    _m0.configure();
}
function isObject(value) {
    return typeof value === "object" && value !== null;
}
function isSet(value) {
    return value !== null && value !== undefined;
}
