// @xmtp/proto/ts/dist/esm/message_contents/composite.pb.js
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v1.181.2
//   protoc               v3.21.12
// source: message_contents/composite.proto
/* eslint-disable */
import Long from "long";
import _m0 from "protobufjs/minimal";
import { EncodedContent } from "./content.pb";
export const protobufPackage = "xmtp.message_contents";
function createBaseComposite() {
    return { parts: [] };
}
export const Composite = {
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.parts) {
            Composite_Part.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseComposite();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.parts.push(Composite_Part.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            parts: globalThis.Array.isArray(object?.parts) ? object.parts.map((e) => Composite_Part.fromJSON(e)) : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.parts?.length) {
            obj.parts = message.parts.map((e) => Composite_Part.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return Composite.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseComposite();
        message.parts = object.parts?.map((e) => Composite_Part.fromPartial(e)) || [];
        return message;
    },
};
function createBaseComposite_Part() {
    return { part: undefined, composite: undefined };
}
export const Composite_Part = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.part !== undefined) {
            EncodedContent.encode(message.part, writer.uint32(10).fork()).ldelim();
        }
        if (message.composite !== undefined) {
            Composite.encode(message.composite, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseComposite_Part();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.part = EncodedContent.decode(reader, reader.uint32());
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.composite = Composite.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            part: isSet(object.part) ? EncodedContent.fromJSON(object.part) : undefined,
            composite: isSet(object.composite) ? Composite.fromJSON(object.composite) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.part !== undefined) {
            obj.part = EncodedContent.toJSON(message.part);
        }
        if (message.composite !== undefined) {
            obj.composite = Composite.toJSON(message.composite);
        }
        return obj;
    },
    create(base) {
        return Composite_Part.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseComposite_Part();
        message.part = (object.part !== undefined && object.part !== null)
            ? EncodedContent.fromPartial(object.part)
            : undefined;
        message.composite = (object.composite !== undefined && object.composite !== null)
            ? Composite.fromPartial(object.composite)
            : undefined;
        return message;
    },
};
if (_m0.util.Long !== Long) {
    _m0.util.Long = Long;
    _m0.configure();
}
function isSet(value) {
    return value !== null && value !== undefined;
}
