// @xmtp/proto/ts/dist/esm/message_contents/conversation_reference.pb.js
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v1.181.2
//   protoc               v3.21.12
// source: message_contents/conversation_reference.proto
/* eslint-disable */
import Long from "long";
import _m0 from "protobufjs/minimal";
import { ConsentProofPayload, InvitationV1_Context } from "./invitation.pb";
export const protobufPackage = "xmtp.message_contents";
function createBaseConversationReference() {
    return { topic: "", peerAddress: "", createdNs: Long.UZERO, context: undefined, consentProofPayload: undefined };
}
export const ConversationReference = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.topic !== "") {
            writer.uint32(10).string(message.topic);
        }
        if (message.peerAddress !== "") {
            writer.uint32(18).string(message.peerAddress);
        }
        if (!message.createdNs.equals(Long.UZERO)) {
            writer.uint32(24).uint64(message.createdNs);
        }
        if (message.context !== undefined) {
            InvitationV1_Context.encode(message.context, writer.uint32(34).fork()).ldelim();
        }
        if (message.consentProofPayload !== undefined) {
            ConsentProofPayload.encode(message.consentProofPayload, writer.uint32(42).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseConversationReference();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.topic = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.peerAddress = reader.string();
                    continue;
                case 3:
                    if (tag !== 24) {
                        break;
                    }
                    message.createdNs = reader.uint64();
                    continue;
                case 4:
                    if (tag !== 34) {
                        break;
                    }
                    message.context = InvitationV1_Context.decode(reader, reader.uint32());
                    continue;
                case 5:
                    if (tag !== 42) {
                        break;
                    }
                    message.consentProofPayload = ConsentProofPayload.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            topic: isSet(object.topic) ? globalThis.String(object.topic) : "",
            peerAddress: isSet(object.peerAddress) ? globalThis.String(object.peerAddress) : "",
            createdNs: isSet(object.createdNs) ? Long.fromValue(object.createdNs) : Long.UZERO,
            context: isSet(object.context) ? InvitationV1_Context.fromJSON(object.context) : undefined,
            consentProofPayload: isSet(object.consentProofPayload)
                ? ConsentProofPayload.fromJSON(object.consentProofPayload)
                : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.topic !== "") {
            obj.topic = message.topic;
        }
        if (message.peerAddress !== "") {
            obj.peerAddress = message.peerAddress;
        }
        if (!message.createdNs.equals(Long.UZERO)) {
            obj.createdNs = (message.createdNs || Long.UZERO).toString();
        }
        if (message.context !== undefined) {
            obj.context = InvitationV1_Context.toJSON(message.context);
        }
        if (message.consentProofPayload !== undefined) {
            obj.consentProofPayload = ConsentProofPayload.toJSON(message.consentProofPayload);
        }
        return obj;
    },
    create(base) {
        return ConversationReference.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseConversationReference();
        message.topic = object.topic ?? "";
        message.peerAddress = object.peerAddress ?? "";
        message.createdNs = (object.createdNs !== undefined && object.createdNs !== null)
            ? Long.fromValue(object.createdNs)
            : Long.UZERO;
        message.context = (object.context !== undefined && object.context !== null)
            ? InvitationV1_Context.fromPartial(object.context)
            : undefined;
        message.consentProofPayload = (object.consentProofPayload !== undefined && object.consentProofPayload !== null)
            ? ConsentProofPayload.fromPartial(object.consentProofPayload)
            : undefined;
        return message;
    },
};
if (_m0.util.Long !== Long) {
    _m0.util.Long = Long;
    _m0.configure();
}
function isSet(value) {
    return value !== null && value !== undefined;
}
