// @xmtp/proto/ts/dist/esm/message_contents/private_key.pb.js
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v1.181.2
//   protoc               v3.21.12
// source: message_contents/private_key.proto
/* eslint-disable */
import Long from "long";
import _m0 from "protobufjs/minimal";
import { Ciphertext } from "./ciphertext.pb";
import { PublicKey, SignedPublicKey } from "./public_key.pb";
export const protobufPackage = "xmtp.message_contents";
function createBaseSignedPrivateKey() {
    return { createdNs: Long.UZERO, secp256k1: undefined, publicKey: undefined };
}
export const SignedPrivateKey = {
    encode(message, writer = _m0.Writer.create()) {
        if (!message.createdNs.equals(Long.UZERO)) {
            writer.uint32(8).uint64(message.createdNs);
        }
        if (message.secp256k1 !== undefined) {
            SignedPrivateKey_Secp256k1.encode(message.secp256k1, writer.uint32(18).fork()).ldelim();
        }
        if (message.publicKey !== undefined) {
            SignedPublicKey.encode(message.publicKey, writer.uint32(26).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSignedPrivateKey();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 8) {
                        break;
                    }
                    message.createdNs = reader.uint64();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.secp256k1 = SignedPrivateKey_Secp256k1.decode(reader, reader.uint32());
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.publicKey = SignedPublicKey.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            createdNs: isSet(object.createdNs) ? Long.fromValue(object.createdNs) : Long.UZERO,
            secp256k1: isSet(object.secp256k1) ? SignedPrivateKey_Secp256k1.fromJSON(object.secp256k1) : undefined,
            publicKey: isSet(object.publicKey) ? SignedPublicKey.fromJSON(object.publicKey) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (!message.createdNs.equals(Long.UZERO)) {
            obj.createdNs = (message.createdNs || Long.UZERO).toString();
        }
        if (message.secp256k1 !== undefined) {
            obj.secp256k1 = SignedPrivateKey_Secp256k1.toJSON(message.secp256k1);
        }
        if (message.publicKey !== undefined) {
            obj.publicKey = SignedPublicKey.toJSON(message.publicKey);
        }
        return obj;
    },
    create(base) {
        return SignedPrivateKey.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseSignedPrivateKey();
        message.createdNs = (object.createdNs !== undefined && object.createdNs !== null)
            ? Long.fromValue(object.createdNs)
            : Long.UZERO;
        message.secp256k1 = (object.secp256k1 !== undefined && object.secp256k1 !== null)
            ? SignedPrivateKey_Secp256k1.fromPartial(object.secp256k1)
            : undefined;
        message.publicKey = (object.publicKey !== undefined && object.publicKey !== null)
            ? SignedPublicKey.fromPartial(object.publicKey)
            : undefined;
        return message;
    },
};
function createBaseSignedPrivateKey_Secp256k1() {
    return { bytes: new Uint8Array(0) };
}
export const SignedPrivateKey_Secp256k1 = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.bytes.length !== 0) {
            writer.uint32(10).bytes(message.bytes);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSignedPrivateKey_Secp256k1();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.bytes = reader.bytes();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { bytes: isSet(object.bytes) ? bytesFromBase64(object.bytes) : new Uint8Array(0) };
    },
    toJSON(message) {
        const obj = {};
        if (message.bytes.length !== 0) {
            obj.bytes = base64FromBytes(message.bytes);
        }
        return obj;
    },
    create(base) {
        return SignedPrivateKey_Secp256k1.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseSignedPrivateKey_Secp256k1();
        message.bytes = object.bytes ?? new Uint8Array(0);
        return message;
    },
};
function createBasePrivateKeyBundleV2() {
    return { identityKey: undefined, preKeys: [] };
}
export const PrivateKeyBundleV2 = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.identityKey !== undefined) {
            SignedPrivateKey.encode(message.identityKey, writer.uint32(10).fork()).ldelim();
        }
        for (const v of message.preKeys) {
            SignedPrivateKey.encode(v, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBasePrivateKeyBundleV2();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.identityKey = SignedPrivateKey.decode(reader, reader.uint32());
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.preKeys.push(SignedPrivateKey.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            identityKey: isSet(object.identityKey) ? SignedPrivateKey.fromJSON(object.identityKey) : undefined,
            preKeys: globalThis.Array.isArray(object?.preKeys)
                ? object.preKeys.map((e) => SignedPrivateKey.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.identityKey !== undefined) {
            obj.identityKey = SignedPrivateKey.toJSON(message.identityKey);
        }
        if (message.preKeys?.length) {
            obj.preKeys = message.preKeys.map((e) => SignedPrivateKey.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return PrivateKeyBundleV2.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBasePrivateKeyBundleV2();
        message.identityKey = (object.identityKey !== undefined && object.identityKey !== null)
            ? SignedPrivateKey.fromPartial(object.identityKey)
            : undefined;
        message.preKeys = object.preKeys?.map((e) => SignedPrivateKey.fromPartial(e)) || [];
        return message;
    },
};
function createBasePrivateKey() {
    return { timestamp: Long.UZERO, secp256k1: undefined, publicKey: undefined };
}
export const PrivateKey = {
    encode(message, writer = _m0.Writer.create()) {
        if (!message.timestamp.equals(Long.UZERO)) {
            writer.uint32(8).uint64(message.timestamp);
        }
        if (message.secp256k1 !== undefined) {
            PrivateKey_Secp256k1.encode(message.secp256k1, writer.uint32(18).fork()).ldelim();
        }
        if (message.publicKey !== undefined) {
            PublicKey.encode(message.publicKey, writer.uint32(26).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBasePrivateKey();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 8) {
                        break;
                    }
                    message.timestamp = reader.uint64();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.secp256k1 = PrivateKey_Secp256k1.decode(reader, reader.uint32());
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.publicKey = PublicKey.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            timestamp: isSet(object.timestamp) ? Long.fromValue(object.timestamp) : Long.UZERO,
            secp256k1: isSet(object.secp256k1) ? PrivateKey_Secp256k1.fromJSON(object.secp256k1) : undefined,
            publicKey: isSet(object.publicKey) ? PublicKey.fromJSON(object.publicKey) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (!message.timestamp.equals(Long.UZERO)) {
            obj.timestamp = (message.timestamp || Long.UZERO).toString();
        }
        if (message.secp256k1 !== undefined) {
            obj.secp256k1 = PrivateKey_Secp256k1.toJSON(message.secp256k1);
        }
        if (message.publicKey !== undefined) {
            obj.publicKey = PublicKey.toJSON(message.publicKey);
        }
        return obj;
    },
    create(base) {
        return PrivateKey.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBasePrivateKey();
        message.timestamp = (object.timestamp !== undefined && object.timestamp !== null)
            ? Long.fromValue(object.timestamp)
            : Long.UZERO;
        message.secp256k1 = (object.secp256k1 !== undefined && object.secp256k1 !== null)
            ? PrivateKey_Secp256k1.fromPartial(object.secp256k1)
            : undefined;
        message.publicKey = (object.publicKey !== undefined && object.publicKey !== null)
            ? PublicKey.fromPartial(object.publicKey)
            : undefined;
        return message;
    },
};
function createBasePrivateKey_Secp256k1() {
    return { bytes: new Uint8Array(0) };
}
export const PrivateKey_Secp256k1 = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.bytes.length !== 0) {
            writer.uint32(10).bytes(message.bytes);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBasePrivateKey_Secp256k1();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.bytes = reader.bytes();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { bytes: isSet(object.bytes) ? bytesFromBase64(object.bytes) : new Uint8Array(0) };
    },
    toJSON(message) {
        const obj = {};
        if (message.bytes.length !== 0) {
            obj.bytes = base64FromBytes(message.bytes);
        }
        return obj;
    },
    create(base) {
        return PrivateKey_Secp256k1.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBasePrivateKey_Secp256k1();
        message.bytes = object.bytes ?? new Uint8Array(0);
        return message;
    },
};
function createBasePrivateKeyBundleV1() {
    return { identityKey: undefined, preKeys: [] };
}
export const PrivateKeyBundleV1 = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.identityKey !== undefined) {
            PrivateKey.encode(message.identityKey, writer.uint32(10).fork()).ldelim();
        }
        for (const v of message.preKeys) {
            PrivateKey.encode(v, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBasePrivateKeyBundleV1();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.identityKey = PrivateKey.decode(reader, reader.uint32());
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.preKeys.push(PrivateKey.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            identityKey: isSet(object.identityKey) ? PrivateKey.fromJSON(object.identityKey) : undefined,
            preKeys: globalThis.Array.isArray(object?.preKeys) ? object.preKeys.map((e) => PrivateKey.fromJSON(e)) : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.identityKey !== undefined) {
            obj.identityKey = PrivateKey.toJSON(message.identityKey);
        }
        if (message.preKeys?.length) {
            obj.preKeys = message.preKeys.map((e) => PrivateKey.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return PrivateKeyBundleV1.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBasePrivateKeyBundleV1();
        message.identityKey = (object.identityKey !== undefined && object.identityKey !== null)
            ? PrivateKey.fromPartial(object.identityKey)
            : undefined;
        message.preKeys = object.preKeys?.map((e) => PrivateKey.fromPartial(e)) || [];
        return message;
    },
};
function createBasePrivateKeyBundle() {
    return { v1: undefined, v2: undefined };
}
export const PrivateKeyBundle = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.v1 !== undefined) {
            PrivateKeyBundleV1.encode(message.v1, writer.uint32(10).fork()).ldelim();
        }
        if (message.v2 !== undefined) {
            PrivateKeyBundleV2.encode(message.v2, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBasePrivateKeyBundle();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.v1 = PrivateKeyBundleV1.decode(reader, reader.uint32());
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.v2 = PrivateKeyBundleV2.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            v1: isSet(object.v1) ? PrivateKeyBundleV1.fromJSON(object.v1) : undefined,
            v2: isSet(object.v2) ? PrivateKeyBundleV2.fromJSON(object.v2) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.v1 !== undefined) {
            obj.v1 = PrivateKeyBundleV1.toJSON(message.v1);
        }
        if (message.v2 !== undefined) {
            obj.v2 = PrivateKeyBundleV2.toJSON(message.v2);
        }
        return obj;
    },
    create(base) {
        return PrivateKeyBundle.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBasePrivateKeyBundle();
        message.v1 = (object.v1 !== undefined && object.v1 !== null)
            ? PrivateKeyBundleV1.fromPartial(object.v1)
            : undefined;
        message.v2 = (object.v2 !== undefined && object.v2 !== null)
            ? PrivateKeyBundleV2.fromPartial(object.v2)
            : undefined;
        return message;
    },
};
function createBaseEncryptedPrivateKeyBundleV1() {
    return { walletPreKey: new Uint8Array(0), ciphertext: undefined };
}
export const EncryptedPrivateKeyBundleV1 = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.walletPreKey.length !== 0) {
            writer.uint32(10).bytes(message.walletPreKey);
        }
        if (message.ciphertext !== undefined) {
            Ciphertext.encode(message.ciphertext, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseEncryptedPrivateKeyBundleV1();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.walletPreKey = reader.bytes();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.ciphertext = Ciphertext.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            walletPreKey: isSet(object.walletPreKey) ? bytesFromBase64(object.walletPreKey) : new Uint8Array(0),
            ciphertext: isSet(object.ciphertext) ? Ciphertext.fromJSON(object.ciphertext) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.walletPreKey.length !== 0) {
            obj.walletPreKey = base64FromBytes(message.walletPreKey);
        }
        if (message.ciphertext !== undefined) {
            obj.ciphertext = Ciphertext.toJSON(message.ciphertext);
        }
        return obj;
    },
    create(base) {
        return EncryptedPrivateKeyBundleV1.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseEncryptedPrivateKeyBundleV1();
        message.walletPreKey = object.walletPreKey ?? new Uint8Array(0);
        message.ciphertext = (object.ciphertext !== undefined && object.ciphertext !== null)
            ? Ciphertext.fromPartial(object.ciphertext)
            : undefined;
        return message;
    },
};
function createBaseEncryptedPrivateKeyBundle() {
    return { v1: undefined };
}
export const EncryptedPrivateKeyBundle = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.v1 !== undefined) {
            EncryptedPrivateKeyBundleV1.encode(message.v1, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseEncryptedPrivateKeyBundle();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.v1 = EncryptedPrivateKeyBundleV1.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { v1: isSet(object.v1) ? EncryptedPrivateKeyBundleV1.fromJSON(object.v1) : undefined };
    },
    toJSON(message) {
        const obj = {};
        if (message.v1 !== undefined) {
            obj.v1 = EncryptedPrivateKeyBundleV1.toJSON(message.v1);
        }
        return obj;
    },
    create(base) {
        return EncryptedPrivateKeyBundle.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseEncryptedPrivateKeyBundle();
        message.v1 = (object.v1 !== undefined && object.v1 !== null)
            ? EncryptedPrivateKeyBundleV1.fromPartial(object.v1)
            : undefined;
        return message;
    },
};
function bytesFromBase64(b64) {
    if (globalThis.Buffer) {
        return Uint8Array.from(globalThis.Buffer.from(b64, "base64"));
    }
    else {
        const bin = globalThis.atob(b64);
        const arr = new Uint8Array(bin.length);
        for (let i = 0; i < bin.length; ++i) {
            arr[i] = bin.charCodeAt(i);
        }
        return arr;
    }
}
function base64FromBytes(arr) {
    if (globalThis.Buffer) {
        return globalThis.Buffer.from(arr).toString("base64");
    }
    else {
        const bin = [];
        arr.forEach((byte) => {
            bin.push(globalThis.String.fromCharCode(byte));
        });
        return globalThis.btoa(bin.join(""));
    }
}
if (_m0.util.Long !== Long) {
    _m0.util.Long = Long;
    _m0.configure();
}
function isSet(value) {
    return value !== null && value !== undefined;
}
