// @xmtp/proto/ts/dist/esm/device_sync/consent_backup.pb.js
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v1.181.2
//   protoc               v3.21.12
// source: device_sync/consent_backup.proto
/* eslint-disable */
import Long from "long";
import _m0 from "protobufjs/minimal";
export const protobufPackage = "xmtp.device_sync.consent_backup";
/** Definitions for backups */
/** Consent record type */
export var ConsentTypeSave;
(function (ConsentTypeSave) {
    ConsentTypeSave[ConsentTypeSave["CONSENT_TYPE_SAVE_UNSPECIFIED"] = 0] = "CONSENT_TYPE_SAVE_UNSPECIFIED";
    ConsentTypeSave[ConsentTypeSave["CONSENT_TYPE_SAVE_CONVERSATION_ID"] = 1] = "CONSENT_TYPE_SAVE_CONVERSATION_ID";
    ConsentTypeSave[ConsentTypeSave["CONSENT_TYPE_SAVE_INBOX_ID"] = 2] = "CONSENT_TYPE_SAVE_INBOX_ID";
    /** @deprecated */
    ConsentTypeSave[ConsentTypeSave["CONSENT_TYPE_SAVE_ADDRESS"] = 3] = "CONSENT_TYPE_SAVE_ADDRESS";
    ConsentTypeSave[ConsentTypeSave["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(ConsentTypeSave || (ConsentTypeSave = {}));
export function consentTypeSaveFromJSON(object) {
    switch (object) {
        case 0:
        case "CONSENT_TYPE_SAVE_UNSPECIFIED":
            return ConsentTypeSave.CONSENT_TYPE_SAVE_UNSPECIFIED;
        case 1:
        case "CONSENT_TYPE_SAVE_CONVERSATION_ID":
            return ConsentTypeSave.CONSENT_TYPE_SAVE_CONVERSATION_ID;
        case 2:
        case "CONSENT_TYPE_SAVE_INBOX_ID":
            return ConsentTypeSave.CONSENT_TYPE_SAVE_INBOX_ID;
        case 3:
        case "CONSENT_TYPE_SAVE_ADDRESS":
            return ConsentTypeSave.CONSENT_TYPE_SAVE_ADDRESS;
        case -1:
        case "UNRECOGNIZED":
        default:
            return ConsentTypeSave.UNRECOGNIZED;
    }
}
export function consentTypeSaveToJSON(object) {
    switch (object) {
        case ConsentTypeSave.CONSENT_TYPE_SAVE_UNSPECIFIED:
            return "CONSENT_TYPE_SAVE_UNSPECIFIED";
        case ConsentTypeSave.CONSENT_TYPE_SAVE_CONVERSATION_ID:
            return "CONSENT_TYPE_SAVE_CONVERSATION_ID";
        case ConsentTypeSave.CONSENT_TYPE_SAVE_INBOX_ID:
            return "CONSENT_TYPE_SAVE_INBOX_ID";
        case ConsentTypeSave.CONSENT_TYPE_SAVE_ADDRESS:
            return "CONSENT_TYPE_SAVE_ADDRESS";
        case ConsentTypeSave.UNRECOGNIZED:
        default:
            return "UNRECOGNIZED";
    }
}
/** Consent record state */
export var ConsentStateSave;
(function (ConsentStateSave) {
    ConsentStateSave[ConsentStateSave["CONSENT_STATE_SAVE_UNSPECIFIED"] = 0] = "CONSENT_STATE_SAVE_UNSPECIFIED";
    ConsentStateSave[ConsentStateSave["CONSENT_STATE_SAVE_UNKNOWN"] = 1] = "CONSENT_STATE_SAVE_UNKNOWN";
    ConsentStateSave[ConsentStateSave["CONSENT_STATE_SAVE_ALLOWED"] = 2] = "CONSENT_STATE_SAVE_ALLOWED";
    ConsentStateSave[ConsentStateSave["CONSENT_STATE_SAVE_DENIED"] = 3] = "CONSENT_STATE_SAVE_DENIED";
    ConsentStateSave[ConsentStateSave["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(ConsentStateSave || (ConsentStateSave = {}));
export function consentStateSaveFromJSON(object) {
    switch (object) {
        case 0:
        case "CONSENT_STATE_SAVE_UNSPECIFIED":
            return ConsentStateSave.CONSENT_STATE_SAVE_UNSPECIFIED;
        case 1:
        case "CONSENT_STATE_SAVE_UNKNOWN":
            return ConsentStateSave.CONSENT_STATE_SAVE_UNKNOWN;
        case 2:
        case "CONSENT_STATE_SAVE_ALLOWED":
            return ConsentStateSave.CONSENT_STATE_SAVE_ALLOWED;
        case 3:
        case "CONSENT_STATE_SAVE_DENIED":
            return ConsentStateSave.CONSENT_STATE_SAVE_DENIED;
        case -1:
        case "UNRECOGNIZED":
        default:
            return ConsentStateSave.UNRECOGNIZED;
    }
}
export function consentStateSaveToJSON(object) {
    switch (object) {
        case ConsentStateSave.CONSENT_STATE_SAVE_UNSPECIFIED:
            return "CONSENT_STATE_SAVE_UNSPECIFIED";
        case ConsentStateSave.CONSENT_STATE_SAVE_UNKNOWN:
            return "CONSENT_STATE_SAVE_UNKNOWN";
        case ConsentStateSave.CONSENT_STATE_SAVE_ALLOWED:
            return "CONSENT_STATE_SAVE_ALLOWED";
        case ConsentStateSave.CONSENT_STATE_SAVE_DENIED:
            return "CONSENT_STATE_SAVE_DENIED";
        case ConsentStateSave.UNRECOGNIZED:
        default:
            return "UNRECOGNIZED";
    }
}
function createBaseConsentSave() {
    return { entityType: 0, state: 0, entity: "", consentedAtNs: Long.ZERO };
}
export const ConsentSave = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.entityType !== 0) {
            writer.uint32(8).int32(message.entityType);
        }
        if (message.state !== 0) {
            writer.uint32(16).int32(message.state);
        }
        if (message.entity !== "") {
            writer.uint32(26).string(message.entity);
        }
        if (!message.consentedAtNs.equals(Long.ZERO)) {
            writer.uint32(32).int64(message.consentedAtNs);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseConsentSave();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 8) {
                        break;
                    }
                    message.entityType = reader.int32();
                    continue;
                case 2:
                    if (tag !== 16) {
                        break;
                    }
                    message.state = reader.int32();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.entity = reader.string();
                    continue;
                case 4:
                    if (tag !== 32) {
                        break;
                    }
                    message.consentedAtNs = reader.int64();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            entityType: isSet(object.entityType) ? consentTypeSaveFromJSON(object.entityType) : 0,
            state: isSet(object.state) ? consentStateSaveFromJSON(object.state) : 0,
            entity: isSet(object.entity) ? globalThis.String(object.entity) : "",
            consentedAtNs: isSet(object.consentedAtNs) ? Long.fromValue(object.consentedAtNs) : Long.ZERO,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.entityType !== 0) {
            obj.entityType = consentTypeSaveToJSON(message.entityType);
        }
        if (message.state !== 0) {
            obj.state = consentStateSaveToJSON(message.state);
        }
        if (message.entity !== "") {
            obj.entity = message.entity;
        }
        if (!message.consentedAtNs.equals(Long.ZERO)) {
            obj.consentedAtNs = (message.consentedAtNs || Long.ZERO).toString();
        }
        return obj;
    },
    create(base) {
        return ConsentSave.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseConsentSave();
        message.entityType = object.entityType ?? 0;
        message.state = object.state ?? 0;
        message.entity = object.entity ?? "";
        message.consentedAtNs = (object.consentedAtNs !== undefined && object.consentedAtNs !== null)
            ? Long.fromValue(object.consentedAtNs)
            : Long.ZERO;
        return message;
    },
};
if (_m0.util.Long !== Long) {
    _m0.util.Long = Long;
    _m0.configure();
}
function isSet(value) {
    return value !== null && value !== undefined;
}
