// @xmtp/proto/ts/dist/esm/device_sync/device_sync.pb.js
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v1.181.2
//   protoc               v3.21.12
// source: device_sync/device_sync.proto
/* eslint-disable */
import Long from "long";
import _m0 from "protobufjs/minimal";
import { ConsentSave } from "./consent_backup.pb";
import { EventSave } from "./event_backup.pb";
import { GroupSave } from "./group_backup.pb";
import { GroupMessageSave } from "./message_backup.pb";
export const protobufPackage = "xmtp.device_sync";
/** Definitions for backups */
/** Elements selected for backup */
export var BackupElementSelection;
(function (BackupElementSelection) {
    BackupElementSelection[BackupElementSelection["BACKUP_ELEMENT_SELECTION_UNSPECIFIED"] = 0] = "BACKUP_ELEMENT_SELECTION_UNSPECIFIED";
    BackupElementSelection[BackupElementSelection["BACKUP_ELEMENT_SELECTION_MESSAGES"] = 1] = "BACKUP_ELEMENT_SELECTION_MESSAGES";
    BackupElementSelection[BackupElementSelection["BACKUP_ELEMENT_SELECTION_CONSENT"] = 2] = "BACKUP_ELEMENT_SELECTION_CONSENT";
    /** @deprecated */
    BackupElementSelection[BackupElementSelection["BACKUP_ELEMENT_SELECTION_EVENT"] = 3] = "BACKUP_ELEMENT_SELECTION_EVENT";
    BackupElementSelection[BackupElementSelection["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(BackupElementSelection || (BackupElementSelection = {}));
export function backupElementSelectionFromJSON(object) {
    switch (object) {
        case 0:
        case "BACKUP_ELEMENT_SELECTION_UNSPECIFIED":
            return BackupElementSelection.BACKUP_ELEMENT_SELECTION_UNSPECIFIED;
        case 1:
        case "BACKUP_ELEMENT_SELECTION_MESSAGES":
            return BackupElementSelection.BACKUP_ELEMENT_SELECTION_MESSAGES;
        case 2:
        case "BACKUP_ELEMENT_SELECTION_CONSENT":
            return BackupElementSelection.BACKUP_ELEMENT_SELECTION_CONSENT;
        case 3:
        case "BACKUP_ELEMENT_SELECTION_EVENT":
            return BackupElementSelection.BACKUP_ELEMENT_SELECTION_EVENT;
        case -1:
        case "UNRECOGNIZED":
        default:
            return BackupElementSelection.UNRECOGNIZED;
    }
}
export function backupElementSelectionToJSON(object) {
    switch (object) {
        case BackupElementSelection.BACKUP_ELEMENT_SELECTION_UNSPECIFIED:
            return "BACKUP_ELEMENT_SELECTION_UNSPECIFIED";
        case BackupElementSelection.BACKUP_ELEMENT_SELECTION_MESSAGES:
            return "BACKUP_ELEMENT_SELECTION_MESSAGES";
        case BackupElementSelection.BACKUP_ELEMENT_SELECTION_CONSENT:
            return "BACKUP_ELEMENT_SELECTION_CONSENT";
        case BackupElementSelection.BACKUP_ELEMENT_SELECTION_EVENT:
            return "BACKUP_ELEMENT_SELECTION_EVENT";
        case BackupElementSelection.UNRECOGNIZED:
        default:
            return "UNRECOGNIZED";
    }
}
function createBaseBackupElement() {
    return { metadata: undefined, group: undefined, groupMessage: undefined, consent: undefined, event: undefined };
}
export const BackupElement = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.metadata !== undefined) {
            BackupMetadataSave.encode(message.metadata, writer.uint32(10).fork()).ldelim();
        }
        if (message.group !== undefined) {
            GroupSave.encode(message.group, writer.uint32(18).fork()).ldelim();
        }
        if (message.groupMessage !== undefined) {
            GroupMessageSave.encode(message.groupMessage, writer.uint32(26).fork()).ldelim();
        }
        if (message.consent !== undefined) {
            ConsentSave.encode(message.consent, writer.uint32(34).fork()).ldelim();
        }
        if (message.event !== undefined) {
            EventSave.encode(message.event, writer.uint32(42).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseBackupElement();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.metadata = BackupMetadataSave.decode(reader, reader.uint32());
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.group = GroupSave.decode(reader, reader.uint32());
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.groupMessage = GroupMessageSave.decode(reader, reader.uint32());
                    continue;
                case 4:
                    if (tag !== 34) {
                        break;
                    }
                    message.consent = ConsentSave.decode(reader, reader.uint32());
                    continue;
                case 5:
                    if (tag !== 42) {
                        break;
                    }
                    message.event = EventSave.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            metadata: isSet(object.metadata) ? BackupMetadataSave.fromJSON(object.metadata) : undefined,
            group: isSet(object.group) ? GroupSave.fromJSON(object.group) : undefined,
            groupMessage: isSet(object.groupMessage) ? GroupMessageSave.fromJSON(object.groupMessage) : undefined,
            consent: isSet(object.consent) ? ConsentSave.fromJSON(object.consent) : undefined,
            event: isSet(object.event) ? EventSave.fromJSON(object.event) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.metadata !== undefined) {
            obj.metadata = BackupMetadataSave.toJSON(message.metadata);
        }
        if (message.group !== undefined) {
            obj.group = GroupSave.toJSON(message.group);
        }
        if (message.groupMessage !== undefined) {
            obj.groupMessage = GroupMessageSave.toJSON(message.groupMessage);
        }
        if (message.consent !== undefined) {
            obj.consent = ConsentSave.toJSON(message.consent);
        }
        if (message.event !== undefined) {
            obj.event = EventSave.toJSON(message.event);
        }
        return obj;
    },
    create(base) {
        return BackupElement.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseBackupElement();
        message.metadata = (object.metadata !== undefined && object.metadata !== null)
            ? BackupMetadataSave.fromPartial(object.metadata)
            : undefined;
        message.group = (object.group !== undefined && object.group !== null)
            ? GroupSave.fromPartial(object.group)
            : undefined;
        message.groupMessage = (object.groupMessage !== undefined && object.groupMessage !== null)
            ? GroupMessageSave.fromPartial(object.groupMessage)
            : undefined;
        message.consent = (object.consent !== undefined && object.consent !== null)
            ? ConsentSave.fromPartial(object.consent)
            : undefined;
        message.event = (object.event !== undefined && object.event !== null)
            ? EventSave.fromPartial(object.event)
            : undefined;
        return message;
    },
};
function createBaseBackupMetadataSave() {
    return { elements: [], exportedAtNs: Long.ZERO, startNs: undefined, endNs: undefined };
}
export const BackupMetadataSave = {
    encode(message, writer = _m0.Writer.create()) {
        writer.uint32(18).fork();
        for (const v of message.elements) {
            writer.int32(v);
        }
        writer.ldelim();
        if (!message.exportedAtNs.equals(Long.ZERO)) {
            writer.uint32(24).int64(message.exportedAtNs);
        }
        if (message.startNs !== undefined) {
            writer.uint32(32).int64(message.startNs);
        }
        if (message.endNs !== undefined) {
            writer.uint32(40).int64(message.endNs);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseBackupMetadataSave();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 2:
                    if (tag === 16) {
                        message.elements.push(reader.int32());
                        continue;
                    }
                    if (tag === 18) {
                        const end2 = reader.uint32() + reader.pos;
                        while (reader.pos < end2) {
                            message.elements.push(reader.int32());
                        }
                        continue;
                    }
                    break;
                case 3:
                    if (tag !== 24) {
                        break;
                    }
                    message.exportedAtNs = reader.int64();
                    continue;
                case 4:
                    if (tag !== 32) {
                        break;
                    }
                    message.startNs = reader.int64();
                    continue;
                case 5:
                    if (tag !== 40) {
                        break;
                    }
                    message.endNs = reader.int64();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            elements: globalThis.Array.isArray(object?.elements)
                ? object.elements.map((e) => backupElementSelectionFromJSON(e))
                : [],
            exportedAtNs: isSet(object.exportedAtNs) ? Long.fromValue(object.exportedAtNs) : Long.ZERO,
            startNs: isSet(object.startNs) ? Long.fromValue(object.startNs) : undefined,
            endNs: isSet(object.endNs) ? Long.fromValue(object.endNs) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.elements?.length) {
            obj.elements = message.elements.map((e) => backupElementSelectionToJSON(e));
        }
        if (!message.exportedAtNs.equals(Long.ZERO)) {
            obj.exportedAtNs = (message.exportedAtNs || Long.ZERO).toString();
        }
        if (message.startNs !== undefined) {
            obj.startNs = (message.startNs || Long.ZERO).toString();
        }
        if (message.endNs !== undefined) {
            obj.endNs = (message.endNs || Long.ZERO).toString();
        }
        return obj;
    },
    create(base) {
        return BackupMetadataSave.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseBackupMetadataSave();
        message.elements = object.elements?.map((e) => e) || [];
        message.exportedAtNs = (object.exportedAtNs !== undefined && object.exportedAtNs !== null)
            ? Long.fromValue(object.exportedAtNs)
            : Long.ZERO;
        message.startNs = (object.startNs !== undefined && object.startNs !== null)
            ? Long.fromValue(object.startNs)
            : undefined;
        message.endNs = (object.endNs !== undefined && object.endNs !== null) ? Long.fromValue(object.endNs) : undefined;
        return message;
    },
};
function createBaseArchiveOptions() {
    return { elements: [], startNs: undefined, endNs: undefined, excludeDisappearingMessages: false };
}
export const ArchiveOptions = {
    encode(message, writer = _m0.Writer.create()) {
        writer.uint32(10).fork();
        for (const v of message.elements) {
            writer.int32(v);
        }
        writer.ldelim();
        if (message.startNs !== undefined) {
            writer.uint32(16).int64(message.startNs);
        }
        if (message.endNs !== undefined) {
            writer.uint32(24).int64(message.endNs);
        }
        if (message.excludeDisappearingMessages !== false) {
            writer.uint32(32).bool(message.excludeDisappearingMessages);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseArchiveOptions();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag === 8) {
                        message.elements.push(reader.int32());
                        continue;
                    }
                    if (tag === 10) {
                        const end2 = reader.uint32() + reader.pos;
                        while (reader.pos < end2) {
                            message.elements.push(reader.int32());
                        }
                        continue;
                    }
                    break;
                case 2:
                    if (tag !== 16) {
                        break;
                    }
                    message.startNs = reader.int64();
                    continue;
                case 3:
                    if (tag !== 24) {
                        break;
                    }
                    message.endNs = reader.int64();
                    continue;
                case 4:
                    if (tag !== 32) {
                        break;
                    }
                    message.excludeDisappearingMessages = reader.bool();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            elements: globalThis.Array.isArray(object?.elements)
                ? object.elements.map((e) => backupElementSelectionFromJSON(e))
                : [],
            startNs: isSet(object.startNs) ? Long.fromValue(object.startNs) : undefined,
            endNs: isSet(object.endNs) ? Long.fromValue(object.endNs) : undefined,
            excludeDisappearingMessages: isSet(object.excludeDisappearingMessages)
                ? globalThis.Boolean(object.excludeDisappearingMessages)
                : false,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.elements?.length) {
            obj.elements = message.elements.map((e) => backupElementSelectionToJSON(e));
        }
        if (message.startNs !== undefined) {
            obj.startNs = (message.startNs || Long.ZERO).toString();
        }
        if (message.endNs !== undefined) {
            obj.endNs = (message.endNs || Long.ZERO).toString();
        }
        if (message.excludeDisappearingMessages !== false) {
            obj.excludeDisappearingMessages = message.excludeDisappearingMessages;
        }
        return obj;
    },
    create(base) {
        return ArchiveOptions.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseArchiveOptions();
        message.elements = object.elements?.map((e) => e) || [];
        message.startNs = (object.startNs !== undefined && object.startNs !== null)
            ? Long.fromValue(object.startNs)
            : undefined;
        message.endNs = (object.endNs !== undefined && object.endNs !== null) ? Long.fromValue(object.endNs) : undefined;
        message.excludeDisappearingMessages = object.excludeDisappearingMessages ?? false;
        return message;
    },
};
if (_m0.util.Long !== Long) {
    _m0.util.Long = Long;
    _m0.configure();
}
function isSet(value) {
    return value !== null && value !== undefined;
}
